---
title: "ccRCC_Tissue"
author: "Christina Schmidt"
date: "January 2025"
output:
  html_document:
    toc: yes
    toc_depth: 3
    toc_float: yes
  pdf_document:
    toc: yes
    toc_depth: '3'
editor_options:
  chunk_output_type: console
---
```{r, warning=FALSE, message=FALSE}
#load the needed:
library(tidyverse) # used for data manipulation
library(rmarkdown) # used for paged_table function
```

# Samples
the processed data was downloaded from the supplementary table 3 of the paper "SiRCle (Signature Regulatory Clustering) model integration reveals mechanisms of phenotype regulation in renal cancer" [Mora&Schmidt et all](https://genomemedicine.biomedcentral.com/articles/10.1186/s13073-024-01415-3). In this work the study from [Clark et all](https://www.sciencedirect.com/science/article/pii/S0092867419311237?via%3Dihub) [PDC000127](https://proteomic.datacommons.cancer.gov/pdc/study/PDC000127) was processed and SiRCle clusters assigned to each gene. Here we load the RNAseq and proteomics data together with the SiRCle cluster information as example data. 

# The Data
```{r, warning=FALSE, message=FALSE}
data <- readxl::read_excel("Input/13073_2024_1415_MOESM3_ESM.xlsx", sheet = "SiRCle")

RNAseq_TvsN <- data%>%
  select("gene_name", "entrezgene_id", "logFC_rna", "pvalue_rna", "padj_rna", "stat_rna", "RG2_Changes_filtered", "Methylation","RNA","Protein" )%>%
  mutate(RG2_Changes_filtered = ifelse(is.na(RG2_Changes_filtered), "None", RG2_Changes_filtered))%>%
  dplyr::rename("Log2FC" = "logFC_rna",
                "p.val" = "pvalue_rna",
                "p.adj" = "padj_rna",
                "t.val" = "stat_rna",
                "SiRCleCluster_RG2" = "RG2_Changes_filtered",
                "SiRCleThreshold_DNA-Methylation"="Methylation",
                "SiRCleThreshold_RNA"="RNA",
                "SiRCleThreshold_Protein"="Protein")%>%
  filter(!is.na(Log2FC))%>%
  arrange(SiRCleCluster_RG2)

Proteomics_TvsN <- data%>%
  select("gene_name", "entrezgene_id","logFC_protein", "pvalue_protein", "padj_protein", "stat_protein", "RG2_Changes_filtered", "Methylation","RNA","Protein" )%>%
  mutate(RG2_Changes_filtered = ifelse(is.na(RG2_Changes_filtered), "None", RG2_Changes_filtered))%>%
  dplyr::rename("Log2FC" = "logFC_protein",
                "p.val" = "pvalue_protein",
                "p.adj" = "padj_protein",
                "t.val" = "stat_protein",
                "SiRCleCluster_RG2" = "RG2_Changes_filtered",
                "SiRCleThreshold_DNA-Methylation"="Methylation",
                "SiRCleThreshold_RNA"="RNA",
                "SiRCleThreshold_Protein"="Protein")%>%
  filter(SiRCleThreshold_Protein != "Protein Undetected")%>%
  arrange(SiRCleCluster_RG2)

```
\
Save DFs as .csv file
```{r}
write.csv(RNAseq_TvsN, "ccRCC-Tissue_TvN_RNAseq.csv", row.names = FALSE)
write.csv(Proteomics_TvsN, "ccRCC-Tissue_TvN_Proteomics.csv", row.names=FALSE)
```

# Information about package used and versions 
```{r session_info, echo=FALSE}
options(width = 120)
sessioninfo::session_info()
```
