---
title: "Enrichment Analysis"
author:
  - name: Christina Schmidt
    affiliation:
      - Heidelberg University
  - name: Macabe Daley
    affiliation:
      - Heidelberg University
output:
  html_document:
    self_contained: true
    toc: true
    toc_float: true
    toc_depth: 5
    code_folding: show
vignette: >
  %\VignetteIndexEntry{Standard Metabolomics}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
bibliography: bibliography.bib
editor_options:
  chunk_output_type: console
  markdown:
    wrap: sentence
---

```{=html}
<style>
.vscroll-plot {
    width: 850px;
    height: 500px;
    overflow-y: scroll;
    overflow-x: hidden;
}
</style>
```
```{r chunk_setup, include = FALSE}
knitr::opts_chunk$set(
    collapse = TRUE,
    comment = "#>"
)
```

# <img src="Hexagon_MetaProViz.png" align="right" width="200"/>

\
[In this tutorial we showcase how to use **MetaProViz**]{style="text-decoration:underline"}:\
- To access metabolite prior knowledge.\
- To deal with many-to-many mapping in your metabolite identifiers.\
- To perform pathway enrichment analysis.\

\
First if you have not done yet, install the required dependencies and load the libraries:

```{r message=FALSE, warning=FALSE}
# 1. Install Rtools if you havenâ€™t done this yet, using the appropriate version (e.g.windows or macOS).
# 2. Install the latest development version from GitHub using devtools
#devtools::install_github("https://github.com/saezlab/MetaProViz")

library(MetaProViz)

#dependencies that need to be loaded:
library(magrittr)
library(dplyr)

library(tidyr)

#Please install the Biocmanager Dependencies:
#BiocManager::install("clusterProfiler")
#BiocManager::install("EnhancedVolcano")
#BiocManager::install("cosmosR")
```

\
\

::: {.progress .progress-striped .active}
::: {.progress-bar .progress-bar-success style="width: 100%"}
:::
:::

# 1. Loading the example data

::: {.progress .progress-striped .active}
::: {.progress-bar .progress-bar-success style="width: 100%"}
:::
:::

\
[As part of the **MetaProViz** package you can load the example data into your global environment using the function `toy_data()`]{style="text-decoration:underline"}:\
`1.`Cell line experiment **(CellLine)**\
Here we choose an example datasets, which is publicly available on [metabolomics workbench project PR001418](https://www.metabolomicsworkbench.org/data/DRCCMetadata.php?Mode=Project&ProjectID=PR001418) including metabolic profiles of human renal epithelial cells HK2 and cell renal cell carcinoma (ccRCC) cell lines cultured in Plasmax cell culture media [@Sciacovelli_Dugourd2022].
The raw data are available via [metabolomics workbench study ST002224](https://www.metabolomicsworkbench.org/data/DRCCMetadata.php?Mode=Study&StudyID=ST002224&StudyType=MS&ResultType=1) were intracellular metabolomics of HK2 and ccRCC cell lines 786-O, 786-M1A and 786-M2A were performed.\
We have performed pre-processing and differential analysis (details can be found in the vignette vignette [Standard Metabolomics](https://saezlab.github.io/MetaProViz/articles/Standard%20Metabolomics.html)) and and here we load the differential metabolite analysis results for the comparison of 786M-1A versus HK2.\

```{r}
#Load the Pre-processed intracellular data:
Intra_DMA_786M1A_vs_HK2<- MetaProViz::ToyData(Data="IntraCells_DMA")
```

`2.` Tissue experiment **(Tissue)**\
Here we chose publicly available data from the [paper](https://www.cell.com/cancer-cell/comments/S1535-6108(15)00468-7#supplementaryMaterial) "An Integrated Metabolic Atlas of Clear Cell Renal Cell Carcinoma", which includes metabolomic profiling on 138 matched clear cell renal cell carcinoma (ccRCC)/normal tissue pairs.\
We have performed differential analysis (details can be found in the vignette [Metadata Analysis](https://saezlab.github.io/MetaProViz/articles/Metadata%20Analysis.html)) and and here we load the differential metabolite analysis results for the comparison of Tumour versus Normal.\

```{r}
# Load the example data:
Tissue_TvsN <- MetaProViz::ToyData(Data="Tissue_DMA")
Tissue_TvsN_Old <- MetaProViz::ToyData(Data="Tissue_DMA_Old")
Tissue_TvsN_Young <- MetaProViz::ToyData(Data="Tissue_DMA_Young")

```

\
\

::: {.progress .progress-striped .active}
::: {.progress-bar .progress-bar-success style="width: 100%"}
:::
:::

# 2. Accessing Prior Knowledge

::: {.progress .progress-striped .active}
::: {.progress-bar .progress-bar-success style="width: 100%"}
:::
:::

\
Metabolite prior knowledge (PK) is essential for the interpretation of metabolomics data.
It can be used to perform pathway enrichment analysis, compound class enrichment analysis, and by using specific PK databases, it can be used to study the connection of metabolites and receptors or transporters.Since the quality and content of the PK will dictate the success of the downstream analysis and biological interpretation, it is important to ensure the PK is used correctly.\
Specifically in metabolite PK, the many different PK databases and resources pose several issues.
Indeed, the metabolite identifiers (e.g. KEGG, HMDB, PubChem, etc.) are not standardized across databases, and the same metabolite can have multiple identifiers in different databases.
This is known as the many-to-many mapping problem.\
Moreover, metabolic pathways that are the basis of the PK databases also includes co-factors such as ions or other small molecules that are not only part of most reactions, but can also not be detected in experimentallly acquired data (e.g. H2O, CO2, etc).

## KEGG pathway-metabolite sets

KEGG pathways that are loaded via KEGG API using the package `KEGGREST` and can be used to perform pathway analysis [@Kanehisa2000].
**(KEGG_Pathways)**\

```{r}
#This will use KEGGREST to query the KEGG API to load the pathways:
MetaProViz::LoadKEGG()
```

\

```{r, echo=FALSE}
# Check how our data looks like:
KEGG_Pathways[c(1:3,52:54),]%>%
  kableExtra::kbl(caption = "Preview of the DF `KEGG_Pathways`.") %>%
  kableExtra::kable_classic(full_width = F, html_font = "Cambria", font_size = 12)
```

## Create pathway-metabolite sets

The function `Make_GeneMetabSet` can be used to translate gene names to metabolite names by using a PK network of metabolic reactions calls CosmosR [@Dugourd2021].
This function is useful if you want to perform pathway enrichment analysis on available gene-sets such as the Hallmarks gene-sets from MSigDB [@Castanza2022].
Moreover, it enables you to perform combined pathway enrichment analysis on metabolite-gene sets, if you have other data types such as proteomics measuring the enzymes expression.\
The Hallmarks gene-set is available in the package `MetaProViz` and can be loaded using the function `LoadHallmarks()`.\

```{r}
#Load the example data:
MetaProViz::LoadHallmarks()
```

```{r, echo=FALSE}
# Check how our data looks like:
Hallmark_Pathways[c(997:998,2916:2917, 2916:2918),]%>%
  kableExtra::kbl(caption = "Preview of the DF `Hallmarks_Pathways` including gene-sets usable for pathway enrichment analysis.", row.names=FALSE) %>%
  kableExtra::kable_classic(full_width = F, html_font = "Cambria", font_size = 12)
```

\
Now we can use the function `Make_GeneMetabSet` to translate the gene names to metabolite names.

```{r}
#Translate gene names to metabolite names
Hallmarks_GeneMetab <- MetaProViz::Make_GeneMetabSet(Input_GeneSet=Hallmark_Pathways,
                                                     SettingsInfo=c(Target="gene"),
                                                     PKName="Hallmarks")
```

```{r, echo=FALSE}
# Check how our data looks like:
Hallmarks_GeneMetab[["GeneMetabSet"]]%>%
  dplyr::filter(term=="HALLMARK_GLYCOLYSIS")%>%
  dplyr::slice(c(59,142,193, 201:203))%>%
  kableExtra::kbl(caption = "Preview of the DF `Hallmarks_GeneMetab` including gene-sets usable for pathway enrichment analysis.") %>%
  kableExtra::kable_classic(full_width = F, html_font = "Cambria", font_size = 12)

```

```{r, echo=FALSE}
#plot out how the metabolite set size is different from the original gene-set size
#Show bar graph of the size of the gene set versus the metabolite set --> Show both tables (pathways gene names and pathways metabolite names) and show the difference in size (e.g. 10% of the genes are translated to metabolites, 90% are not metabolic enzymes and hence only genes)


```

\
Given that we have the gene-metabolite-sets, we can now also run enrichment analysis on combined data types, once including the metabolite Log2FC and one including gene Log2FC from e.g. transcriptomics or proteomics data.
Yet, it is important to keep in mind that generally we detect less metabolites than genes and hence this may bias the results obtained from combined enrichment analysis.

## MetaLinksDB Metabolite-receptor sets
The MetaLinks database is a manually curated database of metabolite-receptor and metabolite-transporter sets that can be used to study the connection of metabolites and receptors or transporters [@Farr_Dimitrov2024].\

```{r}
MetaLinksDB_Res <- MetaProViz::LoadMetalinks()
```

```{r, echo=FALSE}
# Check how our data looks like:
MetaLinksDB_Res[["MetalinksDB"]][c(1:8),]%>%
  kableExtra::kbl(caption = "Preview of the DF `MetaLinksDB` including metabolite-receptor sets.", row.names=FALSE) %>%
  kableExtra::kable_classic(full_width = F, html_font = "Cambria", font_size = 12)

MetaLinksDB_Res[["MetalinksDB_Type"]][c(1,50, 90, 101),]%>%
  kableExtra::kbl(caption = "Preview of the DF `MetaLinksDB_Type` including metabolite-receptor and metabolite-transporter sets.", row.names=FALSE) %>%
  kableExtra::kable_classic(full_width = F, html_font = "Cambria", font_size = 12)
```

\
\


::: {.progress .progress-striped .active}
::: {.progress-bar .progress-bar-success style="width: 100%"}
:::
:::

# 3. Translate IDs

::: {.progress .progress-striped .active}
::: {.progress-bar .progress-bar-success style="width: 100%"}
:::
:::

\

::: {style="border: 1px solid red; padding: 10px; background-color: #ffe6e6;"}
<strong>Important Information:</strong> Translating IDs between formats e.g. KEGG to HMDB is a non-trivial task, and it is expected for one original ID to link to many translated IDs, and vice versa. We discuss the implications throughout this vignette and leave it to user discretion to select the most appropriate ID based on their research question and data.
:::

<br>

Across the different prior knowledge resources (see also tables above) specific metabolite IDs are used and hence depending on the prior knowledge resource a specific metabolite ID is required.\
If we want to convert or 'translate' those IDs to another commonly used form of ID, for instance because our measured data uses another type of ID, we can make use of the `MetaProViz::TranslateID()` function.
This is based on [OmniPathR](https://www.bioconductor.org/packages/release/bioc/html/OmnipathR.html) and  RaMP-DB [@Braisted2023] on the backend of our function and currently supports ID translation of metabolites to and from the following formats:\
-   KEGG\
-   HMDB\
-   ChEBI\
-   PubChem\
\
As an example we are translating the KEGG pathways we loaded with `metaproViz::LoadKEGG` into HMDB ids:\
```{r, echo=TRUE, message=FALSE, warning=FALSE}
KEGG_Pathways_Translated <- MetaProViz::TranslateID(InputData= KEGG_Pathways,
                                                     SettingsInfo = c(InputID="MetaboliteID", GroupingVariable="term"),
                                                     From = c("kegg"),
                                                     To = c("hmdb", "pubchem"))
```

```{r, echo=FALSE, message=FALSE, warning=FALSE, include=TRUE}
KEGG_Pathways_Translated[["TranslatedDF"]][c(1:3, 300:301,600:602),] %>%
  kableExtra::kbl(caption = "Translation of KEGG IDs in KEGG pathways to HMDB & pubchem IDs") %>%
  kableExtra::kable_classic(full_width = F, html_font = "Cambria", font_size = 12)
```
\
Here it becomes apparent that the translation of IDs is not a one-to-one mapping, but rather a one-to-many mapping. In fact it is very common that an ID from one format will have a genuine one-to-many relationship with the other format (e.g. one KEGG ID maps to multiple HMDB IDs) or even a many-to-many relationship, where some of the IDs from the new format link back to multiple IDs in the original format (e.g. two different KEGG IDs map to multiple HMDS IDs, some of which are shared between them).\
This comes with many implications for the analysis that will be discussed in the next section.

## 3.1 Mapping problems
The complexities of translating metabolite IDs are demonstrated here (Fig.2). The relationships between Original IDs (e.g. KEGG) and Translated IDs (e.g. HMDB), can be quite complex and in fact we encounter `one-to-one` (no matches were found for the ID), `one-to-none` ( direct #relationship was established), `one-to-many` (multiple matches were found for the ID. i.e. it is ambiguously mapped) and `many-to-many` ( considers the relationships between the Translated IDs to Original IDs , where a translated ID ambiguously maps back to multiple different Original IDs) mappings.\
For enrichment analysis the translation from KEGG IDs to HMDB IDs increases the pathways size, i.e. how many metabolites are in the pathway "Glycolysis / Gluconeogenesis - Homo sapiens (human)", which would in turn inflate/deflate the enrichment results and hence it would be desired to keep the number of metabolites in a pathway consistent.
\

<center>

![Fig. 2: Mapping problems in prior knowledge metabolite-sets when translating metabolite IDs.](Fig.2.png){width="50%"}

</center>

<br>

Because of this complexity the output of `MetaProViz::TranslateID()` includes not only the Translation table showcased above, but additionally information about the mapping ambiguity as well as a summary of the relationships between the Original and Translated IDs.\
Indeed, the translation of e.g. KEGG to hmdb and pubchem includes multiple data frames including a summary of the mapping occurrences:\
```{r}
names(KEGG_Pathways_Translated)
```

```{r, echo=FALSE, message=FALSE, warning=FALSE, include=TRUE}
KEGG_Pathways_Translated[["MappingSummary_hmdb"]][c(1:3, 7:9),] %>%
  kableExtra::kbl(caption = "MappingSummary_hmdb") %>%
  kableExtra::kable_classic(full_width = F, html_font = "Cambria", font_size = 12)
```
\
\
We also have the ability to extract a long version of the DF that includes a row for each mapping occurrence, which can be useful for downstream analysis. Yet, this can become very large dependent on the amount of many-to-many mappings, hence by default we do not generate this summary. Within the `MetaProViz::TranslateID()` you can set the parameter `Summary =TRUE` or in case you have a dataframe that includes both, original and translated ID, you can use the function `MetaProViz::MappingAmbiguity()` to generate this long summary as well as the mapping summary in general.\
```{r, eval=FALSE, include=TRUE}
# Option 1:
KEGG_Pathways_TranslatedSum <- MetaProViz::TranslateID(InputData= KEGG_Pathways,
                                                    SettingsInfo = c(InputID="MetaboliteID", GroupingVariable="term"),
                                                    From = c("kegg"),
                                                    To = c("hmdb", "pubchem"),
                                                    Summary =TRUE)
```
```{r}
# Option 2:
MappingProblems <- MetaProViz::MappingAmbiguity(InputData= KEGG_Pathways_Translated[["TranslatedDF"]]%>% dplyr::rename("KEGG"="MetaboliteID"),
                                                From = "KEGG",
                                                To = "hmdb",
                                                GroupingVariable = "term",
                                                Summary=TRUE)

```
```{r, echo=FALSE, message=FALSE, warning=FALSE, include=TRUE}
MappingProblems[["Summary"]][c(1:2, 503, 505),c(2:6, 8:9,11)] %>%
  kableExtra::kbl(caption = "Long summary of mapping problems taking into account both directions, From-to-To and To-to-From.", row.names=FALSE) %>%
  kableExtra::kable_classic(full_width = F, html_font = "Cambria", font_size = 12)
```
\
in the table it is shown that the KEGG ID C00002 maps to 3 different HMDB IDs, and it is shown that one of those HMDB IDs HMDB0000538 maps to one KEGG ID, hence this Mapping is one-to-many. The other two HMDB Ids are also in the table and it is checked to how many KEGG IDs they map. Additionally, we have passed `GroupingVariable = "term"` as we have pathways, which also means each of those mappings is checked within a pathway and across pathways (e.g. C00002 is shown twice, for two different terms). \
In fact, to perform enrichment analysis we need a column source (=e.g. term) and we would want to keep the metabolite IDs across pathways consistent, avoid ambiguous mapping as much as possible (many-to-many mapping), and have this metabolite ID selection guided by any IDs we might have available to us in our measured input data (Fig. 3). This is crucial to circumvent inflation or deflation of metabolite-sets, which in turn will affect the enrichment analysis results. In the next section 4., we will elaborate on this.
\

<center>

![Fig. 3: Mapping problems in prior knowledge metabolite-sets when translating metabolite IDs and the connection to detected (measured input metabolites). ](Fig.3.png){width="50%"}

</center>

<br>


::: {.progress .progress-striped .active}
::: {.progress-bar .progress-bar-success style="width: 100%"}
:::
:::

# 4. Metabolite IDs in measured data

::: {.progress .progress-striped .active}
::: {.progress-bar .progress-bar-success style="width: 100%"}
:::
:::

## 4.1 Use measured Metabolite IDs to guide the selection of PK IDs
The measured metabolite IDs can be used to guide the selection of PK IDs. This is crucial to circumvent inflation or deflation of metabolite-sets, which in turn will affect the enrichment analysis results.\
This is something we are currently working on and hope to provide within the next release, so stay tuned.

## 4.2. Assigning Metabolite IDs to measured data
The difficulty with assigning metabolite IDs to measured data is the uncertainty in the detection of metabolites. Indeed, differentiation of structural isomers (both constitutional isomers and stereoisomers) as for example the distinction between enantiomers. This leads to loss of information and hence uncertainty is assigning metabolite IDs.\
One example is the metabolite Alanine, which can occur in its L- or D- form. If in an experiment those enantiomers have not been distinguished, the correct way would be to either assign two metabolite IDs (L- and D-Alanine) or a more general Alanine ID without chiral information. Yet, in reality this is not as trivial:\
```{r, echo=FALSE}
#Create DF for Alanine:
Alanine <- data.frame(
  TrivialName = c("D-Alanine", "L-Alanine", "Alanine", "Alanine zwitterion"),
  HMDB= c("HMDB0001310",  "HMDB0000161", NA, NA),
  ChEBI = c("15570", "16977", "16449", "66916" ),
  KEGG = c("C00133", "C00041", "C01401", NA ),
  PubChem = c("71080" , "5950", "602", "57383916"),
  stringsAsFactors = FALSE)

# Print table:
Alanine%>%
  kableExtra::kbl(caption = "Available Alanine IDs in HMDB and ChEBI.") %>%
  kableExtra::kable_classic(full_width = F, html_font = "Cambria", font_size = 12)
```
\
Indeed, dependent on the database, the Alanine metabolite can have different IDs available:\
```{r, echo=FALSE}
#Print Plot:
AlaninePlot <- Alanine%>%
  tidyr::pivot_longer(cols = c("HMDB", "ChEBI", "KEGG", "PubChem"),
                               names_to = "Conditions",
                               values_to = "ID")%>%
  tidyr::drop_na()
AlaninePlot$`IDs in Database` <- as.numeric(1)

Plot <- ggplot2::ggplot(AlaninePlot, ggplot2::aes(fill=TrivialName, y=`IDs in Database`, x=Conditions)) +
        ggplot2::geom_bar(position="stack", stat="identity")+
        ggplot2::scale_fill_manual(values = c("D-Alanine" = "#9F0162"  , "L-Alanine" = "#006384", "Alanine" = "grey", "Alanine zwitterion" = "#00735C"))+
        ggplot2::theme_classic()+
        ggplot2::ggtitle("Alanine IDs in in different databases")+
        ggplot2::theme(axis.text.x = ggplot2::element_text(angle = 90, hjust = 1))+
        ggplot2::labs(x = "Database", y = "Number of IDs")

Plot_Sized <-  MetaProViz:::plotGrob_Superplot(InputPlot=Plot, SettingsInfo=c(Conditions="Conditions", Superplot="Conditions"), SettingsFile_Sample=AlaninePlot,  PlotName = "Alanine IDs in in different databases", Subtitle = NULL, PlotType="Bar")

PlotHeight <- grid::convertUnit(Plot_Sized$height, 'cm', valueOnly = TRUE)
PlotWidth <- grid::convertUnit(Plot_Sized$width, 'cm', valueOnly = TRUE)
Plot_Sized %<>%
      {ggplot2::ggplot() + ggplot2::annotation_custom(.)} %>%
      add(ggplot2::theme(panel.background = ggplot2::element_rect(fill = "transparent")))

#ggplot2::ggsave(filename = "AlaninePlot.svg", plot = Plot_Sized, width = PlotWidth,  height = PlotHeight, unit="cm")

Plot_Sized
```
\
For instance, if we want to assign a HMDB ID, we have to assign both "HMDB0001310",  "HMDB0000161" to the metabolite Alanine, for ChEBI we could assign only one, "16449", but this may lead to other problems as the ChEBI ID is not specific and may not be part of certain metabolic pathways. The reason for this is that substrate chirality is critical to enzymatic processes and stereo selectivity of enzymes to be homochiral with predominance of one particular enantiomer (e.g. D-sugars, L-amino acids, etc.).\
To showcase the severity of this problem, we can look at the occurrence of those metabolites in metabolic pathways across different databases. To do so we searched for those metabolite IDs in the RaMP database [@Braisted2023] and extracted the pathways they are part of:\
```{r, echo=FALSE}
# devtools::install_github("ncats/RAMP-DB")
rampDB <-  suppressMessages(suppressWarnings(RaMP::RaMP(version = "2.5.4")))

Pathways <- RaMP::getPathwayFromAnalyte(db = rampDB, c("hmdb:HMDB0001310", "hmdb:HMDB0000161", "chebi:16449", "chebi:66916", "chebi:16977", "chebi:15570", "kegg:C00041", "kegg:C00133", "kegg:C01401", "pubchem:71080","pubchem:5950","pubchem:602","pubchem:57383916" ))

Count <- Pathways %>%
  tidyr::unite(Pathway, c("pathwaySource", "pathwayName"), sep = ": ")%>%
  dplyr::group_by(inputId) %>%
  dplyr::summarise(PathwayCount = dplyr::n_distinct(Pathway), .groups = 'drop')%>%
  tidyr::separate(inputId, into = c("Database", "ID"), sep = ":", remove = TRUE)

Alanine <- merge(x = AlaninePlot%>%dplyr::rename("Database"="Conditions") ,
                 y= Count[,2:3],
                 by = "ID",
                 all.x=TRUE)%>%
  dplyr::mutate(PathwayCount = replace(PathwayCount, is.na(PathwayCount), 0))%>%
  dplyr::arrange(desc( TrivialName)) %>%
  dplyr::select(TrivialName, ID, Database, PathwayCount)

# Print table:
Alanine%>%
  kableExtra::kbl(caption = "Alanine IDs in HMDB and ChEBI mapped to pathways from wiki, KEGG and Reactome using RamP.") %>%
  kableExtra::kable_classic(full_width = F, html_font = "Cambria", font_size = 12)
```
\
```{r, echo=FALSE}
#BarGraph
Plot <- ggplot2::ggplot(Alanine, ggplot2::aes(fill=TrivialName, y=`PathwayCount`, x=Database)) +
        ggplot2::geom_bar(position="stack", stat="identity")+
        ggplot2::scale_fill_manual(values = c("D-Alanine" = "#9F0162"  , "L-Alanine" = "#006384", "Alanine" = "grey", "Alanine zwitterion" = "#00735C"))+
        ggplot2::theme_classic()+
        ggplot2::ggtitle("Alanine IDs in in different pathway")+
        ggplot2::theme(axis.text.x = ggplot2::element_text(angle = 90, hjust = 1))+
        ggplot2::labs(x = "Database", y = "Number of Pathways")

Plot_Sized <-  MetaProViz:::plotGrob_Superplot(InputPlot=Plot, SettingsInfo=c(Conditions="Conditions", Superplot="Conditions"), SettingsFile_Sample=AlaninePlot,  PlotName = "Alanine IDs in in different databases", Subtitle = NULL, PlotType="Bar")

PlotHeight <- grid::convertUnit(Plot_Sized$height, 'cm', valueOnly = TRUE)
PlotWidth <- grid::convertUnit(Plot_Sized$width, 'cm', valueOnly = TRUE)
Plot_Sized %<>%
      {ggplot2::ggplot() + ggplot2::annotation_custom(.)} %>%
      add(ggplot2::theme(panel.background = ggplot2::element_rect(fill = "transparent")))

#ggplot2::ggsave(filename = "AlaninePlot_Pathways.svg", plot = Plot_Sized, width = PlotWidth,  height = PlotHeight, unit="cm")

Plot_Sized
```

```{r, eval=FALSE, include=FALSE}
# Make BarGraph per pathway and ID type
Alanine_Path <- Pathways%>%
  tidyr::separate(commonName, into = c("commonName", "secondaryName"), sep = ",A", remove = FALSE)%>%
  tidyr::separate(commonName, into = c("commonName", "secondaryName"), sep = ",D", remove = FALSE)%>%
  dplyr::group_by(pathwaySource, commonName) %>%
  dplyr::summarise(PathwayCount = dplyr::n_distinct(pathwayName),
                   commonNames = paste(unique(commonName), collapse = ", "),
                   .groups = 'drop')


#BarGraph
Plot <- ggplot2::ggplot(Alanine_Path, ggplot2::aes(fill=commonNames, y=PathwayCount , x=pathwaySource)) +
        ggplot2::geom_bar(position="stack", stat="identity")+
        ggplot2::scale_fill_manual(values = c("D-Alanine" = "#9F0162"  , "L-Alanine" = "#006384", "Alanine" = "grey", "Alanine zwitterion" = "#00735C"))+
        ggplot2::theme_classic()+
        ggplot2::ggtitle("Alanine IDs in in different databases")+
        ggplot2::theme(axis.text.x = ggplot2::element_text(angle = 90, hjust = 1))+
        ggplot2::labs(x = "Pathway Source", y = "Number of Pathways")


#* https://www.sciencedirect.com/science/article/pii/S0731708521005410
#* This problem is even excarbated in lipidomics, where different levels of residues....
```
\
This showcases if we choose the ChEBI ID for Alanine (ChEBI ID 16449), if experimentally the distinction was not possible, we will not map to any pathway even though the metabolite is part of many pathways. Hence, we recommend to assign multiple IDs to a measured ID, where specificity in detection is not given.\

::: {.progress .progress-striped .active}
::: {.progress-bar .progress-bar-success style="width: 100%"}
:::
:::

# 5. Run enrichment analysis

::: {.progress .progress-striped .active}
::: {.progress-bar .progress-bar-success style="width: 100%"}
:::
:::

\
There are two options:\
1.
Over Representation Analysis (ORA) that determines if a set of features (=metabolic pathways) are over-represented in the selection of features (=metabolites) from the data in comparison to all measured features (metabolites) using the Fishers exact test `MetaProViz::ClusterORA`.
This can be applied to clusters of metabolites as for example the results from `MetaProViz::MCA_2Cond()` or `MetaProViz::CoRe()` function.
If you want to have more details on these clustering methods please visit the vignette [Standard Metabolomics](https://saezlab.github.io/MetaProViz/articles/Standard%20Metabolomics.html) or [CoRe Metabolomics](https://saezlab.github.io/MetaProViz/articles/CoRe%20Metabolomics.html).\

\
2.
Enrichment analysis on standard differential analysis results.
We offer ORA `MetaProViz::StandardORA`, but there are many other statistical tests that can be used for enrichment analysis.
The full scope of different methods is beyond the scope of MetaProViz, but are available in decoupleR [@Badia-I-Mompel2022] packages from our group.

```{r, echo=FALSE, eval=FALSE}
# Future

#Run metabolite-gene analysis - Maybe add cosmos data - this also has matched transcriptomics (otherwise there are also the CPTAC data for ccRCC) and Cissys data are to be published soon as well.
#- Maybe add the data from the paper of the 3D cell culture model of ccRCC

#Run footprinting analysis - Maybe move to extra vignette - ocean method --\> MetaProViz::RunOcean

#Reactome - As we already did most of the API retrieval, we can consider to add `MetaProViz::LoadReactome` - Remove unspecific metabolites - Translate ChEBI to HMDB and KEGG - FYI: See code I used to access via API

#ocean PK - Add HMDB and CheBI IDs to the KEGG IDs.
#- Make ocean PK accessible via MetaProViz

#Lipid prediction - This goes into the direction of linex where we will use the lipidPKN and the reaction rules to also use lipids that are not in the prior knowledge when doing downstream analsyis!
#- linex2 PKN is manually curated - instead of using this we may want to make am updateable version.
#- Now we can at least do understand the classes present \* LipidMaps IDs, can we link other ID types?

#Viz: - Showcase specific viz options - Add network plot





#*Discuss chemical class enrichment, clustering before/after enrichment analysis. Also showcase metabolite-protein enrichment. Discuss how adding the additional columns (showcase helper function) will aid interpretability!
#*Also discuss that chemical enrichment can be more informative and what are the disadvantages of pathways (specifically in metabolomics/lipidomics) ---> Makes it even more important to make the right visualisation choices as the feature space is not that big (compared to other omics)


```
\
\


::: {.progress .progress-striped .active}
::: {.progress-bar .progress-bar-success style="width: 100%"}
:::
:::

# Session information

```{r session_info, echo=FALSE}
options(width = 120)
sessionInfo()
```

# Bibliography
