<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>core Metabolomics • MetaProViz</title>
<script src="../../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../../deps/headroom-0.11.0/headroom.min.js"></script><script src="../../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../../deps/search-1.0.0/fuse.min.js"></script><script src="../../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../../pkgdown.js"></script><link href="../../extra.css" rel="stylesheet">
<meta property="og:title" content="core Metabolomics">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../../index.html">MetaProViz</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">3.99.24</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true"><span class="fa fa-book-open"></span></button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-">
<li><a class="dropdown-item" href="../../articles/quick-start.html">Getting started</a></li>
    <li><a class="dropdown-item" href="../../articles/vignettes/pkgdown/standard-metabolomics.html">Standard Metabolomics</a></li>
    <li><a class="dropdown-item" href="../../articles/vignettes/pkgdown/core-metabolomics.html">CoRe Metabolomics</a></li>
    <li><a class="dropdown-item" href="../../articles/vignettes/pkgdown/prior-knowledge.html">Prior Knowledge</a></li>
    <li><a class="dropdown-item" href="../../articles/vignettes/pkgdown/sample-metadata.html">Sample Metadata Analysis</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../../reference/index.html"><span class="fa fa-solid fa-rocket"></span> Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/saezlab/MetaProViz"><span class="fa fa-brands fa-github"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://saezlab.org/"><span class="fa fas fa-university"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://frezza.cecad-labs.uni-koeln.de/home"><span class="fa fas fa-university"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">



<script src="core-metabolomics_files/kePrint-0.0.1/kePrint.js"></script><link href="core-metabolomics_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>core Metabolomics</h1>
                        <h4 data-toc-skip class="author">Christina
Schmidt</h4>
            <address class="author_afil">
      Heidelberg
University<br><h4 data-toc-skip class="author">Dimitrios
Prymidis</h4>
            <address class="author_afil">
      Cologne University<br><small class="dont-index">Source: <a href="https://github.com/saezlab/MetaProViz/blob/main/vignettes/pkgdown/core-metabolomics.Rmd" class="external-link"><code>vignettes/pkgdown/core-metabolomics.Rmd</code></a></small>
      <div class="d-none name"><code>core-metabolomics.Rmd</code></div>
    </address>
</address>
</div>

    
    
<style>
.vscroll-plot {
    width: 850px;
    height: 500px;
    overflow-y: scroll;
    overflow-x: hidden;
}
</style>
<div class="section level2">
<h2 id="section">
<img src="Hexagon_MetaProViz.png" align="right" width="200"><a class="anchor" aria-label="anchor" href="#section"></a>
</h2>
<p><br>
A Consumption-Release (core) metabolomics experiment usually refers to a
cell culture experiment where metabolomics is performed on the cell
culture media.<br><br><span style="text-decoration:underline">In this tutorial we showcase how
to use <strong>MetaProViz</strong></span>:<br></p>
<ul>
<li>to process raw peak data and identify outliers.<br>
</li>
<li>to perform differential metabolite analysis (dma) to generate
Log2Distance and statistics and perform pathway analysis using Over
Representation Analysis (ORA) on the results.<br>
</li>
<li>to do metabolite clustering analysis (MCA) to find clusters of
metabolites with similar behaviors and perform pathway analysis using
ORA on each cluster.<br>
</li>
<li>to use specific visualizations to aid biological interpretation of
the results.<br><br><br>
First if you have not done yet, install the required dependencies and
load the libraries:</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Install Rtools if you haven’t done this yet, using the appropriate version (e.g.windows or macOS).</span></span>
<span><span class="co"># 2. Install the latest development version from GitHub using devtools</span></span>
<span><span class="co"># devtools::install_github("https://github.com/saezlab/MetaProViz")</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://saezlab.github.io/MetaProViz" class="external-link">MetaProViz</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># dependencies that need to be loaded:</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rlang.r-lib.org" class="external-link">rlang</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/sinhrks/ggfortify" class="external-link">ggfortify</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Please install the Biocmanager Dependencies:</span></span>
<span><span class="co"># BiocManager::install("clusterProfiler")</span></span>
<span><span class="co"># BiocManager::install("EnhancedVolcano")</span></span></code></pre></div>
<br><br><div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
</div>
<div class="section level2">
<h2 id="loading-the-example-data">1. Loading the example data<a class="anchor" aria-label="anchor" href="#loading-the-example-data"></a>
</h2>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
<p>Here we choose an example datasets, which is publicly available on <a href="https://www.metabolomicsworkbench.org/data/DRCCMetadata.php?Mode=Project&amp;ProjectID=PR001418" class="external-link">metabolomics
workbench project PR001418</a> including metabolic profiles of human
renal epithelial cells HK2 and cell renal cell carcinoma (ccRCC) cell
lines cultured in Plasmax cell culture media. Here we use the integrated
raw peak data as example data using the trivial metabolite name in
combination with the KEGG ID as the metabolite identifiers.<br><br><span style="text-decoration:underline">As part of the
<strong>MetaProViz</strong> package you can load the example data into
your global environment using the function
<code>toy_data()</code></span>:<br><br><code>1.</code> core experiment <strong>(core)</strong><br>
The raw data are available via <a href="https://www.metabolomicsworkbench.org/data/DRCCMetadata.php?Mode=Study&amp;StudyID=ST002226&amp;StudyType=MS&amp;ResultType=1" class="external-link">metabolomics
workbench study ST002226</a> were exometabolomics of HK2 and ccRCC cell
lines 786-O, 786-M1A, 786-M2A, OS-RC-2, OS-LM1 and RFX-631 were
performed.<br></p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">medium_raw</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Media</span> <span class="op">&lt;-</span> <span class="va">medium_raw</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Code"</span><span class="op">)</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>core</code> including columns with sample
information and metabolite ids with their measured values.
</caption>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Conditions
</th>
<th style="text-align:right;">
Biological_Replicates
</th>
<th style="text-align:right;">
GrowthFactor
</th>
<th style="text-align:right;">
valine-d8
</th>
<th style="text-align:right;">
hipppuric acid-d5
</th>
<th style="text-align:right;">
2-hydroxyglutarate
</th>
<th style="text-align:right;">
2-ketoglutarate
</th>
<th style="text-align:right;">
3-Dehydro-L-threonate
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
MS51-06
</td>
<td style="text-align:left;">
HK2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
249.2817
</td>
<td style="text-align:right;">
780552871
</td>
<td style="text-align:right;">
3127630257
</td>
<td style="text-align:right;">
84950547
</td>
<td style="text-align:right;">
169244158
</td>
<td style="text-align:right;">
1807489245
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-07
</td>
<td style="text-align:left;">
HK2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
249.2817
</td>
<td style="text-align:right;">
802602348
</td>
<td style="text-align:right;">
3256031922
</td>
<td style="text-align:right;">
60753859
</td>
<td style="text-align:right;">
151064767
</td>
<td style="text-align:right;">
695228424
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-08
</td>
<td style="text-align:left;">
HK2
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
249.2817
</td>
<td style="text-align:right;">
831984796
</td>
<td style="text-align:right;">
3308009345
</td>
<td style="text-align:right;">
73718363
</td>
<td style="text-align:right;">
171281531
</td>
<td style="text-align:right;">
791442407
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-09
</td>
<td style="text-align:left;">
HK2
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
249.2817
</td>
<td style="text-align:right;">
822744518
</td>
<td style="text-align:right;">
3209731571
</td>
<td style="text-align:right;">
65933166
</td>
<td style="text-align:right;">
112033043
</td>
<td style="text-align:right;">
315209589
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-10
</td>
<td style="text-align:left;">
HK2
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
249.2817
</td>
<td style="text-align:right;">
805565867
</td>
<td style="text-align:right;">
3297793480
</td>
<td style="text-align:right;">
68183576
</td>
<td style="text-align:right;">
170902744
</td>
<td style="text-align:right;">
615035216
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-11
</td>
<td style="text-align:left;">
786-O
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
297.3423
</td>
<td style="text-align:right;">
841873509
</td>
<td style="text-align:right;">
3418515398
</td>
<td style="text-align:right;">
75941661
</td>
<td style="text-align:right;">
215553005
</td>
<td style="text-align:right;">
501977089
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-12
</td>
<td style="text-align:left;">
786-O
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
297.3423
</td>
<td style="text-align:right;">
825462964
</td>
<td style="text-align:right;">
3218049751
</td>
<td style="text-align:right;">
62100210
</td>
<td style="text-align:right;">
195308040
</td>
<td style="text-align:right;">
484681099
</td>
</tr>
</tbody>
</table>
<p><br><code>2.</code> Additional information mapping the trivial metabolite
names to KEGG IDs and selected pathways
<strong>(MappingInfo)</strong><br></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">cellular_meta</span><span class="op">)</span></span>
<span></span>
<span><span class="va">MappingInfo</span> <span class="op">&lt;-</span> <span class="va">cellular_meta</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Metabolite"</span><span class="op">)</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>Pathways</code> including the trivial metabolite
identifiers used in the experiment as well as KEGG IDs and pathway
information.
</caption>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
HMDB
</th>
<th style="text-align:left;">
KEGG.ID
</th>
<th style="text-align:left;">
KEGGCompound
</th>
<th style="text-align:left;">
Pathway
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
N-acetylaspartate
</td>
<td style="text-align:left;">
HMDB0000812
</td>
<td style="text-align:left;">
C01042
</td>
<td style="text-align:left;">
N-Acetyl-L-aspartate
</td>
<td style="text-align:left;">
Alanine, aspartate and glutamate metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
argininosuccinate
</td>
<td style="text-align:left;">
HMDB0000052
</td>
<td style="text-align:left;">
C03406
</td>
<td style="text-align:left;">
N-(L-Arginino)succinate
</td>
<td style="text-align:left;">
Alanine, aspartate and glutamate metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
N-acetylaspartylglutamate
</td>
<td style="text-align:left;">
HMDB0001067
</td>
<td style="text-align:left;">
C12270
</td>
<td style="text-align:left;">
N-Acetylaspartylglutamate
</td>
<td style="text-align:left;">
Alanine, aspartate and glutamate metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
tyrosine
</td>
<td style="text-align:left;">
HMDB0000158
</td>
<td style="text-align:left;">
C00082
</td>
<td style="text-align:left;">
L-Tyrosine
</td>
<td style="text-align:left;">
Amino acid metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
asparagine
</td>
<td style="text-align:left;">
HMDB0000168
</td>
<td style="text-align:left;">
C00152
</td>
<td style="text-align:left;">
L-Asparagine
</td>
<td style="text-align:left;">
Amino acid metabolism
</td>
</tr>
</tbody>
</table>
<p><br><code>3.</code> KEGG pathways that are loaded via KEGG API using the
package <code>KEGGREST</code> and can be used to perform pathway
analysis. <strong>(KEGG_Pathways)</strong><br></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># This will use KEGGREST to query the KEGG API to load the pathways:</span></span>
<span><span class="va">KEGG_Pathways</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/metsigdb_kegg.html">metsigdb_kegg</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>KEGG_Pathways</code>.
</caption>
<thead><tr>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
MetaboliteID
</th>
<th style="text-align:left;">
term
</th>
<th style="text-align:left;">
Metabolite
</th>
<th style="text-align:left;">
pubchem
</th>
<th style="text-align:left;">
compound_names
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
map00010
</td>
<td style="text-align:left;">
C00022
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis
</td>
<td style="text-align:left;">
Pyruvate
</td>
<td style="text-align:left;">
3324
</td>
<td style="text-align:left;">
Pyruvate….
</td>
</tr>
<tr>
<td style="text-align:left;">
map00010
</td>
<td style="text-align:left;">
C00024
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis
</td>
<td style="text-align:left;">
Acetyl-CoA
</td>
<td style="text-align:left;">
3326
</td>
<td style="text-align:left;">
Acetyl-C….
</td>
</tr>
<tr>
<td style="text-align:left;">
map00010
</td>
<td style="text-align:left;">
C00031
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis
</td>
<td style="text-align:left;">
D-Glucose
</td>
<td style="text-align:left;">
3333
</td>
<td style="text-align:left;">
D-Glucos….
</td>
</tr>
<tr>
<td style="text-align:left;">
map00010
</td>
<td style="text-align:left;">
C00033
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis
</td>
<td style="text-align:left;">
Acetate
</td>
<td style="text-align:left;">
3335
</td>
<td style="text-align:left;">
Acetate,….
</td>
</tr>
<tr>
<td style="text-align:left;">
map00010
</td>
<td style="text-align:left;">
C00036
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis
</td>
<td style="text-align:left;">
Oxaloacetate
</td>
<td style="text-align:left;">
3338
</td>
<td style="text-align:left;">
Oxaloace….
</td>
</tr>
</tbody>
</table>
<br><br><div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 550%">

</div>
</div>
</div>
<div class="section level2">
<h2 id="run-metaproviz-analysis">2. Run MetaProViz Analysis<a class="anchor" aria-label="anchor" href="#run-metaproviz-analysis"></a>
</h2>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 550%">

</div>
</div>
<p>Currently, <strong>MetaProViz</strong> contains four different
modules, which include different methods and can be used independently
from each other or in combination (see introduction for more details).
Here we will go trough each of those modules and apply them to the
example data.</p>
<div class="section level3">
<h3 id="pre-processing">Pre-processing<a class="anchor" aria-label="anchor" href="#pre-processing"></a>
</h3>
<strong>MetaProViz</strong> includes a pre-processing module with the
function <code>Preprocessing()</code> that has multiple parameters to
perform customize data processing.<br><code>Feature_Filtering</code> applies the 80%-filtering rule on the
metabolite features either on the whole dataset (=“Standard”) <span class="citation">(Bijlsma et al. 2006)</span> or per condition
(=“Modified”) <span class="citation">(Wei et al. 2018)</span>. This
means that metabolites are removed were more than 20% of the samples
(all or per condition) have no detection. In case of the core
experiment, the blank samples are ignored during feature filtering,
since often metabolites are released from a cell and not naturally
present in the culture media leading to no detection in the blank. With
the parameter <code>Feature_Filt_Value</code> we enable the adaptation
of the stringency of the filtering based on the experimental context.
For instance, patient tumour samples can contain many unknown subgroups
due to gender, age, stage etc., which leads to a metabolite being
detected in only 50% (or even less) of the tumour samples, hence in this
context it could be considered to change the
<code>Feature_Filt_Value</code> from the default (=0.8). If
<code>Feature_Filtering = "None"</code>, no feature filtering is
performed. In the context of <code>Feature_Filtering</code> it is also
noteworthy that the function <code>Pool_Estimation()</code> can be used
to estimate the quality of the metabolite detection and will return a
list of metabolites that are variable across the different pool
measurements (pool = mixture of all experimental samples measured
several times during the LC-MS run) . Variable metabolite in the pool
sample should be removed from the data.<br>
The parameter <code>tic_Normalization</code> refers to total Ion Count
(tic) normalisation, which is often used with LC-MS derived metabolomics
data. If <code>tic_Normalization = TRUE</code>, each feature
(=metabolite) in a sample is divided by the sum of all intensity value
(= total number of ions) for the sample and finally multiplied by a
constant ( = the mean of all samples total number of ions). Noteworthy,
tic normalisation should not be used with small number of features (=
metabolites), since tic assumes that on “average” the ion count of each
sample is equal if there were no instrument batch effects <span class="citation">(Wulff and Mitchell 2018)</span>.<br>
The parameter <code>mvi</code> refers to Missing Value Imputation (mvi)
and if <code>mvi = TRUE</code> half minimum (HM) missing value
imputation is performed per feature (= per metabolite). Here it is
important to mention that HM has been shown to perform well for missing
vales that are missing not at random (MNAR) <span class="citation">(Wei
et al. 2018)</span>.<br>
Lastly, the function <code>Preprocessing()</code> performs outlier
detection and adds a column “Outliers” into the DF, which can be used to
remove outliers. The parameter <code>hotellins_confidence</code> can be
used to choose the confidence interval that should be used for the
Hotellins T2 outlier test <span class="citation">(Hotelling
1931)</span>.<br><br>
Since our example data contains pool samples, we will do
<code>Pool_Estimation()</code> before applying the
<code>Preprocessing()</code> function. This is important, since one
should remove the features (=metabolites) that are too variable prior to
performing any data transformations such as tic as part of the
<code>Preprocessing()</code> function.<br>
It is worth mentioning that the Coefficient of variation (CV) is
calculated by dividing the standard deviation (SD) by the mean. Hence CV
depends on the SD, which in turn works for normally distributed
data.<br><div class="vscroll-plot">
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Pool_Estimation_result</span><span class="op">&lt;-</span> <span class="fu"><a href="../../reference/pool_estimation.html">pool_estimation</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">Media</span><span class="op">[</span>,<span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span><span class="op">]</span>,</span>
<span>                                                    metadata_sample <span class="op">=</span> <span class="va">Media</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span>,</span>
<span>                                                    metadata_info <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>PoolSamples <span class="op">=</span> <span class="st">"Pool"</span>, Conditions<span class="op">=</span><span class="st">"Conditions"</span><span class="op">)</span>,</span>
<span>                                                    cutoff_cv <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></span>
<span><span class="co">#&gt; `stat_bin()` using `bins = 30`. Pick better value `binwidth`.</span></span>
<span><span class="co">#&gt; Bin width defaults to 1/30 of the range of the data. Pick better value with</span></span>
<span><span class="co">#&gt; `binwidth`.</span></span></code></pre></div>
<p><img src="core-metabolomics_files/figure-html/pool-estimation-1.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/pool-estimation-2.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/pool-estimation-3.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">Pool_Estimation_result_DF_CV</span> <span class="op">&lt;-</span><span class="va">Pool_Estimation_result</span><span class="op">[[</span><span class="st">"DF"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"CV"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
</div>
<br><br><br><table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the Pool_Estimation result.
</caption>
<thead><tr>
<th style="text-align:left;">
Metabolite
</th>
<th style="text-align:right;">
CV
</th>
<th style="text-align:left;">
HighVar
</th>
<th style="text-align:right;">
MissingValuepercentage
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
valine-d8
</td>
<td style="text-align:right;">
1.744198
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
hipppuric acid-d5
</td>
<td style="text-align:right;">
1.253983
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
2-hydroxyglutarate
</td>
<td style="text-align:right;">
7.199141
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
2-ketoglutarate
</td>
<td style="text-align:right;">
2.766989
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
3-Dehydro-L-threonate
</td>
<td style="text-align:right;">
9.222780
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
0
</td>
</tr>
</tbody>
</table>
<br>
The results from the <code>Pool_Estimation()</code> is a table that has
the CVs. If there is a high variability, one should consider to remove
those features from the data. For the example data nothing needs to be
removed. If you have used internal standard in your experiment you
should specifically check their CV as this would indicate technical
issues (here valine-d8 and hippuric acid-d5).<br><br>
Now we will apply the <code>Preprocessing()</code> function to the
example data and have a look at the output produced. You will notice
that all the chosen parameters and results are documented in messages.
All the results data tables, the Quality Control (QC) plots and outlier
detection plots are returned and can be easily viewed. Importantly, here
we are able to specify that we have a core experiment setting the
parameter <code>core=TRUE</code>, in which case a few additional data
processing steps are applied:<br><code>1.</code> <strong>Blank sample</strong>: This refers to media
samples where no cells have been cultured in, which will be used as
blank. In detail, the mean of the blank sample of a feature (=
metabolite) will be substracted from the values measured in each sample
for the same feature. In the column “Condition” of the
Experimental_design DF, you will need to label your blank samples with
“blank”.<br><code>2.</code> <strong>Growth factor</strong> or <strong>growth
rate</strong>: This refers to the different conditions and is either
based on cell count or protein quantification at the start of the
experiment (t0) and at the end of the experiment (t1) resulting in the
growth factor (t0/t1). Otherwise, one can experimentally estimate the
growth rate of each condition. Ultimately, this measure is used to
normalize the data, since the amount of growth will impact the
consumption and release of metabolites from the media and hence we need
to account for this. If you do not have this information, this will be
set to 1, yet be aware that this may affect the results.<br><br>
You can pass these additional information via the parameter
<code>Input_metadata_info</code>, by passing the column name for the
<code>core_norm_factor</code> in the <code>Input_SettingsFile</code> and
the condition name for the <code>core_media</code> in the
<code>Input_data</code> file.<br><div class="vscroll-plot">
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Prepare the input:</span></span>
<span><span class="va">Media_input</span> <span class="op">&lt;-</span> <span class="va">Media</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="op">!</span><span class="va">Conditions</span><span class="op">==</span><span class="st">"Pool"</span>, select <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span><span class="co">#remove pool samples and remove the information columns</span></span>
<span></span>
<span><span class="va">Media_Metadata</span> <span class="op">&lt;-</span> <span class="va">Media</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="op">!</span><span class="va">Conditions</span><span class="op">==</span><span class="st">"Pool"</span>, select <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span><span class="co">#remove pool samples and keep the information columns only</span></span>
<span></span>
<span><span class="va">PreProcessing_res</span> <span class="op">&lt;-</span>  <span class="fu"><a href="../../reference/processing.html">processing</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">Media_input</span>,</span>
<span>                                                metadata_sample <span class="op">=</span><span class="va">Media_Metadata</span>,</span>
<span>                                                metadata_info <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>Conditions <span class="op">=</span> <span class="st">"Conditions"</span>,</span>
<span>                                                Biological_Replicates <span class="op">=</span> <span class="st">"Biological_Replicates"</span>,</span>
<span>                                                core_norm_factor <span class="op">=</span> <span class="st">"GrowthFactor"</span>,</span>
<span>                                                core_media <span class="op">=</span> <span class="st">"blank"</span><span class="op">)</span>,</span>
<span>                                                featurefilt <span class="op">=</span> <span class="st">"Modified"</span>,</span>
<span>                                                cutoff_featurefilt <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>                                                tic <span class="op">=</span> <span class="cn">TRUE</span>,<span class="co"># As we have raw data we will perform total ion count norm</span></span>
<span>                                                mvi<span class="op">=</span><span class="cn">TRUE</span>, <span class="co">#We assume the values are not missing at random and perform half minimum mvi</span></span>
<span>                                                mvi_percentage<span class="op">=</span><span class="fl">50</span>,</span>
<span>                                                hotellins_confidence <span class="op">=</span> <span class="fl">0.99</span>,<span class="co"># We perform outlier testing using 0.99 confidence interval</span></span>
<span>                                                core <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; For Consumption Release experiment we are using the method from Jain M.  REF: Jain et. al, (2012), Science 336(6084):1040-4, doi: 10.1126/science.1218595.</span></span>
<span><span class="co">#&gt; feature_filtering: Here we apply the modified 80%-filtering rule that takes the class information (Column `Conditions`) into account, which additionally reduces the effect of missing values (REF: Yang et. al., (2015), doi: 10.3389/fmolb.2015.00004). Filtering value selected: 0.8</span></span>
<span><span class="co">#&gt; 3 metabolites where removed: N-acetylaspartylglutamate, hypotaurine, S-(2-succinyl)cysteine</span></span>
<span><span class="co">#&gt; Missing Value Imputation: Missing value imputation is performed, as a complementary approach to address the missing value problem, where the missing values are imputing using the `half minimum value`. REF: Wei et. al., (2018), Reports, 8, 663, doi:https://doi.org/10.1038/s41598-017-19120-0</span></span>
<span><span class="co">#&gt; NA values were found in Control_media samples for metabolites. For metabolites including NAs mvi is performed unless all samples of a metabolite are NA.</span></span>
<span><span class="co">#&gt; Metabolites with high NA load (&gt;20%) in Control_media samples are: dihydroorotate.</span></span>
<span><span class="co">#&gt; Metabolites with only NAs (= 100%) in Control_media samples are: hydroxyphenylpyruvate. Those NAs are set zero as we consider them true zeros</span></span>
<span><span class="co">#&gt; total Ion Count (tic) normalization: total Ion Count (tic) normalization is used to reduce the variation from non-biological sources, while maintaining the biological variation. REF: Wulff et. al., (2018), Advances in Bioscience and Biotechnology, 9, 339-351, doi:https://doi.org/10.4236/abb.2018.98022</span></span>
<span><span class="co">#&gt; Warning: Removed 5 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_boxplot()`).</span></span>
<span><span class="co">#&gt; Removed 5 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_boxplot()`).</span></span>
<span><span class="co">#&gt; 8 of variables have high variability (CV &gt; 30) in the core_media control samples. Consider checking the pooled samples to decide whether to remove these metabolites or not.</span></span>
<span><span class="co">#&gt; `stat_bin()` using `bins = 30`. Pick better value `binwidth`.</span></span>
<span><span class="co">#&gt; Warning: The following aesthetics were dropped during statistical transformation: <span style="color: #00BB00;">label</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> This can happen when ggplot fails to infer the correct grouping structure in</span></span>
<span><span class="co">#&gt;   the data.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Did you forget to specify a `group` aesthetic or to convert a numerical</span></span>
<span><span class="co">#&gt;   variable into a factor?</span></span>
<span><span class="co">#&gt; Bin width defaults to 1/30 of the range of the data. Pick better value with</span></span>
<span><span class="co">#&gt; `binwidth`.</span></span>
<span><span class="co">#&gt; Warning: The following aesthetics were dropped during statistical transformation: <span style="color: #00BB00;">label</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> This can happen when ggplot fails to infer the correct grouping structure in</span></span>
<span><span class="co">#&gt;   the data.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Did you forget to specify a `group` aesthetic or to convert a numerical</span></span>
<span><span class="co">#&gt;   variable into a factor?</span></span>
<span><span class="co">#&gt; Warning in core_norm(data = ticRes, metadata_sample = metadata_sample,</span></span>
<span><span class="co">#&gt; metadata_info = metadata_info): The core_media samples MS51-06 were found to be</span></span>
<span><span class="co">#&gt; different from the rest. They will not be included in the sum of the core_media</span></span>
<span><span class="co">#&gt; samples.</span></span>
<span><span class="co">#&gt; core data are normalised by substracting mean (blank) from each sample and multiplying with the core_norm_factor</span></span>
<span><span class="co">#&gt; Outlier detection: Identification of outlier samples is performed using Hotellin's T2 test to define sample outliers in a mathematical way (Confidence = 0.99 ~ p.val &lt; 0.01) (REF: Hotelling, H. (1931), Annals of Mathematical Statistics. 2 (3), 360-378, doi:https://doi.org/10.1214/aoms/1177732979). hotellins_confidence value selected: 0.99</span></span>
<span><span class="co">#&gt; There are possible outlier samples in the data</span></span>
<span><span class="co">#&gt; Filtering round  1  Outlier Samples:  MS51-06  </span></span>
<span><span class="co">#&gt; Filtering round  2  Outlier Samples:  MS51-09</span></span>
<span><span class="co">#&gt; Warning: Removed 5 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_boxplot()`).</span></span></code></pre></div>
<p><img src="core-metabolomics_files/figure-html/code-5-1.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Warning: Removed 5 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_boxplot()`).</span></span>
<span><span class="co">#&gt; Warning: Removed 5 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_boxplot()`).</span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/code-5-2.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Warning: Removed 5 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_boxplot()`).</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 9 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/code-5-3.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Warning: ggrepel: 9 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span>
<span><span class="co">#&gt; Warning in geom_bar(stat = "identity", fill = barfill, color = barcolor, :</span></span>
<span><span class="co">#&gt; Ignoring empty aesthetic: `width`.</span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/code-5-4.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Warning in geom_bar(stat = "identity", fill = barfill, color = barcolor, :</span></span>
<span><span class="co">#&gt; Ignoring empty aesthetic: `width`.</span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/code-5-5.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Warning: ggrepel: 3 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/code-5-6.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Warning: ggrepel: 3 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></span>
<span><span class="co">#&gt; Ignoring empty aesthetic: `width`.</span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/code-5-7.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Warning in geom_bar(stat = "identity", fill = barfill, color = barcolor, :</span></span>
<span><span class="co">#&gt; Ignoring empty aesthetic: `width`.</span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/code-5-8.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Warning: ggrepel: 3 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/code-5-9.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Warning: ggrepel: 3 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></span>
<span><span class="co">#&gt; Ignoring empty aesthetic: `width`.</span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/code-5-10.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Warning in geom_bar(stat = "identity", fill = barfill, color = barcolor, :</span></span>
<span><span class="co">#&gt; Ignoring empty aesthetic: `width`.</span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/code-5-11.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Warning: ggrepel: 8 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/code-5-12.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Warning: ggrepel: 8 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 8 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/code-5-13.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Warning: ggrepel: 8 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/code-5-14.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/code-5-15.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/code-5-16.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Now we can have a look at the results table:</span></span>
<span><span class="va">Media_Preprocessed</span> <span class="op">&lt;-</span>  <span class="va">PreProcessing_res</span><span class="op">[[</span><span class="st">"DF"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"Preprocessing_output"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
</div>
<br><br><br><table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the pre-processing results, which has an additional column
<code>Outlier</code> including the results of Hotellins T2.
</caption>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Conditions
</th>
<th style="text-align:right;">
Biological_Replicates
</th>
<th style="text-align:right;">
GrowthFactor
</th>
<th style="text-align:left;">
Outliers
</th>
<th style="text-align:right;">
valine-d8
</th>
<th style="text-align:right;">
hipppuric acid-d5
</th>
<th style="text-align:right;">
2-hydroxyglutarate
</th>
<th style="text-align:right;">
2-ketoglutarate
</th>
<th style="text-align:right;">
3-Dehydro-L-threonate
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
MS51-06
</td>
<td style="text-align:left;">
HK2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
249.2817
</td>
<td style="text-align:left;">
Outlier_filtering_round_1
</td>
<td style="text-align:right;">
6622682287
</td>
<td style="text-align:right;">
26658050746
</td>
<td style="text-align:right;">
5853665424
</td>
<td style="text-align:right;">
7527471565
</td>
<td style="text-align:right;">
318856944286
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-07
</td>
<td style="text-align:left;">
HK2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
249.2817
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
5049281744
</td>
<td style="text-align:right;">
30218506574
</td>
<td style="text-align:right;">
-868601575
</td>
<td style="text-align:right;">
1530912430
</td>
<td style="text-align:right;">
28673393701
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-08
</td>
<td style="text-align:left;">
HK2
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
249.2817
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
-1441406385
</td>
<td style="text-align:right;">
-11561761745
</td>
<td style="text-align:right;">
1108346025
</td>
<td style="text-align:right;">
3684365990
</td>
<td style="text-align:right;">
39313101752
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-09
</td>
<td style="text-align:left;">
HK2
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
249.2817
</td>
<td style="text-align:left;">
Outlier_filtering_round_2
</td>
<td style="text-align:right;">
2563904070
</td>
<td style="text-align:right;">
-10237061776
</td>
<td style="text-align:right;">
-189633550
</td>
<td style="text-align:right;">
-9097094735
</td>
<td style="text-align:right;">
-67782920598
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-10
</td>
<td style="text-align:left;">
HK2
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
249.2817
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
-6171779428
</td>
<td style="text-align:right;">
-8419683053
</td>
<td style="text-align:right;">
-50110899
</td>
<td style="text-align:right;">
3881816314
</td>
<td style="text-align:right;">
-203574855
</td>
</tr>
</tbody>
</table>
<p><br>
In the output table you can now see the column “Outliers” and for the
Condition HK2 CCM, we can see that based on Hotellin’s T2 test, samples
were detected as outliers in the first and second round of
filtering.<br>
As part of the <code>Preprocessing()</code> function several plots are
generated and saved. Additionally, the ggplots are returned into the
list to enable further modifiaction using the ggplot syntax. These plots
include plots showing the outliers for each filtering round and other QC
plots.<br><br>
As part of the <strong>MetaProViz</strong> visualization module one can
easily further customize the PCA plot and adapt color and shape for the
information of interest. You can see more below for the
<code><a href="../../reference/viz_pca.html">viz_pca()</a></code> function.<br>
Before we proceed, we will remove the outlier:<br></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Media_Preprocessed</span> <span class="op">&lt;-</span><span class="va">Media_Preprocessed</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="op">!</span><span class="va">Outliers</span><span class="op">==</span><span class="st">"Outlier_filtering_round_1"</span><span class="op">)</span></span></code></pre></div>
<p><br>
In metabolomics, sometimes samples are injected (=measured) several
times, which can be termed as analytical replicates. The
<strong>MetaProViz</strong> pre-processing module includes the function
<code><a href="../../reference/replicate_sum.html">replicate_sum()</a></code>, which will summarize those and save the
results.</p>
</div>
<div class="section level3">
<h3 id="dma">dma<a class="anchor" aria-label="anchor" href="#dma"></a>
</h3>
Differential Metabolite Analysis (<code>dma</code>) between two
conditions (e.g. Tumour versus Healthy) usually calculates the Log2FC,
p-value, adjusted p-value and t-value. Yet, in a core experiment the
normalized metabolite values can be either a negative value, if the
metabolite has been consumed from the media, or a positive value, if the
metabolite has been released from the cell into the culture media. Since
we can not calculate a Log2FC using negative values, we calculate the
absolute difference between the mean of Condition 1 versus the mean of
Condition 2. The absolute difference is log2 transformed in order to
make the values comparable between the different metabolites, resulting
in the Log2Dist. The result doesn’t consider whether one product is
larger than the other; it only looks at the magnitude of their
difference. to reflect the direction of change between the two
conditions we multiply with -1 if C1 &lt; C2. By setting the paramteter
<code>core</code> = TRUE, instead of calclulating the Log2FC, the Log2
Distance is calculated.<br>
With the different parameters <code>STAT_pval</code> and
<code>STAT_padj</code> one can choose the statistical tests such as
t.test, wilcoxon test, limma, annova, kruskal walles, etc. (see function
reference for more information).<br>
As input one can use the pre-processed data we have generated using the
<code>Preprocessing</code> module, but here one can of course use any DF
including metabolite values, even though we recommend to normalize the
data and remove outliers prior to dma. Moreover, we require the
<code>Input_metadata_sample</code> including the sample metadata with
information which condition a sample corresponds to. Additionally, we
enable the user to provide a <code>Plot_metadata_feature</code>
containing the metadata for the features (metabolites), such as KEGG ID,
pathway, retention time, etc.<br><br>
By defining the numerator and denominator as part of the
<code>Input_metadata_info</code> parameter, it is defined which
comparisons are performed:<br>
1. <strong>one_vs_one</strong> (single comparison):
numerator=“Condition1”, denominator =“Condition2”<br>
2. <strong>all_vs_one</strong> (multiple comparison): numerator=NULL,
denominator =“Condition”<br>
3. <strong>all_vs_all</strong> (multiple comparison): numerator=NULL,
denominator =NULL (=default)<br><br>
As input we will use the pre-processed data we have generated using the
<code>Preprocessing</code> module, but here one can of course use any DF
including metabolite values and information about the conditions that
should be compared (even though we recommend to normalize the data and
remove outliers prior to dma).<br><br>
In the example data we have seven different cell lines, healthy (HK2)
and cancer (ccRCC: 786-M1A, 786-M2A, 786-O, OSRC2, OSLM1B and RFX631)
and hence we can perform multiple different comparisons. The results can
be automatically saved and all the results are returned in a list with
the different data frames. If parameter Plot=TRUE, an overview Volcano
plot is generated and saved.<br><div class="vscroll-plot">
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Perform multiple comparison All_vs_One using annova:</span></span>
<span><span class="va">DMA_Annova</span> <span class="op">&lt;-</span>  <span class="fu"><a href="../../reference/dma.html">dma</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">Media_Preprocessed</span><span class="op">[</span>,<span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">)</span><span class="op">]</span>,</span>
<span>metadata_sample<span class="op">=</span><span class="va">Media_Preprocessed</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">]</span>,</span>
<span>metadata_info <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>Conditions<span class="op">=</span><span class="st">"Conditions"</span>, Numerator<span class="op">=</span><span class="cn">NULL</span>, Denominator <span class="op">=</span> <span class="st">"HK2"</span><span class="op">)</span>,</span>
<span>pval <span class="op">=</span><span class="st">"aov"</span>,</span>
<span>padj<span class="op">=</span><span class="st">"fdr"</span>,</span>
<span>metadata_feature <span class="op">=</span> <span class="va">MappingInfo</span>,</span>
<span>core<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; There are no NA/0 values</span></span>
<span><span class="co">#&gt; For 67.65% of metabolites the group variances are equal.</span></span>
<span><span class="co">#&gt; We added +1 to the mean value of metabolite(s) , since the mean of the replicate values where 0. This was not due to missing values (NA/0).</span></span>
<span><span class="co">#&gt; We added +1 to the mean value of metabolite(s) , since the mean of the replicate values where 0. This was not due to missing values (NA/0).</span></span>
<span><span class="co">#&gt; We added +1 to the mean value of metabolite(s) , since the mean of the replicate values where 0. This was not due to missing values (NA/0).</span></span>
<span><span class="co">#&gt; We added +1 to the mean value of metabolite(s) , since the mean of the replicate values where 0. This was not due to missing values (NA/0).</span></span>
<span><span class="co">#&gt; We added +1 to the mean value of metabolite(s) , since the mean of the replicate values where 0. This was not due to missing values (NA/0).</span></span>
<span><span class="co">#&gt; We added +1 to the mean value of metabolite(s) , since the mean of the replicate values where 0. This was not due to missing values (NA/0).</span></span></code></pre></div>
<p><img src="core-metabolomics_files/figure-html/dma-1.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/dma-2.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/dma-3.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/dma-4.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/dma-5.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/dma-6.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; `stat_bin()` using `bins = 30`. Pick better value `binwidth`.</span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/dma-7.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; `stat_bin()` using `bins = 30`. Pick better value `binwidth`.</span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/dma-8.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/dma-9.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/dma-10.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/dma-11.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/dma-12.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/dma-13.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/dma-14.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Inspect the dma results tables:</span></span>
<span><span class="va">DMA_HK2_vs_786M1A</span> <span class="op">&lt;-</span> <span class="va">DMA_Annova</span><span class="op">[[</span><span class="st">"dma"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"HK2_vs_786-M1A"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">DMA_HK2_vs_786O</span> <span class="op">&lt;-</span> <span class="va">DMA_Annova</span><span class="op">[[</span><span class="st">"dma"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"HK2_vs_786-O"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">shapiro</span> <span class="op">&lt;-</span> <span class="va">DMA_Annova</span><span class="op">[[</span><span class="st">"ShapiroTest"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"DF"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"Shapiro_result"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
</div>
<br><br><br><table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<ol style="list-style-type: decimal"><li>Preview of the Shaprio results for the different conditions.
</li></ol>
</caption>
<thead><tr>
<th style="text-align:left;">
Code
</th>
<th style="text-align:left;">
Metabolites with normal distribution [%]
</th>
<th style="text-align:left;">
Metabolites with not-normal distribution [%]
</th>
<th style="text-align:right;">
shapiro p.val(2-hydroxyglutarate)
</th>
<th style="text-align:right;">
shapiro p.val(2-ketoglutarate)
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
HK2
</td>
<td style="text-align:left;">
82.35
</td>
<td style="text-align:left;">
17.65
</td>
<td style="text-align:right;">
0.6833007
</td>
<td style="text-align:right;">
0.0446492
</td>
</tr>
<tr>
<td style="text-align:left;">
786-O
</td>
<td style="text-align:left;">
95.71
</td>
<td style="text-align:left;">
4.29
</td>
<td style="text-align:right;">
0.4938675
</td>
<td style="text-align:right;">
0.3823712
</td>
</tr>
<tr>
<td style="text-align:left;">
786-M1A
</td>
<td style="text-align:left;">
97.14
</td>
<td style="text-align:left;">
2.86
</td>
<td style="text-align:right;">
0.9979050
</td>
<td style="text-align:right;">
0.1384610
</td>
</tr>
<tr>
<td style="text-align:left;">
786-M2A
</td>
<td style="text-align:left;">
88.57
</td>
<td style="text-align:left;">
11.43
</td>
<td style="text-align:right;">
0.5546558
</td>
<td style="text-align:right;">
0.5369470
</td>
</tr>
<tr>
<td style="text-align:left;">
OSRC2
</td>
<td style="text-align:left;">
92.86
</td>
<td style="text-align:left;">
7.14
</td>
<td style="text-align:right;">
0.8899005
</td>
<td style="text-align:right;">
0.2007242
</td>
</tr>
<tr>
<td style="text-align:left;">
OSLM1B
</td>
<td style="text-align:left;">
85.71
</td>
<td style="text-align:left;">
14.29
</td>
<td style="text-align:right;">
0.4643014
</td>
<td style="text-align:right;">
0.9022803
</td>
</tr>
<tr>
<td style="text-align:left;">
RFX631
</td>
<td style="text-align:left;">
97.14
</td>
<td style="text-align:left;">
2.86
</td>
<td style="text-align:right;">
0.9292099
</td>
<td style="text-align:right;">
0.0247568
</td>
</tr>
</tbody>
</table>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<ol start="2" style="list-style-type: decimal"><li>Preview of the dma results for the comparison of 786-M1A versus HK2
cells.
</li></ol>
</caption>
<thead><tr>
<th style="text-align:left;">
Metabolite
</th>
<th style="text-align:right;">
Log2(Distance)
</th>
<th style="text-align:right;">
p.adj
</th>
<th style="text-align:right;">
t.val
</th>
<th style="text-align:left;">
Mean_786-M1A
</th>
<th style="text-align:left;">
core_786-M1A
</th>
<th style="text-align:left;">
Mean_HK2
</th>
<th style="text-align:left;">
core_HK2
</th>
<th style="text-align:left;">
core_specific
</th>
<th style="text-align:left;">
core
</th>
<th style="text-align:right;">
MS51-16
</th>
<th style="text-align:right;">
MS51-17
</th>
<th style="text-align:right;">
MS51-18
</th>
<th style="text-align:right;">
MS51-19
</th>
<th style="text-align:right;">
MS51-20
</th>
<th style="text-align:right;">
MS51-07
</th>
<th style="text-align:right;">
MS51-08
</th>
<th style="text-align:right;">
MS51-09
</th>
<th style="text-align:right;">
MS51-10
</th>
<th style="text-align:left;">
HMDB
</th>
<th style="text-align:left;">
KEGG.ID
</th>
<th style="text-align:left;">
KEGGCompound
</th>
<th style="text-align:left;">
Pathway
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
aconitate
</td>
<td style="text-align:right;">
32.58145
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
6426780519
</td>
<td style="text-align:left;">
-6426780518.51619
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
-9.31322574615479e-07
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:right;">
-5865088200
</td>
<td style="text-align:right;">
-6431671420
</td>
<td style="text-align:right;">
-6503096254
</td>
<td style="text-align:right;">
-6848155621
</td>
<td style="text-align:right;">
-6485891098
</td>
<td style="text-align:right;">
1186119568
</td>
<td style="text-align:right;">
-610503914
</td>
<td style="text-align:right;">
-406215498
</td>
<td style="text-align:right;">
-169400156
</td>
<td style="text-align:left;">
HMDB0000072
</td>
<td style="text-align:left;">
C00417
</td>
<td style="text-align:left;">
cis-Aconitate
</td>
<td style="text-align:left;">
Citrate cycle (TCA cycle)
</td>
</tr>
<tr>
<td style="text-align:left;">
arginine
</td>
<td style="text-align:right;">
-36.63197
</td>
<td style="text-align:right;">
0.9411031
</td>
<td style="text-align:right;">
-106493120308
</td>
<td style="text-align:left;">
106493120307.837
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
5.7220458984375e-05
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:right;">
216146922501
</td>
<td style="text-align:right;">
123584771692
</td>
<td style="text-align:right;">
-184582255060
</td>
<td style="text-align:right;">
137947177034
</td>
<td style="text-align:right;">
239368985373
</td>
<td style="text-align:right;">
124885985748
</td>
<td style="text-align:right;">
174141953768
</td>
<td style="text-align:right;">
-154239029796
</td>
<td style="text-align:right;">
-144788909720
</td>
<td style="text-align:left;">
HMDB0000517
</td>
<td style="text-align:left;">
C00062
</td>
<td style="text-align:left;">
L-Arginine
</td>
<td style="text-align:left;">
Amino acid metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
aspartate
</td>
<td style="text-align:right;">
34.35422
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
21960976607
</td>
<td style="text-align:left;">
-21960976606.516
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
-1.72853469848633e-06
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:right;">
-20078854995
</td>
<td style="text-align:right;">
-23024873124
</td>
<td style="text-align:right;">
-23695204780
</td>
<td style="text-align:right;">
-22469858974
</td>
<td style="text-align:right;">
-20536091160
</td>
<td style="text-align:right;">
1216173971
</td>
<td style="text-align:right;">
2386511960
</td>
<td style="text-align:right;">
-2796694643
</td>
<td style="text-align:right;">
-805991288
</td>
<td style="text-align:left;">
HMDB0000191
</td>
<td style="text-align:left;">
C00049
</td>
<td style="text-align:left;">
L-Aspartate
</td>
<td style="text-align:left;">
Amino acid metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
betaine
</td>
<td style="text-align:right;">
-39.09360
</td>
<td style="text-align:right;">
0.0017858
</td>
<td style="text-align:right;">
-586606639559
</td>
<td style="text-align:left;">
586606639558.529
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
1.9073486328125e-06
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:right;">
688358341457
</td>
<td style="text-align:right;">
490864850765
</td>
<td style="text-align:right;">
657311755851
</td>
<td style="text-align:right;">
724283773821
</td>
<td style="text-align:right;">
372214475898
</td>
<td style="text-align:right;">
222640096432
</td>
<td style="text-align:right;">
-24221141664
</td>
<td style="text-align:right;">
-466988404785
</td>
<td style="text-align:right;">
268569450018
</td>
<td style="text-align:left;">
HMDB0000043
</td>
<td style="text-align:left;">
C00719
</td>
<td style="text-align:left;">
Betaine
</td>
<td style="text-align:left;">
Not assigned
</td>
</tr>
<tr>
<td style="text-align:left;">
carbamoyl phosphate
</td>
<td style="text-align:right;">
-29.63461
</td>
<td style="text-align:right;">
0.5228459
</td>
<td style="text-align:right;">
-833502731
</td>
<td style="text-align:left;">
833502730.89944
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
9.23871994018555e-07
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:right;">
544741981
</td>
<td style="text-align:right;">
632898115
</td>
<td style="text-align:right;">
1989818101
</td>
<td style="text-align:right;">
816820279
</td>
<td style="text-align:right;">
183235178
</td>
<td style="text-align:right;">
200618839
</td>
<td style="text-align:right;">
-354809141
</td>
<td style="text-align:right;">
549923002
</td>
<td style="text-align:right;">
-395732700
</td>
<td style="text-align:left;">
HMDB0001096
</td>
<td style="text-align:left;">
C00169
</td>
<td style="text-align:left;">
Carbamoyl phosphate
</td>
<td style="text-align:left;">
Purine metabolism
</td>
</tr>
</tbody>
</table>
<p><br>
Using the dma results, we can now use the <strong>MetaProViz</strong>
visualization module and generate further customized Volcano plots
<code><a href="../../reference/viz_volcano.html">viz_volcano()</a></code>. You can see some examples below.<br><br>
Additionally to the individual comparison that there is also a summary
table created including the individual information about metabolite
consumption or release based on the mean measured value:<br></p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">core_MetaInfo</span> <span class="op">&lt;-</span> <span class="va">DMA_Annova</span><span class="op">[[</span><span class="st">"Feature_Metadata"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<ol style="list-style-type: decimal"><li>Preview of the consumption-release information for each metabolite
and cell line.
</li></ol>
</caption>
<thead><tr>
<th style="text-align:left;">
Metabolite
</th>
<th style="text-align:left;">
HMDB
</th>
<th style="text-align:left;">
core_786-M1A
</th>
<th style="text-align:left;">
core_HK2
</th>
<th style="text-align:left;">
core_786-M2A
</th>
<th style="text-align:left;">
core_786-O
</th>
<th style="text-align:left;">
core_OSLM1B
</th>
<th style="text-align:left;">
core_OSRC2
</th>
<th style="text-align:left;">
core_RFX631
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
2-hydroxyglutarate
</td>
<td style="text-align:left;">
HMDB0059655
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Released
</td>
</tr>
<tr>
<td style="text-align:left;">
acetylcarnitine
</td>
<td style="text-align:left;">
HMDB0000201
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Released
</td>
</tr>
<tr>
<td style="text-align:left;">
acetylcholine
</td>
<td style="text-align:left;">
HMDB0000895
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Consumed
</td>
</tr>
<tr>
<td style="text-align:left;">
acetylornithine
</td>
<td style="text-align:left;">
HMDB0003357
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Released
</td>
</tr>
<tr>
<td style="text-align:left;">
aconitate
</td>
<td style="text-align:left;">
HMDB0000072
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Consumed
</td>
</tr>
</tbody>
</table>
<p><br>
We can also visualize this information by assigning -1 to released, +1
to consumed and 0 for NA:
<img src="core-metabolomics_files/figure-html/plot-1.png" class="r-plt" width="700"><img src="core-metabolomics_files/figure-html/plot-2.png" class="r-plt" width="700"></p>
<div class="section level4">
<h4 id="ora-using-the-dma-results">ORA using the dma results<a class="anchor" aria-label="anchor" href="#ora-using-the-dma-results"></a>
</h4>
<p>Over Representation Analysis (ORA) is a pathway enrichment analysis
(PEA) method that determines if a set of features (=metabolic pathways)
are over-represented in the selection of features (=metabolites) from
the data in comparison to all measured features (metabolites) using the
Fishers exact test. The selection of metabolites are usually the most
altered metabolites in the data, which can be selected by the top and
bottom t-values. Given that for core data it is important to consider
weather a metabolite was consumed or released, it is sensible to perform
ORA on each metabolite cluster.<br>
Of course, there are many other PEA methods such as the well known GSEA.
Here we do not aim to provide an extensive tool for different methods to
perform pathway enrichment analysis and only focus on ORA since we can
apply this to perform standard pathway enrichment as well as pathway
enrichment on clusters of metabolites. If you are interested in using
different pathway enrichment methods please check out specialized tools
such as <a href="https://saezlab.github.io/decoupleR/" class="external-link">decopupleR</a>
<span class="citation">(Badia-I-Mompel et al. 2022)</span>.<br><br>
Here we will use the KEGG pathways <span class="citation">(Kanehisa and
Goto 2000)</span>. Before we can perform ORA on the dma results, we have
to ensure that the metabolite names match with the KEGG IDs or KEGG
trivial names. In general, the <code>input_pathway</code> requirements
are column “term”, “Metabolite” and “Description”, and the
<code>Input_data</code> requirements are column “t.val” and column
“Metabolite”.<br></p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Since we have performed multiple comparisons (all_vs_HK2), we will run ORA for each of this comparison</span></span>
<span><span class="va">DM_ORA_res</span><span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">comparisons</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">DMA_Annova</span><span class="op">[[</span><span class="st">"dma"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">comparison</span> <span class="kw">in</span> <span class="va">comparisons</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="co">#Ensure that the Metabolite names match with KEGG IDs or KEGG trivial names.</span></span>
<span>    <span class="va">dma_res</span> <span class="op">&lt;-</span> <span class="va">DMA_Annova</span><span class="op">[[</span><span class="st">"dma"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="va">comparison</span><span class="op">]</span><span class="op">]</span></span>
<span>    <span class="va">dma_res</span> <span class="op">&lt;-</span> <span class="va">dma_res</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/stats/complete.cases.html" class="external-link">complete.cases</a></span><span class="op">(</span><span class="va">dma_res</span><span class="op">)</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="co">#we remove metabolites that do not have a KEGG ID/KEGG pathway</span></span>
<span>    <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">remove_rownames</a></span><span class="op">(</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"KEGGCompound"</span><span class="op">)</span><span class="co">#We use the KEGG trivial names to match with the KEGG pathways</span></span>
<span></span>
<span>    <span class="co">#Perform ORA: Here we use</span></span>
<span>    <span class="va">DM_ORA_res</span><span class="op">[[</span><span class="va">comparison</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/cluster_ora.html">cluster_ora</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">dma_res</span>,</span>
<span>    metadata_info<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>ClusterColumn<span class="op">=</span><span class="st">"core_specific"</span>, PathwayTerm<span class="op">=</span> <span class="st">"term"</span>, PathwayFeature<span class="op">=</span> <span class="st">"Metabolite"</span><span class="op">)</span>,</span>
<span>    remove_background<span class="op">=</span><span class="cn">FALSE</span>,<span class="co">#we do not have any background</span></span>
<span>    input_pathway<span class="op">=</span><span class="va">KEGG_Pathways</span>,</span>
<span>    pathway_name<span class="op">=</span><span class="st">"KEGG"</span>,</span>
<span>    min_gssize<span class="op">=</span><span class="fl">3</span>,</span>
<span>    max_gssize<span class="op">=</span><span class="fl">1000</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Lets check how the results look like:</span></span>
<span><span class="va">MC_ORA_HK2_vs_786M1A_Consumed</span> <span class="op">&lt;-</span> <span class="va">DM_ORA_res</span><span class="op">[[</span><span class="st">"HK2_vs_786-M1A"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"DF"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"Consumed"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the ORA results for the comparison of 786-M1A versus HK2
cells focusing on pathways enriched in <code>consumed</code>
metabolites.
</caption>
<thead><tr>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
RichFactor
</th>
<th style="text-align:right;">
FoldEnrichment
</th>
<th style="text-align:right;">
zScore
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
Metabolites_in_pathway
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
Metabolites_in_Pathway
</th>
<th style="text-align:right;">
percentage_of_Pathway_detected
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
3/13
</td>
<td style="text-align:left;">
3/52
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
4.0000000
</td>
<td style="text-align:right;">
3.0606122
</td>
<td style="text-align:right;">
0.0129412
</td>
<td style="text-align:right;">
0.4464706
</td>
<td style="text-align:right;">
0.4464706
</td>
<td style="text-align:left;">
L-Aspartate/L-Histidine/Pantothenate
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
32
</td>
<td style="text-align:right;">
9.38
</td>
</tr>
<tr>
<td style="text-align:left;">
3/13
</td>
<td style="text-align:left;">
3/52
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
4.0000000
</td>
<td style="text-align:right;">
3.0606122
</td>
<td style="text-align:right;">
0.0129412
</td>
<td style="text-align:right;">
0.4464706
</td>
<td style="text-align:right;">
0.4464706
</td>
<td style="text-align:left;">
(9Z)-Octadecenoic acid/Hexadecanoic acid/Octadecanoic acid
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
58
</td>
<td style="text-align:right;">
5.17
</td>
</tr>
<tr>
<td style="text-align:left;">
3/13
</td>
<td style="text-align:left;">
15/52
</td>
<td style="text-align:right;">
0.2000000
</td>
<td style="text-align:right;">
0.8000000
</td>
<td style="text-align:right;">
-0.5250483
</td>
<td style="text-align:right;">
0.8092578
</td>
<td style="text-align:right;">
0.8880214
</td>
<td style="text-align:right;">
0.8880214
</td>
<td style="text-align:left;">
cis-Aconitate/L-Aspartate/L-Tyrosine
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
144
</td>
<td style="text-align:right;">
2.08
</td>
</tr>
<tr>
<td style="text-align:left;">
4/13
</td>
<td style="text-align:left;">
18/52
</td>
<td style="text-align:right;">
0.2222222
</td>
<td style="text-align:right;">
0.8888889
</td>
<td style="text-align:right;">
-0.3333333
</td>
<td style="text-align:right;">
0.7455615
</td>
<td style="text-align:right;">
0.8880214
</td>
<td style="text-align:right;">
0.8880214
</td>
<td style="text-align:left;">
L-Aspartate/Glycine/L-Histidine/Taurine
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
129
</td>
<td style="text-align:right;">
3.10
</td>
</tr>
<tr>
<td style="text-align:left;">
2/13
</td>
<td style="text-align:left;">
11/52
</td>
<td style="text-align:right;">
0.1818182
</td>
<td style="text-align:right;">
0.7272727
</td>
<td style="text-align:right;">
-0.5824484
</td>
<td style="text-align:right;">
0.8354283
</td>
<td style="text-align:right;">
0.8880214
</td>
<td style="text-align:right;">
0.8880214
</td>
<td style="text-align:left;">
L-Aspartate/Succinate
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
27
</td>
<td style="text-align:right;">
7.41
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level3">
<h3 id="mca">MCA<a class="anchor" aria-label="anchor" href="#mca"></a>
</h3>
<p>Metabolite Clustering Analysis (<code>MCA</code>) is a module, which
includes different functions to enable clustering of metabolites into
groups based on logical regulatory rules. This can be particularly
useful if one has multiple conditions and aims to find patterns in the
data.</p>
<div class="section level4">
<h4 id="mca_core">mca_core<a class="anchor" aria-label="anchor" href="#mca_core"></a>
</h4>
<p>This metabolite clustering method is based on logical regulatory
rules to sort metabolites into metabolite clusters. Here you
additionally need intracellular samples corresponding to the core
samples.<br>
Here we will define if a feature (= metabolite) is assigned into:<br>
1. <strong><em>“UP”</em></strong>, which means a metabolite is
significantly up-regulated in the underlying comparison.<br>
2. <strong><em>“DOWN”</em></strong>, which means a metabolite is
significantly down-regulated in the underlying comparison.<br>
3. <strong><em>“No Change”</em></strong>, which means a metabolite does
not change significantly in the underlying comparison and/or is not
defined as up-regulated/down-regulated based on the Log2FC threshold
chosen.<br><br>
Therebye “No Change” is further subdivided into four states:<br>
1. <strong><em>“Not Detected”</em></strong>, which means a metabolite is
not detected in the underlying comparison.<br>
2. <strong><em>“Not Significant”</em></strong>, which means a metabolite
is not significant in the underlying comparison.<br>
3. <strong><em>“Significant positive”</em></strong>, which means a
metabolite is significant in the underlying comparison and the
differential metabolite abundance is positive, yet does not meet the
threshold set for “UP” (e.g. Log2FC &gt;1 = “UP” and we have a
significant Log2FC=0.8).<br>
4. <strong><em>“Significant negative”</em></strong>, which means a
metabolite is significant in the underlying comparison and the
differential metabolite abundance is negative, yet does not meet the
threshold set for “DOWN”.<br><br>
Lastly, we also take into account the core direction, meaning if a
metabolite was:<br>
1. <strong><em>“Released”</em></strong>, which means is released into
the media in both conditions of the underlying comparison.<br>
2. <strong><em>“Consumed”</em></strong>, which means is consumed from
the media in both conditions of the underlying comparison.<br>
3. <strong><em>“Released/Consumed”</em></strong>, which means is
consumed/released in one condition, whilst the opposite occurs in the
second condition of the underlying comparison.<br>
4. <strong><em>“Not Detected”</em></strong>, which means a metabolite is
not detected in the underlying comparison.<br>
This definition is done individually for each comparison and will impact
in which metabolite cluster a metabolite is sorted into.<br>
Since we have two comparisons (Intracellular and core), we can choose
between different Background settings, which defines which features will
be considered for the clusters (e.g. you could include only features (=
metabolites) that are detected in both comparisons, removing the rest of
the features).The background methods <code>method_background</code> are
the following from <strong><em>1.1. - 1.4.</em></strong> from most
restrictive to least restrictive:<br><strong><em>1.1. Intra&amp;core</em></strong>: Most stringend background
setting and will lead to a small number of metabolites.<br><strong><em>1.2. core</em></strong>: Focus is on the metabolite
abundance of the core.<br><strong><em>1.3. Intra</em></strong>: Focus is on the metabolite
abundance of intracellular.<br><strong><em>1.4. Intra|core</em></strong>: Least stringent background
method, since a metabolite will be included in the input if it has been
detected on one of the two conditions.<br><br>
Lastly, we will get clusters of metabolites that are defined by the
metabolite change in the two conditions. For example, if Alanine is “UP”
based on the thresholds in both comparisons it will be sorted into the
cluster “core_UP”. As there are three 6-state6-state4 transitions
between the comparisons, the flows are summarised into smaller amount of
metabolite clusters using different Regulation Groupings (RG): 1.
RG1_All<br>
2. RG2_Significant taking into account genes that are significant (UP,
DOWN, significant positive, significant negative)<br>
3. RG3_SignificantChange only takes into account genes that have
significant changes (UP, DOWN).<br><br>
In order to define which group a metabolite is assigned to, we set two
different thresholds. For intracellular those are based on the
differential metabolite abundance (<code>Log2FC</code>) and the
<code>significance</code> (e.g. p.adj). For the core data this is based
on the <code>Log2 Distance</code> and the <code>significance</code>
(e.g. p.adj). For <code>Log2FC</code> we recommend a threshold of 0.5 or
1, whilst for the <code>Log2 Distance</code> one should check the
distance ranges and base the threshold on this.<br><br>
Regulatory rules:<br><br></p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example of all possible flows:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">mca_core_rules</span><span class="op">)</span></span>
<span></span>
<span><span class="va">MCA_CoRe_Rule</span> <span class="op">&lt;-</span> <span class="va">mca_core_rules</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Metabolite Clustering Analysis: core.
</caption>
<thead><tr>
<th style="text-align:left;">
Intra
</th>
<th style="text-align:left;">
CoRe
</th>
<th style="text-align:left;">
Core_Direction
</th>
<th style="text-align:left;">
RG1_All
</th>
<th style="text-align:left;">
R2_Significant
</th>
<th style="text-align:left;">
RG3_Change
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra DOWN+ CoRe DOWN_Released
</td>
<td style="text-align:left;">
Both_DOWN (Released)
</td>
<td style="text-align:left;">
Both_DOWN (Released)
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra DOWN+ CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra DOWN+ CoRe Not Significant_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra DOWN+ CoRe Significant Negative_Released
</td>
<td style="text-align:left;">
Both_DOWN (Released)
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra DOWN+ CoRe Significant Positive_Released
</td>
<td style="text-align:left;">
Opposite (Released UP)
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra DOWN+ CoRe UP_Released
</td>
<td style="text-align:left;">
Opposite (Released UP)
</td>
<td style="text-align:left;">
Opposite (Released UP)
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra UP+ CoRe DOWN_Released
</td>
<td style="text-align:left;">
Opposite (Released DOWN)
</td>
<td style="text-align:left;">
Opposite (Released DOWN)
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra UP+ CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra UP+ CoRe Not Significant_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra UP+ CoRe Significant Negative_Released
</td>
<td style="text-align:left;">
Opposite (Released UP)
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra UP+ CoRe Significant Positive_Released
</td>
<td style="text-align:left;">
Both_UP (Released)
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra UP+ CoRe UP_Released
</td>
<td style="text-align:left;">
Both_UP (Released)
</td>
<td style="text-align:left;">
Both_UP (Released)
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Not Detected+ CoRe DOWN_Released
</td>
<td style="text-align:left;">
CoRe_DOWN (Released)
</td>
<td style="text-align:left;">
CoRe_DOWN (Released)
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra Not Detected+ CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Not Detected+ CoRe Not Significant_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Not Detected+ CoRe Significant Negative_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Not Detected+ CoRe Significant Positive_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Not Detected+ CoRe UP_Released
</td>
<td style="text-align:left;">
CoRe_UP (Released)
</td>
<td style="text-align:left;">
CoRe_UP (Released)
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Significant Negative+ CoRe DOWN_Released
</td>
<td style="text-align:left;">
Both_DOWN (Released)
</td>
<td style="text-align:left;">
CoRe_DOWN (Released)
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra Significant Negative+ CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Significant Negative+ CoRe Not Significant_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Significant Negative+ CoRe Significant Negative_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Significant Negative+ CoRe Significant Positive_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Significant Negative+ CoRe UP_Released
</td>
<td style="text-align:left;">
Opposite (Released UP)
</td>
<td style="text-align:left;">
CoRe_UP (Released)
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Significant Positive+ CoRe DOWN_Released
</td>
<td style="text-align:left;">
Opposite (Released DOWN)
</td>
<td style="text-align:left;">
CoRe_DOWN (Released)
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra Significant Positive+ CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Significant Positive+ CoRe Not Significant_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Significant Positive+ CoRe Significant Negative_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Significant Positive+ CoRe Significant Positive_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Significant Positive+ CoRe UP_Released
</td>
<td style="text-align:left;">
Both_UP (Released)
</td>
<td style="text-align:left;">
CoRe_UP (Released)
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Not Significant+ CoRe DOWN_Released
</td>
<td style="text-align:left;">
CoRe_DOWN (Released)
</td>
<td style="text-align:left;">
CoRe_DOWN (Released)
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra Not Significant+ CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Not Significant+ CoRe Not Significant_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Not Significant+ CoRe Significant Negative_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Not Significant+ CoRe Significant Positive_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Not Significant+ CoRe UP_Released
</td>
<td style="text-align:left;">
CoRe_UP (Released)
</td>
<td style="text-align:left;">
CoRe_UP (Released)
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra DOWN+ CoRe DOWN_Consumed
</td>
<td style="text-align:left;">
Both_DOWN (Consumed)
</td>
<td style="text-align:left;">
Both_DOWN (Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra DOWN+ CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra DOWN+ CoRe Not Significant_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra DOWN+ CoRe Significant Negative_Consumed
</td>
<td style="text-align:left;">
Both_DOWN (Consumed)
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra DOWN+ CoRe Significant Positive_Consumed
</td>
<td style="text-align:left;">
Opposite (Consumed UP)
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra DOWN+ CoRe UP_Consumed
</td>
<td style="text-align:left;">
Opposite (Consumed UP)
</td>
<td style="text-align:left;">
Opposite (Consumed UP)
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra UP+ CoRe DOWN_Consumed
</td>
<td style="text-align:left;">
Opposite (Consumed DOWN)
</td>
<td style="text-align:left;">
Opposite (Consumed DOWN)
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra UP+ CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra UP+ CoRe Not Significant_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra UP+ CoRe Significant Negative_Consumed
</td>
<td style="text-align:left;">
Opposite (Consumed UP)
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra UP+ CoRe Significant Positive_Consumed
</td>
<td style="text-align:left;">
Both_UP (Consumed)
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra UP+ CoRe UP_Consumed
</td>
<td style="text-align:left;">
Both_UP (Consumed)
</td>
<td style="text-align:left;">
Both_UP (Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Not Detected+ CoRe DOWN_Consumed
</td>
<td style="text-align:left;">
CoRe_DOWN (Consumed)
</td>
<td style="text-align:left;">
CoRe_DOWN (Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra Not Detected+ CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Not Detected+ CoRe Not Significant_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Not Detected+ CoRe Significant Negative_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Not Detected+ CoRe Significant Positive_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Not Detected+ CoRe UP_Consumed
</td>
<td style="text-align:left;">
CoRe_UP (Consumed)
</td>
<td style="text-align:left;">
CoRe_UP (Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Significant Negative+ CoRe DOWN_Consumed
</td>
<td style="text-align:left;">
Both_DOWN (Consumed)
</td>
<td style="text-align:left;">
CoRe_DOWN (Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra Significant Negative+ CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Significant Negative+ CoRe Not Significant_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Significant Negative+ CoRe Significant Negative_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Significant Negative+ CoRe Significant Positive_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Significant Negative+ CoRe UP_Consumed
</td>
<td style="text-align:left;">
Opposite (Consumed UP)
</td>
<td style="text-align:left;">
CoRe_UP (Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Significant Positive+ CoRe DOWN_Consumed
</td>
<td style="text-align:left;">
Opposite (Consumed DOWN)
</td>
<td style="text-align:left;">
CoRe_DOWN (Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra Significant Positive+ CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Significant Positive+ CoRe Not Significant_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Significant Positive+ CoRe Significant Negative_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Significant Positive+ CoRe Significant Positive_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Significant Positive+ CoRe UP_Consumed
</td>
<td style="text-align:left;">
Both_UP (Consumed)
</td>
<td style="text-align:left;">
CoRe_UP (Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Not Significant+ CoRe DOWN_Consumed
</td>
<td style="text-align:left;">
CoRe_DOWN (Consumed)
</td>
<td style="text-align:left;">
CoRe_DOWN (Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra Not Significant+ CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Not Significant+ CoRe Not Significant_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Not Significant+ CoRe Significant Negative_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Not Significant+ CoRe Significant Positive_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Not Significant+ CoRe UP_Consumed
</td>
<td style="text-align:left;">
CoRe_UP (Consumed)
</td>
<td style="text-align:left;">
CoRe_UP (Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra DOWN + CoRe DOWN_Released/Consumed
</td>
<td style="text-align:left;">
Both_DOWN (Released/Consumed)
</td>
<td style="text-align:left;">
Both_DOWN (Released/Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra DOWN + CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra DOWN + CoRe Not Significant_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra DOWN + CoRe Significant Negative_Released/Consumed
</td>
<td style="text-align:left;">
Both_DOWN (Released/Consumed)
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra DOWN + CoRe Significant Positive_Released/Consumed
</td>
<td style="text-align:left;">
Opposite (Released/Consumed UP)
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra DOWN + CoRe UP_Released/Consumed
</td>
<td style="text-align:left;">
Opposite (Released/Consumed UP)
</td>
<td style="text-align:left;">
Opposite (Released/Consumed UP)
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra UP + CoRe DOWN_Released/Consumed
</td>
<td style="text-align:left;">
Opposite (Released/Consumed DOWN)
</td>
<td style="text-align:left;">
Opposite (Released/Consumed DOWN)
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra UP + CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra UP + CoRe Not Significant_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra UP + CoRe Significant Negative_Released/Consumed
</td>
<td style="text-align:left;">
Opposite (Released/Consumed UP)
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra UP + CoRe Significant Positive_Released/Consumed
</td>
<td style="text-align:left;">
Both_UP (Released/Consumed)
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra UP + CoRe UP_Released/Consumed
</td>
<td style="text-align:left;">
Both_UP (Released/Consumed)
</td>
<td style="text-align:left;">
Both_UP (Released/Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Not Detected + CoRe DOWN_Released/Consumed
</td>
<td style="text-align:left;">
CoRe_DOWN (Released/Consumed)
</td>
<td style="text-align:left;">
CoRe_DOWN (Released/Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra Not Detected + CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Not Detected + CoRe Not Significant_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Not Detected + CoRe Significant Negative_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Not Detected + CoRe Significant Positive_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Not Detected + CoRe UP_Released/Consumed
</td>
<td style="text-align:left;">
CoRe_UP (Released/Consumed)
</td>
<td style="text-align:left;">
CoRe_UP (Released/Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Significant Negative + CoRe DOWN_Released/Consumed
</td>
<td style="text-align:left;">
Both_DOWN (Released/Consumed)
</td>
<td style="text-align:left;">
CoRe_DOWN (Released/Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra Significant Negative + CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Significant Negative + CoRe Not Significant_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Significant Negative + CoRe Significant Negative_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Significant Negative + CoRe Significant Positive_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Significant Negative + CoRe UP_Released/Consumed
</td>
<td style="text-align:left;">
Opposite (Released/Consumed UP)
</td>
<td style="text-align:left;">
CoRe_UP (Released/Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Significant Positive + CoRe DOWN_Released/Consumed
</td>
<td style="text-align:left;">
Opposite (Released/Consumed DOWN)
</td>
<td style="text-align:left;">
CoRe_DOWN (Released/Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra Significant Positive + CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Significant Positive + CoRe Not Significant_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Significant Positive + CoRe Significant Negative_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Significant Positive + CoRe Significant Positive_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Significant Positive + CoRe UP_Released/Consumed
</td>
<td style="text-align:left;">
Both_UP (Released/Consumed)
</td>
<td style="text-align:left;">
CoRe_UP (Released/Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Not Significant + CoRe DOWN_Released/Consumed
</td>
<td style="text-align:left;">
CoRe_DOWN (Released/Consumed)
</td>
<td style="text-align:left;">
CoRe_DOWN (Released/Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra Not Significant + CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Not Significant + CoRe Not Significant_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Not Significant + CoRe Significant Negative_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Not Significant + CoRe Significant Positive_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Not Significant + CoRe UP_Released/Consumed
</td>
<td style="text-align:left;">
CoRe_UP (Released/Consumed)
</td>
<td style="text-align:left;">
CoRe_UP (Released/Consumed)
</td>
</tr>
</tbody>
</table>
<p><br>
Now we can load the corresponding pre-processed intracellular example
data for the comparison of 786M-1A versus HK2 (For the detailed
pre-processing please see the vignette “Standard Metabolomics”).</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the Pre-processed intracellular data:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">intracell_dma</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Intra_DMA_HK2_vs_786M1A</span> <span class="op">&lt;-</span> <span class="va">intracell_dma</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Perform metabolite clustering:</span></span>
<span><span class="va">MCA_core_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/mca_core.html">mca_core</a></span><span class="op">(</span>data_intra <span class="op">=</span><span class="va">Intra_DMA_HK2_vs_786M1A</span>,</span>
<span>data_core <span class="op">=</span> <span class="va">DMA_HK2_vs_786M1A</span>,</span>
<span>metadata_info_intra<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>ValueCol<span class="op">=</span><span class="st">"Log2FC"</span>,StatCol<span class="op">=</span><span class="st">"p.adj"</span>, cutoff_stat<span class="op">=</span> <span class="fl">0.05</span>, ValueCutoff<span class="op">=</span><span class="fl">0.8</span><span class="op">)</span>,</span>
<span>metadata_info_core<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>DirectionCol<span class="op">=</span><span class="st">"core"</span>, ValueCol<span class="op">=</span><span class="st">"Log2(Distance)"</span>,StatCol<span class="op">=</span><span class="st">"p.adj"</span>, cutoff_stat<span class="op">=</span> <span class="fl">0.05</span>, ValueCutoff<span class="op">=</span><span class="fl">28</span><span class="op">)</span>,</span>
<span>feature<span class="op">=</span> <span class="st">"Metabolite"</span>,</span>
<span>method_background<span class="op">=</span><span class="st">"Intra&amp;core"</span>,</span>
<span>path<span class="op">=</span><span class="cn">NULL</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Lets check how the results look like:</span></span>
<span><span class="va">MCA_res</span> <span class="op">&lt;-</span> <span class="va">MCA_core_res</span><span class="op">[[</span><span class="st">"MCA_core_Results"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">Clustersummary</span> <span class="op">&lt;-</span> <span class="va">MCA_core_res</span><span class="op">[[</span><span class="st">"MCA_core_summary"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
mca_core for the comparison of 786-M1A versus HK2 cells in intracellular
and core samples.
</caption>
<thead><tr>
<th style="text-align:left;">
Metabolite
</th>
<th style="text-align:left;">
Intra_DF_Cutoff
</th>
<th style="text-align:left;">
Intra_DF_Cutoff_Specific.x
</th>
<th style="text-align:left;">
core_DF_Detected
</th>
<th style="text-align:left;">
core_DF_Cutoff
</th>
<th style="text-align:left;">
core_DF_Cutoff_Specific
</th>
<th style="text-align:left;">
BG_method
</th>
<th style="text-align:left;">
RG1_All
</th>
<th style="text-align:left;">
RG2_Significant
</th>
<th style="text-align:left;">
RG3_Change
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
adenosine
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
Background = FALSE
</td>
<td style="text-align:left;">
Background = FALSE
</td>
<td style="text-align:left;">
Background = FALSE
</td>
</tr>
<tr>
<td style="text-align:left;">
ADP
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
Background = FALSE
</td>
<td style="text-align:left;">
Background = FALSE
</td>
<td style="text-align:left;">
Background = FALSE
</td>
</tr>
<tr>
<td style="text-align:left;">
betaine
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
Intra DOWN + core DOWN_Released
</td>
<td style="text-align:left;">
Both_DOWN (Released)
</td>
<td style="text-align:left;">
Both_DOWN (Released)
</td>
</tr>
<tr>
<td style="text-align:left;">
creatine
</td>
<td style="text-align:left;">
No Change
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
</tr>
</tbody>
</table>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
mca_core summary of number of metabolites per cluster.
</caption>
<thead><tr>
<th style="text-align:left;">
Regulation Grouping
</th>
<th style="text-align:left;">
SiRCle cluster Name
</th>
<th style="text-align:right;">
Number of Features
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
RG3_Change
</td>
<td style="text-align:left;">
Both_DOWN (Released)
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
RG3_Change
</td>
<td style="text-align:left;">
core_UP (Consumed)
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
RG3_Change
</td>
<td style="text-align:left;">
core_DOWN (Released)
</td>
<td style="text-align:right;">
1
</td>
</tr>
</tbody>
</table>
<p><br>
Now we can also create Bargraphs of the clusters to visualize the
results. Here we create two summary bargraphs for the Regulation
Grouping RG2 and RG3:<br></p>
<pre><code><span><span class="co">#&gt; Warning: No shared levels found between `names(values)` of the manual scale and the</span></span>
<span><span class="co">#&gt; data's <span style="color: #00BB00;">fill</span> values.</span></span>
<span><span class="co">#&gt; No shared levels found between `names(values)` of the manual scale and the</span></span>
<span><span class="co">#&gt; data's <span style="color: #00BB00;">fill</span> values.</span></span>
<span><span class="co">#&gt; No shared levels found between `names(values)` of the manual scale and the</span></span>
<span><span class="co">#&gt; data's <span style="color: #00BB00;">fill</span> values.</span></span>
<span><span class="co">#&gt; No shared levels found between `names(values)` of the manual scale and the</span></span>
<span><span class="co">#&gt; data's <span style="color: #00BB00;">fill</span> values.</span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/core-DOWN-1.png" class="r-plt" width="700"></p>
</div>
<div class="section level4">
<h4 id="metalinksdb-metabolite-receptor-and-metabolite-transporter-sets">MetaLinksDB metabolite-receptor and metabolite-transporter sets<a class="anchor" aria-label="anchor" href="#metalinksdb-metabolite-receptor-and-metabolite-transporter-sets"></a>
</h4>
<p>The MetaLinks database is a manually curated database of
metabolite-receptor and metabolite-transporter sets that can be used to
study the connection of metabolites and receptors or transporters <span class="citation">(Farr et al. 2024)</span>.<br>
to remove potential false positives and decrease the number of putative
metabolite-receptor interactions, we filter the MetalinksDB resource to
metabolites that are annotated as present in the kidney, blood, or urine
in HMDB and known to be extracellular.<br></p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Selection as described in ST2 of Farr_Dimitrov2024:</span></span>
<span><span class="va">MetaLinksDB</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/metsigdb_metalinks.html">metsigdb_metalinks</a></span><span class="op">(</span>cell_location <span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Extracellular"</span><span class="op">)</span>,</span>
<span>tissue_location <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Kidney"</span>, <span class="st">"All Tissues"</span><span class="op">)</span>,</span>
<span>biospecimen_location <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Blood"</span>,  <span class="st">"Urine"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Here we add a UniquePair column combining hmdb-protein connection, removing duplications originating from different PK resources</span></span>
<span><span class="va">MetaLinksDB_Select</span> <span class="op">&lt;-</span> <span class="va">MetaLinksDB</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/unite.html" class="external-link">unite</a></span><span class="op">(</span><span class="st">"UniquePair"</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hmdb"</span>, <span class="st">"gene_symbol"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"_"</span>, remove<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="va">UniquePair</span>, .keep_all <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><br>
Now we can use this information to understand if the metabolites that
are consumed in the 786-M1A cells and Released in the HK2 cells are
connected to specific receptors or transporters and are in the bioRCM
cluster “Both_DOWN (Released/Consumed)<br></p>
<pre><code><span><span class="co">#&gt; The following metabolites are not connected to any receptor or transporter in the MetalinksDB:</span></span>
<span><span class="co">#&gt; The following metabolites are connected to at least one receptor or transporter in the MetalinksDB:</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="ora-on-each-metabolite-cluster">ORA on each metabolite cluster<a class="anchor" aria-label="anchor" href="#ora-on-each-metabolite-cluster"></a>
</h4>
As explained in detail above, Over Representation Analysis (ORA) is a
pathway enrichment analysis (PEA) method. As ORA is based on the Fishers
exact test it is perfect to test if a set of features (=metabolic
pathways) are over-represented in the selection of features (= clusters
of metabolites) from the data in comparison to all measured features
(all metabolites). In detail, <code>MC_ORA()</code> will perform ORA on
each of the metabolite clusters using all metabolites as the background.
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Pathway Input for MC_ORA.
</caption>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
HMDB
</th>
<th style="text-align:left;">
KEGG.ID
</th>
<th style="text-align:left;">
KEGGCompound
</th>
<th style="text-align:left;">
Pathway
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
N-acetylaspartate
</td>
<td style="text-align:left;">
HMDB0000812
</td>
<td style="text-align:left;">
C01042
</td>
<td style="text-align:left;">
N-Acetyl-L-aspartate
</td>
<td style="text-align:left;">
Alanine, aspartate and glutamate metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
argininosuccinate
</td>
<td style="text-align:left;">
HMDB0000052
</td>
<td style="text-align:left;">
C03406
</td>
<td style="text-align:left;">
N-(L-Arginino)succinate
</td>
<td style="text-align:left;">
Alanine, aspartate and glutamate metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
N-acetylaspartylglutamate
</td>
<td style="text-align:left;">
HMDB0001067
</td>
<td style="text-align:left;">
C12270
</td>
<td style="text-align:left;">
N-Acetylaspartylglutamate
</td>
<td style="text-align:left;">
Alanine, aspartate and glutamate metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
tyrosine
</td>
<td style="text-align:left;">
HMDB0000158
</td>
<td style="text-align:left;">
C00082
</td>
<td style="text-align:left;">
L-Tyrosine
</td>
<td style="text-align:left;">
Amino acid metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
asparagine
</td>
<td style="text-align:left;">
HMDB0000168
</td>
<td style="text-align:left;">
C00152
</td>
<td style="text-align:left;">
L-Asparagine
</td>
<td style="text-align:left;">
Amino acid metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
glutamate
</td>
<td style="text-align:left;">
HMDB0000148
</td>
<td style="text-align:left;">
C00025
</td>
<td style="text-align:left;">
L-Glutamate
</td>
<td style="text-align:left;">
Amino acid metabolism
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">MC_ORA_result</span><span class="op">&lt;-</span> <span class="fu"><a href="../../reference/cluster_ora.html">cluster_ora</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">MCA_core_res</span><span class="op">[[</span><span class="st">"MCA_core_Results"</span><span class="op">]</span><span class="op">]</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Metabolite"</span><span class="op">)</span>,</span>
<span>metadata_info<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>ClusterColumn<span class="op">=</span><span class="st">"RG2_Significant"</span>,</span>
<span>                                                        BackgroundColumn<span class="op">=</span><span class="st">"BG_method"</span>,</span>
<span>                                                        PathwayTerm<span class="op">=</span> <span class="st">"Pathway"</span>, <span class="co">#This is the column name including the pathways names</span></span>
<span>                                                        PathwayFeature<span class="op">=</span> <span class="st">"Metabolite"</span><span class="op">)</span>,</span>
<span>                                                        remove_background<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                                                        input_pathway<span class="op">=</span><span class="va">MappingInfo</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span><span class="st">"Metabolite"</span><span class="op">)</span>,</span>
<span>                                                        pathway_name<span class="op">=</span><span class="st">"KEGG"</span>,</span>
<span>                                                        min_gssize<span class="op">=</span><span class="fl">3</span>,</span>
<span>                                                        max_gssize<span class="op">=</span><span class="fl">1000</span> ,</span>
<span>                                                        save_table<span class="op">=</span> <span class="st">"csv"</span><span class="op">)</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
MC_ORA results for the RG2_Significant cluster
<code>Both_DOWN (Consumed)</code>.
</caption>
<tbody><tr></tr></tbody>
</table>
<p><br>
Here we see that the pathways have a low amount of genes included that
were also part of the cluster and the pathways are not significant. This
is due to multiple factors, first we only start with a small number of
metabolites with KEGG IDs and secondly we only included metabolites if
they where detected in both, intracellular and core samples (parameter
<code>method_background="Intra&amp;core"</code>). Hence, by for example
setting parameter <code>method_background="Intra|core"</code>, we will
obtain larger metabolite clusters.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="run-metaproviz-visualisation">3. Run MetaProViz Visualisation<a class="anchor" aria-label="anchor" href="#run-metaproviz-visualisation"></a>
</h2>
<p>The big advantages of the <code>MetaProViz</code> visualization
module is its flexible and easy usage, which we will showcase below and
that the figures are saved in a publication ready style and format. For
instance, the x- and y-axis size will always be adjusted for the amount
of samples or features (=metabolites) plotted, or in the case of Volcano
plot and PCA plot the axis size is fixed and not affected by figure
legends or title. In this way, there is no need for many adjustments and
the figures can just be dropped into the presentation or paper and are
all in the same style.<br><br>
All the <code>VizPlotName()</code> functions are constructed in the same
way. Indeed, with the parameter <code>Plot_metadata_info</code> the user
can pass a named vector with information about the metadata column that
should be used to customize the plot by colour, shape or creating
individual plots, which will all be showcased for the different plot
types. Via the parameter <code>Plot_SettingsFile</code> the user can
pass the metadata DF, which can be dependent on the plot type for the
samples and/or the features (=metabolites). In case of both the
parameter is named <code>Plot_metadata_sample</code> and
<code>Plot_metadata_feature</code>.<br><br>
In each of those Plot_Settings, the user can label color and/or shape
based on additional information (e.g. Pathway information, Cluster
information or other other demographics like gender). Moreover, we also
enable to plot individual plots where applicable based on those MetaData
(e.g. one plot for each metabolic pathway).<br>
For this we need a metadata table including information about our
samples that could be relevant to e.g. color code:<br></p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">MetaData_Sample</span> <span class="op">&lt;-</span> <span class="va">Media_Preprocessed</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span><span class="op">]</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Status <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/case_when.html" class="external-link">case_when</a></span><span class="op">(</span><span class="va">Conditions</span><span class="op">==</span><span class="st">"HK2"</span> <span class="op">~</span> <span class="st">'Healthy'</span>,</span>
<span><span class="cn">TRUE</span> <span class="op">~</span> <span class="st">'Cancer'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Metadata table including additional information about our Samples.
</caption>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Conditions
</th>
<th style="text-align:right;">
Biological_Replicates
</th>
<th style="text-align:left;">
Status
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
MS51-07
</td>
<td style="text-align:left;">
HK2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
Healthy
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-08
</td>
<td style="text-align:left;">
HK2
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
Healthy
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-09
</td>
<td style="text-align:left;">
HK2
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:left;">
Healthy
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-10
</td>
<td style="text-align:left;">
HK2
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:left;">
Healthy
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-11
</td>
<td style="text-align:left;">
786-O
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-12
</td>
<td style="text-align:left;">
786-O
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-13
</td>
<td style="text-align:left;">
786-O
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-14
</td>
<td style="text-align:left;">
786-O
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-15
</td>
<td style="text-align:left;">
786-O
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-16
</td>
<td style="text-align:left;">
786-M1A
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-17
</td>
<td style="text-align:left;">
786-M1A
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-18
</td>
<td style="text-align:left;">
786-M1A
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-19
</td>
<td style="text-align:left;">
786-M1A
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-20
</td>
<td style="text-align:left;">
786-M1A
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-21
</td>
<td style="text-align:left;">
786-M2A
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-23
</td>
<td style="text-align:left;">
786-M2A
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-24
</td>
<td style="text-align:left;">
786-M2A
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-25
</td>
<td style="text-align:left;">
786-M2A
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-26
</td>
<td style="text-align:left;">
OSRC2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-27
</td>
<td style="text-align:left;">
OSRC2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-28
</td>
<td style="text-align:left;">
OSRC2
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-29
</td>
<td style="text-align:left;">
OSRC2
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-30
</td>
<td style="text-align:left;">
OSRC2
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-31
</td>
<td style="text-align:left;">
OSLM1B
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-32
</td>
<td style="text-align:left;">
OSLM1B
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-33
</td>
<td style="text-align:left;">
OSLM1B
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-34
</td>
<td style="text-align:left;">
OSLM1B
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-35
</td>
<td style="text-align:left;">
OSLM1B
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-36
</td>
<td style="text-align:left;">
RFX631
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-37
</td>
<td style="text-align:left;">
RFX631
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-38
</td>
<td style="text-align:left;">
RFX631
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-39
</td>
<td style="text-align:left;">
RFX631
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-40
</td>
<td style="text-align:left;">
RFX631
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
</tbody>
</table>
<p><br>
Moreover, we can use MetaData for our features (=Metabolites), which we
loaded with the <code>MappingInfo</code> and we can also add the
information on which cluster a metabolite was assigned to in the
<code>MCA()</code> analysis above:<br></p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">MetaData_Metab</span> <span class="op">&lt;-</span><span class="va">MappingInfo</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Metadata table including additional information about the Metabolites.
</caption>
<thead><tr>
<th style="text-align:left;">
HMDB
</th>
<th style="text-align:left;">
KEGG.ID
</th>
<th style="text-align:left;">
KEGGCompound
</th>
<th style="text-align:left;">
Pathway
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
HMDB0001067
</td>
<td style="text-align:left;">
C12270
</td>
<td style="text-align:left;">
N-Acetylaspartylglutamate
</td>
<td style="text-align:left;">
Alanine, aspartate and glutamate metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000158
</td>
<td style="text-align:left;">
C00082
</td>
<td style="text-align:left;">
L-Tyrosine
</td>
<td style="text-align:left;">
Amino acid metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000148
</td>
<td style="text-align:left;">
C00025
</td>
<td style="text-align:left;">
L-Glutamate
</td>
<td style="text-align:left;">
Amino acid metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0250980
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Amino acid metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Amino acid metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0004207
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Amino acid metabolism
</td>
</tr>
</tbody>
</table>
<p><br>
Noteworthy, here we can also use the KEGG pathways we used for the
pathway analysis.</p>
<div class="section level4">
<h4 id="pca-plots">PCA plots<a class="anchor" aria-label="anchor" href="#pca-plots"></a>
</h4>
<p>Principal component analysis (PCA) is a dimensionality reduction
method that reduces all the measured features (=metabolites) of one
sample into a few features in the different principal components,
whereby each principal component can explain a certain percentage of the
variance between the different samples. Hence, this enables
interpretation of sample clustering based on the measured features
(=metabolites).<br>
As mentioned above, PCA plots can be quite useful for quality control,
but of course it offers us many more opportunities, which will be
showcased here.<br><br>
As input, we need a DF that contains the samples as rownames and the
features (=metabolites) as column names:<br></p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Input_PCA</span> <span class="op">&lt;-</span> <span class="va">Media_Preprocessed</span><span class="op">[</span>,<span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">]</span> <span class="co">#remove columns that include Metadata such as cell type,...</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Input_data for <code><a href="../../reference/viz_pca.html">viz_pca()</a></code>, with samples as rownames and
metabolites as column names.
</caption>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
valine-d8
</th>
<th style="text-align:right;">
hipppuric acid-d5
</th>
<th style="text-align:right;">
2-hydroxyglutarate
</th>
<th style="text-align:right;">
2-ketoglutarate
</th>
<th style="text-align:right;">
3-Dehydro-L-threonate
</th>
<th style="text-align:right;">
acetylcarnitine
</th>
<th style="text-align:right;">
acetylcholine
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
MS51-07
</td>
<td style="text-align:right;">
5049281744
</td>
<td style="text-align:right;">
30218506574
</td>
<td style="text-align:right;">
-868601575
</td>
<td style="text-align:right;">
1530912430
</td>
<td style="text-align:right;">
28673393701
</td>
<td style="text-align:right;">
-67547592848
</td>
<td style="text-align:right;">
-496576017
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-08
</td>
<td style="text-align:right;">
-1441406385
</td>
<td style="text-align:right;">
-11561761745
</td>
<td style="text-align:right;">
1108346025
</td>
<td style="text-align:right;">
3684365990
</td>
<td style="text-align:right;">
39313101752
</td>
<td style="text-align:right;">
-234677799645
</td>
<td style="text-align:right;">
-124297109
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-09
</td>
<td style="text-align:right;">
2563904070
</td>
<td style="text-align:right;">
-10237061776
</td>
<td style="text-align:right;">
-189633550
</td>
<td style="text-align:right;">
-9097094735
</td>
<td style="text-align:right;">
-67782920598
</td>
<td style="text-align:right;">
218513442831
</td>
<td style="text-align:right;">
526295913
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-10
</td>
<td style="text-align:right;">
-6171779428
</td>
<td style="text-align:right;">
-8419683053
</td>
<td style="text-align:right;">
-50110899
</td>
<td style="text-align:right;">
3881816314
</td>
<td style="text-align:right;">
-203574855
</td>
<td style="text-align:right;">
83711949661
</td>
<td style="text-align:right;">
94577213
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-11
</td>
<td style="text-align:right;">
27378617230
</td>
<td style="text-align:right;">
123649127347
</td>
<td style="text-align:right;">
4311754839
</td>
<td style="text-align:right;">
23286535748
</td>
<td style="text-align:right;">
-16721754062
</td>
<td style="text-align:right;">
-434945308541
</td>
<td style="text-align:right;">
-183109349
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-12
</td>
<td style="text-align:right;">
31998154622
</td>
<td style="text-align:right;">
99965859879
</td>
<td style="text-align:right;">
808357705
</td>
<td style="text-align:right;">
19381509727
</td>
<td style="text-align:right;">
-16349278172
</td>
<td style="text-align:right;">
-583665678018
</td>
<td style="text-align:right;">
-998475065
</td>
</tr>
</tbody>
</table>
<p><br>
Now lets check out the standard plot:</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../../reference/viz_pca.html">viz_pca</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">Input_PCA</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="core-metabolomics_files/figure-html/viz-pca-2-1.png" class="r-plt" alt="Figure: Standard Settings." width="576"><p class="caption">
Figure: Standard Settings.
</p>
</div>
<p>Next, we can interactively choose shape and color using the
additional information of interest from our Metadata. Especially for
complex data, such as patient data, it can be valuable to use different
demographics (e.g. age, gender, medication,…) for this. First lets check
if we have any batch effect by colour coding for the biological
replicates, which would be the case if the replicates cluster
together.<br></p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../../reference/viz_pca.html">viz_pca</a></span><span class="op">(</span>metadata_info<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"Biological_Replicates"</span><span class="op">)</span>,</span>
<span>metadata_sample <span class="op">=</span> <span class="va">MetaData_Sample</span> ,</span>
<span>data<span class="op">=</span><span class="va">Input_PCA</span>,</span>
<span>plot_name <span class="op">=</span> <span class="st">"Batch Effect"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 3 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></span>
<span><span class="co">#&gt; ggrepel: 3 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="core-metabolomics_files/figure-html/viz-pca-3-1.png" class="r-plt" alt="Figure: Do we have a batch effect?" width="576"><p class="caption">
Figure: Do we have a batch effect?
</p>
</div>
<p>Given the biological replicates are numeric, we can also set
<code>color_scale</code> to continuous:</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../../reference/viz_pca.html">viz_pca</a></span><span class="op">(</span>metadata_info<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"Biological_Replicates"</span><span class="op">)</span>,</span>
<span>metadata_sample <span class="op">=</span> <span class="va">MetaData_Sample</span> ,</span>
<span>data<span class="op">=</span><span class="va">Input_PCA</span>,</span>
<span>scale_color <span class="op">=</span> <span class="st">"continuous"</span>,</span>
<span>plot_name <span class="op">=</span> <span class="st">"Batch Effect (continuous color scale)"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 3 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></span>
<span><span class="co">#&gt; ggrepel: 3 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="core-metabolomics_files/figure-html/viz-pca-4-1.png" class="r-plt" alt="Figure: Do we have a batch effect?" width="576"><p class="caption">
Figure: Do we have a batch effect?
</p>
</div>
<p>Next, we can colour code for condition and use the biological
replicates in the shape parameter:<br></p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../../reference/viz_pca.html">viz_pca</a></span><span class="op">(</span>metadata_info<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"Conditions"</span>, shape<span class="op">=</span><span class="st">"Biological_Replicates"</span><span class="op">)</span>,</span>
<span>metadata_sample <span class="op">=</span> <span class="va">MetaData_Sample</span> ,</span>
<span>data<span class="op">=</span><span class="va">Input_PCA</span>,</span>
<span>plot_name <span class="op">=</span> <span class="st">"Sample Conditions"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 3 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></span>
<span><span class="co">#&gt; ggrepel: 3 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="core-metabolomics_files/figure-html/viz-pca-5-1.png" class="r-plt" alt="Figure: Do the samples cluster for the conditions?" width="576"><p class="caption">
Figure: Do the samples cluster for the conditions?
</p>
</div>
<p>The different cell lines we have are either control or cancerous, so
we can display this too.<br></p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../../reference/viz_pca.html">viz_pca</a></span><span class="op">(</span>metadata_info<span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"Status"</span><span class="op">)</span>,</span>
<span>metadata_sample <span class="op">=</span> <span class="va">MetaData_Sample</span> ,</span>
<span>data<span class="op">=</span><span class="va">Input_PCA</span>,</span>
<span>plot_name <span class="op">=</span> <span class="st">"Sample Status"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 3 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></span>
<span><span class="co">#&gt; ggrepel: 3 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="core-metabolomics_files/figure-html/viz-pca-6-1.png" class="r-plt" alt="Figure: Do the samples cluster for the Cell status?" width="576"><p class="caption">
Figure: Do the samples cluster for the Cell status?
</p>
</div>
</div>
<div class="section level4">
<h4 id="heatmaps">Heatmaps<a class="anchor" aria-label="anchor" href="#heatmaps"></a>
</h4>
<p>Clustered heatmaps can be useful to understand the patterns in the
data, which will be showcased on different examples.<br>
As input, we need a DF that contains the samples as rownames and the
features (=metabolites) as column names:<br></p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Input_Heatmap</span> <span class="op">&lt;-</span>   <span class="va">Media_Preprocessed</span><span class="op">[</span>,<span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">)</span><span class="op">]</span> <span class="co">#remove columns that include Metadata such as cell type,...</span></span>
<span></span>
<span><span class="co"># Add consumption-release information of each cell type:</span></span>
<span><span class="va">MetaData_Metab</span> <span class="op">&lt;-</span> <span class="va">DMA_Annova</span><span class="op">[[</span><span class="st">"Feature_Metadata"</span><span class="op">]</span><span class="op">]</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Metabolite"</span><span class="op">)</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Input for <code><a href="../../reference/viz_heatmap.html">viz_heatmap()</a></code>, with samples as rownames and
metabolites as column names.
</caption>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
2-hydroxyglutarate
</th>
<th style="text-align:right;">
2-ketoglutarate
</th>
<th style="text-align:right;">
3-Dehydro-L-threonate
</th>
<th style="text-align:right;">
acetylcarnitine
</th>
<th style="text-align:right;">
acetylcholine
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
MS51-07
</td>
<td style="text-align:right;">
-868601575
</td>
<td style="text-align:right;">
1530912430
</td>
<td style="text-align:right;">
28673393701
</td>
<td style="text-align:right;">
-67547592848
</td>
<td style="text-align:right;">
-496576017
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-08
</td>
<td style="text-align:right;">
1108346025
</td>
<td style="text-align:right;">
3684365990
</td>
<td style="text-align:right;">
39313101752
</td>
<td style="text-align:right;">
-234677799645
</td>
<td style="text-align:right;">
-124297109
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-09
</td>
<td style="text-align:right;">
-189633550
</td>
<td style="text-align:right;">
-9097094735
</td>
<td style="text-align:right;">
-67782920598
</td>
<td style="text-align:right;">
218513442831
</td>
<td style="text-align:right;">
526295913
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-10
</td>
<td style="text-align:right;">
-50110899
</td>
<td style="text-align:right;">
3881816314
</td>
<td style="text-align:right;">
-203574855
</td>
<td style="text-align:right;">
83711949661
</td>
<td style="text-align:right;">
94577213
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-11
</td>
<td style="text-align:right;">
4311754839
</td>
<td style="text-align:right;">
23286535748
</td>
<td style="text-align:right;">
-16721754062
</td>
<td style="text-align:right;">
-434945308541
</td>
<td style="text-align:right;">
-183109349
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-12
</td>
<td style="text-align:right;">
808357705
</td>
<td style="text-align:right;">
19381509727
</td>
<td style="text-align:right;">
-16349278172
</td>
<td style="text-align:right;">
-583665678018
</td>
<td style="text-align:right;">
-998475065
</td>
</tr>
</tbody>
</table>
<p><br>
Now we can generate an overview heatmap. Since we plot all metabolites
the metabolite names are not plotted since this would get too crowded
(You can enforce this by changing the parameter
<code>enforce_featurenames = TRUE</code>).</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../../reference/viz_heatmap.html">viz_heatmap</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">Input_Heatmap</span>,</span>
<span>plot_name <span class="op">=</span> <span class="st">"Overview"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="core-metabolomics_files/figure-html/viz-heatmap-3-1.png" class="r-plt" alt="Overview heatmap." width="960"><p class="caption">
Overview heatmap.
</p>
</div>
<p><br>
Here we can add as many sample metadata information as needed at the
same time:<br></p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../../reference/viz_heatmap.html">viz_heatmap</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">Input_Heatmap</span>,</span>
<span>metadata_sample <span class="op">=</span> <span class="va">MetaData_Sample</span>,</span>
<span>metadata_info <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>color_Sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"Conditions"</span>,<span class="st">"Biological_Replicates"</span>, <span class="st">"Status"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>plot_name <span class="op">=</span> <span class="st">"Colour Samples"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="core-metabolomics_files/figure-html/viz-heatmap-4-1.png" class="r-plt" alt="Colour for sample metadata." width="960"><p class="caption">
Colour for sample metadata.
</p>
</div>
<p><br>
Moreover, we can also add metabolite metadata information:<br></p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../../reference/viz_heatmap.html">viz_heatmap</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">Input_Heatmap</span>,</span>
<span>metadata_sample <span class="op">=</span> <span class="va">MetaData_Sample</span>,</span>
<span>metadata_info <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>color_Metab <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"Pathway"</span>,  <span class="st">"core_786-M1A"</span>, <span class="st">"core_HK2"</span>, <span class="st">"core_786-M2A"</span>, <span class="st">"core_786-O"</span>, <span class="st">"core_OSLM1B"</span>, <span class="st">"core_OSRC2"</span>, <span class="st">"core_RFX631"</span><span class="op">)</span>,</span>
<span>                                        color_Sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"Conditions"</span>,<span class="st">"Biological_Replicates"</span>, <span class="st">"Status"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                                        metadata_feature <span class="op">=</span>  <span class="va">MetaData_Metab</span>,</span>
<span>                                        plot_name <span class="op">=</span> <span class="st">"Colour Metabolites"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="core-metabolomics_files/figure-html/viz-heatmap-5-1.png" class="r-plt" alt="Colour for metabolite metadata." width="1152"><p class="caption">
Colour for metabolite metadata.
</p>
</div>
<br>
Lastly, by generate individual plot for e.g. each pathway or the
metabolite clusters by adding individual (<code>individual_Sample</code>
or <code>individual_Metab</code>) to <code>Plot_metadata_info</code>. At
the same time we can still maintain the metadata information for both,
the samples and the metabolites. together this can help us to draw
biological conclusions about the different pathways: Indeed, we can
observe for the <code>D-Amino acid metabolism</code> many metabolites
fall into the MCA-Cluster <code>core_DOWN</code>, meaning in comparison
to HK2 cells we have a negative Log2FC for 786-O and 786-M1A.
<div class="vscroll-plot">
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../../reference/viz_heatmap.html">viz_heatmap</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">Input_Heatmap</span>,</span>
<span>metadata_sample <span class="op">=</span> <span class="va">MetaData_Sample</span>,</span>
<span>metadata_info <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>individual_Metab <span class="op">=</span> <span class="st">"Pathway"</span>,</span>
<span>                                        color_Sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"Conditions"</span>,<span class="st">"Biological_Replicates"</span><span class="op">)</span>,</span>
<span>                                        color_Metab <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"core_786-M1A"</span>, <span class="st">"core_HK2"</span>, <span class="st">"core_786-M2A"</span>, <span class="st">"core_786-O"</span>, <span class="st">"core_OSLM1B"</span>, <span class="st">"core_OSRC2"</span>, <span class="st">"core_RFX631"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                                        metadata_feature <span class="op">=</span>  <span class="va">MetaData_Metab</span>,</span>
<span>                                        plot_name <span class="op">=</span> <span class="st">"Pathway"</span><span class="op">)</span></span></code></pre></div>
</div>
<p><br><br><br></p>
<p>You can also choose to make individual plots for any Sample Metadata
using <code>individual_Sample</code> (e.g. in patients you may want to
plot male and female separately). Moreover, you can also use both at the
same time.</p>
</div>
<div class="section level4">
<h4 id="superplots">Superplots<a class="anchor" aria-label="anchor" href="#superplots"></a>
</h4>
<p>Sometimes one might be interested to create individual plots for each
metabolite to understand the differences between specific conditions.
For this common plot types are bargraphs, boxplots or violin plots. As
input, we need a DF that contains the samples as rownames and the
features (=metabolites) as column names:<br></p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Input_Superplot</span> <span class="op">&lt;-</span>  <span class="va">Media_Preprocessed</span><span class="op">[</span>,<span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">]</span><span class="co">#remove columns that include Metadata such as cell type,...</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Input for <code><a href="../../reference/viz_superplot.html">viz_superplot()</a></code>, with samples as rownames and
metabolites as column names.
</caption>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
valine-d8
</th>
<th style="text-align:right;">
hipppuric acid-d5
</th>
<th style="text-align:right;">
2-hydroxyglutarate
</th>
<th style="text-align:right;">
2-ketoglutarate
</th>
<th style="text-align:right;">
3-Dehydro-L-threonate
</th>
<th style="text-align:right;">
acetylcarnitine
</th>
<th style="text-align:right;">
acetylcholine
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
MS51-07
</td>
<td style="text-align:right;">
5049281744
</td>
<td style="text-align:right;">
30218506574
</td>
<td style="text-align:right;">
-868601575
</td>
<td style="text-align:right;">
1530912430
</td>
<td style="text-align:right;">
28673393701
</td>
<td style="text-align:right;">
-67547592848
</td>
<td style="text-align:right;">
-496576017
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-08
</td>
<td style="text-align:right;">
-1441406385
</td>
<td style="text-align:right;">
-11561761745
</td>
<td style="text-align:right;">
1108346025
</td>
<td style="text-align:right;">
3684365990
</td>
<td style="text-align:right;">
39313101752
</td>
<td style="text-align:right;">
-234677799645
</td>
<td style="text-align:right;">
-124297109
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-09
</td>
<td style="text-align:right;">
2563904070
</td>
<td style="text-align:right;">
-10237061776
</td>
<td style="text-align:right;">
-189633550
</td>
<td style="text-align:right;">
-9097094735
</td>
<td style="text-align:right;">
-67782920598
</td>
<td style="text-align:right;">
218513442831
</td>
<td style="text-align:right;">
526295913
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-10
</td>
<td style="text-align:right;">
-6171779428
</td>
<td style="text-align:right;">
-8419683053
</td>
<td style="text-align:right;">
-50110899
</td>
<td style="text-align:right;">
3881816314
</td>
<td style="text-align:right;">
-203574855
</td>
<td style="text-align:right;">
83711949661
</td>
<td style="text-align:right;">
94577213
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-11
</td>
<td style="text-align:right;">
27378617230
</td>
<td style="text-align:right;">
123649127347
</td>
<td style="text-align:right;">
4311754839
</td>
<td style="text-align:right;">
23286535748
</td>
<td style="text-align:right;">
-16721754062
</td>
<td style="text-align:right;">
-434945308541
</td>
<td style="text-align:right;">
-183109349
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-12
</td>
<td style="text-align:right;">
31998154622
</td>
<td style="text-align:right;">
99965859879
</td>
<td style="text-align:right;">
808357705
</td>
<td style="text-align:right;">
19381509727
</td>
<td style="text-align:right;">
-16349278172
</td>
<td style="text-align:right;">
-583665678018
</td>
<td style="text-align:right;">
-998475065
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-13
</td>
<td style="text-align:right;">
26220971337
</td>
<td style="text-align:right;">
99972874145
</td>
<td style="text-align:right;">
-235593734
</td>
<td style="text-align:right;">
19536285942
</td>
<td style="text-align:right;">
-28131696536
</td>
<td style="text-align:right;">
-645802166950
</td>
<td style="text-align:right;">
-822951440
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-14
</td>
<td style="text-align:right;">
27882417489
</td>
<td style="text-align:right;">
115689376712
</td>
<td style="text-align:right;">
1115432087
</td>
<td style="text-align:right;">
17101698081
</td>
<td style="text-align:right;">
-26309877624
</td>
<td style="text-align:right;">
-275257639669
</td>
<td style="text-align:right;">
-1272434260
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-15
</td>
<td style="text-align:right;">
24806237271
</td>
<td style="text-align:right;">
132343157722
</td>
<td style="text-align:right;">
-1179339178
</td>
<td style="text-align:right;">
17846663766
</td>
<td style="text-align:right;">
-40351828895
</td>
<td style="text-align:right;">
-321374884275
</td>
<td style="text-align:right;">
-1318142384
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-16
</td>
<td style="text-align:right;">
11094140749
</td>
<td style="text-align:right;">
57354194293
</td>
<td style="text-align:right;">
5445671697
</td>
<td style="text-align:right;">
28056643231
</td>
<td style="text-align:right;">
9905712215
</td>
<td style="text-align:right;">
-356638276041
</td>
<td style="text-align:right;">
-929117918
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-17
</td>
<td style="text-align:right;">
34144526544
</td>
<td style="text-align:right;">
116670253290
</td>
<td style="text-align:right;">
3041583921
</td>
<td style="text-align:right;">
22341957663
</td>
<td style="text-align:right;">
-1871784883
</td>
<td style="text-align:right;">
-379917701289
</td>
<td style="text-align:right;">
-1217951807
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-18
</td>
<td style="text-align:right;">
30843226841
</td>
<td style="text-align:right;">
131915991104
</td>
<td style="text-align:right;">
2364979960
</td>
<td style="text-align:right;">
19876068811
</td>
<td style="text-align:right;">
-12106631836
</td>
<td style="text-align:right;">
-297445077871
</td>
<td style="text-align:right;">
-907725948
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-19
</td>
<td style="text-align:right;">
26032777849
</td>
<td style="text-align:right;">
93842230196
</td>
<td style="text-align:right;">
600387750
</td>
<td style="text-align:right;">
22338213828
</td>
<td style="text-align:right;">
-15996801077
</td>
<td style="text-align:right;">
-325213646459
</td>
<td style="text-align:right;">
-1216114751
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-20
</td>
<td style="text-align:right;">
15413411654
</td>
<td style="text-align:right;">
47939629182
</td>
<td style="text-align:right;">
4007911817
</td>
<td style="text-align:right;">
20555474456
</td>
<td style="text-align:right;">
13695406411
</td>
<td style="text-align:right;">
-438100232873
</td>
<td style="text-align:right;">
-1430510133
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-21
</td>
<td style="text-align:right;">
53001553101
</td>
<td style="text-align:right;">
127450060294
</td>
<td style="text-align:right;">
4809729749
</td>
<td style="text-align:right;">
39166236259
</td>
<td style="text-align:right;">
-38896232803
</td>
<td style="text-align:right;">
-125539074625
</td>
<td style="text-align:right;">
-943874817
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-23
</td>
<td style="text-align:right;">
47001796040
</td>
<td style="text-align:right;">
161667174110
</td>
<td style="text-align:right;">
6707051392
</td>
<td style="text-align:right;">
38517108736
</td>
<td style="text-align:right;">
-6979130432
</td>
<td style="text-align:right;">
-673387467221
</td>
<td style="text-align:right;">
-664880164
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-24
</td>
<td style="text-align:right;">
8217942926
</td>
<td style="text-align:right;">
37130736106
</td>
<td style="text-align:right;">
5857701472
</td>
<td style="text-align:right;">
39276732227
</td>
<td style="text-align:right;">
-10740427138
</td>
<td style="text-align:right;">
-333936377093
</td>
<td style="text-align:right;">
-1206364609
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-25
</td>
<td style="text-align:right;">
7242347148
</td>
<td style="text-align:right;">
57539536462
</td>
<td style="text-align:right;">
4568079831
</td>
<td style="text-align:right;">
40583987397
</td>
<td style="text-align:right;">
-18377163802
</td>
<td style="text-align:right;">
-282195052525
</td>
<td style="text-align:right;">
-1314339238
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-26
</td>
<td style="text-align:right;">
1631567949
</td>
<td style="text-align:right;">
36681096085
</td>
<td style="text-align:right;">
1121378276
</td>
<td style="text-align:right;">
-2694080144
</td>
<td style="text-align:right;">
-60644185117
</td>
<td style="text-align:right;">
-198965081888
</td>
<td style="text-align:right;">
-843128099
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-27
</td>
<td style="text-align:right;">
1209476862
</td>
<td style="text-align:right;">
-8930444074
</td>
<td style="text-align:right;">
3405336890
</td>
<td style="text-align:right;">
-2060735139
</td>
<td style="text-align:right;">
-51783244894
</td>
<td style="text-align:right;">
-209186167980
</td>
<td style="text-align:right;">
-365874316
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-28
</td>
<td style="text-align:right;">
-4518925715
</td>
<td style="text-align:right;">
-2435870916
</td>
<td style="text-align:right;">
7007395646
</td>
<td style="text-align:right;">
-2525310574
</td>
<td style="text-align:right;">
-30952324005
</td>
<td style="text-align:right;">
-408114545906
</td>
<td style="text-align:right;">
-609942826
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-29
</td>
<td style="text-align:right;">
1484077336
</td>
<td style="text-align:right;">
-2058725307
</td>
<td style="text-align:right;">
3515115140
</td>
<td style="text-align:right;">
-5164411612
</td>
<td style="text-align:right;">
-42307838881
</td>
<td style="text-align:right;">
-287356849930
</td>
<td style="text-align:right;">
-912778912
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-30
</td>
<td style="text-align:right;">
4368619286
</td>
<td style="text-align:right;">
15502685767
</td>
<td style="text-align:right;">
-1093856576
</td>
<td style="text-align:right;">
-4703700433
</td>
<td style="text-align:right;">
-60704125766
</td>
<td style="text-align:right;">
-260908005179
</td>
<td style="text-align:right;">
-383882500
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-31
</td>
<td style="text-align:right;">
7907484382
</td>
<td style="text-align:right;">
12534261776
</td>
<td style="text-align:right;">
-2452294215
</td>
<td style="text-align:right;">
-8563910085
</td>
<td style="text-align:right;">
-75125518110
</td>
<td style="text-align:right;">
-289811675340
</td>
<td style="text-align:right;">
-918369787
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-32
</td>
<td style="text-align:right;">
10026207227
</td>
<td style="text-align:right;">
8766901671
</td>
<td style="text-align:right;">
1953315759
</td>
<td style="text-align:right;">
-7520737233
</td>
<td style="text-align:right;">
-55168263224
</td>
<td style="text-align:right;">
-890597864901
</td>
<td style="text-align:right;">
-314644061
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-33
</td>
<td style="text-align:right;">
1320088121
</td>
<td style="text-align:right;">
-19794580878
</td>
<td style="text-align:right;">
-541809575
</td>
<td style="text-align:right;">
-6750277825
</td>
<td style="text-align:right;">
-52231898984
</td>
<td style="text-align:right;">
-770337735939
</td>
<td style="text-align:right;">
-182991027
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-34
</td>
<td style="text-align:right;">
4019534162
</td>
<td style="text-align:right;">
-21662185450
</td>
<td style="text-align:right;">
1428924068
</td>
<td style="text-align:right;">
-7578531520
</td>
<td style="text-align:right;">
-47605477000
</td>
<td style="text-align:right;">
-741976996697
</td>
<td style="text-align:right;">
-550044215
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-35
</td>
<td style="text-align:right;">
-11674356179
</td>
<td style="text-align:right;">
-75718957578
</td>
<td style="text-align:right;">
972005613
</td>
<td style="text-align:right;">
-6094793780
</td>
<td style="text-align:right;">
-57165852971
</td>
<td style="text-align:right;">
-822161480737
</td>
<td style="text-align:right;">
-503488121
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-36
</td>
<td style="text-align:right;">
9892941550
</td>
<td style="text-align:right;">
46138316334
</td>
<td style="text-align:right;">
4954019648
</td>
<td style="text-align:right;">
5560697649
</td>
<td style="text-align:right;">
-23867649475
</td>
<td style="text-align:right;">
-71475573825
</td>
<td style="text-align:right;">
-1119704368
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-37
</td>
<td style="text-align:right;">
14414593887
</td>
<td style="text-align:right;">
78736973172
</td>
<td style="text-align:right;">
-257560661
</td>
<td style="text-align:right;">
-2341634721
</td>
<td style="text-align:right;">
-51914744178
</td>
<td style="text-align:right;">
-35405004462
</td>
<td style="text-align:right;">
-687055540
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-38
</td>
<td style="text-align:right;">
7716702900
</td>
<td style="text-align:right;">
3492507671
</td>
<td style="text-align:right;">
-2307137058
</td>
<td style="text-align:right;">
-4331637640
</td>
<td style="text-align:right;">
-67900220172
</td>
<td style="text-align:right;">
230840109765
</td>
<td style="text-align:right;">
-408157136
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-39
</td>
<td style="text-align:right;">
-583337199
</td>
<td style="text-align:right;">
-9483365742
</td>
<td style="text-align:right;">
706088109
</td>
<td style="text-align:right;">
-2757559248
</td>
<td style="text-align:right;">
-56366820622
</td>
<td style="text-align:right;">
-263196399957
</td>
<td style="text-align:right;">
-227087862
</td>
</tr>
<tr>
<td style="text-align:left;">
MS51-40
</td>
<td style="text-align:right;">
11097184979
</td>
<td style="text-align:right;">
24818171463
</td>
<td style="text-align:right;">
3186690665
</td>
<td style="text-align:right;">
-4462779576
</td>
<td style="text-align:right;">
-50563367720
</td>
<td style="text-align:right;">
145901488878
</td>
<td style="text-align:right;">
-319389629
</td>
</tr>
</tbody>
</table>
We also need the Metadata as we will need to know which conditions to
plot for together. If you have further information such as replicates or
patient ID, we can use this for the colour of the plotted samples per
condition as in the superplots style as described in by Lord et al <span class="citation">(Lord et al. 2020)</span>.<br><br><div class="vscroll-plot">
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metabolite_list</span> <span class="op">&lt;-</span> <span class="va">MCA_res</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="va">RG2_Significant</span>, <span class="st">"Opposite"</span><span class="op">)</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">Metabolite</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="../../reference/viz_superplot.html">viz_superplot</a></span><span class="op">(</span>data <span class="op">=</span><span class="va">Input_Superplot</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/all_of.html" class="external-link">any_of</a></span><span class="op">(</span><span class="va">metabolite_list</span><span class="op">)</span><span class="op">)</span>,<span class="co">#We just plot selected metabolites</span></span>
<span>metadata_sample <span class="op">=</span><span class="va">MetaData_Sample</span>,</span>
<span>metadata_info <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>Conditions<span class="op">=</span><span class="st">"Conditions"</span>, Superplot <span class="op">=</span> <span class="st">"Biological_Replicates"</span><span class="op">)</span>,</span>
<span>plot_type <span class="op">=</span> <span class="st">"Bar"</span>, <span class="co">#Bar, Box, Violin</span></span>
<span>plot_conditions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HK2"</span>, <span class="st">"786-O"</span>, <span class="st">"786-M1A"</span>, <span class="st">"786-M2A"</span>, <span class="st">"OSRC2"</span>, <span class="st">"OSLM1B"</span>, <span class="st">"RFX631"</span><span class="op">)</span>,<span class="co">#sets the order in which the samples should be plotted</span></span>
<span>stat_comparison <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">4</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="co">#Stat comparisons to be included on the plot</span></span>
<span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span>
<span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span></code></pre></div>
<p><img src="core-metabolomics_files/figure-html/viz-superplot-2-1.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span>
<span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/viz-superplot-2-2.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"></p>
</div>
<br><br><br>
Now, if we for instance prefer boxplots over bargraphs we can simply
change the parameter <code>plot_type</code>:
<div class="vscroll-plot">
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../../reference/viz_superplot.html">viz_superplot</a></span><span class="op">(</span>data <span class="op">=</span><span class="va">Input_Superplot</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">)</span><span class="op">]</span>,<span class="co">#We just plot six metabolites</span></span>
<span>metadata_sample <span class="op">=</span><span class="va">MetaData_Sample</span>,</span>
<span>metadata_info <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>Conditions<span class="op">=</span><span class="st">"Conditions"</span>, Superplot <span class="op">=</span> <span class="st">"Biological_Replicates"</span><span class="op">)</span>,</span>
<span>plot_type <span class="op">=</span> <span class="st">"Box"</span>, <span class="co">#Bar, Box, Violin</span></span>
<span>plot_conditions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HK2"</span>, <span class="st">"786-O"</span>, <span class="st">"786-M1A"</span>, <span class="st">"786-M2A"</span>, <span class="st">"OSRC2"</span>, <span class="st">"OSLM1B"</span>, <span class="st">"RFX631"</span><span class="op">)</span>,<span class="co">#sets the order in which the samples should be plotted</span></span>
<span>stat_comparison <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">4</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="co">#Stat comparisons to be included on the plot</span></span>
<span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span>
<span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span></code></pre></div>
<p><img src="core-metabolomics_files/figure-html/viz-superplot-3-1.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span>
<span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/viz-superplot-3-2.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span>
<span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/viz-superplot-3-3.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span>
<span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/viz-superplot-3-4.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span>
<span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/viz-superplot-3-5.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span>
<span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/viz-superplot-3-6.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"></p>
</div>
<br><br><br>
We can also change it to violin plots:
<div class="vscroll-plot">
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../../reference/viz_superplot.html">viz_superplot</a></span><span class="op">(</span>data <span class="op">=</span><span class="va">Input_Superplot</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">)</span><span class="op">]</span>,<span class="co">#We just plot six metabolites</span></span>
<span>metadata_sample <span class="op">=</span><span class="va">MetaData_Sample</span>,</span>
<span>metadata_info <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>Conditions<span class="op">=</span><span class="st">"Conditions"</span>, Superplot <span class="op">=</span> <span class="st">"Biological_Replicates"</span><span class="op">)</span>,</span>
<span>plot_type <span class="op">=</span> <span class="st">"Violin"</span>, <span class="co">#Bar, Box, Violin</span></span>
<span>plot_conditions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HK2"</span>, <span class="st">"786-O"</span>, <span class="st">"786-M1A"</span>, <span class="st">"786-M2A"</span>, <span class="st">"OSRC2"</span>, <span class="st">"OSLM1B"</span>, <span class="st">"RFX631"</span><span class="op">)</span>,<span class="co">#sets the order in which the samples should be plotted</span></span>
<span>stat_comparison <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">4</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="co">#Stat comparisons to be included on the plot</span></span>
<span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span>
<span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span></code></pre></div>
<p><img src="core-metabolomics_files/figure-html/viz-superplot-4-1.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span>
<span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/viz-superplot-4-2.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span>
<span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/viz-superplot-4-3.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span>
<span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/viz-superplot-4-4.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span>
<span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/viz-superplot-4-5.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span>
<span><span class="co">#&gt; Ignoring unknown labels:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> <span style="color: #00BB00;">fill</span> : <span style="color: #0000BB;">"Biological_Replicates"</span></span></span></code></pre>
<p><img src="core-metabolomics_files/figure-html/viz-superplot-4-6.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"></p>
</div>
<p><br><br><br></p>
</div>
<div class="section level4">
<h4 id="volcano-plot">Volcano plot<a class="anchor" aria-label="anchor" href="#volcano-plot"></a>
</h4>
In general,we have three different <code>Plot_Settings</code>, which
will also be used for other plot types such as lollipop graphs.<br><code>1.</code> <code>"Standard"</code> is the standard version of the
plot, with one dataset being plotted.<br><code>2.</code> <code>"Conditions"</code> here two or more datasets will
be plotted together. How datasets can be plotted together depends on the
plot type.<br><code>3.</code> <code>"PEA"</code> stands for Pathway Enrichment
Analysis, and is used if the results of an GSE analysis should be
plotted as here the figure legends will be adapted. You can find an
example for this in the vignette
<code>Standard Metabolomics</code><br><br>
Here we will look at all the different options we have to display our
results from the different analysis, which will help us to interpret our
results as this can be sometimes difficult to do from the many data
tables.<br>
Just a quick reminder, how the input data look like:<br>
1. Results of Differential Metabolite Analysis (dma): Log2(Distance) and
stats:<br><table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Input_data for <code><a href="../../reference/viz_volcano.html">viz_volcano()</a></code> are for example differential
analysis results from <code><a href="../../reference/dma.html">dma()</a></code>.
</caption>
<thead><tr>
<th style="text-align:left;">
Metabolite
</th>
<th style="text-align:right;">
Log2(Distance)
</th>
<th style="text-align:right;">
p.adj
</th>
<th style="text-align:right;">
t.val
</th>
<th style="text-align:left;">
Mean_786-M1A
</th>
<th style="text-align:left;">
core_786-M1A
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
2-hydroxyglutarate
</td>
<td style="text-align:right;">
-31.52594
</td>
<td style="text-align:right;">
0.3447870
</td>
<td style="text-align:right;">
-3092107029
</td>
<td style="text-align:left;">
3092107028.86536
</td>
<td style="text-align:left;">
Released
</td>
</tr>
<tr>
<td style="text-align:left;">
2-ketoglutarate
</td>
<td style="text-align:right;">
-34.39775
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
-22633671598
</td>
<td style="text-align:left;">
22633671597.7763
</td>
<td style="text-align:left;">
Released
</td>
</tr>
<tr>
<td style="text-align:left;">
3-Dehydro-L-threonate
</td>
<td style="text-align:right;">
30.24765
</td>
<td style="text-align:right;">
0.9999999
</td>
<td style="text-align:right;">
1274819834
</td>
<td style="text-align:left;">
-1274819834.18323
</td>
<td style="text-align:left;">
Consumed
</td>
</tr>
<tr>
<td style="text-align:left;">
acetylcarnitine
</td>
<td style="text-align:right;">
38.38705
</td>
<td style="text-align:right;">
0.0653736
</td>
<td style="text-align:right;">
359462986907
</td>
<td style="text-align:left;">
-359462986906.515
</td>
<td style="text-align:left;">
Consumed
</td>
</tr>
<tr>
<td style="text-align:left;">
acetylcholine
</td>
<td style="text-align:right;">
30.08675
</td>
<td style="text-align:right;">
0.0004996
</td>
<td style="text-align:right;">
1140284111
</td>
<td style="text-align:left;">
-1140284111.31361
</td>
<td style="text-align:left;">
Consumed
</td>
</tr>
<tr>
<td style="text-align:left;">
acetylornithine
</td>
<td style="text-align:right;">
-32.65274
</td>
<td style="text-align:right;">
0.0012793
</td>
<td style="text-align:right;">
-6752323351
</td>
<td style="text-align:left;">
6752323351.13854
</td>
<td style="text-align:left;">
Released
</td>
</tr>
<tr>
<td style="text-align:left;">
aconitate
</td>
<td style="text-align:right;">
32.58145
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
6426780519
</td>
<td style="text-align:left;">
-6426780518.51619
</td>
<td style="text-align:left;">
Consumed
</td>
</tr>
</tbody>
</table>
<div class="section level5">
<h5 id="standard">
<strong>Standard</strong><a class="anchor" aria-label="anchor" href="#standard"></a>
</h5>
<p>Here we will first look into the results from the differential
analysis (see section <code>dma</code> above) for the comparison of
<code>HK2_vs_786M1A</code>:</p>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run with default parameter --&gt; only need to provide Input_data and the title we like</span></span>
<span><span class="fu"><a href="../../reference/viz_volcano.html">viz_volcano</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">DMA_HK2_vs_786M1A</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Metabolite"</span><span class="op">)</span>,</span>
<span>x<span class="op">=</span> <span class="st">"Log2(Distance)"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="core-metabolomics_files/figure-html/viz-volcano-2-1.png" class="r-plt" alt="Figure: Standard figure displaying dma results." width="672"><p class="caption">
Figure: Standard figure displaying dma results.
</p>
</div>
<p><br>
If you seek to plot the metabolite names you can change the paramter
<code>select_label</code> from its default
(<code>select_label=""</code>) to NULL and the metabolite names will be
plotted randomly.</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run with default parameter --&gt; only need to provide Input_data and the title we like</span></span>
<span><span class="fu"><a href="../../reference/viz_volcano.html">viz_volcano</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">DMA_HK2_vs_786M1A</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Metabolite"</span><span class="op">)</span>,</span>
<span>x<span class="op">=</span> <span class="st">"Log2(Distance)"</span>,</span>
<span>select_label <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="core-metabolomics_files/figure-html/viz-volcano-3-1.png" class="r-plt" alt="Figure: Standard figure displaying dma results." width="672"><p class="caption">
Figure: Standard figure displaying dma results.
</p>
</div>
<p><br>
With the parameter <code>select_label</code> you can also pass a vector
with Metabolite names that should be labeled:</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run with default parameter --&gt; only need to provide Input_data and the title we like</span></span>
<span><span class="fu"><a href="../../reference/viz_volcano.html">viz_volcano</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">DMA_HK2_vs_786M1A</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Metabolite"</span><span class="op">)</span>,</span>
<span>x<span class="op">=</span> <span class="st">"Log2(Distance)"</span>,</span>
<span>select_label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"histidine"</span>, <span class="st">"phenylalanine"</span>, <span class="st">"lactate"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="core-metabolomics_files/figure-html/viz-volcano-4-1.png" class="r-plt" alt="Figure: Standard figure displaying dma results." width="672"><p class="caption">
Figure: Standard figure displaying dma results.
</p>
</div>
<p><br>
As explained above, when analyzing core data it is important to take
into account if a metabolite is consumed or released. we can use this
information to colour code and or shape the metabolites on the
plot.<br>
For this we need to add this information into the Metadata_Metabolite
file:</p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># colour for consumption and release: For this we need to add this information into the Metadata_Metabolite file</span></span>
<span><span class="va">MetaData_Metab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">MappingInfo</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span><span class="st">"Metabolite"</span><span class="op">)</span>, <span class="va">DMA_HK2_vs_786M1A</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">6</span>,<span class="fl">8</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span><span class="op">]</span>, by<span class="op">=</span><span class="st">"Metabolite"</span>, all.y<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Metabolite"</span><span class="op">)</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Metadata table including additional information about the Metabolites.
</caption>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
HMDB
</th>
<th style="text-align:left;">
KEGG.ID
</th>
<th style="text-align:left;">
KEGGCompound
</th>
<th style="text-align:left;">
Pathway
</th>
<th style="text-align:left;">
core_786-M1A
</th>
<th style="text-align:left;">
core_HK2
</th>
<th style="text-align:left;">
core_specific
</th>
<th style="text-align:left;">
core
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
3-Dehydro-L-threonate
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Amino acid metabolism
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Consumed
</td>
</tr>
<tr>
<td style="text-align:left;">
acetylcarnitine
</td>
<td style="text-align:left;">
HMDB0000201
</td>
<td style="text-align:left;">
C02571
</td>
<td style="text-align:left;">
O-Acetylcarnitine
</td>
<td style="text-align:left;">
Fatty acyl carnitines
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Consumed in 786-M1A and Released HK2
</td>
<td style="text-align:left;">
Released / Consumed
</td>
</tr>
<tr>
<td style="text-align:left;">
acetylornithine
</td>
<td style="text-align:left;">
HMDB0003357
</td>
<td style="text-align:left;">
C00437
</td>
<td style="text-align:left;">
N-Acetylornithine
</td>
<td style="text-align:left;">
Arginine and proline metabolism
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Released in 786-M1A and Consumed HK2
</td>
<td style="text-align:left;">
Released / Consumed
</td>
</tr>
<tr>
<td style="text-align:left;">
glutamate
</td>
<td style="text-align:left;">
HMDB0000148
</td>
<td style="text-align:left;">
C00025
</td>
<td style="text-align:left;">
L-Glutamate
</td>
<td style="text-align:left;">
Amino acid metabolism
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Released in 786-M1A and Consumed HK2
</td>
<td style="text-align:left;">
Released / Consumed
</td>
</tr>
<tr>
<td style="text-align:left;">
glutamine
</td>
<td style="text-align:left;">
HMDB0000641
</td>
<td style="text-align:left;">
C00064
</td>
<td style="text-align:left;">
L-Glutamine
</td>
<td style="text-align:left;">
Amino acid metabolism
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Released in 786-M1A and Consumed HK2
</td>
<td style="text-align:left;">
Released / Consumed
</td>
</tr>
<tr>
<td style="text-align:left;">
glycerylphosphorylcholine
</td>
<td style="text-align:left;">
HMDB0252858
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Not assigned
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Consumed
</td>
</tr>
</tbody>
</table>
<p><br>
Now we can make the different plots:</p>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Now we need to add our Plot_SettingsFile and the Plot_metadata_info:</span></span>
<span><span class="fu"><a href="../../reference/viz_volcano.html">viz_volcano</a></span><span class="op">(</span>plot_types<span class="op">=</span><span class="st">"Standard"</span>,</span>
<span>metadata_info<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"core_specific"</span><span class="op">)</span>,</span>
<span>metadata_feature<span class="op">=</span> <span class="va">MetaData_Metab</span>,</span>
<span>data<span class="op">=</span><span class="va">DMA_HK2_vs_786M1A</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Metabolite"</span><span class="op">)</span>,</span>
<span>x<span class="op">=</span> <span class="st">"Log2(Distance)"</span>,</span>
<span>plot_name<span class="op">=</span> <span class="st">"786M1A versus HK2"</span>,</span>
<span>subtitle<span class="op">=</span> <span class="st">"Results of dma. Colour coded for consumption/release"</span> <span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="core-metabolomics_files/figure-html/viz-volcano-5-1.png" class="r-plt" alt="Figure: Standard figure displaying dma results." width="672"><p class="caption">
Figure: Standard figure displaying dma results.
</p>
</div>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># If we want to use the shape instead of the colour for the cluster info, we can just change our Plot_metadata_info</span></span>
<span><span class="fu"><a href="../../reference/viz_volcano.html">viz_volcano</a></span><span class="op">(</span>plot_types<span class="op">=</span><span class="st">"Standard"</span>,</span>
<span>metadata_info<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>shape<span class="op">=</span><span class="st">"core_specific"</span><span class="op">)</span>,</span>
<span>metadata_feature<span class="op">=</span> <span class="va">MetaData_Metab</span>,</span>
<span>data<span class="op">=</span><span class="va">DMA_HK2_vs_786M1A</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Metabolite"</span><span class="op">)</span>,</span>
<span>x<span class="op">=</span> <span class="st">"Log2(Distance)"</span>,</span>
<span>plot_name<span class="op">=</span> <span class="st">"786M1A versus HK2"</span>,</span>
<span>subtitle<span class="op">=</span> <span class="st">"Results of dma. Shape for consumption/release, color for significance."</span> <span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="core-metabolomics_files/figure-html/viz-volcano-5-2.png" class="r-plt" alt="Figure: Standard figure displaying dma results." width="672"><p class="caption">
Figure: Standard figure displaying dma results.
</p>
</div>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Of course, we can also adapt both, color and shape for the same parameter:</span></span>
<span><span class="fu"><a href="../../reference/viz_volcano.html">viz_volcano</a></span><span class="op">(</span>plot_types<span class="op">=</span><span class="st">"Standard"</span>,</span>
<span>metadata_info<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>shape<span class="op">=</span><span class="st">"core_specific"</span>, color<span class="op">=</span><span class="st">"core_specific"</span><span class="op">)</span>,</span>
<span>metadata_feature<span class="op">=</span> <span class="va">MetaData_Metab</span>,</span>
<span>data<span class="op">=</span><span class="va">DMA_HK2_vs_786M1A</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Metabolite"</span><span class="op">)</span>,</span>
<span>x<span class="op">=</span> <span class="st">"Log2(Distance)"</span>,</span>
<span>plot_name<span class="op">=</span> <span class="st">"786M1A versus HK2"</span>,</span>
<span>subtitle<span class="op">=</span> <span class="st">"Results of dma. Shape and color for consumption/release."</span> <span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="core-metabolomics_files/figure-html/viz-volcano-5-3.png" class="r-plt" alt="Figure: Standard figure displaying dma results." width="672"><p class="caption">
Figure: Standard figure displaying dma results.
</p>
</div>
<p><br>
Of course, here we may also want an individual plot for each of the
consumption/release metabolites.</p>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># individual plot for each metabolite behaviour:</span></span>
<span><span class="fu"><a href="../../reference/viz_volcano.html">viz_volcano</a></span><span class="op">(</span>plot_types<span class="op">=</span><span class="st">"Standard"</span>,</span>
<span>metadata_info<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>individual<span class="op">=</span><span class="st">"core"</span>, shape<span class="op">=</span><span class="st">"core_specific"</span><span class="op">)</span>,</span>
<span>metadata_feature<span class="op">=</span> <span class="va">MetaData_Metab</span>,</span>
<span>data<span class="op">=</span><span class="va">DMA_HK2_vs_786M1A</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Metabolite"</span><span class="op">)</span>,</span>
<span>x<span class="op">=</span> <span class="st">"Log2(Distance)"</span>,</span>
<span>plot_name<span class="op">=</span> <span class="st">"786M1A versus HK2"</span>,</span>
<span>subtitle<span class="op">=</span> <span class="st">"Results of dma."</span> <span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="core-metabolomics_files/figure-html/viz-volcano-6-1.png" class="r-plt" alt="Figure: Standard figure displaying dma results." width="672"><p class="caption">
Figure: Standard figure displaying dma results.
</p>
</div>
<pre><code><span><span class="co">#&gt; Warning: Removed 2 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_vline()`).</span></span>
<span><span class="co">#&gt; Removed 2 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_vline()`).</span></span></code></pre>
<div class="figure" style="text-align: left">
<img src="core-metabolomics_files/figure-html/viz-volcano-6-2.png" class="r-plt" alt="Figure: Standard figure displaying dma results." width="672"><p class="caption">
Figure: Standard figure displaying dma results.
</p>
</div>
<pre><code><span><span class="co">#&gt; Warning: Removed 2 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_vline()`).</span></span>
<span><span class="co">#&gt; Removed 2 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_vline()`).</span></span></code></pre>
<div class="figure" style="text-align: left">
<img src="core-metabolomics_files/figure-html/viz-volcano-6-3.png" class="r-plt" alt="Figure: Standard figure displaying dma results." width="672"><p class="caption">
Figure: Standard figure displaying dma results.
</p>
</div>
<p><br>
Given that we also know, which metabolic pathway the metabolites
correspond to, we can add this information into the plot. This is also a
good example to showcase the flexibility of the visualisation function:
Either you use the parameter
<code>Plot_SettingsFile= MetaData_Metab</code> as above, but as we have
the column “Pathway” also in our Input_data you can also pass
<code>Plot_SettingsFile= DMA_HK2_vs_786M1A</code> or simply use the
default <code>Plot_SettingsFile=NULL</code>, in which case the
<code>Plot_metadata_info</code> information (here <code>color</code>)
will be used from Input_data.<br></p>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Now we can use color for the pathways and shape for the metabolite clusters:</span></span>
<span><span class="fu"><a href="../../reference/viz_volcano.html">viz_volcano</a></span><span class="op">(</span>plot_types<span class="op">=</span><span class="st">"Standard"</span>,</span>
<span>metadata_info<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>individual<span class="op">=</span><span class="st">"core"</span>, shape<span class="op">=</span><span class="st">"core_specific"</span>, color<span class="op">=</span><span class="st">"Pathway"</span><span class="op">)</span>,</span>
<span>metadata_feature<span class="op">=</span> <span class="va">MetaData_Metab</span>,</span>
<span>data<span class="op">=</span><span class="va">DMA_HK2_vs_786M1A</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Metabolite"</span><span class="op">)</span>,</span>
<span>x<span class="op">=</span> <span class="st">"Log2(Distance)"</span>,</span>
<span>plot_name<span class="op">=</span> <span class="st">"786M1A versus HK2"</span>,</span>
<span>subtitle<span class="op">=</span> <span class="st">"Results of dma."</span> <span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="core-metabolomics_files/figure-html/viz-volcano-7-1.png" class="r-plt" alt="Figure: Standard figure displaying dma results colour coded for metabolic pathways and shaped for metabolic clusters." width="672"><p class="caption">
Figure: Standard figure displaying dma results colour coded for
metabolic pathways and shaped for metabolic clusters.
</p>
</div>
<pre><code><span><span class="co">#&gt; Warning: Removed 2 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_vline()`).</span></span>
<span><span class="co">#&gt; Removed 2 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_vline()`).</span></span></code></pre>
<div class="figure" style="text-align: left">
<img src="core-metabolomics_files/figure-html/viz-volcano-7-2.png" class="r-plt" alt="Figure: Standard figure displaying dma results colour coded for metabolic pathways and shaped for metabolic clusters." width="672"><p class="caption">
Figure: Standard figure displaying dma results colour coded for
metabolic pathways and shaped for metabolic clusters.
</p>
</div>
<pre><code><span><span class="co">#&gt; Warning: Removed 2 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_vline()`).</span></span>
<span><span class="co">#&gt; Removed 2 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_vline()`).</span></span></code></pre>
<div class="figure" style="text-align: left">
<img src="core-metabolomics_files/figure-html/viz-volcano-7-3.png" class="r-plt" alt="Figure: Standard figure displaying dma results colour coded for metabolic pathways and shaped for metabolic clusters." width="672"><p class="caption">
Figure: Standard figure displaying dma results colour coded for
metabolic pathways and shaped for metabolic clusters.
</p>
</div>
</div>
<div class="section level5">
<h5 id="comparison">
<strong>Comparison</strong><a class="anchor" aria-label="anchor" href="#comparison"></a>
</h5>
<p>The parameter <code>Plot_Settings="Compare"</code> is helpful if you
have performed multiple comparisons and seek to compare two of them in
one plot:<br></p>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Make the plot</span></span>
<span><span class="fu"><a href="../../reference/viz_volcano.html">viz_volcano</a></span><span class="op">(</span>plot_types<span class="op">=</span><span class="st">"Compare"</span>,</span>
<span>data<span class="op">=</span><span class="va">DMA_HK2_vs_786M1A</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Metabolite"</span><span class="op">)</span>,</span>
<span>data2<span class="op">=</span> <span class="va">DMA_Annova</span><span class="op">[[</span><span class="st">"dma"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"HK2_vs_786-O"</span><span class="op">]</span><span class="op">]</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Metabolite"</span><span class="op">)</span>,</span>
<span>name_comparison<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>data<span class="op">=</span><span class="st">"HK2_vs_786-M1A"</span>, data2<span class="op">=</span> <span class="st">"HK2_vs_786-O"</span><span class="op">)</span>,</span>
<span>x<span class="op">=</span> <span class="st">"Log2(Distance)"</span>,</span>
<span>plot_name<span class="op">=</span> <span class="st">"786M1A vs HK2 compared to 7860 vs HK2"</span>,</span>
<span>subtitle<span class="op">=</span> <span class="st">"Results of dma"</span> <span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="core-metabolomics_files/figure-html/viz-volcano-8-1.png" class="r-plt" alt="Figure: Comparison." width="672"><p class="caption">
Figure: Comparison.
</p>
</div>
<p><br>
Of course you have option to use shape or color to further customize
your graph as well as make individual plots:<br></p>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Make the plot</span></span>
<span><span class="fu"><a href="../../reference/viz_volcano.html">viz_volcano</a></span><span class="op">(</span>plot_types<span class="op">=</span><span class="st">"Compare"</span>,</span>
<span>metadata_info<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"Pathway"</span><span class="op">)</span>,</span>
<span>metadata_feature<span class="op">=</span> <span class="va">MetaData_Metab</span>,</span>
<span>data<span class="op">=</span><span class="va">DMA_HK2_vs_786M1A</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Metabolite"</span><span class="op">)</span>,</span>
<span>data2<span class="op">=</span> <span class="va">DMA_Annova</span><span class="op">[[</span><span class="st">"dma"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"HK2_vs_786-O"</span><span class="op">]</span><span class="op">]</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Metabolite"</span><span class="op">)</span>,</span>
<span>name_comparison<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>data<span class="op">=</span><span class="st">"HK2_vs_786-M1A"</span>, data2<span class="op">=</span> <span class="st">"HK2_vs_786-O"</span><span class="op">)</span>,</span>
<span>x<span class="op">=</span> <span class="st">"Log2(Distance)"</span>,</span>
<span>plot_name<span class="op">=</span> <span class="st">"786M1A vs HK2 compared to 7860 vs HK2"</span>,</span>
<span>subtitle<span class="op">=</span> <span class="st">"Results of dma"</span> <span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="core-metabolomics_files/figure-html/viz-volcano-9-1.png" class="r-plt" alt="Figure: Comparison." width="672"><p class="caption">
Figure: Comparison.
</p>
</div>
Now we do individual plots again:
<div class="vscroll-plot">
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../../reference/viz_volcano.html">viz_volcano</a></span><span class="op">(</span>plot_types<span class="op">=</span><span class="st">"Compare"</span>,</span>
<span>metadata_info<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>individual<span class="op">=</span><span class="st">"Pathway"</span><span class="op">)</span>,</span>
<span>metadata_feature<span class="op">=</span> <span class="va">MetaData_Metab</span>,</span>
<span>data<span class="op">=</span><span class="va">DMA_HK2_vs_786M1A</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Metabolite"</span><span class="op">)</span>,</span>
<span>data2<span class="op">=</span> <span class="va">DMA_Annova</span><span class="op">[[</span><span class="st">"dma"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"HK2_vs_786-O"</span><span class="op">]</span><span class="op">]</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Metabolite"</span><span class="op">)</span>,</span>
<span>name_comparison<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>data<span class="op">=</span><span class="st">"HK2_vs_786-M1A"</span>, data2<span class="op">=</span> <span class="st">"HK2_vs_786-O"</span><span class="op">)</span>,</span>
<span>x<span class="op">=</span> <span class="st">"Log2(Distance)"</span>,</span>
<span>plot_name<span class="op">=</span> <span class="st">"786M1A vs HK2 compared to 7860 vs HK2"</span>,</span>
<span>subtitle<span class="op">=</span> <span class="st">"Results of dma"</span> <span class="op">)</span></span></code></pre></div>
<p><img src="core-metabolomics_files/figure-html/viz-volcano-10-1.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-10-2.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-10-3.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-10-4.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-10-5.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-10-6.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-10-7.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-10-8.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-10-9.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-10-10.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-10-11.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-10-12.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-10-13.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-10-14.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-10-15.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-10-16.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"></p>
</div>
<p><br><br></p>
</div>
<div class="section level5">
<h5 id="pathwayenrichmentanalysis">
<strong>PathwayEnrichmentAnalysis</strong><a class="anchor" aria-label="anchor" href="#pathwayenrichmentanalysis"></a>
</h5>
<p>If you have performed Pathway Enrichment Analysis (PEA) such as ORA
or GSEA, we can also plot the results and add the information into the
Figure legends.<br>
Here we can for example use the results of the ORA we have performed on
the differential expression results. Indeed for DMA_HK2_vs_786M1A we
have performed ORA on each cluster (consumed, released,
consumed/released). Here, I will plot the ORA results of the metabolites
that are released in both conditions, HK2 and 786-M1A.</p>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Prepare the Input:</span></span>
<span><span class="co"># 1. data=Pathway analysis input: Must have features as column names. Those feature names need to match features in the pathway analysis file metadata_feature.</span></span>
<span><span class="va">InputPEA</span> <span class="op">&lt;-</span> <span class="va">DMA_HK2_vs_786M1A</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">KEGGCompound</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"KEGGCompound"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. data2=Pathway analysis output: Must have same column names as metadata_feature for Pathway name</span></span>
<span><span class="va">InputPEA2</span> <span class="op">&lt;-</span> <span class="va">MC_ORA_HK2_vs_786M1A_Consumed</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span><span class="st">"term"</span><span class="op">=</span><span class="st">"ID"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 3. metadata_feature= Pathways used for pathway analysis: Must have same column names as metadata_feature for Pathway name and feature names need to match features in the data. PEA_Feature passes this column name!</span></span></code></pre></div>
<br><div class="vscroll-plot">
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../../reference/viz_volcano.html">viz_volcano</a></span><span class="op">(</span>plot_types<span class="op">=</span><span class="st">"PEA"</span>,</span>
<span>metadata_info<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>PEA_Pathway<span class="op">=</span><span class="st">"term"</span>,<span class="co"># Needs to be the same in both, metadata_feature and data2.</span></span>
<span>PEA_stat<span class="op">=</span><span class="st">"p.adjust"</span>,<span class="co">#Column data2</span></span>
<span>PEA_score<span class="op">=</span><span class="st">"GeneRatio"</span>,<span class="co">#Column data2</span></span>
<span>PEA_Feature<span class="op">=</span><span class="st">"Metabolite"</span><span class="op">)</span>,<span class="co"># Column metadata_feature (needs to be the same as row names in data)</span></span>
<span>metadata_feature<span class="op">=</span> <span class="va">KEGG_Pathways</span>,<span class="co">#Must be the pathways used for pathway analysis</span></span>
<span>data<span class="op">=</span> <span class="va">InputPEA</span>, <span class="co">#Must be the data you have used as an input for the pathway analysis</span></span>
<span>data2<span class="op">=</span> <span class="va">InputPEA2</span>, <span class="co">#Must be the results of the pathway analysis</span></span>
<span>x<span class="op">=</span> <span class="st">"Log2(Distance)"</span>,</span>
<span>plot_name<span class="op">=</span> <span class="st">"KEGG"</span>,</span>
<span>subtitle<span class="op">=</span> <span class="st">"PEA"</span> ,</span>
<span>select_label <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<p><img src="core-metabolomics_files/figure-html/viz-volcano-11-1.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-2.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-3.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-4.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-5.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-6.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-7.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-8.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-9.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-10.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-11.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-12.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-13.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-14.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-15.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-16.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-17.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-18.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-19.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-20.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-21.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-22.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-23.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-24.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-25.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-26.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-27.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-28.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-29.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-30.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-31.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-32.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-33.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-34.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-35.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-36.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-37.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-38.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-39.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-40.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-41.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-42.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-43.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-44.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-45.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-46.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-47.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-48.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-49.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-50.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-51.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-52.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-53.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-54.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-55.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-56.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-57.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-58.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-59.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-60.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-61.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-62.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-63.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-64.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-65.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-66.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-67.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-68.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-69.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-70.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-71.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-72.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-73.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-74.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-75.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-76.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-77.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-78.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-79.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-80.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-81.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-82.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-83.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-84.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-85.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-86.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-87.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-88.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-89.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-90.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-91.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-92.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-93.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-94.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-95.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-96.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-97.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-98.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-99.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-100.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-101.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-102.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-103.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-104.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-105.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-106.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-107.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-108.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-109.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-110.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-111.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-112.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-113.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-114.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-115.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-116.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-117.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-118.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-119.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-120.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-121.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-122.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-123.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-124.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-125.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-126.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-127.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-128.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-129.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-130.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-131.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-132.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-133.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-134.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-135.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-136.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-137.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-138.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-139.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-140.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-141.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-142.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-143.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-144.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-145.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-146.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-147.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-148.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-149.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-150.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-151.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-152.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-153.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-154.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-155.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-156.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-157.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-158.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-159.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-160.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-161.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-162.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-163.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-164.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-165.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-166.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-167.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-168.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-169.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-170.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-171.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-172.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-173.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-174.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-175.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-176.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-177.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-178.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-179.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-180.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-181.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-182.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-183.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-184.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-185.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="core-metabolomics_files/figure-html/viz-volcano-11-186.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"></p>
</div>
<br><br><div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<pre><code><span><span class="co">#&gt; R version 4.5.2 (2025-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.3 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Etc/UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] stringr_1.6.0      ggfortify_0.4.19   ggplot2_4.0.1      rlang_1.1.6        tibble_3.3.0       dplyr_1.1.4       </span></span>
<span><span class="co">#&gt; [7] magrittr_2.0.4     MetaProViz_3.99.24 BiocStyle_2.38.0  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] RColorBrewer_1.1-3          rstudioapi_0.17.1           jsonlite_2.0.0              ggbeeswarm_0.7.2           </span></span>
<span><span class="co">#&gt;   [5] farver_2.1.2                rmarkdown_2.30              fs_1.6.6                    ragg_1.5.0                 </span></span>
<span><span class="co">#&gt;   [9] vctrs_0.6.5                 memoise_2.0.1               rstatix_0.7.3               htmltools_0.5.8.1          </span></span>
<span><span class="co">#&gt;  [13] S4Arrays_1.10.0             progress_1.2.3              curl_7.0.0                  ComplexUpset_1.3.3         </span></span>
<span><span class="co">#&gt;  [17] decoupleR_2.16.0            broom_1.0.10                cellranger_1.1.0            SparseArray_1.10.1         </span></span>
<span><span class="co">#&gt;  [21] Formula_1.2-5               sass_0.4.10                 parallelly_1.45.1           bslib_0.9.0                </span></span>
<span><span class="co">#&gt;  [25] htmlwidgets_1.6.4           desc_1.4.3                  plyr_1.8.9                  httr2_1.2.1                </span></span>
<span><span class="co">#&gt;  [29] lubridate_1.9.4             cachem_1.1.0                igraph_2.2.1                lifecycle_1.0.4            </span></span>
<span><span class="co">#&gt;  [33] pkgconfig_2.0.3             Matrix_1.7-4                R6_2.6.1                    fastmap_1.2.0              </span></span>
<span><span class="co">#&gt;  [37] MatrixGenerics_1.22.0       digest_0.6.38               colorspace_2.1-2            patchwork_1.3.2            </span></span>
<span><span class="co">#&gt;  [41] S4Vectors_0.48.0            textshaping_1.0.4           GenomicRanges_1.62.0        RSQLite_2.4.4              </span></span>
<span><span class="co">#&gt;  [45] ggpubr_0.6.2                labeling_0.4.3              timechange_0.3.0            httr_1.4.7                 </span></span>
<span><span class="co">#&gt;  [49] abind_1.4-8                 compiler_4.5.2              bit64_4.6.0-1               withr_3.0.2                </span></span>
<span><span class="co">#&gt;  [53] S7_0.2.1                    backports_1.5.0             BiocParallel_1.44.0         carData_3.0-5              </span></span>
<span><span class="co">#&gt;  [57] DBI_1.2.3                   logger_0.4.1                OmnipathR_3.19.2            R.utils_2.13.0             </span></span>
<span><span class="co">#&gt;  [61] ggsignif_0.6.4              cosmosR_1.18.0              MASS_7.3-65                 rappdirs_0.3.3             </span></span>
<span><span class="co">#&gt;  [65] DelayedArray_0.36.0         sessioninfo_1.2.3           scatterplot3d_0.3-44        gtools_3.9.5               </span></span>
<span><span class="co">#&gt;  [69] tools_4.5.2                 vipor_0.4.7                 beeswarm_0.4.0              qcc_2.7                    </span></span>
<span><span class="co">#&gt;  [73] zip_2.3.3                   R.oo_1.27.1                 glue_1.8.0                  grid_4.5.2                 </span></span>
<span><span class="co">#&gt;  [77] checkmate_2.3.3             reshape2_1.4.5              generics_0.1.4              gtable_0.3.6               </span></span>
<span><span class="co">#&gt;  [81] tzdb_0.5.0                  R.methodsS3_1.8.2           tidyr_1.3.1                 hms_1.1.4                  </span></span>
<span><span class="co">#&gt;  [85] xml2_1.5.0                  car_3.1-3                   XVector_0.50.0              BiocGenerics_0.56.0        </span></span>
<span><span class="co">#&gt;  [89] ggrepel_0.9.6               pillar_1.11.1               vroom_1.6.6                 limma_3.66.0               </span></span>
<span><span class="co">#&gt;  [93] later_1.4.4                 splines_4.5.2               lattice_0.22-7              bit_4.6.0                  </span></span>
<span><span class="co">#&gt;  [97] tidyselect_1.2.1            knitr_1.50                  gridExtra_2.3               bookdown_0.45              </span></span>
<span><span class="co">#&gt; [101] IRanges_2.44.0              Seqinfo_1.0.0               SummarizedExperiment_1.40.0 svglite_2.2.2              </span></span>
<span><span class="co">#&gt; [105] stats4_4.5.2                xfun_0.54                   Biobase_2.70.0              statmod_1.5.1              </span></span>
<span><span class="co">#&gt; [109] factoextra_1.0.7            matrixStats_1.5.0           pheatmap_1.0.13             stringi_1.8.7              </span></span>
<span><span class="co">#&gt; [113] yaml_2.3.10                 kableExtra_1.4.0            evaluate_1.0.5              codetools_0.2-20           </span></span>
<span><span class="co">#&gt; [117] tcltk_4.5.2                 qvalue_2.42.0               hash_2.2.6.3                BiocManager_1.30.27        </span></span>
<span><span class="co">#&gt; [121] Polychrome_1.5.4            cli_3.6.5                   systemfonts_1.3.1           jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt; [125] EnhancedVolcano_1.13.2      Rcpp_1.1.0                  readxl_1.4.5                XML_3.99-0.20              </span></span>
<span><span class="co">#&gt; [129] parallel_4.5.2              pkgdown_2.2.0               readr_2.1.6                 blob_1.2.4                 </span></span>
<span><span class="co">#&gt; [133] prettyunits_1.2.0           viridisLite_0.4.2           scales_1.4.0                writexl_1.5.4              </span></span>
<span><span class="co">#&gt; [137] inflection_1.3.7            purrr_1.2.0                 crayon_1.5.3                rvest_1.0.5</span></span></code></pre>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-Badia-I-Mompel2022" class="csl-entry">
Badia-I-Mompel, Pau, Jesús Vélez Santiago, Jana Braunger, Celina Geiss,
Daniel Dimitrov, Sophia Müller-Dott, Petr Taus, et al. 2022.
<span>“decoupleR: Ensemble of Computational Methods to Infer Biological
Activities from Omics Data.”</span> <em>Bioinformatics Advances</em> 2
(1): vbac016. <a href="https://doi.org/10.1093/bioadv/vbac016" class="external-link">https://doi.org/10.1093/bioadv/vbac016</a>.
</div>
<div id="ref-Bijlsma2006" class="csl-entry">
Bijlsma, Sabina, Ivana Bobeldijk, Elwin R Verheij, Raymond Ramaker,
Sunil Kochhar, Ian A Macdonald, Ben van Ommen, and Age K Smilde. 2006.
<span>“Large-Scale Human Metabolomics Studies: A Strategy for Data
(Pre-) Processing and Validation.”</span> <em>Analytical Chemistry</em>
78 (2): 567–74. <a href="https://doi.org/10.1021/ac051495j" class="external-link">https://doi.org/10.1021/ac051495j</a>.
</div>
<div id="ref-Farr_Dimitrov2024" class="csl-entry">
Farr, Elias, Daniel Dimitrov, Christina Schmidt, Denes Turei, Sebastian
Lobentanzer, Aurelien Dugourd, and Julio Saez-Rodriguez. 2024.
<span>“MetalinksDB: A Flexible and Contextualizable Resource of
Metabolite-Protein Interactions.”</span> <em>Briefings in
Bioinformatics</em>, no. 4 (May). <a href="https://doi.org/10.1093/bib/bbae347" class="external-link">https://doi.org/10.1093/bib/bbae347</a>.
</div>
<div id="ref-Hotelling1931" class="csl-entry">
Hotelling, Harold. 1931. <span>“The Generalization of Student’s
Ratio.”</span> <em>The Annals of Mathematical Statistics</em> 2 (3):
360–78. <a href="https://doi.org/10.1214/aoms/1177732979" class="external-link">https://doi.org/10.1214/aoms/1177732979</a>.
</div>
<div id="ref-Kanehisa2000" class="csl-entry">
Kanehisa, M, and S Goto. 2000. <span>“KEGG: Kyoto Encyclopedia of Genes
and Genomes.”</span> <em>Nucleic Acids Research</em> 28 (1): 27–30. <a href="https://doi.org/10.1093/nar/28.1.27" class="external-link">https://doi.org/10.1093/nar/28.1.27</a>.
</div>
<div id="ref-Lord2020" class="csl-entry">
Lord, Samuel J, Katrina B Velle, R Dyche Mullins, and Lillian K
Fritz-Laylin. 2020. <span>“SuperPlots: Communicating Reproducibility and
Variability in Cell Biology.”</span> <em>The Journal of Cell
Biology</em>, no. 6 (June). <a href="https://doi.org/10.1083/jcb.202001064" class="external-link">https://doi.org/10.1083/jcb.202001064</a>.
</div>
<div id="ref-Wei2018" class="csl-entry">
Wei, Runmin, Jingye Wang, Mingming Su, Erik Jia, Shaoqiu Chen, Tianlu
Chen, and Yan Ni. 2018. <span>“Missing Value Imputation Approach for
Mass Spectrometry-Based Metabolomics Data.”</span> <em>Scientific
Reports</em> 8 (1): 663. <a href="https://doi.org/10.1038/s41598-017-19120-0" class="external-link">https://doi.org/10.1038/s41598-017-19120-0</a>.
</div>
<div id="ref-Wulff2018" class="csl-entry">
Wulff, Jacob E., and Matthew W. Mitchell. 2018. <span>“A Comparison of
Various Normalization Methods for LC/MS Metabolomics Data.”</span>
<em>Advances in Bioscience and Biotechnology</em> 09 (08): 339–51. <a href="https://doi.org/10.4236/abb.2018.98022" class="external-link">https://doi.org/10.4236/abb.2018.98022</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Christina Schmidt, Denes Turei, Dimitrios Prymidis, Macabe Daley, Jannik Franken, Julio Saez-Rodriguez, Christian Frezza.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
