<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Prior Knowledge - Access &amp; Integration • MetaProViz</title>
<script src="../../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../../deps/headroom-0.11.0/headroom.min.js"></script><script src="../../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../../deps/search-1.0.0/fuse.min.js"></script><script src="../../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../../pkgdown.js"></script><link href="../../extra.css" rel="stylesheet">
<meta property="og:title" content="Prior Knowledge - Access &amp; Integration">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../../index.html">MetaProViz</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">3.99.26</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true"><span class="fa fa-book-open"></span></button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-">
<li><a class="dropdown-item" href="../../articles/quick-start.html">Getting started</a></li>
    <li><a class="dropdown-item" href="../../articles/pkgdown/standard-metabolomics.html">Standard Metabolomics</a></li>
    <li><a class="dropdown-item" href="../../articles/pkgdown/core-metabolomics.html">CoRe Metabolomics</a></li>
    <li><a class="dropdown-item" href="../../articles/pkgdown/prior-knowledge.html">Prior Knowledge</a></li>
    <li><a class="dropdown-item" href="../../articles/pkgdown/sample-metadata.html">Sample Metadata Analysis</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../../reference/index.html"><span class="fa fa-solid fa-rocket"></span> Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/saezlab/MetaProViz"><span class="fa fa-brands fa-github"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://saezlab.org/"><span class="fa fas fa-university"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://frezza.cecad-labs.uni-koeln.de/home"><span class="fa fas fa-university"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">



<script src="prior-knowledge_files/kePrint-0.0.1/kePrint.js"></script><link href="prior-knowledge_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Prior Knowledge - Access &amp; Integration</h1>
                        <h4 data-toc-skip class="author">Christina
Schmidt</h4>
            <address class="author_afil">
      Heidelberg
University<br><h4 data-toc-skip class="author">Macabe
Daley</h4>
            <address class="author_afil">
      Heidelberg University<br><small class="dont-index">Source: <a href="https://github.com/saezlab/MetaProViz/blob/main/vignettes/pkgdown/prior-knowledge.Rmd" class="external-link"><code>vignettes/pkgdown/prior-knowledge.Rmd</code></a></small>
      <div class="d-none name"><code>prior-knowledge.Rmd</code></div>
    </address>
</address>
</div>

    
    
<style>
.vscroll-plot {
    width: 850px;
    height: 500px;
    overflow-y: scroll;
    overflow-x: hidden;
}
</style>
<div class="section level2">
<h2 id="section">
<img src="Hexagon_MetaProViz.png" align="right" width="200"><a class="anchor" aria-label="anchor" href="#section"></a>
</h2>
<p><br><span style="text-decoration:underline">In this tutorial we showcase how
to use <strong>MetaProViz</strong> prior knowledge</span>:<br>
- 1. <a href="#sect1">to understand detected metabolite IDs in measured
data.</a></p>
<p>- 2. <a href="#sect2">to access metabolite prior knowledge and
metabolite-gene prior knowledge networks.</a></p>
<p>- 3. <a href="#sect3">to link experimental data to prior knowledge -
the Do’s and Don’ts.</a></p>
<p>- 4. <a href="#sect4">to deal with many-to-many mapping in your
metabolite identifiers.</a></p>
<p>- 5. <a href="#sect5">to perform pathway enrichment
analysis.</a><br></p>
<p><br>
First if you have not done yet, install the required dependencies and
load the libraries:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Install Rtools if you haven’t done this yet, using the appropriate version (e.g.windows or macOS).</span></span>
<span><span class="co"># 2. Install the latest development version from GitHub using devtools</span></span>
<span><span class="co"># devtools::install_github("https://github.com/saezlab/MetaProViz")</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://saezlab.github.io/MetaProViz" class="external-link">MetaProViz</a></span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rlang.r-lib.org" class="external-link">rlang</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://purrr.tidyverse.org/" class="external-link">purrr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Please install the Biocmanager Dependencies:</span></span>
<span><span class="co"># BiocManager::install("clusterProfiler")</span></span>
<span><span class="co"># BiocManager::install("EnhancedVolcano")</span></span>
<span><span class="co"># BiocManager::install("cosmosR")</span></span></code></pre></div>
<p><br><br></p>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
</div>
<div class="section level2">
<h2 id="loading-the-example-data">Loading the example data<a class="anchor" aria-label="anchor" href="#loading-the-example-data"></a>
</h2>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
<p><br></p>
<p><span style="text-decoration:underline">As part of the
<strong>MetaProViz</strong> package you can load the example feature
metadata using the function <code>toy_data()</code></span>:<br><code>1.</code> Metadata of cell line experiment
<strong>(CellLine)</strong><br>
Example datasets is publicly available on <a href="https://www.metabolomicsworkbench.org/data/DRCCMetadata.php?Mode=Project&amp;ProjectID=PR001418" class="external-link">metabolomics
workbench project PR001418</a> including metabolic profiles of human
renal epithelial cells HK2 and cell renal cell carcinoma (ccRCC) cell
lines cultured in Plasmax cell culture media <span class="citation">(Sciacovelli et al. 2022)</span>.<br></p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the Pre-processed intracellular data:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">cellular_meta</span><span class="op">)</span></span>
<span></span>
<span><span class="va">FeatureMetadata_Cells</span> <span class="op">&lt;-</span> <span class="va">cellular_meta</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Metabolite"</span><span class="op">)</span></span></code></pre></div>
<p><br><code>2.</code> Metadata of patients tissue experiment
<strong>(Tissue)</strong><br>
Example datasets is publicly available in the <a href="https://www.cell.com/cancer-cell/comments/S1535-6108(15)00468-7#supplementaryMaterial" class="external-link">paper</a>
“An Integrated Metabolic Atlas of Clear Cell Renal Cell Carcinoma”,
which includes metabolomic profiling on 138 matched clear cell renal
cell carcinoma (ccRCC)/normal tissue pairs.<br></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the example data:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">tissue_meta</span><span class="op">)</span></span>
<span></span>
<span><span class="va">FeatureMetadata_Tissue</span> <span class="op">&lt;-</span> <span class="va">tissue_meta</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Metabolite"</span><span class="op">)</span></span></code></pre></div>
<p><br><code>3.</code> Metadata of Biocrates kit
<strong>(Biocrates)</strong><br>
Here we use the Biocrates kit feature information of the <a href="https://biocrates.com/mxp-quant-500-xl/" class="external-link">“MxP® Quant 500 XL
kit”</a> that covers more than 1,000 metabolites from various
biochemical classes.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the example data:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">biocrates_features</span><span class="op">)</span></span>
<span><span class="va">FeatureMetadata_Biocrates</span> <span class="op">&lt;-</span> <span class="va">biocrates_features</span></span></code></pre></div>
<p><br><br></p>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
</div>
<div class="section level2">
<h2 id="sect1">1. Metabolite IDs in measured data<a class="anchor" aria-label="anchor" href="#sect1"></a>
</h2>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
<p><strong><em>Assigning Metabolite IDs to measured
data</em></strong><br>
The difficulty with assigning metabolite IDs to measured data is the
uncertainty in the detection of metabolites. Indeed, differentiation of
structural isomers (both constitutional isomers and stereoisomers) as
for example the distinction between enantiomers. This leads to loss of
information and hence uncertainty is assigning metabolite IDs.<br>
One example is the metabolite Alanine, which can occur in its L- or D-
form. If in an experiment those enantiomers have not been distinguished,
the correct way would be to either assign two metabolite IDs (L- and
D-Alanine) or a more general Alanine ID without chiral information. Yet,
in reality this is not as trivial:<br></p>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Available Alanine IDs in HMDB and ChEBI.
</caption>
<thead><tr>
<th style="text-align:left;">
TrivialName
</th>
<th style="text-align:left;">
HMDB
</th>
<th style="text-align:left;">
ChEBI
</th>
<th style="text-align:left;">
KEGG
</th>
<th style="text-align:left;">
PubChem
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
D-Alanine
</td>
<td style="text-align:left;">
HMDB0001310
</td>
<td style="text-align:left;">
15570
</td>
<td style="text-align:left;">
C00133
</td>
<td style="text-align:left;">
71080
</td>
</tr>
<tr>
<td style="text-align:left;">
L-Alanine
</td>
<td style="text-align:left;">
HMDB0000161
</td>
<td style="text-align:left;">
16977
</td>
<td style="text-align:left;">
C00041
</td>
<td style="text-align:left;">
5950
</td>
</tr>
<tr>
<td style="text-align:left;">
Alanine
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
16449
</td>
<td style="text-align:left;">
C01401
</td>
<td style="text-align:left;">
602
</td>
</tr>
<tr>
<td style="text-align:left;">
Alanine zwitterion
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
66916
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
57383916
</td>
</tr>
</tbody>
</table>
<p><br>
Indeed, dependent on the database, the Alanine metabolite can have
different IDs available:<br></p>
<p><img src="prior-knowledge_files/figure-html/plot-1.png" class="r-plt" width="700"></p>
<p><br>
For instance, if we want to assign a HMDB ID, we have to assign both
“HMDB0001310”, “HMDB0000161” to the metabolite Alanine, for ChEBI we
could assign only one, “16449”, but this may lead to other problems as
the ChEBI ID is not specific and may not be part of certain metabolic
pathways. The reason for this is that substrate chirality is critical to
enzymatic processes and stereo selectivity of enzymes to be homochiral
with predominance of one particular enantiomer (e.g. D-sugars, L-amino
acids, etc.).<br>
to showcase the severity of this problem, we can look at the occurrence
of those metabolites in metabolic pathways across different databases.
to do so we searched for those metabolite IDs in the RaMP database <span class="citation">(Braisted et al. 2023)</span> and extracted the
pathways they are part of:<br></p>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Alanine IDs in HMDB and ChEBI mapped to pathways from wiki, KEGG and
Reactome using RamP.
</caption>
<thead><tr>
<th style="text-align:left;">
TrivialName
</th>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Database
</th>
<th style="text-align:right;">
PathwayCount
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
L-Alanine
</td>
<td style="text-align:left;">
16977
</td>
<td style="text-align:left;">
ChEBI
</td>
<td style="text-align:right;">
44
</td>
</tr>
<tr>
<td style="text-align:left;">
L-Alanine
</td>
<td style="text-align:left;">
5950
</td>
<td style="text-align:left;">
PubChem
</td>
<td style="text-align:right;">
44
</td>
</tr>
<tr>
<td style="text-align:left;">
L-Alanine
</td>
<td style="text-align:left;">
C00041
</td>
<td style="text-align:left;">
KEGG
</td>
<td style="text-align:right;">
44
</td>
</tr>
<tr>
<td style="text-align:left;">
L-Alanine
</td>
<td style="text-align:left;">
HMDB0000161
</td>
<td style="text-align:left;">
HMDB
</td>
<td style="text-align:right;">
44
</td>
</tr>
<tr>
<td style="text-align:left;">
D-Alanine
</td>
<td style="text-align:left;">
15570
</td>
<td style="text-align:left;">
ChEBI
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
D-Alanine
</td>
<td style="text-align:left;">
71080
</td>
<td style="text-align:left;">
PubChem
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
D-Alanine
</td>
<td style="text-align:left;">
C00133
</td>
<td style="text-align:left;">
KEGG
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
D-Alanine
</td>
<td style="text-align:left;">
HMDB0001310
</td>
<td style="text-align:left;">
HMDB
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
Alanine zwitterion
</td>
<td style="text-align:left;">
57383916
</td>
<td style="text-align:left;">
PubChem
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
Alanine zwitterion
</td>
<td style="text-align:left;">
66916
</td>
<td style="text-align:left;">
ChEBI
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
Alanine
</td>
<td style="text-align:left;">
16449
</td>
<td style="text-align:left;">
ChEBI
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
Alanine
</td>
<td style="text-align:left;">
602
</td>
<td style="text-align:left;">
PubChem
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
Alanine
</td>
<td style="text-align:left;">
C01401
</td>
<td style="text-align:left;">
KEGG
</td>
<td style="text-align:right;">
4
</td>
</tr>
</tbody>
</table>
<p><br></p>
<p><img src="prior-knowledge_files/figure-html/plot-2-1.png" class="r-plt" width="700"></p>
<p><br></p>
<p><br>
This showcases if we choose the ChEBI ID for Alanine (ChEBI ID 16449),
if experimentally the distinction was not possible, we will not map to
any pathway even though the metabolite is part of many pathways. Hence,
we recommend to assign multiple IDs to a measured ID, where specificity
in detection is not given.<br></p>
<p><br>
Yet, many metabolomics studies do not report multiple IDs, but rather
one ID per measured peak. In some cases the chosen ID is correct in
representing the degree of ambiguity in the detection,
e.g. “Alanine-Zwitterion”, whilst in other cases a specific ID, which
has not been detected in this specificity, e.g. “L-Alanine”. In both
cases this can lead to missed maps to prior knowledge and hence a loss
of information as discussed above.<br>
Hence, we assign multiple IDs to a measured peak, where specificity in
detection may not be given (i.e. amino acids).<br></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example Cell-line data:</span></span>
<span><span class="va">Input_HMDB</span> <span class="op">&lt;-</span> <span class="va">FeatureMetadata_Cells</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="va">HMDB</span> <span class="op">==</span> <span class="st">"NA"</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="co"># ID in the measured data we want to use, hence we remove NA's</span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="st">"HMDB"</span>, <span class="st">"Pathway"</span><span class="op">)</span> <span class="co"># only keep relevant columns</span></span>
<span></span>
<span><span class="co"># Add equivalent IDs:</span></span>
<span><span class="va">FeatureMetadata_Cells_AddIDs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/equivalent_id.html">equivalent_id</a></span><span class="op">(</span>data<span class="op">=</span> <span class="va">Input_HMDB</span>,</span>
<span>metadata_info <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>InputID<span class="op">=</span><span class="st">"HMDB"</span><span class="op">)</span>,<span class="co"># ID in the measured data, here we use the HMDB ID</span></span>
<span>from <span class="op">=</span> <span class="st">"hmdb"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in equivalent_id(data = Input_HMDB, metadata_info = c(InputID =</span></span>
<span><span class="co">#&gt; "HMDB"), : The following IDs are duplicated and removed: HMDB0000725,</span></span>
<span><span class="co">#&gt; HMDB0002013, HMDB0000267, HMDB0000755</span></span>
<span><span class="co">#&gt; chebi is used to find additional potential IDs for hmdb.</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>FeatureMetadata_Cells_AddIDs</code> including
new columns with potential additional HMDB IDs assign using
equivalent_id() function.
</caption>
<thead><tr>
<th style="text-align:left;">
HMDB
</th>
<th style="text-align:left;">
Pathway
</th>
<th style="text-align:left;">
PotentialAdditionalIDs
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
HMDB0000014
</td>
<td style="text-align:left;">
Pyrimidine metabolism
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000033
</td>
<td style="text-align:left;">
Not assigned
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000043
</td>
<td style="text-align:left;">
Not assigned
</td>
<td style="text-align:left;">
HMDB0059606
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000045
</td>
<td style="text-align:left;">
Purine metabolism
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000050
</td>
<td style="text-align:left;">
Purine metabolism
</td>
<td style="text-align:left;">
HMDB0004401, HMDB0004402, HMDB0004421
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000052
</td>
<td style="text-align:left;">
Alanine, aspartate and glutamate metabolism
</td>
<td style="text-align:left;">
HMDB0001006
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000056
</td>
<td style="text-align:left;">
Pyrimidine metabolism
</td>
<td style="text-align:left;">
NA
</td>
</tr>
</tbody>
</table>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
</div>
<div class="section level2">
<h2 id="sect2">2. Accessing Prior Knowledge<a class="anchor" aria-label="anchor" href="#sect2"></a>
</h2>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
<p>Metabolite prior knowledge (PK) is essential for the interpretation
of metabolomics data. It can be used to perform pathway enrichment
analysis, compound class enrichment analysis, and by using specific PK
databases, it can be used to study the connection of metabolites and
receptors or transporters.Since the quality and content of the PK will
dictate the success of the downstream analysis and biological
interpretation, it is important to ensure the PK is used
correctly.<br>
Specifically in metabolite PK, the many different PK databases and
resources pose several issues. Indeed, the metabolite identifiers
(e.g. KEGG, HMDB, PubChem, etc.) are not standardized across databases,
and the same metabolite can have multiple identifiers in different
databases. This is known as the many-to-many mapping problem. Moreover,
metabolic pathways that are the basis of the PK databases also includes
co-factors such as ions or other small molecules that are not only part
of most reactions, but can also not be detected in experimentally
acquired data (e.g. H2O, CO2, etc).<br>
to address these issues and provide a standardized way to access and
integrate metabolite PK, we provide access to several prior knowledge
resources and removed molecules such as water. We terme this collection
of metabolite sets MetSigDB (Metabolite signature database).</p>
<div class="section level3">
<h3 id="kegg-pathway-metabolite-sets">KEGG pathway-metabolite sets<a class="anchor" aria-label="anchor" href="#kegg-pathway-metabolite-sets"></a>
</h3>
<p>KEGG pathways that are loaded via KEGG API using the package
<code>KEGGREST</code> and can be used to perform pathway analysis <span class="citation">(Kanehisa and Goto 2000)</span>.
<strong>(KEGG_Pathways)</strong><br></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># This will use KEGGREST to query the KEGG API to load the pathways:</span></span>
<span><span class="va">KEGG_Pathways</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/metsigdb_kegg.html">metsigdb_kegg</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>KEGG_Pathways</code>.
</caption>
<thead><tr>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
MetaboliteID
</th>
<th style="text-align:left;">
term
</th>
<th style="text-align:left;">
Metabolite
</th>
<th style="text-align:left;">
pubchem
</th>
<th style="text-align:left;">
compound_names
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
map00010
</td>
<td style="text-align:left;">
C00022
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis
</td>
<td style="text-align:left;">
Pyruvate
</td>
<td style="text-align:left;">
3324
</td>
<td style="text-align:left;">
Pyruvate….
</td>
</tr>
<tr>
<td style="text-align:left;">
map00010
</td>
<td style="text-align:left;">
C00024
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis
</td>
<td style="text-align:left;">
Acetyl-CoA
</td>
<td style="text-align:left;">
3326
</td>
<td style="text-align:left;">
Acetyl-C….
</td>
</tr>
<tr>
<td style="text-align:left;">
map00010
</td>
<td style="text-align:left;">
C00031
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis
</td>
<td style="text-align:left;">
D-Glucose
</td>
<td style="text-align:left;">
3333
</td>
<td style="text-align:left;">
D-Glucos….
</td>
</tr>
<tr>
<td style="text-align:left;">
map00030
</td>
<td style="text-align:left;">
C00022
</td>
<td style="text-align:left;">
Pentose phosphate pathway
</td>
<td style="text-align:left;">
Pyruvate
</td>
<td style="text-align:left;">
3324
</td>
<td style="text-align:left;">
Pyruvate….
</td>
</tr>
<tr>
<td style="text-align:left;">
map00030
</td>
<td style="text-align:left;">
C00031
</td>
<td style="text-align:left;">
Pentose phosphate pathway
</td>
<td style="text-align:left;">
D-Glucose
</td>
<td style="text-align:left;">
3333
</td>
<td style="text-align:left;">
D-Glucos….
</td>
</tr>
<tr>
<td style="text-align:left;">
map00030
</td>
<td style="text-align:left;">
C00085
</td>
<td style="text-align:left;">
Pentose phosphate pathway
</td>
<td style="text-align:left;">
D-Fructose 6-phosphate
</td>
<td style="text-align:left;">
3385
</td>
<td style="text-align:left;">
D-Fructo….
</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="chemical-class-metabolite-sets">Chemical class-metabolite sets<a class="anchor" aria-label="anchor" href="#chemical-class-metabolite-sets"></a>
</h3>
<p>The chemical class-metabolite sets are based on the classification of
metabolites into chemical classes, which can be used to perform compound
class enrichment analysis.<br>
The chemical class-metabolite sets were curated by RaMP-DB, which used
ClassyFire <span class="citation">(Braisted et al. 2023)</span>. Here we
access them via OmnipathR.<br></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ChemicalClass_MetabSet</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/metsigdb_chemicalclass.html">metsigdb_chemicalclass</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Cached file loaded from: ~/.cache/RaMP-ChemicalClass_Metabolite.rds</span></span></code></pre></div>
<p><br></p>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>ChemicalClass_MetabSet</code>.
</caption>
<thead><tr>
<th style="text-align:left;">
class_source_id
</th>
<th style="text-align:left;">
common_name
</th>
<th style="text-align:left;">
ClassyFire_class
</th>
<th style="text-align:left;">
ClassyFire_super_class
</th>
<th style="text-align:left;">
ClassyFire_sub_class
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
HMDB0000001
</td>
<td style="text-align:left;">
3-Methylhistidine; 1-Methylhistidine
</td>
<td style="text-align:left;">
Carboxylic acids and derivatives
</td>
<td style="text-align:left;">
Organic acids and derivatives
</td>
<td style="text-align:left;">
Amino acids, peptides, and analogues
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000479
</td>
<td style="text-align:left;">
3-Methylhistidine; 1-Methylhistidine
</td>
<td style="text-align:left;">
Carboxylic acids and derivatives
</td>
<td style="text-align:left;">
Organic acids and derivatives
</td>
<td style="text-align:left;">
Amino acids, peptides, and analogues
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000005
</td>
<td style="text-align:left;">
2-Ketobutyric acid
</td>
<td style="text-align:left;">
Keto acids and derivatives
</td>
<td style="text-align:left;">
Organic acids and derivatives
</td>
<td style="text-align:left;">
Short-chain keto acids and derivatives
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000008
</td>
<td style="text-align:left;">
2-Hydroxybutyric acid
</td>
<td style="text-align:left;">
Hydroxy acids and derivatives
</td>
<td style="text-align:left;">
Organic acids and derivatives
</td>
<td style="text-align:left;">
Alpha hydroxy acids and derivatives
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000010
</td>
<td style="text-align:left;">
2-Methoxyestrone
</td>
<td style="text-align:left;">
Steroids and steroid derivatives
</td>
<td style="text-align:left;">
Lipids and lipid-like molecules
</td>
<td style="text-align:left;">
Estrane steroids
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000163
</td>
<td style="text-align:left;">
D-Maltose
</td>
<td style="text-align:left;">
Organooxygen compounds
</td>
<td style="text-align:left;">
Organic oxygen compounds
</td>
<td style="text-align:left;">
Carbohydrates and carbohydrate conjugates
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000169
</td>
<td style="text-align:left;">
D-Mannose
</td>
<td style="text-align:left;">
Organooxygen compounds
</td>
<td style="text-align:left;">
Organic oxygen compounds
</td>
<td style="text-align:left;">
Carbohydrates and carbohydrate conjugates
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000172
</td>
<td style="text-align:left;">
L-Isoleucine
</td>
<td style="text-align:left;">
Carboxylic acids and derivatives
</td>
<td style="text-align:left;">
Organic acids and derivatives
</td>
<td style="text-align:left;">
Amino acids, peptides, and analogues
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000174
</td>
<td style="text-align:left;">
L-Fucose
</td>
<td style="text-align:left;">
Organooxygen compounds
</td>
<td style="text-align:left;">
Organic oxygen compounds
</td>
<td style="text-align:left;">
Carbohydrates and carbohydrate conjugates
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0004940
</td>
<td style="text-align:left;">
N-Glycoloylganglioside GM1; Ganglioside GM2 (d18:1/20:0);
N-Glycoloylganglioside GM2
</td>
<td style="text-align:left;">
Sphingolipids
</td>
<td style="text-align:left;">
Lipids and lipid-like molecules
</td>
<td style="text-align:left;">
Glycosphingolipids
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000772
</td>
<td style="text-align:left;">
Nonadecanoic acid
</td>
<td style="text-align:left;">
Fatty Acyls
</td>
<td style="text-align:left;">
Lipids and lipid-like molecules
</td>
<td style="text-align:left;">
Fatty acids and conjugates
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000780
</td>
<td style="text-align:left;">
Porphinehexacarboxylic acid
</td>
<td style="text-align:left;">
Tetrapyrroles and derivatives
</td>
<td style="text-align:left;">
Organoheterocyclic compounds
</td>
<td style="text-align:left;">
Porphyrins
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000781
</td>
<td style="text-align:left;">
N-Acetylgalactosamine 4-sulphate
</td>
<td style="text-align:left;">
Organooxygen compounds
</td>
<td style="text-align:left;">
Organic oxygen compounds
</td>
<td style="text-align:left;">
Carbohydrates and carbohydrate conjugates
</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="create-pathway-metabolite-sets">Create pathway-metabolite sets<a class="anchor" aria-label="anchor" href="#create-pathway-metabolite-sets"></a>
</h3>
<p>The function <code>make_gene_metab_set</code> can be used to convert
gene names to metabolite names by using a PK network of metabolic
reactions calls CosmosR <span class="citation">(Dugourd et al.
2021)</span>. This function is useful if you want to perform pathway
enrichment analysis on available gene-sets such as the Hallmarks
gene-sets from MSigDB <span class="citation">(Castanza et al.
2022)</span>. Moreover, it enables you to perform combined pathway
enrichment analysis on metabolite-gene sets, if you have other data
types such as proteomics measuring the enzymes expression.<br>
The Hallmarks <span class="citation">(Liberzon et al. 2015)</span>
gene-set and Gaude <span class="citation">(Gaude and Frezza 2016)</span>
gene-set is available in the package <code>MetaProViz</code> and can be
loaded using the function <code>metsigdb_hallmarks()</code> and
<code>metsigdb_gaude</code> respectively.<br></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the example gene-sets:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">hallmarks</span><span class="op">)</span></span>
<span><span class="va">Hallmark_Pathways</span> <span class="op">&lt;-</span> <span class="va">hallmarks</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gaude_pathways</span><span class="op">)</span></span>
<span><span class="va">Gaude_Pathways</span> <span class="op">&lt;-</span> <span class="va">gaude_pathways</span></span></code></pre></div>
<p><br></p>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>Hallmarks_Pathways</code> including gene-sets
usable for pathway enrichment analysis.
</caption>
<thead><tr>
<th style="text-align:left;">
term
</th>
<th style="text-align:left;">
gene
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
HALLMARK_BILE_ACID_METABOLISM
</td>
<td style="text-align:left;">
GSTK1
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_BILE_ACID_METABOLISM
</td>
<td style="text-align:left;">
ABCG4
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
LDHC
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
ARPP19
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
LDHC
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
ARPP19
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
CENPA
</td>
</tr>
</tbody>
</table>
<p><br></p>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>gaude_Pathways</code> including gene-sets usable
for pathway enrichment analysis.
</caption>
<thead><tr>
<th style="text-align:left;">
gene
</th>
<th style="text-align:left;">
term
</th>
<th style="text-align:left;">
UniqueGene
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
CPT1A
</td>
<td style="text-align:left;">
Carnitine shuttle
</td>
<td style="text-align:left;">
Unique
</td>
</tr>
<tr>
<td style="text-align:left;">
CPT1B
</td>
<td style="text-align:left;">
Carnitine shuttle
</td>
<td style="text-align:left;">
Unique
</td>
</tr>
<tr>
<td style="text-align:left;">
CPT1C
</td>
<td style="text-align:left;">
Carnitine shuttle
</td>
<td style="text-align:left;">
Unique
</td>
</tr>
<tr>
<td style="text-align:left;">
CPT2
</td>
<td style="text-align:left;">
Carnitine shuttle
</td>
<td style="text-align:left;">
Unique
</td>
</tr>
<tr>
<td style="text-align:left;">
CRAT
</td>
<td style="text-align:left;">
Carnitine shuttle
</td>
<td style="text-align:left;">
Unique
</td>
</tr>
<tr>
<td style="text-align:left;">
SLC22A4
</td>
<td style="text-align:left;">
Carnitine shuttle
</td>
<td style="text-align:left;">
In multiple Pathways
</td>
</tr>
<tr>
<td style="text-align:left;">
SLC22A5
</td>
<td style="text-align:left;">
Carnitine shuttle
</td>
<td style="text-align:left;">
In multiple Pathways
</td>
</tr>
<tr>
<td style="text-align:left;">
SLC25A20
</td>
<td style="text-align:left;">
Carnitine shuttle
</td>
<td style="text-align:left;">
In multiple Pathways
</td>
</tr>
<tr>
<td style="text-align:left;">
ACO1
</td>
<td style="text-align:left;">
Citric Acid Cycle
</td>
<td style="text-align:left;">
Unique
</td>
</tr>
<tr>
<td style="text-align:left;">
ACO2
</td>
<td style="text-align:left;">
Citric Acid Cycle
</td>
<td style="text-align:left;">
Unique
</td>
</tr>
<tr>
<td style="text-align:left;">
GOT1
</td>
<td style="text-align:left;">
Citric Acid Cycle
</td>
<td style="text-align:left;">
In multiple Pathways
</td>
</tr>
</tbody>
</table>
<p><br>
Now we can use the function <code>make_gene_metab_set</code> to
translate the gene names to metabolite names.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Translate gene names to metabolite names</span></span>
<span><span class="va">Hallmarks_GeneMetab</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/make_gene_metab_set.html">make_gene_metab_set</a></span><span class="op">(</span>input_pk<span class="op">=</span><span class="va">Hallmark_Pathways</span>,</span>
<span>metadata_info<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>Target<span class="op">=</span><span class="st">"gene"</span><span class="op">)</span>,</span>
<span>pk_name<span class="op">=</span><span class="st">"Hallmarks"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Gaude_GeneMetab</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/make_gene_metab_set.html">make_gene_metab_set</a></span><span class="op">(</span>input_pk<span class="op">=</span><span class="va">Gaude_Pathways</span>,</span>
<span>metadata_info<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>Target<span class="op">=</span><span class="st">"gene"</span><span class="op">)</span>,</span>
<span>pk_name<span class="op">=</span><span class="st">"Gaude"</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>Hallmarks_GeneMetab</code> including gene-sets
usable for pathway enrichment analysis.
</caption>
<thead><tr>
<th style="text-align:left;">
term
</th>
<th style="text-align:left;">
feature
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
ME2
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
LDHC
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
FKBP4
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
HMDB0000241
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
HMDB0000570
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
HMDB0000122
</td>
</tr>
</tbody>
</table>
<p><br></p>
<p><br>
Given that we have the gene-metabolite-sets, we can now also run
enrichment analysis on combined data types, once including the
metabolite Log2FC and one including gene Log2FC from
e.g. transcriptomics or proteomics data. Yet, it is important to keep in
mind that generally we detect less metabolites than genes and hence this
may bias the results obtained from combined enrichment analysis.</p>
</div>
<div class="section level3">
<h3 id="metalinksdb-metabolite-receptor-metabolite-transporter-sets">MetaLinksDB metabolite-receptor &amp; metabolite-transporter
sets<a class="anchor" aria-label="anchor" href="#metalinksdb-metabolite-receptor-metabolite-transporter-sets"></a>
</h3>
<p>The MetaLinks database is a manually curated database of
metabolite-receptor and metabolite-transporter sets that can be used to
study the connection of metabolites and receptors or transporters <span class="citation">(Farr et al. 2024)</span>.<br></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">MetaLinksDB</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/metsigdb_metalinks.html">metsigdb_metalinks</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>MetaLinksDB</code> including metabolite-receptor
sets.
</caption>
<thead><tr>
<th style="text-align:left;">
hmdb
</th>
<th style="text-align:left;">
metabolite
</th>
<th style="text-align:left;">
gene_symbol
</th>
<th style="text-align:left;">
protein_type
</th>
<th style="text-align:left;">
source
</th>
<th style="text-align:right;">
mor
</th>
<th style="text-align:left;">
type
</th>
<th style="text-align:left;">
transport_direction
</th>
<th style="text-align:left;">
mode_of_regulation
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
HMDB0000538
</td>
<td style="text-align:left;">
Adenosine triphosphate
</td>
<td style="text-align:left;">
UBA6
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
hmr
</td>
<td style="text-align:right;">
-1
</td>
<td style="text-align:left;">
Production-Degradation
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Inhibiting
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000538
</td>
<td style="text-align:left;">
Adenosine triphosphate
</td>
<td style="text-align:left;">
UBA6
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
recon
</td>
<td style="text-align:right;">
-1
</td>
<td style="text-align:left;">
Production-Degradation
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Inhibiting
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000122
</td>
<td style="text-align:left;">
D-Glucose
</td>
<td style="text-align:left;">
SLC5A10
</td>
<td style="text-align:left;">
“transporter”
</td>
<td style="text-align:left;">
recon
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
Production-Degradation
</td>
<td style="text-align:left;">
out
</td>
<td style="text-align:left;">
Activating
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000143
</td>
<td style="text-align:left;">
D-Galactose
</td>
<td style="text-align:left;">
SLC5A10
</td>
<td style="text-align:left;">
“transporter”
</td>
<td style="text-align:left;">
recon
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
Production-Degradation
</td>
<td style="text-align:left;">
out
</td>
<td style="text-align:left;">
Activating
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0001341
</td>
<td style="text-align:left;">
ADP
</td>
<td style="text-align:left;">
TLR4
</td>
<td style="text-align:left;">
“catalytic_receptor”
</td>
<td style="text-align:left;">
Stitch
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
Ligand-Receptor
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Binding
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000151
</td>
<td style="text-align:left;">
Estradiol
</td>
<td style="text-align:left;">
TNFRSF11B
</td>
<td style="text-align:left;">
“catalytic_receptor”
</td>
<td style="text-align:left;">
Stitch
</td>
<td style="text-align:right;">
-1
</td>
<td style="text-align:left;">
Ligand-Receptor
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Inhibiting
</td>
</tr>
</tbody>
</table>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the metabolite-receptor and metabolite-transporter sets.
</caption>
<thead><tr>
<th style="text-align:left;">
hmdb
</th>
<th style="text-align:left;">
metabolite
</th>
<th style="text-align:left;">
pubchem
</th>
<th style="text-align:left;">
metabolite_subclass
</th>
<th style="text-align:left;">
uniprot
</th>
<th style="text-align:left;">
gene_symbol
</th>
<th style="text-align:left;">
protein_type
</th>
<th style="text-align:left;">
source
</th>
<th style="text-align:right;">
db_score
</th>
<th style="text-align:right;">
experiment_score
</th>
<th style="text-align:right;">
combined_score
</th>
<th style="text-align:right;">
mor
</th>
<th style="text-align:left;">
type
</th>
<th style="text-align:left;">
transport_direction
</th>
<th style="text-align:left;">
mode_of_regulation
</th>
<th style="text-align:left;">
term_specific
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
HMDB0000660
</td>
<td style="text-align:left;">
D-Fructose
</td>
<td style="text-align:left;">
439709
</td>
<td style="text-align:left;">
Carbohydrates and carbohydrate conjugates
</td>
<td style="text-align:left;">
A0PJK1
</td>
<td style="text-align:left;">
SLC5A10
</td>
<td style="text-align:left;">
“transporter”
</td>
<td style="text-align:left;">
recon
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
Production-Degradation
</td>
<td style="text-align:left;">
out
</td>
<td style="text-align:left;">
Activating
</td>
<td style="text-align:left;">
transporter_Production-Degradation
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0002212
</td>
<td style="text-align:left;">
Arachidic acid
</td>
<td style="text-align:left;">
10467
</td>
<td style="text-align:left;">
Fatty acids and conjugates
</td>
<td style="text-align:left;">
A6NFH5
</td>
<td style="text-align:left;">
FABP12
</td>
<td style="text-align:left;">
“other_protein”
</td>
<td style="text-align:left;">
hmr
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
Production-Degradation
</td>
<td style="text-align:left;">
out
</td>
<td style="text-align:left;">
Activating
</td>
<td style="text-align:left;">
other_protein_Production-Degradation
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0037790
</td>
<td style="text-align:left;">
Polyethylene glycol
</td>
<td style="text-align:left;">
174
</td>
<td style="text-align:left;">
Alcohols and polyols
</td>
<td style="text-align:left;">
O00206
</td>
<td style="text-align:left;">
TLR4
</td>
<td style="text-align:left;">
“catalytic_receptor”
</td>
<td style="text-align:left;">
Stitch
</td>
<td style="text-align:right;">
413
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
483
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
Ligand-Receptor
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Binding
</td>
<td style="text-align:left;">
catalytic_receptor_Ligand-Receptor
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0034442
</td>
<td style="text-align:left;">
L-trans-alpha-Amino-2-carboxycyclopropaneacetic acid
</td>
<td style="text-align:left;">
1271
</td>
<td style="text-align:left;">
Amino acids, peptides, and analogues
</td>
<td style="text-align:left;">
O00222
</td>
<td style="text-align:left;">
GRM8
</td>
<td style="text-align:left;">
“gpcr”
</td>
<td style="text-align:left;">
Stitch
</td>
<td style="text-align:right;">
367
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
888
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
Ligand-Receptor
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Binding
</td>
<td style="text-align:left;">
gpcr_Ligand-Receptor
</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="comparison-of-pk-coverage">Comparison of PK coverage<a class="anchor" aria-label="anchor" href="#comparison-of-pk-coverage"></a>
</h3>
<p>We have now loaded a number of different PK sources (details above)
and aim to compare the overlap in coverage between these PK sources to
understand if there are certain genes or metabolites covered by one PK
resource but not the others. As an example, we will compare the
resources containing gene-metabolite sets (Hallmarks, Gaude,
MetalinksDB) using <code><a href="../../reference/compare_pk.html">compare_pk()</a></code> function, which will
generate a combined datatable and visualises it as an Upset plot. The
Upset plot shows the overlap of coverage, similar to how a Venn diagram
works, but enable to visualize multiple combinations clearly.</p>
<pre><code><span><span class="co">#&gt; Warning: `aes_string()` was deprecated in ggplot2 3.0.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use tidy evaluation idioms with `aes()`.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> See also `vignette("ggplot2-in-packages")` for more information.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">MetaProViz</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue at <span style="color: #0000BB; font-style: italic;">&lt;https://github.com/saezlab/MetaProViz/issues&gt;</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use `linewidth` instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">ComplexUpset</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue at</span></span>
<span><span class="co">#&gt;   <span style="color: #0000BB; font-style: italic;">&lt;https://github.com/krassowski/complex-upset/issues&gt;</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; Warning: themes$intersections_matrix is not a valid theme.</span></span>
<span><span class="co">#&gt; Please use `theme()` to construct themes.</span></span>
<span><span class="co">#&gt; Warning: `legend.margin` must be specified using `margin()`</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> For the old behavior use `legend.spacing`</span></span>
<span><span class="co">#&gt; Warning: themes$intersections_matrix is not a valid theme.</span></span>
<span><span class="co">#&gt; Please use `theme()` to construct themes.</span></span>
<span><span class="co">#&gt; Warning: selected_theme is not a valid theme.</span></span>
<span><span class="co">#&gt; Please use `theme()` to construct themes.</span></span>
<span><span class="co">#&gt; Warning: `legend.margin` must be specified using `margin()`</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> For the old behavior use `legend.spacing`</span></span>
<span><span class="co">#&gt; Warning: selected_theme is not a valid theme.</span></span>
<span><span class="co">#&gt; Please use `theme()` to construct themes.</span></span>
<span><span class="co">#&gt; Warning: themes$overall_sizes is not a valid theme.</span></span>
<span><span class="co">#&gt; Please use `theme()` to construct themes.</span></span>
<span><span class="co">#&gt; Warning: `legend.margin` must be specified using `margin()`</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> For the old behavior use `legend.spacing`</span></span>
<span><span class="co">#&gt; Warning: themes$overall_sizes is not a valid theme.</span></span>
<span><span class="co">#&gt; Please use `theme()` to construct themes.</span></span>
<span><span class="co">#&gt; Note: Could not render upset plot to screen due to theme compatibility issues. Plot was saved to file successfully.</span></span></code></pre>
<img src="prior-knowledge.Rmd_compare-pk.svg"><!-- --><br>
The table includes all features of the three PK resources we decided to
compare. Additionally, it includes a column for each PK resource with
value 1 if the feature is present in the PK resource and 0 if it is not.
The column “Type” specifies with ID type the feature is corresponding
to, here either metabolite (HMDB) or gene, which is used for
visualisation purposes in the upset plot.<br><table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>pk_comp_res$summary_table</code> showing
coverage of features (genes or metabolites) across PK sources.
</caption>
<thead><tr>
<th style="text-align:left;">
Feature
</th>
<th style="text-align:right;">
Hallmarks
</th>
<th style="text-align:right;">
Gaude
</th>
<th style="text-align:right;">
MetalinksDB
</th>
<th style="text-align:left;">
Type
</th>
<th style="text-align:right;">
None
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
FABP4
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
gene
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
ADIPOQ
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
gene
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
PPARG
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
gene
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
LIPE
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
gene
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
DGAT1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
gene
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
LPL
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
gene
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
CPT2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
gene
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
CD36
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
gene
</td>
<td style="text-align:right;">
0
</td>
</tr>
</tbody>
</table>
<p><br></p>
<p><br>
The upset plot shows that MetalinksDB has unique genes and metabolites
not present in the other resources. This is to be expected, since
MetalinksDB focuses on receptors, transporters and metabolic enzymes,
whilst Gaude and Hallmakrs focus on pathways.<br>
Since Gaude only focuses on metabolic enzymes of pathways, it contains a
low number of genes and hence has a small amount of unique genes and
metabolites. In this regard it makes sense that Hallmarks includes many
unique genes, since Hallmarks also includes other genes and pathways not
related to metabolism. Between Hallmarks and Gaude we observe a high
overlap with metabolites, which can be explained since both gene sets
are assigned metabolites by using the same metabolic pathways in the
backend of <code>make_gene_metab_set</code> and it is important to keep
in mind that this dictates the gene-metabolite associations added.</p>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="sect3">3. Linking experimental data to prior knowledge<a class="anchor" aria-label="anchor" href="#sect3"></a>
</h2>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
<p>Now that we have loaded the prior knowledge and inspected the overlap
of genes and metabolites between the different PK resources, in this
section we want to link our experimental data to the PK.<br>
The experimental data in this example will be the data loaded in section
1 (Loading the example data). Whichever resource the experimental data
resembles, we recommend to follow this overall approach:<br><br>
1. Determine the metabolite identifiers from the experimental data and
inspect their coverage.<br>
2. Select the metabolite identifiers from the experimental data, which
can be used to link to the PK of interest.<br>
3. Connect metabolite identifiers to the PK and assess overlap.<br></p>
<p><br>
Here one can select any combination of experimental data and PK. This
selection should be based on the experimental data at hand and the
biological research question.<br>
For this vignette, we will focus on the combination of Biocrates
experimental data and MetalinksDB PK. The reason for using the landscape
of metabolites included in the Biocrates Kit is that Biocrates has
assigned mutliple metabolite IDs to each measurement (Lipid maps, HMDB,
ChEBI) and if needed assigned multiple IDs to one measurement (e.g. the
two HMDB IDs of L- and D-Alanine).</p>
<div class="section level3">
<h3 id="determine-identifiers-and-inspect-coverage">1. Determine identifiers and inspect coverage<a class="anchor" aria-label="anchor" href="#determine-identifiers-and-inspect-coverage"></a>
</h3>
<p>First we determine the identifiers and inspect the coverage.<br>
In many experimentally aquired data, there can be different metabolite
IDs used to describe the metabolite. In the Biocrates data the available
metabolite identifiers are mostly CHEBI, HMDB, and LIMID that could be
linked to PK resources. Now we select these metabolite identifier
columns to count the coverage and look at the combinations of the
coverage grouped by the class of metabolite. This helps us to understand
what to expect, when linking the data to prior knowledge.<br></p>
<pre><code><span><span class="co">#&gt; Warning: themes$intersections_matrix is not a valid theme.</span></span>
<span><span class="co">#&gt; Please use `theme()` to construct themes.</span></span>
<span><span class="co">#&gt; Warning: `legend.margin` must be specified using `margin()`</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> For the old behavior use `legend.spacing`</span></span>
<span><span class="co">#&gt; Warning: themes$intersections_matrix is not a valid theme.</span></span>
<span><span class="co">#&gt; Please use `theme()` to construct themes.</span></span>
<span><span class="co">#&gt; Warning: selected_theme is not a valid theme.</span></span>
<span><span class="co">#&gt; Please use `theme()` to construct themes.</span></span>
<span><span class="co">#&gt; Warning: `legend.margin` must be specified using `margin()`</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> For the old behavior use `legend.spacing`</span></span>
<span><span class="co">#&gt; Warning: selected_theme is not a valid theme.</span></span>
<span><span class="co">#&gt; Please use `theme()` to construct themes.</span></span>
<span><span class="co">#&gt; Warning: themes$overall_sizes is not a valid theme.</span></span>
<span><span class="co">#&gt; Please use `theme()` to construct themes.</span></span>
<span><span class="co">#&gt; Warning: `legend.margin` must be specified using `margin()`</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> For the old behavior use `legend.spacing`</span></span>
<span><span class="co">#&gt; Warning: themes$overall_sizes is not a valid theme.</span></span>
<span><span class="co">#&gt; Please use `theme()` to construct themes.</span></span>
<span><span class="co">#&gt; Note: Could not render upset plot to screen due to theme compatibility issues. Plot was saved to file successfully.</span></span></code></pre>
<p><img src="prior-knowledge.Rmd_compare-pk-2.svg"><!-- --><br></p>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>pk_comp_res_biocft$summary_table</code> showing
coverage of identifiers.
</caption>
<thead><tr>
<th style="text-align:left;">
TrivialName
</th>
<th style="text-align:right;">
CHEBI
</th>
<th style="text-align:right;">
HMDB
</th>
<th style="text-align:right;">
LIMID
</th>
<th style="text-align:right;">
None
</th>
<th style="text-align:left;">
Class
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
1-Met-His
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
Aminoacids Related
</td>
</tr>
<tr>
<td style="text-align:left;">
3-IAA
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
Indoles Derivatives
</td>
</tr>
<tr>
<td style="text-align:left;">
3-IPA
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
Indoles Derivatives
</td>
</tr>
<tr>
<td style="text-align:left;">
3-Met-His
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
Aminoacids Related
</td>
</tr>
<tr>
<td style="text-align:left;">
5-AVA
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
Aminoacids Related
</td>
</tr>
<tr>
<td style="text-align:left;">
AABA
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
Aminoacids Related
</td>
</tr>
<tr>
<td style="text-align:left;">
AbsAcid
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
Hormones
</td>
</tr>
<tr>
<td style="text-align:left;">
Ac-Orn
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
Aminoacids Related
</td>
</tr>
</tbody>
</table>
<p><br>
These results tell us that:<br></p>
<ul>
<li><p>Less than half the metabolites had <strong><em>CHEBI+HMDB+LIMID
(n=413)</em></strong>, which isn’t great but probably higher than we
have expected.<br></p></li>
<li>
<p>If you relied only upon -<br></p>
<ul>
<li><p>CHEBI: then you would miss at least 256 metabolites with
HMDB+LIMID</p></li>
<li><p>HMDB: then you would miss at least 54 metabolites with
CHEBI+LIMID</p></li>
<li><p>LIMID: then you would miss at least 89 metabolites with
CHEBI+HMDB</p></li>
</ul>
</li>
</ul>
<p>However there are two things to note about these observations
above.</p>
<p>The first is that these numbers are minimums, the real values are a
little higher once you add for instance metabolites with just CHEBI or
just LIMID, etc. So actually, relying upon only LIMID means we miss 151
biocrates metabolites (89 with only HMDB/ChEBI + 15 with only CHEBI + 8
with only HMDB and 39 with no LIMID/CHEBI/HMDB identifiers).</p>
<p>The second point is more nuanced but just as important: we need to
keep in mind that there is not a direct 1:1 relationship between the
number of Biocrates metabolites with, for instance, a ChEBI ID, and the
number of unique ChEBI IDs that they map to. This upset plot is treating
each Biocrates metabolite as an individual entry and categorising
whether it has a ChEBI, HMDB, LIMID ID, etc. However we are not
considering what those ChEBI, HMDB, LIMIDs actually are. So it is
possible that there are cases where the number of metabolites is higher,
or lower, than the number of unique IDs due to multimapping.</p>
<p>For example, if we have 3 Biocrates metabolites with only HMDB IDs.
These would count as 3 entries in the upset plot. But two of these
Biocrates metabolites could both map to the same HMDB ID, whereas the
other Biocrates metabolite could map to 6 different HMDBs. In this case,
there are 3 Biocrates metabolites with HMDBs but 7 unique HMDBs. Hence,
it is important to keep this in mind when interpreting the upset
plot.</p>
</div>
<div class="section level3">
<h3 id="select-identifiers-to-link-to-pk-of-interest">2. Select identifiers to link to PK of interest<a class="anchor" aria-label="anchor" href="#select-identifiers-to-link-to-pk-of-interest"></a>
</h3>
<p>We have inspected the experimental coverage of the metabolite IDs
from the Biocrates kit. Next, we have to choose which metabolite
identifier to use to link to the PK of choice. Often this choice is
dictated by the prior knowledge resource as most of them use a specific
identifier.<br>
Here we will use MetalinksDB, which uses HMDB IDs as metabolite
identifiers, hence it is best to use the HMDB IDs to link the Biocrates
features to the resource. Noteworthy, in some cases the Biocrates
metabolites have multiple HMDB IDs listed per metabolite:<br></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Count entries and record NA information</span></span>
<span><span class="va">result_bioc_hmdb_count</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/count_id.html">count_id</a></span><span class="op">(</span><span class="va">FeatureMetadata_Biocrates</span>, <span class="st">"HMDB"</span><span class="op">)</span></span></code></pre></div>
<p><img src="prior-knowledge_files/figure-html/code-11-1.png" class="r-plt" width="768"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Access the processed data:</span></span>
<span><span class="va">processed_df_bioc_hmdb_count</span> <span class="op">&lt;-</span> <span class="va">result_bioc_hmdb_count</span><span class="op">$</span><span class="va">result</span></span></code></pre></div>
<p>From this plot we can make a few observations: 1. The number of
biocrates cells without any HMDB at all is very high (n=253). 2. Whilst
a number of biocrates cells have only a single HMDB ID associated with
them (n=314), the majority of biocrates metabolites actually have
multiple entries (n=452). 3. For those with multiple, this ranges from
2-20 HMDB entries per biocrates cell!</p>
<p>With so many biocrates cells linked to multiple HMDB IDs there are
different ways to deal with this (e.g. select a single HMDB ID in cases
where we have multiple). We discuss the pros and cons of this later in
section: <a href="#sect4_bonus">Bonus: Are multiple metabolite IDs
helpful or a hindrance?</a>. But for now, we will proceed with making
the connections to the PK using all HMDBs, which the MetaProViz function
<code><a href="../../reference/checkmatch_pk_to_data.html">checkmatch_pk_to_data()</a></code> has been designed to handle.</p>
</div>
<div class="section level3">
<h3 id="make-connection-to-pk-and-assess-overlap">3. Make connection to PK and assess overlap<a class="anchor" aria-label="anchor" href="#make-connection-to-pk-and-assess-overlap"></a>
</h3>
<p>Here we connect the experimental Biocrates table to the MetalinksDB
PK using the HMDB IDs using the <code><a href="../../reference/checkmatch_pk_to_data.html">checkmatch_pk_to_data()</a></code>
function.<br></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Biocrates_to_MetalinksDB</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/checkmatch_pk_to_data.html">checkmatch_pk_to_data</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">FeatureMetadata_Biocrates</span>,</span>
<span>input_pk <span class="op">=</span> <span class="va">MetaLinksDB</span>,</span>
<span>metadata_info <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>InputID<span class="op">=</span><span class="st">"HMDB"</span>, PriorID<span class="op">=</span><span class="st">"hmdb"</span>, grouping_variable<span class="op">=</span><span class="cn">NULL</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in checkmatch_pk_to_data(data = FeatureMetadata_Biocrates, input_pk =</span></span>
<span><span class="co">#&gt; MetaLinksDB, : 253 NA values were removed from column HMDB</span></span>
<span><span class="co">#&gt; Warning in checkmatch_pk_to_data(data = FeatureMetadata_Biocrates, input_pk =</span></span>
<span><span class="co">#&gt; MetaLinksDB, : 4 duplicated IDs were removed from column HMDB</span></span>
<span><span class="co">#&gt; No metadata_info grouping_variable provided. If this was not intentional, please check your input.</span></span>
<span><span class="co">#&gt; Warning in checkmatch_pk_to_data(data = FeatureMetadata_Biocrates, input_pk =</span></span>
<span><span class="co">#&gt; MetaLinksDB, : 40766 duplicated IDs were removed from PK column hmdb</span></span>
<span><span class="co">#&gt; data has multiple IDs per measurement = TRUE. input_pk has multiple IDs per entry = FALSE.</span></span>
<span><span class="co">#&gt; data has 762 unique entries with 2027 unique HMDB IDs. Of those IDs, 176 match, which is 8.68278243709916%.</span></span>
<span><span class="co">#&gt; input_pk has 1128 unique entries with 1128 unique hmdb IDs. Of those IDs, 176 are detected in the data, which is 15.6028368794326%.</span></span>
<span><span class="co">#&gt; Warning in checkmatch_pk_to_data(data = FeatureMetadata_Biocrates, input_pk =</span></span>
<span><span class="co">#&gt; MetaLinksDB, : There are cases where multiple detected IDs match to multiple</span></span>
<span><span class="co">#&gt; prior knowledge IDs of the same category</span></span></code></pre></div>
<p><br>
This returns some warning messages and 4 tables:<br></p>
<ul>
<li><p><code>data_summary</code>: a summary table with links to the
prior knowledge. Includes pointers if any action is required This is
however with any NA values or duplicates of the InputID (e.g. ‘HMDB’)
removed from the table.</p></li>
<li><p><code>GroupingVariable_summary</code>: This is an extended
version of the summary, where GroupingVariable is extended. e.g. for
pathway-metabolite sets this would include duplicated InputIDs as they
are in multiple pathways</p></li>
<li><p><code>data_long</code>: all versus all commparison to enable
checking in a case-by-case basis.</p></li>
</ul>
<p>In some cases we will expect to have only input ID linked to one
entry in the PK. But this won’t always be the case, and as already
discussed in the last section above, in many cases here we will have
multiple HMDB IDs per single biocrates metabolite cell. The
<code><a href="../../reference/checkmatch_pk_to_data.html">checkmatch_pk_to_data()</a></code> has been designed with this in
mind, by default splitting any comma separated values in the InputID (or
PriorID) into separate entities, counting the number of links between
these, and reporting this to the user.</p>
<p>Let’s take a look at the results:</p>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>data_Matched</code> showing coverage of
identifiers (some columns hidden)
</caption>
<thead><tr>
<th style="text-align:left;">
HMDB
</th>
<th style="text-align:left;">
matches
</th>
<th style="text-align:right;">
original_count
</th>
<th style="text-align:right;">
matches_count
</th>
<th style="text-align:left;">
GroupingVariable
</th>
<th style="text-align:left;">
Count_FeatureIDs_to_GroupingVariable
</th>
<th style="text-align:left;">
Group_Conflict_Notes
</th>
<th style="text-align:right;">
Unique_GroupingVariable_count
</th>
<th style="text-align:left;">
ActionRequired
</th>
<th style="text-align:left;">
InputID_select
</th>
<th style="text-align:left;">
Action_Specific
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
HMDB0000001
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
HMDB
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000033
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
HMDB
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000036
</td>
<td style="text-align:left;">
HMDB0000036
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
OneGroup
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
HMDB
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000043
</td>
<td style="text-align:left;">
HMDB0000043
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
OneGroup
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
HMDB
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000056
</td>
<td style="text-align:left;">
HMDB0000056
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
OneGroup
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
HMDB
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000062
</td>
<td style="text-align:left;">
HMDB0000062
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
OneGroup
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
HMDB
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000063
</td>
<td style="text-align:left;">
HMDB0000063
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
OneGroup
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
HMDB
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0000072, HMDB0000958
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NA, NA
</td>
<td style="text-align:left;">
NA, NA
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
HMDB
</td>
<td style="text-align:left;">
None
</td>
</tr>
</tbody>
</table>
<p>For a few metabolite classes from the biocrates kit every metabolite
has a HMDB ID and a corresponding entry in MetalinksDB. Whilst for a
number of classes of metabolites, we see that they will not be
represented in the MetalinksDB PK at all, either because we do not have
a HMDB ID associated with the Biocrates metabolite, or because
MetalinksDB does not include a HMDB ID for that metabolite.<br>
This poor coverage could be a concern for the analysis if we are
interested in analysing many of these classes, since it will mean in
this case that our experimental results for Phosphatidylglycerols for
instance will not be linked to PK. In any case we will certainly need to
keep this in mind for downstream analysis and the interpretation of
results, so that we don’t overinterpret results that for instance have a
large number of Amino Acids, or falsely assume that the absence of
Phosphatidylinositols in our PK integration results means that they are
not present or important in our data.<br><br>
Let’s turn back to consider the number of HMDB IDs we have in the
Biocrates data and ask ourself a question: is it helpful or detrimental
to have multiple IDs? To answer this, we will take only the first HMDB
of each cell with multiple HMDB IDs and test to see this has worked.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Number of matches:</span></span>
<span><span class="va">MultipleIDs</span> <span class="op">&lt;-</span> <span class="va">Biocrates_to_MetalinksDB</span><span class="op">$</span><span class="va">data_summary</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>total_matches_count <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">matches_count</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Name <span class="op">=</span> <span class="st">"MultipleIDs"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">######################################</span></span>
<span><span class="co"># Extract first ID:</span></span>
<span><span class="va">extract_first_id</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">id_col</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map_chr</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">id_col</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="co"># Check for NA or empty string</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">||</span> <span class="va">x</span> <span class="op">==</span> <span class="st">""</span><span class="op">)</span> <span class="op">{</span></span>
<span>        <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="cn">NA</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span>    <span class="co"># Split on comma (adjust the delimiter if needed)</span></span>
<span>    <span class="va">parts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="va">x</span>, split <span class="op">=</span> <span class="st">","</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="co"># Return the first value after trimming any whitespace</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/trimws.html" class="external-link">trimws</a></span><span class="op">(</span><span class="va">parts</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="op">}</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Create a copy of the df</span></span>
<span><span class="va">FeatureMetadata_Biocrates_singleHMDB</span> <span class="op">&lt;-</span> <span class="va">FeatureMetadata_Biocrates</span></span>
<span><span class="co"># Get the first entry of each HMDB ID</span></span>
<span><span class="va">FeatureMetadata_Biocrates_singleHMDB</span><span class="op">$</span><span class="va">HMDB_single</span> <span class="op">&lt;-</span> <span class="fu">extract_first_id</span><span class="op">(</span><span class="va">FeatureMetadata_Biocrates</span><span class="op">$</span><span class="va">HMDB</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Visually check that the single ID function has worked</span></span>
<span><span class="co"># Count entries and record NA information</span></span>
<span><span class="va">result_bioc_hmdb_count_single</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/count_id.html">count_id</a></span><span class="op">(</span><span class="va">FeatureMetadata_Biocrates_singleHMDB</span>, <span class="st">"HMDB_single"</span><span class="op">)</span></span></code></pre></div>
<p><img src="prior-knowledge_files/figure-html/data-prep-6-1.png" class="r-plt" width="700"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Check the macthes:</span></span>
<span><span class="va">Biocrates_to_MetalinksDB_singleHMDB</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/checkmatch_pk_to_data.html">checkmatch_pk_to_data</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">FeatureMetadata_Biocrates_singleHMDB</span>,</span>
<span>input_pk <span class="op">=</span> <span class="va">MetaLinksDB</span>,</span>
<span>metadata_info <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>InputID<span class="op">=</span><span class="st">"HMDB_single"</span>, PriorID<span class="op">=</span><span class="st">"hmdb"</span>, grouping_variable<span class="op">=</span><span class="cn">NULL</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in checkmatch_pk_to_data(data = FeatureMetadata_Biocrates_singleHMDB, :</span></span>
<span><span class="co">#&gt; 253 NA values were removed from column HMDB_single</span></span>
<span><span class="co">#&gt; Warning in checkmatch_pk_to_data(data = FeatureMetadata_Biocrates_singleHMDB, :</span></span>
<span><span class="co">#&gt; 49 duplicated IDs were removed from column HMDB_single</span></span>
<span><span class="co">#&gt; No metadata_info grouping_variable provided. If this was not intentional, please check your input.</span></span>
<span><span class="co">#&gt; Warning in checkmatch_pk_to_data(data = FeatureMetadata_Biocrates_singleHMDB, :</span></span>
<span><span class="co">#&gt; 40766 duplicated IDs were removed from PK column hmdb</span></span>
<span><span class="co">#&gt; data has multiple IDs per measurement = FALSE. input_pk has multiple IDs per entry = FALSE.</span></span>
<span><span class="co">#&gt; data has 717 unique entries with 717 unique HMDB_single IDs. Of those IDs, 147 match, which is 20.5020920502092%.</span></span>
<span><span class="co">#&gt; input_pk has 1128 unique entries with 1128 unique hmdb IDs. Of those IDs, 147 are detected in the data, which is 13.031914893617%.</span></span>
<span></span>
<span></span>
<span><span class="co">#Number of matches:</span></span>
<span><span class="va">SingleIDs</span> <span class="op">&lt;-</span> <span class="va">Biocrates_to_MetalinksDB_singleHMDB</span><span class="op">$</span><span class="va">data_summary</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>total_matches_count <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">matches_count</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Name <span class="op">=</span> <span class="st">"SingleID"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">##############################</span></span>
<span><span class="co">#Compare:</span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind_rows.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="va">MultipleIDs</span>, <span class="va">SingleIDs</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu">kableExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kbl.html" class="external-link">kbl</a></span><span class="op">(</span>caption <span class="op">=</span> <span class="st">"Comparison of IDs found in MetalinksDB PK when having multiple versus a random single HMDB ID in the measured data."</span>, row.names<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu">kableExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_classic.html" class="external-link">kable_classic</a></span><span class="op">(</span>full_width <span class="op">=</span> <span class="cn">FALSE</span>, html_font <span class="op">=</span> <span class="st">"Cambria"</span>, font_size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Comparison of IDs found in MetalinksDB PK when having multiple versus a
random single HMDB ID in the measured data.
</caption>
<thead><tr>
<th style="text-align:right;">
total_matches_count
</th>
<th style="text-align:left;">
Name
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:right;">
208
</td>
<td style="text-align:left;">
MultipleIDs
</td>
</tr>
<tr>
<td style="text-align:right;">
147
</td>
<td style="text-align:left;">
SingleID
</td>
</tr>
</tbody>
</table>
<p>This shows using Multi HMDBs has resulted in more metabolites from
Biocrates being able to be linked to MetalinksDB than what would have
been possible if we only used the first HMDB available to us. Hence
while having multiple IDs for a single detected peak may add complexity,
we would recommend against prematurely dropping any IDs until you map to
the PK or have thoroughly assessed what impact the removal may have.</p>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="sect4">4. Translate IDs<a class="anchor" aria-label="anchor" href="#sect4"></a>
</h2>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
<p><br></p>
<div style="border: 1px solid red; padding: 10px; background-color: #ffe6e6;">
<p><strong>Important Information:</strong> Translating IDs between
databases e.g. KEGG to HMDB is a non-trivial task, and it is expected
for one original ID to link to many translated IDs, and vice versa. We
discuss the implications throughout this vignette and leave it to user
discretion to select the most appropriate ID based on their research
question and data.</p>
</div>
<p><br></p>
<p>Across the different prior knowledge resources (see also tables
above) specific metabolite IDs are used and hence depending on the prior
knowledge resource a specific metabolite ID is required.<br>
If we want to convert or ‘translate’ those IDs to another commonly used
form of ID, for instance because our measured data uses another type of
ID, we can make use of the <code><a href="../../reference/translate_id.html">translate_id()</a></code> function. This is
based on <a href="https://www.bioconductor.org/packages/release/bioc/html/OmnipathR.html" class="external-link">OmniPathR</a>
and RaMP-DB <span class="citation">(Braisted et al. 2023)</span> on the
backend of our function and currently supports ID translation of
metabolites to and from the following formats:<br>
- KEGG<br>
- HMDB<br>
- ChEBI<br>
- PubChem<br><br>
As an example we are translating the KEGG pathways we loaded with
<code>metsigdb_kegg</code> into HMDB ids:<br></p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">KEGG_Pathways_Translated</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/translate_id.html">translate_id</a></span><span class="op">(</span>data<span class="op">=</span> <span class="va">KEGG_Pathways</span>,</span>
<span>metadata_info <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>InputID<span class="op">=</span><span class="st">"MetaboliteID"</span>, grouping_variable<span class="op">=</span><span class="st">"term"</span><span class="op">)</span>,</span>
<span>from <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"kegg"</span><span class="op">)</span>,</span>
<span>to <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hmdb"</span>, <span class="st">"pubchem"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Translation of KEGG IDs in KEGG pathways to HMDB &amp; pubchem IDs
</caption>
<thead><tr>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
MetaboliteID
</th>
<th style="text-align:left;">
term
</th>
<th style="text-align:left;">
Metabolite
</th>
<th style="text-align:left;">
pubchem
</th>
<th style="text-align:left;">
compound_names
</th>
<th style="text-align:left;">
hmdb
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
map00010
</td>
<td style="text-align:left;">
C00022
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis
</td>
<td style="text-align:left;">
Pyruvate
</td>
<td style="text-align:left;">
1060, 107735
</td>
<td style="text-align:left;">
Pyruvate….
</td>
<td style="text-align:left;">
HMDB0000243
</td>
</tr>
<tr>
<td style="text-align:left;">
map00010
</td>
<td style="text-align:left;">
C00024
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis
</td>
<td style="text-align:left;">
Acetyl-CoA
</td>
<td style="text-align:left;">
444493, 181
</td>
<td style="text-align:left;">
Acetyl-C….
</td>
<td style="text-align:left;">
HMDB0001206, HMDB0247926
</td>
</tr>
<tr>
<td style="text-align:left;">
map00010
</td>
<td style="text-align:left;">
C00031
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis
</td>
<td style="text-align:left;">
D-Glucose
</td>
<td style="text-align:left;">
64689, 5793, 107526
</td>
<td style="text-align:left;">
D-Glucos….
</td>
<td style="text-align:left;">
HMDB0000122, HMDB0304632, HMDB0000516, HMDB0003340, HMDB0006564,
HMDB0062170
</td>
</tr>
<tr>
<td style="text-align:left;">
map00053
</td>
<td style="text-align:left;">
C15926
</td>
<td style="text-align:left;">
Ascorbate and aldarate metabolism
</td>
<td style="text-align:left;">
beta-L-Galactose 1-phosphate
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
beta-L-G….
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
map00053
</td>
<td style="text-align:left;">
C15930
</td>
<td style="text-align:left;">
Ascorbate and aldarate metabolism
</td>
<td style="text-align:left;">
L-Galactonate
</td>
<td style="text-align:left;">
604
</td>
<td style="text-align:left;">
L-Galact….
</td>
<td style="text-align:left;">
HMDB0253886
</td>
</tr>
<tr>
<td style="text-align:left;">
map00120
</td>
<td style="text-align:left;">
C13550
</td>
<td style="text-align:left;">
Primary bile acid biosynthesis
</td>
<td style="text-align:left;">
Cerebrosterol
</td>
<td style="text-align:left;">
121948
</td>
<td style="text-align:left;">
Cerebros….
</td>
<td style="text-align:left;">
HMDB0001419
</td>
</tr>
<tr>
<td style="text-align:left;">
map00120
</td>
<td style="text-align:left;">
C15518
</td>
<td style="text-align:left;">
Primary bile acid biosynthesis
</td>
<td style="text-align:left;">
(24S)-Cholest-5-ene-3beta,7alpha,24-triol
</td>
<td style="text-align:left;">
11954196
</td>
<td style="text-align:left;">
(24S)-Ch….
</td>
<td style="text-align:left;">
HMDB0060136
</td>
</tr>
<tr>
<td style="text-align:left;">
map00120
</td>
<td style="text-align:left;">
C15519
</td>
<td style="text-align:left;">
Primary bile acid biosynthesis
</td>
<td style="text-align:left;">
25-Hydroxycholesterol
</td>
<td style="text-align:left;">
53477807, 65094, 3247051, 42608393
</td>
<td style="text-align:left;">
25-Hydro….
</td>
<td style="text-align:left;">
HMDB0006247
</td>
</tr>
</tbody>
</table>
<p><br>
Here we can immediately see that despite the ID translation, some of the
translations between the KEGG MetaboliteID and the HMDB or PubChem IDs
have failed, resulting in NA values. to get a better understanding of
the combinations of these, let’s visualise the translation for each of
the ID types.<br></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pk_comp_res_keggtrans</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/compare_pk.html">compare_pk</a></span><span class="op">(</span>data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>kegg_translated <span class="op">=</span> <span class="va">KEGG_Pathways_Translated</span><span class="op">$</span><span class="va">TranslatedDF</span><span class="op">)</span>,</span>
<span>                                                metadata_info <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>kegg_translated <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hmdb"</span>, <span class="st">"pubchem"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                                                plot_name <span class="op">=</span> <span class="st">"IDs available after KEGG ID Translation"</span><span class="op">)</span></span></code></pre></div>
<p><img src="prior-knowledge.Rmd_compare-pk-3.svg"><!-- --><br>
We can also note from the previous table that it becomes apparent that
the translation of IDs is not a one-to-one mapping, but rather a
one-to-many mapping. In fact it is very common that an ID from one
format will have a genuine one-to-many relationship with the other
format (e.g. one KEGG ID maps to multiple HMDB IDs) or even a
many-to-many relationship, where some of the IDs from the new format
link back to multiple IDs in the original format (e.g. two different
KEGG IDs map to multiple HMDS IDs, some of which are shared between
them).<br>
This comes with many implications for the analysis that will be
discussed in the next section.</p>
<div class="section level3">
<h3 id="mapping-problems">4.1 Mapping problems<a class="anchor" aria-label="anchor" href="#mapping-problems"></a>
</h3>
<p>The complexities of translating metabolite IDs are demonstrated here
(Fig.2). The relationships between Original IDs (e.g. KEGG) and
Translated IDs (e.g. HMDB), can be quite complex and in fact we
encounter <code>one-to-one</code> (no matches were found for the ID),
<code>one-to-none</code> ( direct #relationship was established),
<code>one-to-many</code> (multiple matches were found for the ID.
i.e. it is ambiguously mapped) and <code>many-to-many</code> ( considers
the relationships between the Translated IDs to Original IDs , where a
translated ID ambiguously maps back to multiple different Original IDs)
mappings.<br>
For enrichment analysis the translation from KEGG IDs to HMDB IDs
increases the pathways size, i.e. how many metabolites are in the
pathway “Glycolysis / Gluconeogenesis - Homo sapiens (human)”, which
would in turn inflate/deflate the enrichment results and hence it would
be desired to keep the number of metabolites in a pathway
consistent.<br></p>
<center>
<div class="float">
<img src="Fig.2.png" style="width:50.0%" alt="Fig. 2: Mapping problems in prior knowledge metabolite-sets when translating metabolite IDs."><div class="figcaption">Fig. 2: Mapping problems in prior knowledge
metabolite-sets when translating metabolite IDs.</div>
</div>
</center>
<p><br></p>
<p>Because of this complexity the output of <code><a href="../../reference/translate_id.html">translate_id()</a></code>
includes not only the Translation table showcased above, but
additionally information about the mapping ambiguity as well as a
summary of the relationships between the Original and Translated
IDs.<br>
Indeed, the translation of e.g. KEGG to hmdb and pubchem includes
multiple data frames including a summary of the mapping
occurrences:<br></p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">KEGG_Pathways_Translated</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "TranslatedDF"             "TranslatedDF_MappingInfo"</span></span>
<span><span class="co">#&gt; [3] "Mappingsummary_hmdb"      "Mappingsummary_pubchem"</span></span></code></pre></div>
<p><br></p>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Mappingsummary_hmdb
</caption>
<thead><tr>
<th style="text-align:left;">
term
</th>
<th style="text-align:right;">
many_to_many
</th>
<th style="text-align:right;">
one_to_many
</th>
<th style="text-align:right;">
one_to_none
</th>
<th style="text-align:right;">
one_to_one
</th>
<th style="text-align:right;">
total
</th>
<th style="text-align:right;">
to_many
</th>
<th style="text-align:right;">
to_none
</th>
<th style="text-align:right;">
to_one
</th>
<th style="text-align:right;">
to_total
</th>
<th style="text-align:right;">
from_many
</th>
<th style="text-align:right;">
from_one
</th>
<th style="text-align:right;">
from_total
</th>
<th style="text-align:right;">
many_to_one
</th>
<th style="text-align:left;">
scope
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
54
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
31
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
37
</td>
<td style="text-align:right;">
43
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
group
</td>
</tr>
<tr>
<td style="text-align:left;">
Citrate cycle (TCA cycle)
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
27
</td>
<td style="text-align:right;">
27
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
group
</td>
</tr>
<tr>
<td style="text-align:left;">
Pentose phosphate pathway
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
63
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
37
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
34
</td>
<td style="text-align:right;">
44
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
group
</td>
</tr>
<tr>
<td style="text-align:left;">
Ascorbate and aldarate metabolism
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
39
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
90
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
57
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
55
</td>
<td style="text-align:right;">
60
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
group
</td>
</tr>
<tr>
<td style="text-align:left;">
Fatty acid biosynthesis
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
46
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
66
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
46
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
58
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
21
</td>
<td style="text-align:right;">
21
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
group
</td>
</tr>
<tr>
<td style="text-align:left;">
Fatty acid elongation
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
48
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
70
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
39
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
52
</td>
<td style="text-align:right;">
56
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
group
</td>
</tr>
</tbody>
</table>
<p><br><br>
We also have the ability to extract a long version of the DF that
includes a row for each mapping occurrence, which can be useful for
downstream analysis. Yet, this can become very large dependent on the
amount of many-to-many mappings, hence by default we do not generate
this summary. Within the <code><a href="../../reference/translate_id.html">translate_id()</a></code> you can set the
parameter <code>summary =TRUE</code> or in case you have a dataframe
that includes both, original and translated ID, you can use the function
<code><a href="../../reference/mapping_ambiguity.html">mapping_ambiguity()</a></code> to generate this long summary as well
as the mapping summary in general.<br></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Option 1:</span></span>
<span><span class="va">KEGG_Pathways_TranslatedSum</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/translate_id.html">translate_id</a></span><span class="op">(</span>data<span class="op">=</span> <span class="va">KEGG_Pathways</span>,</span>
<span>                                                    metadata_info <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>InputID<span class="op">=</span><span class="st">"MetaboliteID"</span>, grouping_variable<span class="op">=</span><span class="st">"term"</span><span class="op">)</span>,</span>
<span>                                                    from <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"kegg"</span><span class="op">)</span>,</span>
<span>                                                    to <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hmdb"</span>, <span class="st">"pubchem"</span><span class="op">)</span>,</span>
<span>                                                    summary <span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Option 2:</span></span>
<span><span class="va">MappingProblems</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/mapping_ambiguity.html">mapping_ambiguity</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span></span>
<span>        <span class="va">KEGG_Pathways_Translated</span><span class="op">[[</span><span class="st">"TranslatedDF"</span><span class="op">]</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span><span class="st">"KEGG"</span><span class="op">=</span><span class="st">"MetaboliteID"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Description</span>, <span class="va">KEGG</span>, <span class="va">term</span>, <span class="va">Metabolite</span>, <span class="va">hmdb</span><span class="op">)</span>,</span>
<span>    from <span class="op">=</span> <span class="st">"KEGG"</span>,</span>
<span>    to <span class="op">=</span> <span class="st">"hmdb"</span>,</span>
<span>    grouping_variable <span class="op">=</span> <span class="st">"term"</span>,</span>
<span>    summary<span class="op">=</span><span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><br></p>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Long summary of mapping problems taking into account both directions,
from-to-to and to-to-from.
</caption>
<thead><tr>
<th style="text-align:left;">
KEGG
</th>
<th style="text-align:left;">
hmdb
</th>
<th style="text-align:left;">
term
</th>
<th style="text-align:left;">
KEGG_to_hmdb
</th>
<th style="text-align:right;">
Count(KEGG_to_hmdb)
</th>
<th style="text-align:left;">
hmdb_to_KEGG
</th>
<th style="text-align:right;">
Count(hmdb_to_KEGG)
</th>
<th style="text-align:left;">
Mapping
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
C00002
</td>
<td style="text-align:left;">
HMDB0000538
</td>
<td style="text-align:left;">
Bacterial secretion system
</td>
<td style="text-align:left;">
C00002 –&gt; HMDB0000538, HMDB0257997
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
HMDB0000538 –&gt; C00002
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
one-to-many
</td>
</tr>
<tr>
<td style="text-align:left;">
C00002
</td>
<td style="text-align:left;">
HMDB0000538
</td>
<td style="text-align:left;">
Biosynthesis of cofactors
</td>
<td style="text-align:left;">
C00002 –&gt; HMDB0000538, HMDB0257997
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
HMDB0000538 –&gt; C00002
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
one-to-many
</td>
</tr>
<tr>
<td style="text-align:left;">
C00024
</td>
<td style="text-align:left;">
HMDB0247926
</td>
<td style="text-align:left;">
Ethylbenzene degradation
</td>
<td style="text-align:left;">
C00024 –&gt; HMDB0001206, HMDB0247926
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
HMDB0247926 –&gt; C00024
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
one-to-many
</td>
</tr>
<tr>
<td style="text-align:left;">
C00024
</td>
<td style="text-align:left;">
HMDB0247926
</td>
<td style="text-align:left;">
Fatty acid degradation
</td>
<td style="text-align:left;">
C00024 –&gt; HMDB0001206, HMDB0247926
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
HMDB0247926 –&gt; C00024
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
one-to-many
</td>
</tr>
</tbody>
</table>
<p><br>
In the table it is shown that the KEGG ID C00002 maps to 3 different
HMDB IDs, and it is shown that one of those HMDB IDs HMDB0000538 maps to
one KEGG ID, hence this Mapping is one-to-many. The other two HMDB Ids
are also in the table and it is checked to how many KEGG IDs they map.
Additionally, we have passed <code>grouping_variable = "term"</code> as
we have pathways, which also means each of those mappings is checked
within a pathway and across pathways (e.g. C00002 is shown twice, for
two different terms).<br></p>
<p><img src="prior-knowledge_files/figure-html/plot-4-1.png" class="r-plt" width="700"></p>
<p><br>
In fact, to perform enrichment analysis we need a column source
(=e.g. term) and we would want to keep the metabolite IDs across
pathways consistent, avoid ambiguous mapping as much as possible
(many-to-many mapping), and have this metabolite ID selection guided by
any IDs we might have available to us in our measured input data (Fig.
3). This is crucial to circumvent inflation or deflation of
metabolite-sets, which in turn will affect the enrichment analysis
results. Hence, we can use measured metabolite IDs to guide the
selection of PK IDs. This is crucial to circumvent inflation or
deflation of metabolite-sets, which in turn will affect the enrichment
analysis results.<br>
This is something we are currently working on and hope to provide within
the next release, so stay tuned.<br></p>
<center>
<div class="float">
<img src="Fig.3.png" style="width:50.0%" alt="Fig. 3: Mapping problems in prior knowledge metabolite-sets when translating metabolite IDs and the connection to detected (measured input metabolites)."><div class="figcaption">Fig. 3: Mapping problems in prior knowledge
metabolite-sets when translating metabolite IDs and the connection to
detected (measured input metabolites).</div>
</div>
</center>
<p><br></p>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="sect5">5. Run enrichment analysis<a class="anchor" aria-label="anchor" href="#sect5"></a>
</h2>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
<p><br>
There are two options:<br>
1. Over Representation Analysis (ORA) that determines if a set of
features (=metabolic pathways) are over-represented in the selection of
features (=metabolites) from the data in comparison to all measured
features (metabolites) using the Fishers exact test
<code>cluster_ora</code>. This can be applied to clusters of metabolites
as for example the results from <code><a href="../../reference/mca_2cond.html">mca_2cond()</a></code> or
<code>core()</code> function. If you want to have more details on these
clustering methods please visit the vignette <a href="https://saezlab.github.io/MetaProViz/articles/Standard%20Metabolomics.html" class="external-link">Standard
Metabolomics</a> or <a href="https://saezlab.github.io/MetaProViz/articles/core%20Metabolomics.html" class="external-link">core
Metabolomics</a>.<br></p>
<p><br>
2. Enrichment analysis on standard differential analysis results. We
offer ORA <code>standard_ora</code>, but there are many other
statistical tests that can be used for enrichment analysis. The full
scope of different methods is beyond the scope of MetaProViz, but are
available in decoupleR <span class="citation">(Badia-I-Mompel et al.
2022)</span> packages from our group.</p>
<p><br><br><span style="text-decoration:underline">As part of the
<strong>MetaProViz</strong> package you can load the example data using
the function <code>toy_data()</code></span> to test the enrichment
analysis:<br><code>1.</code>Cell line experiment <strong>(CellLine)</strong><br>
Here we choose an example datasets, which is publicly available on <a href="https://www.metabolomicsworkbench.org/data/DRCCMetadata.php?Mode=Project&amp;ProjectID=PR001418" class="external-link">metabolomics
workbench project PR001418</a> including metabolic profiles of human
renal epithelial cells HK2 and cell renal cell carcinoma (ccRCC) cell
lines cultured in Plasmax cell culture media <span class="citation">(Sciacovelli et al. 2022)</span>. The raw data are
available via <a href="https://www.metabolomicsworkbench.org/data/DRCCMetadata.php?Mode=Study&amp;StudyID=ST002224&amp;StudyType=MS&amp;ResultType=1" class="external-link">metabolomics
workbench study ST002224</a> were intracellular metabolomics of HK2 and
ccRCC cell lines 786-O, 786-M1A and 786-M2A were performed.<br>
We have performed pre-processing and differential analysis (details can
be found in the vignette vignette <a href="https://saezlab.github.io/MetaProViz/articles/Standard%20Metabolomics.html" class="external-link">Standard
Metabolomics</a>) and and here we load the differential metabolite
analysis results for the comparison of 786M-1A versus HK2.<br></p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the Pre-processed intracellular data:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">intracell_dma</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Intra_DMA_786M1A_vs_HK2</span> <span class="op">&lt;-</span> <span class="va">intracell_dma</span></span></code></pre></div>
<p><code>2.</code> Tissue experiment <strong>(Tissue)</strong><br>
Here we chose publicly available data from the <a href="https://www.cell.com/cancer-cell/comments/S1535-6108(15)00468-7#supplementaryMaterial" class="external-link">paper</a>
“An Integrated Metabolic Atlas of Clear Cell Renal Cell Carcinoma”,
which includes metabolomic profiling on 138 matched clear cell renal
cell carcinoma (ccRCC)/normal tissue pairs.<br>
We have performed differential analysis (details can be found in the
vignette <a href="https://saezlab.github.io/MetaProViz/articles/Metadata%20Analysis.html" class="external-link">Metadata
Analysis</a>) and and here we load the differential metabolite analysis
results for the comparison of Tumour versus Normal.<br></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the example data:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">tissue_dma</span><span class="op">)</span></span>
<span><span class="va">Tissue_TvsN</span> <span class="op">&lt;-</span> <span class="va">tissue_dma</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">tissue_dma_old</span><span class="op">)</span></span>
<span><span class="va">Tissue_TvsN_Old</span> <span class="op">&lt;-</span> <span class="va">tissue_dma_old</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">tissue_dma_young</span><span class="op">)</span></span>
<span><span class="va">Tissue_TvsN_Young</span> <span class="op">&lt;-</span> <span class="va">tissue_dma_young</span></span></code></pre></div>
<p><br><br></p>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<pre><code><span><span class="co">#&gt; R version 4.5.2 (2025-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.3 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Etc/UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] tibble_3.3.0       stringr_1.6.0      dplyr_1.1.4        purrr_1.2.0        rlang_1.1.6        magrittr_2.0.4    </span></span>
<span><span class="co">#&gt; [7] MetaProViz_3.99.26 BiocStyle_2.38.0  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] RColorBrewer_1.1-3          rstudioapi_0.17.1           jsonlite_2.0.0              ggbeeswarm_0.7.2           </span></span>
<span><span class="co">#&gt;   [5] farver_2.1.2                rmarkdown_2.30              fs_1.6.6                    ragg_1.5.0                 </span></span>
<span><span class="co">#&gt;   [9] vctrs_0.6.5                 memoise_2.0.1               rstatix_0.7.3               htmltools_0.5.8.1          </span></span>
<span><span class="co">#&gt;  [13] S4Arrays_1.10.0             progress_1.2.3              curl_7.0.0                  ComplexUpset_1.3.3         </span></span>
<span><span class="co">#&gt;  [17] decoupleR_2.16.0            broom_1.0.10                cellranger_1.1.0            SparseArray_1.10.2         </span></span>
<span><span class="co">#&gt;  [21] Formula_1.2-5               sass_0.4.10                 parallelly_1.45.1           bslib_0.9.0                </span></span>
<span><span class="co">#&gt;  [25] htmlwidgets_1.6.4           desc_1.4.3                  CARNIVAL_2.20.0             plyr_1.8.9                 </span></span>
<span><span class="co">#&gt;  [29] httr2_1.2.1                 lubridate_1.9.4             cachem_1.1.0                igraph_2.2.1               </span></span>
<span><span class="co">#&gt;  [33] lifecycle_1.0.4             pkgconfig_2.0.3             Matrix_1.7-4                R6_2.6.1                   </span></span>
<span><span class="co">#&gt;  [37] fastmap_1.2.0               MatrixGenerics_1.22.0       selectr_0.5-0               digest_0.6.39              </span></span>
<span><span class="co">#&gt;  [41] colorspace_2.1-2            patchwork_1.3.2             S4Vectors_0.48.0            textshaping_1.0.4          </span></span>
<span><span class="co">#&gt;  [45] GenomicRanges_1.62.0        RSQLite_2.4.4               ggpubr_0.6.2                labeling_0.4.3             </span></span>
<span><span class="co">#&gt;  [49] timechange_0.3.0            httr_1.4.7                  abind_1.4-8                 compiler_4.5.2             </span></span>
<span><span class="co">#&gt;  [53] bit64_4.6.0-1               withr_3.0.2                 S7_0.2.1                    backports_1.5.0            </span></span>
<span><span class="co">#&gt;  [57] BiocParallel_1.44.0         carData_3.0-5               DBI_1.2.3                   logger_0.4.1               </span></span>
<span><span class="co">#&gt;  [61] OmnipathR_3.19.2            R.utils_2.13.0              ggsignif_0.6.4              cosmosR_1.18.0             </span></span>
<span><span class="co">#&gt;  [65] MASS_7.3-65                 rappdirs_0.3.3              DelayedArray_0.36.0         sessioninfo_1.2.3          </span></span>
<span><span class="co">#&gt;  [69] rjson_0.2.23                scatterplot3d_0.3-44        gtools_3.9.5                tools_4.5.2                </span></span>
<span><span class="co">#&gt;  [73] vipor_0.4.7                 beeswarm_0.4.0              qcc_2.7                     zip_2.3.3                  </span></span>
<span><span class="co">#&gt;  [77] R.oo_1.27.1                 glue_1.8.0                  grid_4.5.2                  checkmate_2.3.3            </span></span>
<span><span class="co">#&gt;  [81] reshape2_1.4.5              lpSolve_5.6.23              generics_0.1.4              gtable_0.3.6               </span></span>
<span><span class="co">#&gt;  [85] tzdb_0.5.0                  R.methodsS3_1.8.2           tidyr_1.3.1                 hms_1.1.4                  </span></span>
<span><span class="co">#&gt;  [89] xml2_1.5.0                  car_3.1-3                   XVector_0.50.0              BiocGenerics_0.56.0        </span></span>
<span><span class="co">#&gt;  [93] ggrepel_0.9.6               pillar_1.11.1               vroom_1.6.6                 limma_3.66.0               </span></span>
<span><span class="co">#&gt;  [97] later_1.4.4                 splines_4.5.2               lattice_0.22-7              bit_4.6.0                  </span></span>
<span><span class="co">#&gt; [101] tidyselect_1.2.1            knitr_1.50                  gridExtra_2.3               bookdown_0.45              </span></span>
<span><span class="co">#&gt; [105] IRanges_2.44.0              Seqinfo_1.0.0               SummarizedExperiment_1.40.0 svglite_2.2.2              </span></span>
<span><span class="co">#&gt; [109] stats4_4.5.2                xfun_0.54                   Biobase_2.70.0              statmod_1.5.1              </span></span>
<span><span class="co">#&gt; [113] factoextra_1.0.7            matrixStats_1.5.0           pheatmap_1.0.13             stringi_1.8.7              </span></span>
<span><span class="co">#&gt; [117] yaml_2.3.10                 kableExtra_1.4.0            evaluate_1.0.5              codetools_0.2-20           </span></span>
<span><span class="co">#&gt; [121] tcltk_4.5.2                 qvalue_2.42.0               hash_2.2.6.3                BiocManager_1.30.27        </span></span>
<span><span class="co">#&gt; [125] Polychrome_1.5.4            cli_3.6.5                   systemfonts_1.3.1           jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt; [129] EnhancedVolcano_1.13.2      Rcpp_1.1.0                  readxl_1.4.5                XML_3.99-0.20              </span></span>
<span><span class="co">#&gt; [133] parallel_4.5.2              ggfortify_0.4.19            pkgdown_2.2.0               ggplot2_4.0.1              </span></span>
<span><span class="co">#&gt; [137] readr_2.1.6                 blob_1.2.4                  prettyunits_1.2.0           viridisLite_0.4.2          </span></span>
<span><span class="co">#&gt; [141] scales_1.4.0                writexl_1.5.4               inflection_1.3.7            crayon_1.5.3               </span></span>
<span><span class="co">#&gt; [145] rvest_1.0.5</span></span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-Badia-I-Mompel2022" class="csl-entry">
Badia-I-Mompel, Pau, Jesús Vélez Santiago, Jana Braunger, Celina Geiss,
Daniel Dimitrov, Sophia Müller-Dott, Petr Taus, et al. 2022.
<span>“decoupleR: Ensemble of Computational Methods to Infer Biological
Activities from Omics Data.”</span> <em>Bioinformatics Advances</em> 2
(1): vbac016. <a href="https://doi.org/10.1093/bioadv/vbac016" class="external-link">https://doi.org/10.1093/bioadv/vbac016</a>.
</div>
<div id="ref-Braisted2023" class="csl-entry">
Braisted, John, Andrew Patt, Cole Tindall, Timothy Sheils, Jorge Neyra,
Kyle Spencer, Tara Eicher, and Ewy A Mathé. 2023. <span>“RaMP-DB 2.0: A
Renovated Knowledgebase for Deriving Biological and Chemical Insight
from Metabolites, Proteins, and Genes.”</span> <em>Bioinformatics</em>,
no. 1 (January). <a href="https://doi.org/10.1093/bioinformatics/btac726" class="external-link">https://doi.org/10.1093/bioinformatics/btac726</a>.
</div>
<div id="ref-Castanza2022" class="csl-entry">
Castanza, Anthony Scott, Jill Marie Recla, David Eby, Helga
Thorvaldsdottir, Carol J. Bult, and Jill P Mesirov. 2022. <span>“The
Molecular Signatures Database Revisited: Extending Support for Mouse
Data.”</span> <em>BioRxiv</em>, October. <a href="https://doi.org/10.1101/2022.10.24.513539" class="external-link">https://doi.org/10.1101/2022.10.24.513539</a>.
</div>
<div id="ref-Dugourd2021" class="csl-entry">
Dugourd, Aurelien, Christoph Kuppe, Marco Sciacovelli, Enio Gjerga,
Attila Gabor, Kristina B Emdal, Vitor Vieira, et al. 2021. <span>“Causal
Integration of Multi-Omics Data with Prior Knowledge to Generate
Mechanistic Hypotheses.”</span> <em>Molecular Systems Biology</em>, no.
1 (January): e9730. <a href="https://doi.org/10.15252/msb.20209730" class="external-link">https://doi.org/10.15252/msb.20209730</a>.
</div>
<div id="ref-Farr_Dimitrov2024" class="csl-entry">
Farr, Elias, Daniel Dimitrov, Christina Schmidt, Denes Turei, Sebastian
Lobentanzer, Aurelien Dugourd, and Julio Saez-Rodriguez. 2024.
<span>“MetalinksDB: A Flexible and Contextualizable Resource of
Metabolite-Protein Interactions.”</span> <em>Briefings in
Bioinformatics</em>, no. 4 (May). <a href="https://doi.org/10.1093/bib/bbae347" class="external-link">https://doi.org/10.1093/bib/bbae347</a>.
</div>
<div id="ref-Gaude2016" class="csl-entry">
Gaude, Edoardo, and Christian Frezza. 2016. <span>“Tissue-Specific and
Convergent Metabolic Transformation of Cancer Correlates with Metastatic
Potential and Patient Survival.”</span> <em>Nature Communications</em> 7
(October): 13041. <a href="https://doi.org/10.1038/ncomms13041" class="external-link">https://doi.org/10.1038/ncomms13041</a>.
</div>
<div id="ref-Kanehisa2000" class="csl-entry">
Kanehisa, M, and S Goto. 2000. <span>“KEGG: Kyoto Encyclopedia of Genes
and Genomes.”</span> <em>Nucleic Acids Research</em> 28 (1): 27–30. <a href="https://doi.org/10.1093/nar/28.1.27" class="external-link">https://doi.org/10.1093/nar/28.1.27</a>.
</div>
<div id="ref-Liberzon2015" class="csl-entry">
Liberzon, Arthur, Chet Birger, Helga Thorvaldsdóttir, Mahmoud Ghandi,
Jill P Mesirov, and Pablo Tamayo. 2015. <span>“The Molecular Signatures
Database (MSigDB) Hallmark Gene Set Collection.”</span> <em>Cell
Systems</em> 1 (6): 417–25. <a href="https://doi.org/10.1016/j.cels.2015.12.004" class="external-link">https://doi.org/10.1016/j.cels.2015.12.004</a>.
</div>
<div id="ref-Sciacovelli_Dugourd2022" class="csl-entry">
Sciacovelli, Marco, Aurelien Dugourd, Lorea Valcarcel Jimenez, Ming
Yang, Efterpi Nikitopoulou, Ana S H Costa, Laura Tronci, et al. 2022.
<span>“Dynamic Partitioning of Branched-Chain Amino Acids-Derived
Nitrogen Supports Renal Cancer Progression.”</span> <em>Nature
Communications</em> 13 (1): 7830. <a href="https://doi.org/10.1038/s41467-022-35036-4" class="external-link">https://doi.org/10.1038/s41467-022-35036-4</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Christina Schmidt, Denes Turei, Dimitrios Prymidis, Macabe Daley, Jannik Franken, Julio Saez-Rodriguez, Christian Frezza.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
