---
title: "Prior Knowledge - Access & Integration"
author:
  - name: Christina Schmidt
    affiliation:
      - Heidelberg University
  - name: Macabe Daley
    affiliation:
      - Heidelberg University
output:
  BiocStyle::html_document:
    self_contained: true
    toc: true
    toc_float: true
    toc_depth: 5
    code_folding: show
vignette: >
  %\VignetteIndexEntry{Standard Metabolomics}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
bibliography: bibliography.bib
editor_options:
  chunk_output_type: console
  markdown:
    wrap: sentence
---

```{=html}
<style>
.vscroll-plot {
    width: 850px;
    height: 500px;
    overflow-y: scroll;
    overflow-x: hidden;
}
</style>
```

```{r chunk_setup, include = FALSE}
knitr::opts_chunk$set(
    collapse = TRUE,
    comment = "#>"
)
```

# <img src="Hexagon_MetaProViz.png" align="right" width="200"/>

\
[In this tutorial we showcase how to use **MetaProViz** prior knowledge]{style="text-decoration:underline"}:\
- 1.
[to understand detected metabolite IDs in measured data.](#sect1)

\- 2.
[to access metabolite prior knowledge and metabolite-gene prior knowledge networks.](#sect2)

\- 3.
[to link experimental data to prior knowledge - the Do's and Don'ts.](#sect3)

\- 4.
[to deal with many-to-many mapping in your metabolite identifiers.](#sect4)

\- 5.
[to perform pathway enrichment analysis.](#sect5)\

\
First if you have not done yet, install the required dependencies and load the libraries:

```{r load-packages message=FALSE, warning=FALSE}
# 1. Install Rtools if you haven’t done this yet, using the appropriate version (e.g.windows or macOS).
# 2. Install the latest development version from GitHub using devtools
# devtools::install_github("https://github.com/saezlab/MetaProViz")

library(MetaProViz)

library(magrittr)
library(rlang)
library(purrr)
library(dplyr)
library(stringr)
library(tibble)

# Please install the Biocmanager Dependencies:
# BiocManager::install("clusterProfiler")
# BiocManager::install("EnhancedVolcano")
# BiocManager::install("cosmosR")
```

\
\

:::: {.progress .progress-striped .active}
::: {.progress-bar .progress-bar-success style="width: 100%"}
:::
::::

# Loading the example data

:::: {.progress .progress-striped .active}
::: {.progress-bar .progress-bar-success style="width: 100%"}
:::
::::

\

[As part of the **MetaProViz** package you can load the example feature metadata using the function `toy_data()`]{style="text-decoration:underline"}:\
`1.` Metadata of cell line experiment **(CellLine)**\
Example datasets is publicly available on [metabolomics workbench project PR001418](https://www.metabolomicsworkbench.org/data/DRCCMetadata.php?Mode=Project&ProjectID=PR001418) including metabolic profiles of human renal epithelial cells HK2 and cell renal cell carcinoma (ccRCC) cell lines cultured in Plasmax cell culture media [@Sciacovelli_Dugourd2022].\

```{r load-data}
# Load the Pre-processed intracellular data:
data(cellular_meta)

FeatureMetadata_Cells <- cellular_meta%>%
column_to_rownames("Metabolite")
```

\
`2.` Metadata of patients tissue experiment **(Tissue)**\
Example datasets is publicly available in the [paper](https://www.cell.com/cancer-cell/comments/S1535-6108(15)00468-7#supplementaryMaterial) "An Integrated Metabolic Atlas of Clear Cell Renal Cell Carcinoma", which includes metabolomic profiling on 138 matched clear cell renal cell carcinoma (ccRCC)/normal tissue pairs.\

```{r load-data-2}
# Load the example data:
data(tissue_meta)

FeatureMetadata_Tissue <- tissue_meta%>%
column_to_rownames("Metabolite")

```

\
`3.` Metadata of Biocrates kit **(Biocrates)**\
Here we use the Biocrates kit feature information of the ["MxP® Quant 500 XL kit"](https://biocrates.com/mxp-quant-500-xl/) that covers more than 1,000 metabolites from various biochemical classes.

```{r load-data-3}
# Load the example data:
data(biocrates_features)
FeatureMetadata_Biocrates <- biocrates_features
```

\
\

:::: {.progress .progress-striped .active}
::: {.progress-bar .progress-bar-success style="width: 100%"}
:::
::::

# 1. Metabolite IDs in measured data {#sect1}

:::: {.progress .progress-striped .active}
::: {.progress-bar .progress-bar-success style="width: 100%"}
:::
::::

***Assigning Metabolite IDs to measured data***\
The difficulty with assigning metabolite IDs to measured data is the uncertainty in the detection of metabolites.
Indeed, differentiation of structural isomers (both constitutional isomers and stereoisomers) as for example the distinction between enantiomers.
This leads to loss of information and hence uncertainty is assigning metabolite IDs.\
One example is the metabolite Alanine, which can occur in its L- or D- form.
If in an experiment those enantiomers have not been distinguished, the correct way would be to either assign two metabolite IDs (L- and D-Alanine) or a more general Alanine ID without chiral information.
Yet, in reality this is not as trivial:\

```{r code, echo=FALSE}
# Create DF for Alanine:
Alanine <- data.frame(
    TrivialName = c("D-Alanine", "L-Alanine", "Alanine", "Alanine zwitterion"),
    HMDB= c("HMDB0001310",  "HMDB0000161", NA, NA),
    ChEBI = c("15570", "16977", "16449", "66916" ),
    KEGG = c("C00133", "C00041", "C01401", NA ),
    PubChem = c("71080" , "5950", "602", "57383916"),
    stringsAsFactors = FALSE)

# Print table:
Alanine%>%
kableExtra::kbl(caption = "Available Alanine IDs in HMDB and ChEBI.") %>%
kableExtra::kable_classic(full_width = FALSE, html_font = "Cambria", font_size = 12)
```

\
Indeed, dependent on the database, the Alanine metabolite can have different IDs available:\

```{r plot, echo=FALSE}
# Print Plot:
AlaninePlot <- Alanine%>%
tidyr::pivot_longer(cols = c("HMDB", "ChEBI", "KEGG", "PubChem"),
names_to = "Conditions",
values_to = "ID")%>%
tidyr::drop_na()
AlaninePlot$`IDs in Database` <- as.numeric(1)

Plot <- ggplot2::ggplot(AlaninePlot, ggplot2::aes(fill=TrivialName, y=`IDs in Database`, x=Conditions)) +
        ggplot2::geom_bar(position="stack", stat="identity")+
        ggplot2::scale_fill_manual(values = c("D-Alanine" = "#9F0162"  , "L-Alanine" = "#006384", "Alanine" = "grey", "Alanine zwitterion" = "#00735C"))+
        ggplot2::theme_classic()+
        ggplot2::ggtitle("Alanine IDs in in different databases")+
        ggplot2::theme(axis.text.x = ggplot2::element_text(angle = 90, hjust = 1))+
        ggplot2::labs(x = "Database", y = "Number of IDs")

Plot_Sized <-  MetaProViz:::plot_grob_superplot(input_plot=Plot, metadata_info=c(Conditions="Conditions", Superplot="Conditions"), metadata_sample=AlaninePlot, plot_name = "Alanine IDs in in different databases", subtitle = NULL, plot_type="Bar")

plot_height <- grid::convertUnit(Plot_Sized$height, 'cm', valueOnly = TRUE)
plot_width <- grid::convertUnit(Plot_Sized$width, 'cm', valueOnly = TRUE)
Plot_Sized %<>%
{ggplot2::ggplot() + ggplot2::annotation_custom(.)} %>%
add(ggplot2::theme(panel.background = ggplot2::element_rect(fill = "transparent")))

# ggplot2::ggsave(filename = "AlaninePlot.svg", plot = Plot_Sized, width = plot_width,  height = plot_height, unit="cm")

Plot_Sized
```

\
For instance, if we want to assign a HMDB ID, we have to assign both "HMDB0001310", "HMDB0000161" to the metabolite Alanine, for ChEBI we could assign only one, "16449", but this may lead to other problems as the ChEBI ID is not specific and may not be part of certain metabolic pathways.
The reason for this is that substrate chirality is critical to enzymatic processes and stereo selectivity of enzymes to be homochiral with predominance of one particular enantiomer (e.g. D-sugars, L-amino acids, etc.).\
to showcase the severity of this problem, we can look at the occurrence of those metabolites in metabolic pathways across different databases.
to do so we searched for those metabolite IDs in the RaMP database [@Braisted2023] and extracted the pathways they are part of:\

```{r data-prep, echo=FALSE}
data(alanine_pathways)

Pathways <- alanine_pathways

Count <- Pathways %>%
tidyr::unite(Pathway, c("pathwaySource", "pathwayName"), sep = ": ")%>%
dplyr::group_by(inputId) %>%
dplyr::summarise(PathwayCount = dplyr::n_distinct(Pathway), .groups = 'drop')%>%
tidyr::separate(inputId, into = c("Database", "ID"), sep = ":", remove = TRUE)

Alanine <- merge(x = AlaninePlot%>%dplyr::rename("Database"="Conditions") ,
y= Count[,2:3],
by = "ID",
all.x=TRUE)%>%
dplyr::mutate(PathwayCount = replace(PathwayCount, is.na(PathwayCount), 0))%>%
dplyr::arrange(desc( TrivialName)) %>%
dplyr::select(TrivialName, ID, Database, PathwayCount)

# Print table:
Alanine%>%
kableExtra::kbl(caption = "Alanine IDs in HMDB and ChEBI mapped to pathways from wiki, KEGG and Reactome using RamP.") %>%
kableExtra::kable_classic(full_width = FALSE, html_font = "Cambria", font_size = 12)
```

\

```{r plot-2, echo=FALSE}
# BarGraph
Plot <- ggplot2::ggplot(Alanine, ggplot2::aes(fill=TrivialName, y=`PathwayCount`, x=Database)) +
        ggplot2::geom_bar(position="stack", stat="identity")+
        ggplot2::scale_fill_manual(values = c("D-Alanine" = "#9F0162"  , "L-Alanine" = "#006384", "Alanine" = "grey", "Alanine zwitterion" = "#00735C"))+
        ggplot2::theme_classic()+
        ggplot2::ggtitle("Alanine IDs in in different pathway")+
        ggplot2::theme(axis.text.x = ggplot2::element_text(angle = 90, hjust = 1))+
        ggplot2::labs(x = "Database", y = "Number of Pathways")

Plot_Sized <-  MetaProViz:::plot_grob_superplot(input_plot=Plot, metadata_info=c(Conditions="Conditions", Superplot="Conditions"), metadata_sample=AlaninePlot, plot_name = "Alanine IDs in in different databases", subtitle = NULL, plot_type="Bar")

plot_height <- grid::convertUnit(Plot_Sized$height, 'cm', valueOnly = TRUE)
plot_width <- grid::convertUnit(Plot_Sized$width, 'cm', valueOnly = TRUE)
Plot_Sized %<>%
{ggplot2::ggplot() + ggplot2::annotation_custom(.)} %>%
add(ggplot2::theme(panel.background = ggplot2::element_rect(fill = "transparent")))

# ggplot2::ggsave(filename = "AlaninePlot_Pathways.svg", plot = Plot_Sized, width = plot_width,  height = plot_height, unit="cm")

Plot_Sized
```

\

```{r plot-3, eval=FALSE, include=FALSE}
# Make BarGraph per pathway and ID type
Alanine_Path <- Pathways%>%
tidyr::separate(commonName, into = c("commonName", "secondaryName"), sep = ",A", remove = FALSE)%>%
tidyr::separate(commonName, into = c("commonName", "secondaryName"), sep = ",D", remove = FALSE)%>%
dplyr::group_by(pathwaySource, commonName) %>%
dplyr::summarise(PathwayCount = dplyr::n_distinct(pathwayName),
commonNames = paste(unique(commonName), collapse = ", "),
.groups = 'drop')


# BarGraph
Plot <- ggplot2::ggplot(Alanine_Path, ggplot2::aes(fill=commonNames, y=PathwayCount , x=pathwaySource)) +
        ggplot2::geom_bar(position="stack", stat="identity")+
        ggplot2::scale_fill_manual(values = c("D-Alanine" = "#9F0162"  , "L-Alanine" = "#006384", "Alanine" = "grey", "Alanine zwitterion" = "#00735C"))+
        ggplot2::theme_classic()+
        ggplot2::ggtitle("Alanine IDs in in different databases")+
        ggplot2::theme(axis.text.x = ggplot2::element_text(angle = 90, hjust = 1))+
        ggplot2::labs(x = "Pathway Source", y = "Number of Pathways")


#* https://www.sciencedirect.com/science/article/pii/S0731708521005410
#* This problem is even excarbated in lipidomics, where different levels of residues....
```

\
This showcases if we choose the ChEBI ID for Alanine (ChEBI ID 16449), if experimentally the distinction was not possible, we will not map to any pathway even though the metabolite is part of many pathways.
Hence, we recommend to assign multiple IDs to a measured ID, where specificity in detection is not given.\

\
Yet, many metabolomics studies do not report multiple IDs, but rather one ID per measured peak.
In some cases the chosen ID is correct in representing the degree of ambiguity in the detection, e.g. "Alanine-Zwitterion", whilst in other cases a specific ID, which has not been detected in this specificity, e.g. "L-Alanine".
In both cases this can lead to missed maps to prior knowledge and hence a loss of information as discussed above.\
Hence, we assign multiple IDs to a measured peak, where specificity in detection may not be given (i.e. amino acids).\

```{r data-prep-2}
# Example Cell-line data:
Input_HMDB <- FeatureMetadata_Cells%>%
dplyr::filter(!HMDB == "NA")%>% # ID in the measured data we want to use, hence we remove NA's
dplyr::select("HMDB", "Pathway") # only keep relevant columns

# Add equivalent IDs:
FeatureMetadata_Cells_AddIDs <- equivalent_id(data= Input_HMDB,
metadata_info = c(InputID="HMDB"),# ID in the measured data, here we use the HMDB ID
from = "hmdb")
```

```{r code-2, echo=FALSE}
# Check how our data looks like:
FeatureMetadata_Cells_AddIDs [2:8, 1:3]%>%
kableExtra::kbl(caption = "Preview of the DF `FeatureMetadata_Cells_AddIDs` including new columns with potential additional HMDB IDs assign using equivalent_id() function.") %>%
kableExtra::kable_classic(full_width = FALSE, html_font = "Cambria", font_size = 12)
```

:::: {.progress .progress-striped .active}
::: {.progress-bar .progress-bar-success style="width: 100%"}
:::
::::

# 2. Accessing Prior Knowledge {#sect2}

:::: {.progress .progress-striped .active}
::: {.progress-bar .progress-bar-success style="width: 100%"}
:::
::::

Metabolite prior knowledge (PK) is essential for the interpretation of metabolomics data.
It can be used to perform pathway enrichment analysis, compound class enrichment analysis, and by using specific PK databases, it can be used to study the connection of metabolites and receptors or transporters.Since the quality and content of the PK will dictate the success of the downstream analysis and biological interpretation, it is important to ensure the PK is used correctly.\
Specifically in metabolite PK, the many different PK databases and resources pose several issues.
Indeed, the metabolite identifiers (e.g. KEGG, HMDB, PubChem, etc.) are not standardized across databases, and the same metabolite can have multiple identifiers in different databases.
This is known as the many-to-many mapping problem.
Moreover, metabolic pathways that are the basis of the PK databases also includes co-factors such as ions or other small molecules that are not only part of most reactions, but can also not be detected in experimentally acquired data (e.g. H2O, CO2, etc).\
to address these issues and provide a standardized way to access and integrate metabolite PK, we provide access to several prior knowledge resources and removed molecules such as water.
We terme this collection of metabolite sets MetSigDB (Metabolite signature database).

## KEGG pathway-metabolite sets

KEGG pathways that are loaded via KEGG API using the package `KEGGREST` and can be used to perform pathway analysis [@Kanehisa2000].
**(KEGG_Pathways)**\

```{r metsigdb-kegg}
# This will use KEGGREST to query the KEGG API to load the pathways:
KEGG_Pathways <- metsigdb_kegg()
```

\

```{r code-3, echo=FALSE}
# Check how our data looks like:
KEGG_Pathways[c(1:3,52:54),]%>%
kableExtra::kbl(caption = "Preview of the DF `KEGG_Pathways`.") %>%
kableExtra::kable_classic(full_width = FALSE, html_font = "Cambria", font_size = 12)
```

## Chemical class-metabolite sets

The chemical class-metabolite sets are based on the classification of metabolites into chemical classes, which can be used to perform compound class enrichment analysis.\
The chemical class-metabolite sets were curated by RaMP-DB, which used ClassyFire [@Braisted2023].
Here we access them via OmnipathR.\

```{r metsigdb-chemicalclass}
ChemicalClass_MetabSet <- metsigdb_chemicalclass()
```

\

```{r code-4, echo=FALSE}
# Check how our data looks like:
ChemicalClass_MetabSet[c(1:5,80:83, 500:503),]%>%
kableExtra::kbl(caption = "Preview of the DF `ChemicalClass_MetabSet`.") %>%
kableExtra::kable_classic(full_width = FALSE, html_font = "Cambria", font_size = 12)
```

## Create pathway-metabolite sets

The function `make_gene_metab_set` can be used to convert gene names to metabolite names by using a PK network of metabolic reactions calls CosmosR [@Dugourd2021].
This function is useful if you want to perform pathway enrichment analysis on available gene-sets such as the Hallmarks gene-sets from MSigDB [@Castanza2022].
Moreover, it enables you to perform combined pathway enrichment analysis on metabolite-gene sets, if you have other data types such as proteomics measuring the enzymes expression.\
The Hallmarks [@Liberzon2015] gene-set and Gaude [@Gaude2016] gene-set is available in the package `MetaProViz` and can be loaded using the function `metsigdb_hallmarks()` and `metsigdb_gaude` respectively.\

```{r load-data-4}
# Load the example gene-sets:
data(hallmarks)
Hallmark_Pathways <- hallmarks

data(gaude_pathways)
Gaude_Pathways <- gaude_pathways
```

\

```{r code-5, echo=FALSE}
# Check how our data looks like:
Hallmark_Pathways[c(997:998,2916:2917, 2916:2918),]%>%
kableExtra::kbl(caption = "Preview of the DF `Hallmarks_Pathways` including gene-sets usable for pathway enrichment analysis.", row.names=FALSE) %>%
kableExtra::kable_classic(full_width = FALSE, html_font = "Cambria", font_size = 12)
```

\

```{r code-6, echo=FALSE}
# Check how our data looks like:
Gaude_Pathways[c(486:490, 1504, 1507, 1531, 49:50, 779),]%>%
kableExtra::kbl(caption = "Preview of the DF `gaude_Pathways` including gene-sets usable for pathway enrichment analysis.", row.names=FALSE) %>%
kableExtra::kable_classic(full_width = FALSE, html_font = "Cambria", font_size = 12)
```

\
Now we can use the function `make_gene_metab_set` to translate the gene names to metabolite names.

```{r make-gene-metab-set}
# Translate gene names to metabolite names
Hallmarks_GeneMetab <- make_gene_metab_set(input_pk=Hallmark_Pathways,
metadata_info=c(Target="gene"),
pk_name="Hallmarks")

Gaude_GeneMetab <- make_gene_metab_set(input_pk=Gaude_Pathways,
metadata_info=c(Target="gene"),
pk_name="Gaude")
```

\

```{r data-prep-3, echo=FALSE}
# Check how our data looks like:
Hallmarks_GeneMetab[["GeneMetabSet"]]%>%
dplyr::filter(term=="HALLMARK_GLYCOLYSIS")%>%
dplyr::slice(c(59,142,193, 201:203))%>%
kableExtra::kbl(caption = "Preview of the DF `Hallmarks_GeneMetab` including gene-sets usable for pathway enrichment analysis.") %>%
kableExtra::kable_classic(full_width = FALSE, html_font = "Cambria", font_size = 12)

```

\

```{r code-7, echo=FALSE, include =FALSE}
# plot out how the metabolite set size is different from the original gene-set size
# Show bar graph of the size of the gene set versus the metabolite set --> Show both tables (pathways gene names and pathways metabolite names) and show the difference in size (e.g. 10% of the genes are translated to metabolites, 90% are not metabolic enzymes and hence only genes)


```

\
Given that we have the gene-metabolite-sets, we can now also run enrichment analysis on combined data types, once including the metabolite Log2FC and one including gene Log2FC from e.g. transcriptomics or proteomics data.
Yet, it is important to keep in mind that generally we detect less metabolites than genes and hence this may bias the results obtained from combined enrichment analysis.

## MetaLinksDB metabolite-receptor & metabolite-transporter sets

The MetaLinks database is a manually curated database of metabolite-receptor and metabolite-transporter sets that can be used to study the connection of metabolites and receptors or transporters [@Farr_Dimitrov2024].\

```{r metsigdb-metalinks}
MetaLinksDB <- metsigdb_metalinks()
```

\

```{r data-prep-4, echo=FALSE}
# Check how our data looks like:
MetaLinksDB[
    c(12:13,17:18,432,466),
    c(1:2,6:8, 12:15)
    ] %>%
    kableExtra::kbl(caption = "Preview of the DF `MetaLinksDB` including metabolite-receptor sets.", row.names=FALSE) %>%
    kableExtra::kable_classic(full_width = FALSE, html_font = "Cambria", font_size = 12)

dplyr::filter(MetaLinksDB, !is.na(term_specific))[c(1,50, 90, 101),] %>%
kableExtra::kbl(caption = "Preview of the metabolite-receptor and metabolite-transporter sets.", row.names=FALSE) %>%
kableExtra::kable_classic(full_width = FALSE, html_font = "Cambria", font_size = 12)
```

## Comparison of PK coverage

We have now loaded a number of different PK sources (details above) and aim to compare the overlap in coverage between these PK sources to understand if there are certain genes or metabolites covered by one PK resource but not the others.
As an example, we will compare the resources containing gene-metabolite sets (Hallmarks, Gaude, MetalinksDB) using `compare_pk()` function, which will generate a combined datatable and visualises it as an Upset plot. The Upset plot shows the overlap of coverage, similar to how a Venn diagram works, but enable to visualize multiple combinations clearly.
```{r compare-pk, eval=FALSE}
# Compare the PK resources
pk_comp_res <- compare_pk(data = list(Hallmarks = as.data.frame(Hallmarks_GeneMetab[["GeneMetabSet"]]),
Gaude = as.data.frame(Gaude_GeneMetab[["GeneMetabSet"]]),
MetalinksDB = as.data.frame(MetaLinksDB))
)
```
```{r, echo=FALSE}
knitr::include_graphics("prior-knowledge.Rmd_compare-pk chunk.svg")
```
\
The table includes all features of the three PK resources we decided to compare. Additionally, it includes a column for each PK resource with value 1 if the feature is present in the PK resource and 0 if it is not. The column "Type" specifies with ID type the feature is corresponding to, here either metabolite (HMDB) or gene, which is used for visualisation purposes in the upset plot.\
```{r code-8, echo=FALSE}
# Check how our data looks like:
pk_comp_res$summary_table[c(1:8),]%>%
kableExtra::kbl(caption = "Preview of the DF `pk_comp_res$summary_table` showing coverage of features (genes or metabolites) across PK sources.", row.names=FALSE) %>%
kableExtra::kable_classic(full_width = FALSE, html_font = "Cambria", font_size = 12)
```
\
```{r code-9, echo=FALSE, fig.width=8, fig.height=6}
pk_comp_res$upset_plot
```
\
The upset plot shows that MetalinksDB has unique genes and metabolites not present in the other resources. This is to be expected, since MetalinksDB focuses on receptors, transporters and metabolic enzymes, whilst Gaude and Hallmakrs focus on pathways.\
Since Gaude only focuses on metabolic enzymes of pathways, it contains a low number of genes and hence has a small amount of unique genes and metabolites. In this regard it makes sense that Hallmarks includes many unique genes, since Hallmarks also includes other genes and pathways not related to metabolism.
Between Hallmarks and Gaude we observe a high overlap with metabolites, which can be explained since both gene sets are assigned metabolites by using the same metabolic pathways in the backend of `make_gene_metab_set` and it is important to keep in mind that this dictates the gene-metabolite associations added.

:::: {.progress .progress-striped .active}
::: {.progress-bar .progress-bar-success style="width: 100%"}
:::
::::

# 3. Linking experimental data to prior knowledge {#sect3}

:::: {.progress .progress-striped .active}
::: {.progress-bar .progress-bar-success style="width: 100%"}
:::
::::

Now that we have loaded the prior knowledge and inspected the overlap of genes and metabolites between the different PK resources, in this section we want to link our experimental data to the PK.\
The experimental data in this example will be the data loaded in section 1 (Loading the example data).
Whichever resource the experimental data resembles, we recommend to follow this overall approach:\
\
1.
Determine the metabolite identifiers from the experimental data and inspect their coverage.\
2.
Select the metabolite identifiers from the experimental data, which can be used to link to the PK of interest.\
3.
Connect metabolite identifiers to the PK and assess overlap.\

\
Here one can select any combination of experimental data and PK.
This selection should be based on the experimental data at hand and the biological research question.\
For this vignette, we will focus on the combination of Biocrates experimental data and MetalinksDB PK. The reason for using the landscape of metabolites included in the Biocrates Kit is that Biocrates has assigned mutliple metabolite IDs to each measurement (Lipid maps, HMDB, ChEBI) and if needed assigned multiple IDs to one measurement (e.g. the two HMDB IDs of L- and D-Alanine).

## 1. Determine identifiers and inspect coverage

First we determine the identifiers and inspect the coverage.\
In many experimentally aquired data, there can be different metabolite IDs used to describe the metabolite.
In the Biocrates data the available metabolite identifiers are mostly CHEBI, HMDB, and LIMID that could be linked to PK resources.
Now we select these metabolite identifier columns to count the coverage and look at the combinations of the coverage grouped by the class of metabolite.
This helps us to understand what to expect, when linking the data to prior knowledge.\

```{r compare-pk-2, eval=FALSE}
pk_comp_res_biocft <- compare_pk(data = list(Biocft = FeatureMetadata_Biocrates),
metadata_info = list(Biocft = c("CHEBI", "HMDB", "LIMID")),
plot_name = "Overlap of BioCrates Columns")

```
```{r, echo=FALSE}
knitr::include_graphics("prior-knowledge.Rmd_compare-pk-2 chunk.svg")
```
\

```{r code-10, echo=FALSE}
# Check how our data looks like:
pk_comp_res_biocft$summary_table[c(1:8),]%>%
kableExtra::kbl(caption = "Preview of the DF `pk_comp_res_biocft$summary_table` showing coverage of identifiers.", row.names=FALSE) %>%
kableExtra::kable_classic(full_width = FALSE, html_font = "Cambria", font_size = 12)
```

\
These results tell us that:\

-   Less than half the metabolites had ***CHEBI+HMDB+LIMID (n=413)***, which isn't great but probably higher than we have expected.\

-   If you relied only upon -\

    -   CHEBI: then you would miss at least 256 metabolites with HMDB+LIMID

    -   HMDB: then you would miss at least 54 metabolites with CHEBI+LIMID

    -   LIMID: then you would miss at least 89 metabolites with CHEBI+HMDB

However there are two things to note about these observations above.

The first is that these numbers are minimums, the real values are a little higher once you add for instance metabolites with just CHEBI or just LIMID, etc.
So actually, relying upon only LIMID means we miss 151 biocrates metabolites (89 with only HMDB/ChEBI + 15 with only CHEBI + 8 with only HMDB and 39 with no LIMID/CHEBI/HMDB identifiers).

The second point is more nuanced but just as important: we need to keep in mind that there is not a direct 1:1 relationship between the number of Biocrates metabolites with, for instance, a ChEBI ID, and the number of unique ChEBI IDs that they map to.
This upset plot is treating each Biocrates metabolite as an individual entry and categorising whether it has a ChEBI, HMDB, LIMID ID, etc.
However we are not considering what those ChEBI, HMDB, LIMIDs actually are.
So it is possible that there are cases where the number of metabolites is higher, or lower, than the number of unique IDs due to multimapping.

For example, if we have 3 Biocrates metabolites with only HMDB IDs. These would count as 3 entries in the upset plot.
But two of these Biocrates metabolites could both map to the same HMDB ID, whereas the other Biocrates metabolite could map to 6 different HMDBs.
In this case, there are 3 Biocrates metabolites with HMDBs but 7 unique HMDBs.
Hence, it is important to keep this in mind when interpreting the upset plot.

## 2. Select identifiers to link to PK of interest

We have inspected the experimental coverage of the metabolite IDs from the Biocrates kit.
Next, we have to choose which metabolite identifier to use to link to the PK of choice.
Often this choice is dictated by the prior knowledge resource as most of them use a specific identifier.\
Here we will use MetalinksDB, which uses HMDB IDs as metabolite identifiers, hence it is best to use the HMDB IDs to link the Biocrates features to the resource.
Noteworthy, in some cases the Biocrates metabolites have multiple HMDB IDs listed per metabolite:\

```{r code-11, fig.width=8, fig.height=6}
# Count entries and record NA information
result_bioc_hmdb_count <- count_id(FeatureMetadata_Biocrates, "HMDB")
# Access the processed data:
processed_df_bioc_hmdb_count <- result_bioc_hmdb_count$result

```

From this plot we can make a few observations:
1. The number of biocrates cells without any HMDB at all is very high (n=253).
2. Whilst a number of biocrates cells have only a single HMDB ID associated with them (n=314), the majority of biocrates metabolites actually have multiple entries (n=452).
3. For those with multiple, this ranges from 2-20 HMDB entries per biocrates cell!

With so many biocrates cells linked to multiple HMDB IDs there are different ways to deal with this (e.g. select a single HMDB ID in cases where we have multiple). We discuss the pros and cons of this later in section: [Bonus: Are multiple metabolite IDs helpful or a hindrance?](#sect4_bonus).
But for now, we will proceed with making the connections to the PK using all HMDBs, which the MetaProViz function `checkmatch_pk_to_data()` has been designed to handle.

## 3. Make connection to PK and assess overlap

Here we connect the experimental Biocrates table to the MetalinksDB PK using the HMDB IDs using the `checkmatch_pk_to_data()` function.\

```{r load-data-5}
Biocrates_to_MetalinksDB <- checkmatch_pk_to_data(data = FeatureMetadata_Biocrates,
input_pk = MetaLinksDB,
metadata_info = c(InputID="HMDB", PriorID="hmdb", grouping_variable=NULL))
```

\
This returns some warning messages and 4 tables:\

-   `data_summary`: a summary table with links to the prior knowledge. Includes pointers if any action is required
    This is however with any NA values or duplicates of the InputID (e.g. 'HMDB') removed from the table.

-   `GroupingVariable_summary`: This is an extended version of the summary, where GroupingVariable is extended. e.g. for pathway-metabolite sets this would include duplicated InputIDs as they are in multiple pathways

-   `data_long`: all versus all commparison to enable checking in a case-by-case basis.

In some cases we will expect to have only input ID linked to one entry in the PK.
But this won't always be the case, and as already discussed in the last section above, in many cases here we will have multiple HMDB IDs per single biocrates metabolite cell.
The `checkmatch_pk_to_data()` has been designed with this in mind, by default splitting any comma separated values in the InputID (or PriorID) into separate entities, counting the number of links between these, and reporting this to the user.

Let's take a look at the results:

```{r data-prep-5, echo=FALSE}
irrelevant_cols <- c('INCHI','SMILES','Key','IUPAC','Molecule','MESH','TrivialName_Prior2023', 'Detection','CID')

# Check how our data looks like:
Biocrates_to_MetalinksDB$data_summary[c(1:8),-c(2,4)] %>%
dplyr::select(-any_of(irrelevant_cols)) %>%
kableExtra::kbl(caption = "Preview of the DF `data_Matched` showing coverage of identifiers (some columns hidden)", row.names=FALSE) %>%
kableExtra::kable_classic(full_width = FALSE, html_font = "Cambria", font_size = 12)
```


For a few metabolite classes from the biocrates kit every metabolite has a HMDB ID and a corresponding entry in MetalinksDB. Whilst for a number of classes of metabolites, we see that they will not be represented in the MetalinksDB PK at all, either because we do not have a HMDB ID associated with the Biocrates metabolite, or because MetalinksDB does not include a HMDB ID for that metabolite.\
This poor coverage could be a concern for the analysis if we are interested in analysing many of these classes, since it will mean in this case that our experimental results for Phosphatidylglycerols for instance will not be linked to PK.
In any case we will certainly need to keep this in mind for downstream analysis and the interpretation of results, so that we don't overinterpret results that for instance have a large number of Amino Acids, or falsely assume that the absence of Phosphatidylinositols in our PK integration results means that they are not present or important in our data.\
\
Let's turn back to consider the number of HMDB IDs we have in the Biocrates data and ask ourself a question: is it helpful or detrimental to have multiple IDs?
To answer this, we will take only the first HMDB of each cell with multiple HMDB IDs and test to see this has worked.

```{r data-prep-6}
#Number of matches:
MultipleIDs <- Biocrates_to_MetalinksDB$data_summary%>%
summarise(total_matches_count = sum(matches_count, na.rm = TRUE))%>%
mutate(Name = "MultipleIDs")

######################################
# Extract first ID:
extract_first_id <- function(id_col) {
    map_chr(as.character(id_col), function(x) {
    # Check for NA or empty string
    if (is.na(x) || x == "") {
        return(NA)
    }
    # Split on comma (adjust the delimiter if needed)
    parts <- unlist(strsplit(x, split = ","))
    # Return the first value after trimming any whitespace
    return(trimws(parts[1]))
    })
}

# Create a copy of the df
FeatureMetadata_Biocrates_singleHMDB <- FeatureMetadata_Biocrates
# Get the first entry of each HMDB ID
FeatureMetadata_Biocrates_singleHMDB$HMDB_single <- extract_first_id(FeatureMetadata_Biocrates$HMDB)

# Visually check that the single ID function has worked
# Count entries and record NA information
result_bioc_hmdb_count_single <- count_id(FeatureMetadata_Biocrates_singleHMDB, "HMDB_single")

# Check the macthes:
Biocrates_to_MetalinksDB_singleHMDB <- checkmatch_pk_to_data(data = FeatureMetadata_Biocrates_singleHMDB,
input_pk = MetaLinksDB,
metadata_info = c(InputID="HMDB_single", PriorID="hmdb", grouping_variable=NULL))


#Number of matches:
SingleIDs <- Biocrates_to_MetalinksDB_singleHMDB$data_summary%>%
summarise(total_matches_count = sum(matches_count, na.rm = TRUE))%>%
mutate(Name = "SingleID")

##############################
#Compare:
bind_rows(MultipleIDs, SingleIDs)%>%
kableExtra::kbl(caption = "Comparison of IDs found in MetalinksDB PK when having multiple versus a random single HMDB ID in the measured data.", row.names=FALSE) %>%
kableExtra::kable_classic(full_width = FALSE, html_font = "Cambria", font_size = 12)

```

This shows using Multi HMDBs has resulted in more metabolites from Biocrates being able to be linked to MetalinksDB than what would have been possible if we only used the first HMDB available to us.
Hence while having multiple IDs for a single detected peak may add complexity, we would recommend against prematurely dropping any IDs until you map to the PK or have thoroughly assessed what impact the removal may have.

:::: {.progress .progress-striped .active}
::: {.progress-bar .progress-bar-success style="width: 100%"}
:::
::::

# 4. Translate IDs {#sect4}

:::: {.progress .progress-striped .active}
::: {.progress-bar .progress-bar-success style="width: 100%"}
:::
::::

\

::: {style="border: 1px solid red; padding: 10px; background-color: #ffe6e6;"}
<strong>Important Information:</strong> Translating IDs between databases e.g.
KEGG to HMDB is a non-trivial task, and it is expected for one original ID to link to many translated IDs, and vice versa.
We discuss the implications throughout this vignette and leave it to user discretion to select the most appropriate ID based on their research question and data.
:::

<br>

Across the different prior knowledge resources (see also tables above) specific metabolite IDs are used and hence depending on the prior knowledge resource a specific metabolite ID is required.\
If we want to convert or 'translate' those IDs to another commonly used form of ID, for instance because our measured data uses another type of ID, we can make use of the `translate_id()` function.
This is based on [OmniPathR](https://www.bioconductor.org/packages/release/bioc/html/OmnipathR.html) and RaMP-DB [@Braisted2023] on the backend of our function and currently supports ID translation of metabolites to and from the following formats:\
- KEGG\
- HMDB\
- ChEBI\
- PubChem\
\
As an example we are translating the KEGG pathways we loaded with `metsigdb_kegg` into HMDB ids:\

```{r code-12, echo=TRUE, message=FALSE, warning=FALSE}
KEGG_Pathways_Translated <- translate_id(data= KEGG_Pathways,
metadata_info = c(InputID="MetaboliteID", grouping_variable="term"),
from = c("kegg"),
to = c("hmdb", "pubchem"))
```

\

```{r code-13, echo=FALSE, message=FALSE, warning=FALSE, include=TRUE}
KEGG_Pathways_Translated[["TranslatedDF"]][c(1:3, 300:301,600:602),] %>%
kableExtra::kbl(caption = "Translation of KEGG IDs in KEGG pathways to HMDB & pubchem IDs") %>%
kableExtra::kable_classic(full_width = FALSE, html_font = "Cambria", font_size = 12)
```

\
Here we can immediately see that despite the ID translation, some of the translations between the KEGG MetaboliteID and the HMDB or PubChem IDs have failed, resulting in NA values.
to get a better understanding of the combinations of these, let's visualise the translation for each of the ID types.\

```{r compare-pk-3, eval=FALSE}
pk_comp_res_keggtrans <- compare_pk(data = list(kegg_translated = KEGG_Pathways_Translated$TranslatedDF),
                                                metadata_info = list(kegg_translated = c("hmdb", "pubchem")),
                                                plot_name = "IDs available after KEGG ID Translation")
```
```{r, echo=FALSE}
knitr::include_graphics("prior-knowledge.Rmd_compare-pk-3 chunk.svg")
```
\
We can also note from the previous table that it becomes apparent that the translation of IDs is not a one-to-one mapping, but rather a one-to-many mapping.
In fact it is very common that an ID from one format will have a genuine one-to-many relationship with the other format (e.g. one KEGG ID maps to multiple HMDB IDs) or even a many-to-many relationship, where some of the IDs from the new format link back to multiple IDs in the original format (e.g. two different KEGG IDs map to multiple HMDS IDs, some of which are shared between them).\
This comes with many implications for the analysis that will be discussed in the next section.

## 4.1 Mapping problems

The complexities of translating metabolite IDs are demonstrated here (Fig.2).
The relationships between Original IDs (e.g. KEGG) and Translated IDs (e.g. HMDB), can be quite complex and in fact we encounter `one-to-one` (no matches were found for the ID), `one-to-none` ( direct #relationship was established), `one-to-many` (multiple matches were found for the ID. i.e. it is ambiguously mapped) and `many-to-many` ( considers the relationships between the Translated IDs to Original IDs , where a translated ID ambiguously maps back to multiple different Original IDs) mappings.\
For enrichment analysis the translation from KEGG IDs to HMDB IDs increases the pathways size, i.e. how many metabolites are in the pathway "Glycolysis / Gluconeogenesis - Homo sapiens (human)", which would in turn inflate/deflate the enrichment results and hence it would be desired to keep the number of metabolites in a pathway consistent.\

<center>

![Fig. 2: Mapping problems in prior knowledge metabolite-sets when translating metabolite IDs.](Fig.2.png){width="50%"}

</center>

<br>

Because of this complexity the output of `translate_id()` includes not only the Translation table showcased above, but additionally information about the mapping ambiguity as well as a summary of the relationships between the Original and Translated IDs.\
Indeed, the translation of e.g.
KEGG to hmdb and pubchem includes multiple data frames including a summary of the mapping occurrences:\

```{r code-14}
names(KEGG_Pathways_Translated)
```

\

```{r code-15, echo=FALSE, message=FALSE, warning=FALSE, include=TRUE}
KEGG_Pathways_Translated[["Mappingsummary_hmdb"]][c(1:3, 7:9),] %>%
kableExtra::kbl(caption = "Mappingsummary_hmdb") %>%
kableExtra::kable_classic(full_width = FALSE, html_font = "Cambria", font_size = 12)
```

\
\
We also have the ability to extract a long version of the DF that includes a row for each mapping occurrence, which can be useful for downstream analysis.
Yet, this can become very large dependent on the amount of many-to-many mappings, hence by default we do not generate this summary.
Within the `translate_id()` you can set the parameter `summary =TRUE` or in case you have a dataframe that includes both, original and translated ID, you can use the function `mapping_ambiguity()` to generate this long summary as well as the mapping summary in general.\

```{r code-16, eval=FALSE, include=TRUE}
# Option 1:
KEGG_Pathways_TranslatedSum <- translate_id(data= KEGG_Pathways,
                                                    metadata_info = c(InputID="MetaboliteID", grouping_variable="term"),
                                                    from = c("kegg"),
                                                    to = c("hmdb", "pubchem"),
                                                    summary =TRUE)
```

\

```{r data-prep-7}
# Option 2:
MappingProblems <- mapping_ambiguity(
    data =
        KEGG_Pathways_Translated[["TranslatedDF"]] %>%
        dplyr::rename("KEGG"="MetaboliteID") %>%
        dplyr::select(Description, KEGG, term, Metabolite, hmdb),
    from = "KEGG",
    to = "hmdb",
    grouping_variable = "term",
    summary=TRUE
)

```

\

```{r code-17, echo=FALSE, message=FALSE, warning=FALSE, include=TRUE}
MappingProblems[["summary"]][c(1:2, 503, 505),c(2:6, 8:9,11)] %>%
kableExtra::kbl(caption = "Long summary of mapping problems taking into account both directions, from-to-to and to-to-from.", row.names=FALSE) %>%
kableExtra::kable_classic(full_width = FALSE, html_font = "Cambria", font_size = 12)
```

\
In the table it is shown that the KEGG ID C00002 maps to 3 different HMDB IDs, and it is shown that one of those HMDB IDs HMDB0000538 maps to one KEGG ID, hence this Mapping is one-to-many.
The other two HMDB Ids are also in the table and it is checked to how many KEGG IDs they map.
Additionally, we have passed `grouping_variable = "term"` as we have pathways, which also means each of those mappings is checked within a pathway and across pathways (e.g. C00002 is shown twice, for two different terms).\

```{r plot-4,echo=FALSE, message=FALSE, warning=FALSE, include=TRUE}
Plot <- MappingProblems[["summary"]]%>%
filter(term == "Glycolysis / Gluconeogenesis")%>%
filter(!KEGG=="NA")%>%
distinct(KEGG, .keep_all = TRUE) %>%
dplyr::select(term, KEGG,  KEGG_to_hmdb, `Count(KEGG_to_hmdb)`, `AcrossGroupMappingIssue(KEGG_to_hmdb)`, hmdb_to_KEGG, `Count(hmdb_to_KEGG)`, `AcrossGroupMappingIssue(hmdb_to_KEGG)`, Mapping)
Plot$Count <- 1

PlotA <- ggplot2::ggplot(Plot, ggplot2::aes(fill=Mapping, y=Count, x=as.character(`Count(KEGG_to_hmdb)`))) +
        ggplot2::geom_bar(position="stack", stat="identity")+
        ggplot2::scale_fill_manual(values = c("many-to-many" = "#006384"  , "one-to-many" = "#d03293", "one-to-one"= "#630026", "one-to-none" = "#00735C"))+
        ggplot2::theme_classic()+
        ggplot2::ggtitle("KEGG_Glycolysis/Gluconeogenesis")+
        #ggplot2::theme(axis.text.x = ggplot2::element_text(angle = 90, hjust = 1))+
        ggplot2::labs(x = "One KEGG ID to n HMDB IDs", y = "Number of KEGG IDs",  fill ="Mapping")

Plot_Sized <-  MetaProViz:::plot_grob_superplot(input_plot=PlotA, metadata_info=c(Conditions="Conditions", Superplot="Mapping"), metadata_sample=Plot%>%dplyr::rename("Conditions"="Count(KEGG_to_hmdb)"), plot_name = "KEGG_Glycolysis-Gluconeogenesis", subtitle = NULL, plot_type="Bar")

plot_height <- grid::convertUnit(Plot_Sized$height, 'cm', valueOnly = TRUE)
plot_width <- grid::convertUnit(Plot_Sized$width, 'cm', valueOnly = TRUE)
Plot_Sized %<>%
{ggplot2::ggplot() + ggplot2::annotation_custom(.)} %>%
magrittr::add(ggplot2::theme(panel.background = ggplot2::element_rect(fill = "transparent")))
Plot_Sized_A <- Plot_Sized
# ggplot2::ggsave(filename = "KEGG_Glycolysis-Gluconeogenesis_KEGG-to-HMDB.svg", plot = Plot_Sized, width = plot_width,  height = plot_height, unit="cm")

PlotB <- ggplot2::ggplot(Plot, ggplot2::aes(fill=Mapping, y=Count, x=as.character(`Count(hmdb_to_KEGG)`))) +
        ggplot2::geom_bar(position="stack", stat="identity")+
        ggplot2::scale_fill_manual(values = c("many-to-many" = "#006384"  , "one-to-many" = "#d03293", "one-to-one"= "#630026", "one-to-none" = "#00735C"))+
        ggplot2::theme_classic()+
        ggplot2::ggtitle("KEGG_Glycolysis/Gluconeogenesis")+
        #ggplot2::theme(axis.text.x = ggplot2::element_text(angle = 90, hjust = 1))+
        ggplot2::labs(x = "HMDB ID to n KEGG IDs", y = "Number of KEGG IDs",  fill ="Mapping")

Plot_Sized <-  MetaProViz:::plot_grob_superplot(input_plot=PlotB, metadata_info=c(Conditions="Conditions", Superplot="Mapping"), metadata_sample=Plot%>%dplyr::rename("Conditions"="Count(hmdb_to_KEGG)"), plot_name = "KEGG_Glycolysis-Gluconeogenesis", subtitle = NULL, plot_type="Bar")

plot_height <- grid::convertUnit(Plot_Sized$height, 'cm', valueOnly = TRUE)
plot_width <- grid::convertUnit(Plot_Sized$width, 'cm', valueOnly = TRUE)
Plot_Sized %<>%
{ggplot2::ggplot() + ggplot2::annotation_custom(.)} %>%
magrittr::add(ggplot2::theme(panel.background = ggplot2::element_rect(fill = "transparent")))

# ggplot2::ggsave(filename = "KEGG_Glycolysis-Gluconeogenesis_HMDB-to-KEGG.svg", plot = Plot_Sized, width = plot_width,  height = plot_height, unit="cm")

Plot_Sized_B <- Plot_Sized

gridExtra::grid.arrange(Plot_Sized_A, Plot_Sized_B, ncol=2)

```

\
In fact, to perform enrichment analysis we need a column source (=e.g. term) and we would want to keep the metabolite IDs across pathways consistent, avoid ambiguous mapping as much as possible (many-to-many mapping), and have this metabolite ID selection guided by any IDs we might have available to us in our measured input data (Fig. 3).
This is crucial to circumvent inflation or deflation of metabolite-sets, which in turn will affect the enrichment analysis results.
Hence, we can use measured metabolite IDs to guide the selection of PK IDs.
This is crucial to circumvent inflation or deflation of metabolite-sets, which in turn will affect the enrichment analysis results.\
This is something we are currently working on and hope to provide within the next release, so stay tuned.\

<center>

![Fig. 3: Mapping problems in prior knowledge metabolite-sets when translating metabolite IDs and the connection to detected (measured input metabolites).](Fig.3.png){width="50%"}

</center>

<br>

:::: {.progress .progress-striped .active}
::: {.progress-bar .progress-bar-success style="width: 100%"}
:::
::::

:::: {.progress .progress-striped .active}
::: {.progress-bar .progress-bar-success style="width: 100%"}
:::
::::

# 5. Run enrichment analysis {#sect5}

:::: {.progress .progress-striped .active}
::: {.progress-bar .progress-bar-success style="width: 100%"}
:::
::::

\
There are two options:\
1.
Over Representation Analysis (ORA) that determines if a set of features (=metabolic pathways) are over-represented in the selection of features (=metabolites) from the data in comparison to all measured features (metabolites) using the Fishers exact test `cluster_ora`.
This can be applied to clusters of metabolites as for example the results from `mca_2cond()` or `core()` function.
If you want to have more details on these clustering methods please visit the vignette [Standard Metabolomics](https://saezlab.github.io/MetaProViz/articles/Standard%20Metabolomics.html) or [core Metabolomics](https://saezlab.github.io/MetaProViz/articles/core%20Metabolomics.html).\

\
2.
Enrichment analysis on standard differential analysis results.
We offer ORA `standard_ora`, but there are many other statistical tests that can be used for enrichment analysis.
The full scope of different methods is beyond the scope of MetaProViz, but are available in decoupleR [@Badia-I-Mompel2022] packages from our group.

```{r code-18, echo=FALSE, eval=FALSE}
# Future

# Run metabolite-gene analysis - Maybe add cosmos data - this also has matched transcriptomics (otherwise there are also the CPTAC data for ccRCC) and Cissys data are to be published soon as well.
#- Maybe add the data from the paper of the 3D cell culture model of ccRCC

# Run footprinting analysis - Maybe move to extra vignette - ocean method --\> RunOcean

# Reactome - As we already did most of the API retrieval, we can consider to add `LoadReactome` - Remove unspecific metabolites - Translate ChEBI to HMDB and KEGG - FYI: See code I used to access via API

# ocean PK - Add HMDB and CheBI IDs to the KEGG IDs.
#- Make ocean PK accessible via MetaProViz

# Lipid prediction - This goes into the direction of linex where we will use the lipidPKN and the reaction rules to also use lipids that are not in the prior knowledge when doing downstream analsyis!
#- linex2 PKN is manually curated - instead of using this we may want to make am updateable version.
#- Now we can at least do understand the classes present \* LipidMaps IDs, can we link other ID types?

# Viz: - Showcase specific viz options - Add network plot





#*Discuss chemical class enrichment, clustering before/after enrichment analysis. Also showcase metabolite-protein enrichment. Discuss how adding the additional columns (showcase helper function) will aid interpretability!
#*Also discuss that chemical enrichment can be more informative and what are the disadvantages of pathways (specifically in metabolomics/lipidomics) ---> Makes it even more important to make the right visualisation choices as the feature space is not that big (compared to other omics)


```

\
\
[As part of the **MetaProViz** package you can load the example data using the function `toy_data()`]{style="text-decoration:underline"} to test the enrichment analysis:\
`1.`Cell line experiment **(CellLine)**\
Here we choose an example datasets, which is publicly available on [metabolomics workbench project PR001418](https://www.metabolomicsworkbench.org/data/DRCCMetadata.php?Mode=Project&ProjectID=PR001418) including metabolic profiles of human renal epithelial cells HK2 and cell renal cell carcinoma (ccRCC) cell lines cultured in Plasmax cell culture media [@Sciacovelli_Dugourd2022].
The raw data are available via [metabolomics workbench study ST002224](https://www.metabolomicsworkbench.org/data/DRCCMetadata.php?Mode=Study&StudyID=ST002224&StudyType=MS&ResultType=1) were intracellular metabolomics of HK2 and ccRCC cell lines 786-O, 786-M1A and 786-M2A were performed.\
We have performed pre-processing and differential analysis (details can be found in the vignette vignette [Standard Metabolomics](https://saezlab.github.io/MetaProViz/articles/Standard%20Metabolomics.html)) and and here we load the differential metabolite analysis results for the comparison of 786M-1A versus HK2.\

```{r load-data-6}
# Load the Pre-processed intracellular data:
data(intracell_dma)

Intra_DMA_786M1A_vs_HK2 <- intracell_dma
```

`2.` Tissue experiment **(Tissue)**\
Here we chose publicly available data from the [paper](https://www.cell.com/cancer-cell/comments/S1535-6108(15)00468-7#supplementaryMaterial) "An Integrated Metabolic Atlas of Clear Cell Renal Cell Carcinoma", which includes metabolomic profiling on 138 matched clear cell renal cell carcinoma (ccRCC)/normal tissue pairs.\
We have performed differential analysis (details can be found in the vignette [Metadata Analysis](https://saezlab.github.io/MetaProViz/articles/Metadata%20Analysis.html)) and and here we load the differential metabolite analysis results for the comparison of Tumour versus Normal.\

```{r load-data-7}
# Load the example data:
data(tissue_dma)
Tissue_TvsN <- tissue_dma

data(tissue_dma_old)
Tissue_TvsN_Old <- tissue_dma_old

data(tissue_dma_new)
Tissue_TvsN_Young <- tissue_dma_young
```

\
\

:::: {.progress .progress-striped .active}
::: {.progress-bar .progress-bar-success style="width: 100%"}
:::
::::

# Session information

```{r session_info, echo=FALSE}
options(width = 120)
sessionInfo()
```

# Bibliography
