<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="MetaProViz">
<title>Standard Metabolomics • MetaProViz</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Standard Metabolomics">
<meta property="og:description" content="MetaProViz">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MetaProViz</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/MetaProViz.html">
    <span class="fa fa-home"></span>
     
    Introduction
  </a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown--tutorials">
    <span class="fa fa-book-open"></span>
     
    Tutorials
  </a>
  <div class="dropdown-menu" aria-labelledby="dropdown--tutorials">
    <a class="dropdown-item" href="../articles/Standard%20Metabolomics.html">Standard Metabolomics</a>
    <a class="dropdown-item" href="../articles/CoRe%20Metabolomics.html">CoRe Metabolomics</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">
    <span class="fa fa-solid fa-rocket"></span>
     
    Reference
  </a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/saezlab">
    <span class="fa fa-brands fa-github"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://twitter.com/saezlab">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://twitter.com/FrezzaLab/">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://saezlab.org/">
    <span class="fas fa fas fa-university"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://frezza.cecad-labs.uni-koeln.de/home">
    <span class="fas fa fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="Standard%20Metabolomics_files/kePrint-0.0.1/kePrint.js"></script><link href="Standard%20Metabolomics_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Standard Metabolomics</h1>
                        <h4 data-toc-skip class="author">Christina
Schmidt</h4>
            <address class="author_afil">
      Heidelberg
Universiy<br><h4 data-toc-skip class="author">Dimitrios
Prymidis</h4>
            <address class="author_afil">
      Heidelberg Universiy<br><div class="d-none name"><code>Standard Metabolomics.Rmd</code></div>
    </address>
</address>
</div>

    
    
<style>
.vscroll-plot {
    width: 1000px;
    height: 500px;
    overflow-y: scroll;
    overflow-x: hidden;
}
</style>
<div class="section level2">
<h2 id="section">
<img src="Hexagon_MetaProViz.png" align="right" width="200"><a class="anchor" aria-label="anchor" href="#section"></a>
</h2>
<p><br>
A standard metabolomics experiment refers to intracellular extracts
(e.g. cell or bacteria culture), to tissue samples (e.g. from animals or
patients), to plasma samples (e.g. blood) and many other types of
experimental setups.<br><br><span style="text-decoration:underline">In this tutorial we showcase how
to use <strong>MetaProViz</strong></span>:<br></p>
<ul>
<li>To process raw peak data and identify outliers.<br>
</li>
<li>To perform differential metabolite analysis (DMA) to generate Log2FC
and statistics and perform pathway analysis using Over Representation
Analysis (ORA) on the results.<br>
</li>
<li>To do metabolite clustering analysis (MCA) to find clusters of
metabolites with similar behaviors and perform pathway analysis using
ORA on each cluster.<br>
</li>
<li>To use specific visualizations to aid biological interpretation of
the results.<br><br>
First if you have not done yet, install the required dependencies and
load the libraries:</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#MetaProViz</span></span>
<span></span>
<span><span class="co">#library(MetaProViz)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/load_all.html" class="external-link">load_all</a></span><span class="op">(</span><span class="st">"C:/Users/chris/Documents/GitHub/MetaProViz"</span><span class="op">)</span><span class="co">#only loaded until package is online</span></span>
<span></span>
<span><span class="co">#dependencies</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span></code></pre></div>
<br><br><div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
</div>
<div class="section level2">
<h2 id="loading-the-example-data">1. Loading the example data<a class="anchor" aria-label="anchor" href="#loading-the-example-data"></a>
</h2>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
<p><br>
Here we choose an example datasets, which is publicly available on <a href="https://www.metabolomicsworkbench.org/data/DRCCMetadata.php?Mode=Project&amp;ProjectID=PR001418" class="external-link">metabolomics
workbench project PR001418</a> including metabolic profiles of human
renal epithelial cells HK2 and cell renal cell carcinoma (ccRCC) cell
lines cultured in Plasmax cell culture media <span class="citation">(Sciacovelli et al. 2022)</span>. Here we use the
integrated raw peak data as example data using the trivial metabolite
name in combination with the KEGG ID as the metabolite
identifiers.<br><br><span style="text-decoration:underline">As part of the
<strong>MetaProViz</strong> package you can load the example data into
your global environment using the function
<code><a href="../reference/toy_data.html">toy_data()</a></code></span>:<br><code>1.</code> Intracellular experiment <strong>(Intra)</strong><br>
The raw data are available via <a href="https://www.metabolomicsworkbench.org/data/DRCCMetadata.php?Mode=Study&amp;StudyID=ST002224&amp;StudyType=MS&amp;ResultType=1" class="external-link">metabolomics
workbench study ST002224</a> were intracellular metabolomics of HK2 and
ccRCC cell lines 786-O, 786-M1A and 786-M2A were performed.<br></p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/toy_data.html">toy_data</a></span><span class="op">(</span>data<span class="op">=</span><span class="st">"Standard"</span><span class="op">)</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>Intra</code> including columns with sample
information and metabolite ids with their measured values.
</caption>
<thead><tr>
<th style="text-align:left;">
Code
</th>
<th style="text-align:left;">
Conditions
</th>
<th style="text-align:right;">
Analytical_Replicates
</th>
<th style="text-align:right;">
Biological_Replicates
</th>
<th style="text-align:right;">
ADP
</th>
<th style="text-align:right;">
citrate
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
MS55_01
</td>
<td style="text-align:left;">
HK2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
77105413
</td>
<td style="text-align:right;">
2187070956
</td>
</tr>
<tr>
<td style="text-align:left;">
MS55_02
</td>
<td style="text-align:left;">
HK2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
78102480
</td>
<td style="text-align:right;">
2203043273
</td>
</tr>
<tr>
<td style="text-align:left;">
MS55_03
</td>
<td style="text-align:left;">
HK2
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
76592597
</td>
<td style="text-align:right;">
2184242278
</td>
</tr>
<tr>
<td style="text-align:left;">
MS55_04
</td>
<td style="text-align:left;">
HK2
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
70633916
</td>
<td style="text-align:right;">
1967204606
</td>
</tr>
<tr>
<td style="text-align:left;">
MS55_05
</td>
<td style="text-align:left;">
786-O
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
61574159
</td>
<td style="text-align:right;">
2213563310
</td>
</tr>
</tbody>
</table>
<p><br>
We will split the experimental data into the sample information and the
metabolites values using the the column “Code” as the unique
identifier:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Intra_M</span> <span class="op">&lt;-</span> <span class="va">Intra</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">47</span>,<span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">]</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Code"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">SampleInfo</span> <span class="op">&lt;-</span><span class="va">Intra</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">47</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">]</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Code"</span><span class="op">)</span></span></code></pre></div>
<p><br><code>2.</code> Additional information mapping the trivial metabolite
names to KEGG IDs and selected pathways
<strong>(MappingInfo)</strong><br></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/toy_data.html">toy_data</a></span><span class="op">(</span>data<span class="op">=</span><span class="st">"MappingInfo"</span><span class="op">)</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>Pathways</code> including the trivial metabolite
identifiers used in the experiment as well as KEGG IDs and pathway
information.
</caption>
<thead><tr>
<th style="text-align:left;">
Metabolite
</th>
<th style="text-align:left;">
KEGG.ID
</th>
<th style="text-align:left;">
KEGGCompound
</th>
<th style="text-align:left;">
Pathway
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
N-acetylaspartate
</td>
<td style="text-align:left;">
C01042
</td>
<td style="text-align:left;">
N-Acetyl-L-aspartate
</td>
<td style="text-align:left;">
Alanine, aspartate and glutamate metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
argininosuccinate
</td>
<td style="text-align:left;">
C03406
</td>
<td style="text-align:left;">
N-(L-Arginino)succinate
</td>
<td style="text-align:left;">
Alanine, aspartate and glutamate metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
N-acetylaspartylglutamate
</td>
<td style="text-align:left;">
C12270
</td>
<td style="text-align:left;">
N-Acetylaspartylglutamate
</td>
<td style="text-align:left;">
Alanine, aspartate and glutamate metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
creatinine
</td>
<td style="text-align:left;">
C00791
</td>
<td style="text-align:left;">
Creatinine
</td>
<td style="text-align:left;">
Arginine and proline metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
4-guanidinobutanoate
</td>
<td style="text-align:left;">
C01035
</td>
<td style="text-align:left;">
4-Guanidinobutanoate
</td>
<td style="text-align:left;">
Arginine and proline metabolism
</td>
</tr>
</tbody>
</table>
<p><br><code>3.</code> KEGG pathways that are loaded via KEGG API using the
package <code>KEGGREST</code> and can be used to perform pathway
analysis <span class="citation">(Kanehisa and Goto 2000)</span>.
<strong>(KEGG_Pathways)</strong><br></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#This will use KEGGREST to query the KEGG API to load the pathways:</span></span>
<span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/Load_KEGG.html">Load_KEGG</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Cached file loaded from: C:\Users\chris\AppData\Local/Cache/KEGG_Metabolite.rds</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>KEGG_Pathways</code>.
</caption>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
term
</th>
<th style="text-align:left;">
Metabolite
</th>
<th style="text-align:left;">
MetaboliteID
</th>
<th style="text-align:left;">
Description
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis - Homo sapiens (human)
</td>
<td style="text-align:left;">
Pyruvate
</td>
<td style="text-align:left;">
C00022
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis - Homo sapiens (human)
</td>
</tr>
<tr>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis - Homo sapiens (human)
</td>
<td style="text-align:left;">
Acetyl-CoA
</td>
<td style="text-align:left;">
C00024
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis - Homo sapiens (human)
</td>
</tr>
<tr>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis - Homo sapiens (human)
</td>
<td style="text-align:left;">
D-Glucose
</td>
<td style="text-align:left;">
C00031
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis - Homo sapiens (human)
</td>
</tr>
<tr>
<td style="text-align:left;">
52
</td>
<td style="text-align:left;">
Pentose phosphate pathway - Homo sapiens (human)
</td>
<td style="text-align:left;">
Pyruvate
</td>
<td style="text-align:left;">
C00022
</td>
<td style="text-align:left;">
Pentose phosphate pathway - Homo sapiens (human)
</td>
</tr>
<tr>
<td style="text-align:left;">
53
</td>
<td style="text-align:left;">
Pentose phosphate pathway - Homo sapiens (human)
</td>
<td style="text-align:left;">
D-Glucose
</td>
<td style="text-align:left;">
C00031
</td>
<td style="text-align:left;">
Pentose phosphate pathway - Homo sapiens (human)
</td>
</tr>
<tr>
<td style="text-align:left;">
54
</td>
<td style="text-align:left;">
Pentose phosphate pathway - Homo sapiens (human)
</td>
<td style="text-align:left;">
D-Fructose 6-phosphate
</td>
<td style="text-align:left;">
C00085
</td>
<td style="text-align:left;">
Pentose phosphate pathway - Homo sapiens (human)
</td>
</tr>
</tbody>
</table>
<br><br><div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 550%">

</div>
</div>
</div>
<div class="section level2">
<h2 id="run-metaproviz-analysis">2. Run MetaProViz Analysis<a class="anchor" aria-label="anchor" href="#run-metaproviz-analysis"></a>
</h2>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 550%">

</div>
</div>
<p>Currently, <strong>MetaProViz</strong> contains four different
modules, which include different methods and can be used independently
from each other or in combination (see introduction for more details).
Here we will go trough each of those modules and apply them to the
example data.</p>
<div class="section level3">
<h3 id="pre-processing">Pre-processing<a class="anchor" aria-label="anchor" href="#pre-processing"></a>
</h3>
<p><strong>MetaProViz</strong> includes a pre-processing module with the
function <code><a href="../reference/Preprocessing.html">Preprocessing()</a></code> that has multiple parameters to
perform customize data processing.<br><code>Feature_Filtering</code> applies the 80%-filtering rule on the
metabolite features either on the whole dataset (=“Standard”) <span class="citation">(Bijlsma et al. 2006)</span> or per condition
(=“Modified”) <span class="citation">(Wei et al. 2018)</span>. This
means that metabolites are removed were more than 20% of the samples
(all or per condition) have no detection. With the parameter
<code>Feature_Filt_Value</code> we enable the adaptation of the
stringency of the filtering based on the experimental context. For
instance, patient tumour samples can contain many unknown subgroups due
to gender, age, stage etc., which leads to a metabolite being detected
in only 50% (or even less) of the tumour samples, hence in this context
it could be considered to change the <code>Feature_Filt_Value</code>
from the default (=0.8). If <code>Feature_Filtering = "None"</code>, no
feature filtering is performed. In the context of
<code>Feature_Filtering</code> it is also noteworthy that the function
<code><a href="../reference/Pool_Estimation.html">Pool_Estimation()</a></code> can be used to estimate the quality of
the metabolite detection and will return a list of metabolites that are
variable across the different pool measurements (pool = mixture of all
experimental samples measured several times during the LC-MS run) .
Variable metabolite in the pool sample should be removed from the
data.<br>
The parameter <code>TIC_Normalization</code> refers to Total Ion Count
(TIC) normalisation, which is often used with LC-MS derived metabolomics
data. If <code>TIC_Normalization = TRUE</code>, each feature
(=metabolite) in a sample is divided by the sum of all intensity value
(= total number of ions) for the sample and finally multiplied by a
constant ( = the mean of all samples total number of ions). Noteworthy,
TIC normalisation should not be used with small number of features (=
metabolites), since TIC assumes that on “average” the ion count of each
sample is equal if there were no instrument batch effects <span class="citation">(Wulff and Mitchell 2018)</span>.<br>
The parameter <code>MVI</code> refers to Missing Value Imputation (MVI)
and if <code>MVI = TRUE</code> half minimum (HM) missing value
imputation is performed per feature (= per metabolite). Here it is
important to mention that HM has been shown to perform well for missing
vales that are missing not at random (MNAR) <span class="citation">(Wei
et al. 2018)</span>.<br>
Lastly, the function <code><a href="../reference/Preprocessing.html">Preprocessing()</a></code> performs outlier
detection and adds a column “Outliers” into the DF, which can be used to
remove outliers. The parameter <code>HotellinsConfidence</code> can be
used to choose the confidence interval that should be used for the
Hotellins T2 outlier test <span class="citation">(Hotelling
1931)</span>.<br><br>
Since our example data contains pool samples, we will do
<code><a href="../reference/Pool_Estimation.html">Pool_Estimation()</a></code> before applying the
<code><a href="../reference/Preprocessing.html">Preprocessing()</a></code> function. This is important, since one
should remove the features (=metabolites) that are too variable prior to
performing any data transformations such as TIC as part of the
<code><a href="../reference/Preprocessing.html">Preprocessing()</a></code> function.<br>
It is worth mentioning that the Coefficient of variation (CV) is
calculated by dividing the standard deviation (SD) by the mean. Hence CV
depends on the SD, which in turn works for normally distributed
data.<br></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#### Select Pool samples:</span></span>
<span><span class="co">#Get the Pool data</span></span>
<span><span class="va">PoolData</span> <span class="op">&lt;-</span> <span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/toy_data.html">toy_data</a></span><span class="op">(</span>data<span class="op">=</span><span class="st">"Standard"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">Conditions</span><span class="op">==</span><span class="st">"Pool"</span>, select <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="co"># we remove the columns "Conditions", "Analytical_Replicates" and "Biological_Replicates"</span></span>
<span>  <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">remove_rownames</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Code"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check the metabolite variability</span></span>
<span><span class="va">Pool_Estimation_result</span><span class="op">&lt;-</span> <span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/Pool_Estimation.html">Pool_Estimation</a></span><span class="op">(</span>Input_data <span class="op">=</span> <span class="va">PoolData</span>,</span>
<span>                            Input_SettingsFile <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                            Input_SettingsInfo <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                            Unstable_feature_remove <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                            Threshold_cv <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: package 'ggfortify' was built under R version 4.2.3</span></span></code></pre></div>
<p><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">#--------------------------------------------------------------------</span></span>
<span><span class="co">#### Alternatively a full dataset can be added. Here, the Conditions and PoolSamples name have to be specified in the Input_SettingsInfo  </span></span>
<span><span class="va">Pool_Estimation_result</span><span class="op">&lt;-</span> <span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/Pool_Estimation.html">Pool_Estimation</a></span><span class="op">(</span>Input_data <span class="op">=</span> <span class="va">Intra</span><span class="op">[</span>,<span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Code"</span><span class="op">)</span>,</span>
<span>                            Input_SettingsFile <span class="op">=</span> <span class="va">Intra</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Code"</span><span class="op">)</span>,</span>
<span>                            Input_SettingsInfo <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>PoolSamples <span class="op">=</span> <span class="st">"Pool"</span>, Conditions<span class="op">=</span><span class="st">"Conditions"</span><span class="op">)</span>,</span>
<span>                            Unstable_feature_remove <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                            Threshold_cv <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 35 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 35 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre></div>
<p><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-9-2.png" width="700"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">Pool_Estimation_result_DF_CV</span> <span class="op">&lt;-</span><span class="va">Pool_Estimation_result</span><span class="op">[[</span><span class="st">"DF"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"CV_result"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the Pool_Estimation result.
</caption>
<thead><tr>
<th style="text-align:left;">
Metabolite
</th>
<th style="text-align:right;">
CV
</th>
<th style="text-align:left;">
HighVar
</th>
<th style="text-align:right;">
MissingValuePercentage
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
valine-d8
</td>
<td style="text-align:right;">
2.425263
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
hippuric acid-d5
</td>
<td style="text-align:right;">
1.579717
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
2/3-phosphoglycerate
</td>
<td style="text-align:right;">
2.933440
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
2-aminoadipic acid
</td>
<td style="text-align:right;">
5.461470
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
2-hydroxyglutarate
</td>
<td style="text-align:right;">
1.366187
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
0
</td>
</tr>
</tbody>
</table>
<p><br>
The results from the <code><a href="../reference/Pool_Estimation.html">Pool_Estimation()</a></code> is a table that has
the CVs. If there is a high variability, one should consider to remove
those features from the data. For the example data nothing needs to be
removed. If you have used internal standard in your experiment you
should specifically check their CV as this would indicate technical
issues (here valine-d8 and hippuric acid-d5).<br><br>
Now we will apply the <code><a href="../reference/Preprocessing.html">Preprocessing()</a></code> function to example
data and have a look at the output produced. You will notice that all
the chosen parameters and results are documented in messages. All the
results data tables, the Quality Control (QC) plots and outlier
detection plots are returned and can be easily viewed.<br></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">PreprocessingResults</span> <span class="op">&lt;-</span> <span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/Preprocessing.html">Preprocessing</a></span><span class="op">(</span>Input_data<span class="op">=</span><span class="va">Intra_M</span>, </span>
<span>                                                  Input_SettingsFile<span class="op">=</span><span class="va">SampleInfo</span>,</span>
<span>                                                  Input_SettingsInfo <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>Conditions <span class="op">=</span> <span class="st">"Conditions"</span>,</span>
<span>                                                                         Biological_Replicates <span class="op">=</span> <span class="st">"Biological_Replicates"</span><span class="op">)</span>,</span>
<span>                                                  Feature_Filtering <span class="op">=</span> <span class="st">"Modified"</span>,</span>
<span>                                                  Feature_Filt_Value <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>                                                  TIC_Normalization <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                                  MVI<span class="op">=</span><span class="cn">TRUE</span>, </span>
<span>                                                  HotellinsConfidence <span class="op">=</span> <span class="fl">0.99</span>,<span class="co"># We perform outlier testing using 0.99 confidence intervall</span></span>
<span>                                                  ExportQCPlots <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                                  CoRe <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                                                  Plot <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> </span>
<span><span class="co">#&gt; Here we apply the modified 80%-filtering rule that takes the class information (Column `Conditions`) into account, which additionally reduces the effect of missing values. REF: Yang et. al., (2015), doi: 10.3389/fmolb.2015.00004)</span></span>
<span><span class="co">#&gt; filtering value selected: 0.8</span></span>
<span><span class="co">#&gt; 3 metabolites where removed: AICAR, FAICAR, SAICAR</span></span>
<span><span class="co">#&gt; Missing value imputation is performed, as a complementary approach to address the missing value problem, where the missing values are imputing using the `half minimum value`. REF: Wei et. al., (2018), Reports, 8, 663, doi:https://doi.org/10.1038/s41598-017-19120-0</span></span>
<span><span class="co">#&gt; Total Ion Count (TIC) normalization is used to reduce the variation from non-biological sources, while maintaining the biological variation. REF: Wulff et. al., (2018), Advances in Bioscience and Biotechnology, 9, 339-351, doi:https://doi.org/10.4236/abb.2018.98022</span></span>
<span><span class="co">#&gt; Identification of outlier samples is performed using Hotellin's T2 test to define sample outliers in a mathematical way (Confidence = 0.99 ~ p.val &lt; 0.01) REF: Hotelling, H. (1931), Annals of Mathematical Statistics. 2 (3), 360–378, doi:https://doi.org/10.1214/aoms/1177732979.</span></span>
<span><span class="co">#&gt; HotellinsConfidence value selected: 0.99</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 25 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></span>
<span><span class="co">#&gt; ggrepel: 25 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></span>
<span><span class="co">#&gt; ***Checking for outliers***</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 25 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 14 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></span>
<span><span class="co">#&gt; ggrepel: 14 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></span>
<span><span class="co">#&gt; ***Checking for outliers***</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 14 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span>
<span><span class="co">#&gt; There are possible outlier samples in the data</span></span>
<span><span class="co">#&gt; Filtering round 1 Outlier Samples: MS55_29</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 25 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 25 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></span>
<span><span class="co">#&gt; ggrepel: 25 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></span>
<span><span class="co">#&gt; ggrepel: 25 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></span></code></pre></div>
<p><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># This is the results table:</span></span>
<span><span class="va">Intra_Preprocessed</span> <span class="op">&lt;-</span> <span class="va">PreprocessingResults</span><span class="op">[[</span><span class="st">"DF"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"Preprocessing_output"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"Processed_data"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the pre-processing results, which has an additional column
<code>Outlier</code> including the results of Hotellins T2.
</caption>
<thead><tr>
<th style="text-align:left;">
Code
</th>
<th style="text-align:left;">
Sample_ID
</th>
<th style="text-align:left;">
Conditions
</th>
<th style="text-align:right;">
Analytical_Replicates
</th>
<th style="text-align:right;">
Biological_Replicates
</th>
<th style="text-align:left;">
Outliers
</th>
<th style="text-align:right;">
valine-d8
</th>
<th style="text-align:right;">
hippuric acid-d5
</th>
<th style="text-align:right;">
2/3-phosphoglycerate
</th>
<th style="text-align:right;">
2-aminoadipic acid
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
29
</td>
<td style="text-align:left;">
MS55_29
</td>
<td style="text-align:left;">
786-M2A
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
Outlier_filtering_round_1
</td>
<td style="text-align:right;">
2389973677
</td>
<td style="text-align:right;">
4573652296
</td>
<td style="text-align:right;">
40224284
</td>
<td style="text-align:right;">
6070770
</td>
</tr>
<tr>
<td style="text-align:left;">
30
</td>
<td style="text-align:left;">
MS55_30
</td>
<td style="text-align:left;">
786-M2A
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
2131636828
</td>
<td style="text-align:right;">
3913339561
</td>
<td style="text-align:right;">
40942215
</td>
<td style="text-align:right;">
5934169
</td>
</tr>
<tr>
<td style="text-align:left;">
31
</td>
<td style="text-align:left;">
MS55_31
</td>
<td style="text-align:left;">
786-M2A
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
2010263533
</td>
<td style="text-align:right;">
3823948950
</td>
<td style="text-align:right;">
45701920
</td>
<td style="text-align:right;">
6128537
</td>
</tr>
<tr>
<td style="text-align:left;">
32
</td>
<td style="text-align:left;">
MS55_32
</td>
<td style="text-align:left;">
786-M2A
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
2025374089
</td>
<td style="text-align:right;">
3812717474
</td>
<td style="text-align:right;">
46212143
</td>
<td style="text-align:right;">
6640611
</td>
</tr>
</tbody>
</table>
<p><br>
In the output table you can now see the column “Outliers” and for the
Condition 786-M2A, we can see that based on Hotellin’s T2 test, one
sample was detected as an outlier in the first round of filtering.<br>
As part of the <code><a href="../reference/Preprocessing.html">Preprocessing()</a></code> function several plots are
generated and saved. Additionally, the ggplots are returned into the
list to enable further modifiaction using the ggplot syntax. These plots
include plots showing the outliers for each filtering round and other QC
plots.<br><br>
As part of the <strong>MetaProViz</strong> visualization module one can
easily further customize the PCA plot and adapt color and shape for the
information of interest. You can see more below for the
<code><a href="../reference/VizPCA.html">VizPCA()</a></code> function.<br>
Before we proceed, we will remove the outlier:<br></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Intra_Preprocessed</span> <span class="op">&lt;-</span> <span class="va">Intra_Preprocessed</span><span class="op">[</span><span class="op">-</span><span class="fl">29</span>,<span class="op">]</span><span class="co">#remove MS55_29</span></span></code></pre></div>
<p><br>
As you may have noticed, in this example dataset we have several
biological replicates that where injected (=measured) several times,
which can be termed as analytical replicates. The
<strong>MetaProViz</strong> pre-processing module includes the function
<code><a href="../reference/ReplicateSum.html">ReplicateSum()</a></code>, which will do this task and save the
results:<br></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Intra_Preprocessed</span> <span class="op">&lt;-</span> <span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/ReplicateSum.html">ReplicateSum</a></span><span class="op">(</span>Input_data<span class="op">=</span><span class="va">Intra_Preprocessed</span><span class="op">)</span></span></code></pre></div>
<p><br>
Using the pre-processed data, we can now use the
<strong>MetaProViz</strong> visualization module and generate some
overview Heatmaps <code><a href="../reference/VizHeatmap.html">VizHeatmap()</a></code> or PCA plots
<code><a href="../reference/VizPCA.html">VizPCA()</a></code>. You can see some examples below.</p>
</div>
<div class="section level3">
<h3 id="dma">DMA<a class="anchor" aria-label="anchor" href="#dma"></a>
</h3>
<p>Differential Metabolite Analysis (<code>DMA</code>) is used to
compare two conditions (e.g. Tumour versus Healthy) by calculating the
Log2FC, p-value, adjusted p-value and t-value. With the different
parameters <code>STAT_pval</code> and <code>STAT_padj</code> one can
choose the statistical tests such as t.test, wilcoxon test, limma,
annova, kruskal walles, etc. (see function reference for more
information).<br>
As input one can use the pre-processed data we have generated using the
<code>Preprocessing</code> module, but here one can of course use any DF
including metabolite values, even though we recommend to normalize the
data and remove outliers prior to DMA. Moreover, we require the
<code>Input_SettingsFile_Sample</code> including the sample metadata
with information which condition a sample corresponds to. Additionally,
we enable the user to provide a <code>Plot_SettingsFile_Metab</code>
containing the metadata for the features (metabolites), such as KEGG ID,
pathway, retention time, etc.<br><br>
By defining the numerator and denominator as part of the
<code>Input_SettingsInfo</code> parameter, it is defined which
comparisons are performed:<br>
1. <strong>one_vs_one</strong> (single comparison):
numerator=“Condition1”, denominator =“Condition2”<br>
2. <strong>all_vs_one</strong> (multiple comparison): numerator=NULL,
denominator =“Condition”<br>
3. <strong>all_vs_all</strong> (multiple comparison): numerator=NULL,
denominator =NULL (=default)<br><br>
Noteworthy, if you have not performed missing value imputation and hence
your data includes NAs or 0 values for some features, this is how we
deal with this in the <code><a href="../reference/DMA.html">DMA()</a></code> function:<br>
1. If you use the parameter <code>STAT_pval="lmFit"</code>, limma is
performed. Limma does a baesian fit of the data and substracts
Mean(Condition1 fit) - Mean(Condition2 fit). As such, unless all values
of a feature are NA, Limma can deal with NAs. 2. Standard Log2FC:
log2(Mean(Condition1)) - log2(Mean(Condition2)) a. If all values of the
replicates of one condition are NA/0 for a feature (=metabolite):
Log2FC= Inf/-Inf and the statistics will be NA<br>
b. If some values of the replicates of one condition are NA/0 for a
feature (=metabolite): Log2FC= positive or negative value, but the
statistics will be NA<br></p>
<p>It is important to mention that in case of
<code>STAT_pval="lmFit"</code>, we perform log2 transformation of the
data as prior to running limma to enable the calculation of the log2FC,
hence do not provide log2 transformed data.<br><br>
Here, the example data we have four different cell lines, healthy (HK2)
and cancer (ccRCC: 786-M1A, 786-M2A and 786-O), hence we can perform
multiple different comparisons. The results can be automatically saved
and all the results are returned in a list with the different data
frames. If parameter Plot=TRUE, an overview Volcano plot is generated
and saved.<br></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Perform multiple comparison All_vs_One using annova:</span></span>
<span><span class="va">DMA_Annova</span> <span class="op">&lt;-</span> <span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/DMA.html">DMA</a></span><span class="op">(</span>Input_data<span class="op">=</span><span class="va">Intra_Preprocessed</span><span class="op">[</span>,<span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span><span class="op">]</span>, <span class="co">#we need to remove columns that do not include metabolite measurements</span></span>
<span>                              Input_SettingsFile_Sample<span class="op">=</span><span class="va">Intra_Preprocessed</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span><span class="op">]</span>,<span class="co">#only maintain the information about condition and replicates</span></span>
<span>                              Input_SettingsInfo <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>conditions<span class="op">=</span><span class="st">"Conditions"</span>, numerator<span class="op">=</span><span class="cn">NULL</span>, denominator <span class="op">=</span> <span class="st">"HK2"</span><span class="op">)</span>,<span class="co"># we compar all_vs_HK2</span></span>
<span>                              STAT_pval <span class="op">=</span><span class="st">"aov"</span>,</span>
<span>                              STAT_padj<span class="op">=</span><span class="st">"fdr"</span>,</span>
<span>                              Input_SettingsFile_Metab <span class="op">=</span> <span class="va">MappingInfo</span>,<span class="co"># Adds metadata for the metabolites such as KEGG_ID, Pathway, retention time,...</span></span>
<span>                              OutputName<span class="op">=</span><span class="st">'Annova'</span><span class="op">)</span></span>
<span><span class="co">#&gt; There are no NA/0 values</span></span>
<span><span class="co">#&gt; For the condition 786-M1A 94.41 % of the metabolites follow a normal distribution and 5.59 % of the metabolites are not-normally distributed according to the shapiro test. You have chosen aov, which is for parametric Hypothesis testing. `shapiro.test` ignores missing values in the calculation.</span></span>
<span><span class="co">#&gt; For the condition 786-M2A 97.24 % of the metabolites follow a normal distribution and 2.76 % of the metabolites are not-normally distributed according to the shapiro test. You have chosen aov, which is for parametric Hypothesis testing. `shapiro.test` ignores missing values in the calculation.</span></span></code></pre></div>
<p><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
<pre><code><span><span class="co">#&gt; For the condition 786-O 95.03 % of the metabolites follow a normal distribution and 4.97 % of the metabolites are not-normally distributed according to the shapiro test. You have chosen aov, which is for parametric Hypothesis testing. `shapiro.test` ignores missing values in the calculation.</span></span></code></pre>
<p><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-15-2.png" width="700"></p>
<pre><code><span><span class="co">#&gt; For the condition HK2 96.13 % of the metabolites follow a normal distribution and 3.87 % of the metabolites are not-normally distributed according to the shapiro test. You have chosen aov, which is for parametric Hypothesis testing. `shapiro.test` ignores missing values in the calculation.</span></span></code></pre>
<p><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-15-3.png" width="700"></p>
<pre><code><span><span class="co">#&gt; For 84.36% of metabolites the group variances are equal.</span></span>
<span><span class="co">#&gt; No condition was specified as numerator and HK2 was selected as a denominator. Performing multiple testing `all-vs-one` using aov.</span></span></code></pre>
<p><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-15-4.png" width="700"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-15-5.png" width="700"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-15-6.png" width="700"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-15-7.png" width="700"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">#Inspect the DMA results tables:</span></span>
<span><span class="va">DMA_786M1A_vs_HK2</span> <span class="op">&lt;-</span> <span class="va">DMA_Annova</span><span class="op">[[</span><span class="st">"DF"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"DMA_result"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"786-M1A_vs_HK2"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">Shapiro</span> <span class="op">&lt;-</span> <span class="va">DMA_Annova</span><span class="op">[[</span><span class="st">"DF"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"Shapiro_result"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<ol style="list-style-type: decimal"><li>Preview of the Shaprio results for the different conditions.
</li></ol>
</caption>
<thead><tr>
<th style="text-align:left;">
Metabolites with normal distribution [%]
</th>
<th style="text-align:left;">
Metabolites with not-normal distribution [%]
</th>
<th style="text-align:right;">
Shapiro p.val(uric acid)
</th>
<th style="text-align:right;">
Shapiro p.val(uridine)
</th>
<th style="text-align:right;">
Shapiro p.val(UTP)
</th>
<th style="text-align:right;">
Shapiro p.val(valine)
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
94.41
</td>
<td style="text-align:left;">
5.59
</td>
<td style="text-align:right;">
0.1526546
</td>
<td style="text-align:right;">
0.3814773
</td>
<td style="text-align:right;">
0.1619815
</td>
<td style="text-align:right;">
0.2443471
</td>
</tr>
<tr>
<td style="text-align:left;">
97.24
</td>
<td style="text-align:left;">
2.76
</td>
<td style="text-align:right;">
0.2337966
</td>
<td style="text-align:right;">
0.4514400
</td>
<td style="text-align:right;">
0.8671553
</td>
<td style="text-align:right;">
0.6408601
</td>
</tr>
<tr>
<td style="text-align:left;">
95.03
</td>
<td style="text-align:left;">
4.97
</td>
<td style="text-align:right;">
0.1809369
</td>
<td style="text-align:right;">
0.7696208
</td>
<td style="text-align:right;">
0.0390758
</td>
<td style="text-align:right;">
0.4732762
</td>
</tr>
<tr>
<td style="text-align:left;">
96.13
</td>
<td style="text-align:left;">
3.87
</td>
<td style="text-align:right;">
0.0648389
</td>
<td style="text-align:right;">
0.9830021
</td>
<td style="text-align:right;">
0.0934053
</td>
<td style="text-align:right;">
0.2683982
</td>
</tr>
</tbody>
</table>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<ol start="2" style="list-style-type: decimal"><li>Preview of the DMA results for the comparison of 786-M1A versus HK2
cells.
</li></ol>
</caption>
<thead><tr>
<th style="text-align:left;">
Metabolite
</th>
<th style="text-align:right;">
Log2FC
</th>
<th style="text-align:right;">
p.adj
</th>
<th style="text-align:right;">
t.val
</th>
<th style="text-align:right;">
786-M1A_1
</th>
<th style="text-align:right;">
786-M1A_2
</th>
<th style="text-align:right;">
786-M1A_3
</th>
<th style="text-align:right;">
HK2_1
</th>
<th style="text-align:right;">
HK2_2
</th>
<th style="text-align:right;">
HK2_3
</th>
<th style="text-align:left;">
KEGG.ID
</th>
<th style="text-align:left;">
KEGGCompound
</th>
<th style="text-align:left;">
Pathway
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
isovalerylcarnitine
</td>
<td style="text-align:right;">
0.9170474
</td>
<td style="text-align:right;">
0.0063473
</td>
<td style="text-align:right;">
-14524759.5
</td>
<td style="text-align:right;">
29686278
</td>
<td style="text-align:right;">
30650177
</td>
<td style="text-align:right;">
32294321
</td>
<td style="text-align:right;">
19772945
</td>
<td style="text-align:right;">
15552910
</td>
<td style="text-align:right;">
13730642
</td>
<td style="text-align:left;">
C20826
</td>
<td style="text-align:left;">
Isovalerylcarnitine
</td>
<td style="text-align:left;">
Fatty acyl carnitines
</td>
</tr>
<tr>
<td style="text-align:left;">
lactate
</td>
<td style="text-align:right;">
0.3015843
</td>
<td style="text-align:right;">
0.1582174
</td>
<td style="text-align:right;">
-134576190.2
</td>
<td style="text-align:right;">
722511110
</td>
<td style="text-align:right;">
655030333
</td>
<td style="text-align:right;">
762675334
</td>
<td style="text-align:right;">
565613263
</td>
<td style="text-align:right;">
499634998
</td>
<td style="text-align:right;">
671239945
</td>
<td style="text-align:left;">
C00186
</td>
<td style="text-align:left;">
(S)-Lactate
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis
</td>
</tr>
<tr>
<td style="text-align:left;">
lysine
</td>
<td style="text-align:right;">
0.4006104
</td>
<td style="text-align:right;">
0.0686223
</td>
<td style="text-align:right;">
-9560501.1
</td>
<td style="text-align:right;">
39886351
</td>
<td style="text-align:right;">
37865442
</td>
<td style="text-align:right;">
40540849
</td>
<td style="text-align:right;">
27947113
</td>
<td style="text-align:right;">
25896511
</td>
<td style="text-align:right;">
35767515
</td>
<td style="text-align:left;">
C00047
</td>
<td style="text-align:left;">
L-Lysine
</td>
<td style="text-align:left;">
D-Amino acid metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
malate
</td>
<td style="text-align:right;">
0.0981352
</td>
<td style="text-align:right;">
0.7498870
</td>
<td style="text-align:right;">
-144362286.0
</td>
<td style="text-align:right;">
2246450554
</td>
<td style="text-align:right;">
2279097256
</td>
<td style="text-align:right;">
2060300868
</td>
<td style="text-align:right;">
1942335526
</td>
<td style="text-align:right;">
2066498001
</td>
<td style="text-align:right;">
2143928293
</td>
<td style="text-align:left;">
C00149
</td>
<td style="text-align:left;">
(S)-Malate
</td>
<td style="text-align:left;">
Citrate cycle (TCA cycle)
</td>
</tr>
<tr>
<td style="text-align:left;">
malonate
</td>
<td style="text-align:right;">
0.1701143
</td>
<td style="text-align:right;">
0.7650646
</td>
<td style="text-align:right;">
-817063.4
</td>
<td style="text-align:right;">
7464618
</td>
<td style="text-align:right;">
8335831
</td>
<td style="text-align:right;">
6237131
</td>
<td style="text-align:right;">
5923914
</td>
<td style="text-align:right;">
6204769
</td>
<td style="text-align:right;">
7457707
</td>
<td style="text-align:left;">
C00383
</td>
<td style="text-align:left;">
Malonate
</td>
<td style="text-align:left;">
Pyrimidine metabolism
</td>
</tr>
</tbody>
</table>
<p><br>
Using the DMA results, we can now use the <strong>MetaProViz</strong>
visualization module and generate further customized Volcano plots
<code><a href="../reference/VizVolcano.html">VizVolcano()</a></code>. You can see some examples below.</p>
<div class="section level4">
<h4 id="ora-using-the-dma-results">ORA using the DMA results<a class="anchor" aria-label="anchor" href="#ora-using-the-dma-results"></a>
</h4>
<p>Over Representation Analysis (ORA) is a pathway enrichment analysis
(PEA) method that determines if a set of features (=metabolic pathways)
are over-represented in the selection of features (=metabolites) from
the data in comparison to all measured features (metabolites) using the
Fishers exact test. The selection of metabolites are usually the most
altered metabolites in the data, which can be selected by the top and
bottom t-values.<br>
Of course, there are many other PEA methods such as the well known GSEA.
Here we do not aim to provide an extensive tool for different methods to
perform pathway enrichment analysis and only focus on ORA since we can
apply this to perform standard pathway enrichment as well as pathway
enrichment on clusters of metabolites (see MCA below). If you are
interested in using different pathway enrichment methods please check
out specialized tools such as <a href="https://saezlab.github.io/decoupleR/" class="external-link">decopupleR</a> <span class="citation">(Badia-I-Mompel et al. 2022)</span>.<br><br>
Here we will use the KEGG pathways <span class="citation">(Kanehisa and
Goto 2000)</span>. Before we can perform ORA on the DMA results, we have
to ensure that the metabolite names match with the KEGG IDs or KEGG
trivial names. In general, the <code>PathwayFile</code> requirements are
column “term”, “Metabolite” and “Description”, and the
<code>Input_data</code> requirements are column “t.val” and column
“Metabolite”.<br></p>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the ORA results for the comparison of 786-M1A versus HK2
cells.
</caption>
<thead><tr>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
MetaboliteIDs_in_pathway
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
Metabolites_in_Pathway
</th>
<th style="text-align:right;">
Percentage_of_Pathway_detected
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
ABC transporters - Homo sapiens (human)
</td>
<td style="text-align:left;">
11/24
</td>
<td style="text-align:left;">
32/122
</td>
<td style="text-align:right;">
0.0172642
</td>
<td style="text-align:right;">
0.2956418
</td>
<td style="text-align:right;">
0.2677783
</td>
<td style="text-align:left;">
Betaine/Glutathione/L-Alanine/L-Aspartate/L-Glutamate/L-Histidine/L-Leucine/L-Proline/L-Threonine/myo-Inositol/Taurine
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
130
</td>
<td style="text-align:right;">
8.46
</td>
</tr>
<tr>
<td style="text-align:left;">
Alanine, aspartate and glutamate metabolism - Homo sapiens (human)
</td>
<td style="text-align:left;">
6/24
</td>
<td style="text-align:left;">
14/122
</td>
<td style="text-align:right;">
0.0313864
</td>
<td style="text-align:right;">
0.2956418
</td>
<td style="text-align:right;">
0.2677783
</td>
<td style="text-align:left;">
2-Oxoglutarate/Citrate/L-Alanine/L-Aspartate/L-Glutamate/N-Acetyl-L-aspartate
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
27
</td>
<td style="text-align:right;">
22.22
</td>
</tr>
<tr>
<td style="text-align:left;">
D-Amino acid metabolism - Homo sapiens (human)
</td>
<td style="text-align:left;">
7/24
</td>
<td style="text-align:left;">
18/122
</td>
<td style="text-align:right;">
0.0343769
</td>
<td style="text-align:right;">
0.2956418
</td>
<td style="text-align:right;">
0.2677783
</td>
<td style="text-align:left;">
2-Oxoglutarate/L-Alanine/L-Aspartate/L-Glutamate/L-Histidine/L-Proline/L-Threonine
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
69
</td>
<td style="text-align:right;">
10.14
</td>
</tr>
<tr>
<td style="text-align:left;">
Histidine metabolism - Homo sapiens (human)
</td>
<td style="text-align:left;">
4/24
</td>
<td style="text-align:left;">
6/122
</td>
<td style="text-align:right;">
0.0135586
</td>
<td style="text-align:right;">
0.2956418
</td>
<td style="text-align:right;">
0.2677783
</td>
<td style="text-align:left;">
2-Oxoglutarate/L-Aspartate/L-Glutamate/L-Histidine
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
47
</td>
<td style="text-align:right;">
8.51
</td>
</tr>
<tr>
<td style="text-align:left;">
Taurine and hypotaurine metabolism - Homo sapiens (human)
</td>
<td style="text-align:left;">
4/24
</td>
<td style="text-align:left;">
7/122
</td>
<td style="text-align:right;">
0.0273459
</td>
<td style="text-align:right;">
0.2956418
</td>
<td style="text-align:right;">
0.2677783
</td>
<td style="text-align:left;">
2-Oxoglutarate/L-Alanine/L-Glutamate/Taurine
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
24
</td>
<td style="text-align:right;">
16.67
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level3">
<h3 id="mca">MCA<a class="anchor" aria-label="anchor" href="#mca"></a>
</h3>
<p>Metabolite Clustering Analysis (<code>MCA</code>) is a module, which
includes different functions to enable clustering of metabolites into
groups either based on logical regulatory rules. This can be
particularly useful if one has multiple conditions and aims to find
patterns in the data.</p>
<div class="section level4">
<h4 id="mca-2cond">MCA-2Cond<a class="anchor" aria-label="anchor" href="#mca-2cond"></a>
</h4>
<p>This metabolite clustering method is based on the Regulatory
Clustering Method (RCM) that was developed as part of the Signature
Regulatory Clustering (SiRCle) model (<span class="citation">Mora et al.
(2022)</span>). As part of the <a href="https://github.com/ArianeMora/SiRCleR/tree/main" class="external-link">SiRCleR
package</a>, also variation of the initial RCM method are proposed as
clustering based on two comparisons (e.g. KO versus WT in hypoxia and in
normoxia).<br>
Here we set two different thresholds, one for the differential
metabolite abundance (<code>Log2FC</code>) and one for the
<code>significance</code> (e.g. p.adj). This will define if a feature (=
metabolite) is assigned into:<br>
1. <strong><em>“UP”</em></strong>, which means a metabolite is
significantly up-regulated in the underlying comparison.<br>
2. <strong><em>“DOWN”</em></strong>, which means a metabolite is
significantly down-regulated in the underlying comparison.<br>
3. <strong><em>“No Change”</em></strong>, which means a metabolite does
not change significantly in the underlying comparison and/or is not
defined as up-regulated/down-regulated based on the Log2FC threshold
chosen.<br><br>
Therebye “No Change” is further subdivided into four states:<br>
1. <strong><em>“Not Detected”</em></strong>, which means a metabolite is
not detected in the underlying comparison.<br>
2. <strong><em>“Not Significant”</em></strong>, which means a metabolite
is not significant in the underlying comparison.<br>
3. <strong><em>“Significant positive”</em></strong>, which means a
metabolite is significant in the underlying comparison and the
differential metabolite abundance is positive, yet does not meet the
threshold set for “UP” (e.g. Log2FC &gt;1 = “UP” and we have a
significant Log2FC=0.8).<br>
4. <strong><em>“Significant negative”</em></strong>, which means a
metabolite is significant in the underlying comparison and the
differential metabolite abundance is negative, yet does not meet the
threshold set for “DOWN”.<br><br>
This definition is done individually for each comparison and will impact
in which metabolite cluster a metabolite is sorted into.<br>
Since we have two comparisons, we can choose between different
Background settings, which defines which features will be considered for
the clusters (e.g. you could include only features (= metabolites) that
are detected in both comparisons, removing the rest of the features).The
background methods <code>backgroundMethod</code> are the following from
<strong><em>1.1. - 1.4.</em></strong> from most restrictive to least
restrictive:<br><strong><em>1.1. C1&amp;C2</em></strong>: Most stringend background
setting and will lead to a small number of metabolites.<br><strong><em>1.2. C1</em></strong>: Focus is on the metabolite abundance
of Condition 1 (C1).<br><strong><em>1.3. C2</em></strong>: Focus is on the metabolite abundance
of Condition 2 (C2).<br><strong><em>1.4. C1|C2</em></strong>: Least stringent background method,
since a metabolite will be included in the input if it has been detected
on one of the two conditions.<br><br>
Lastly, we will get clusters of metabolites that are defined by the
metabolite change in the two conditions. For example, if Alanine is “UP”
based on the thresholds in both comparisons it will be sorted into the
cluster “Core_UP”. As there are two 6-state6 transitions between the
comparisons, the flows are summarised into smaller amount of metabolite
clusters using different Regulation Groupings (RG): 1. RG1_All<br>
2. RG2_Significant taking into account genes that are significant (UP,
DOWN, significant positive, significant negative)<br>
3. RG3_SignificantChange only takes into account genes that have
significant changes (UP, DOWN).<br><br></p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Example of all possible flows:</span></span>
<span><span class="va">MCA_2Cond</span> <span class="op">&lt;-</span> <span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/MCA_rules.html">MCA_rules</a></span><span class="op">(</span>data<span class="op">=</span><span class="st">"2Cond"</span><span class="op">)</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Metabolite Clustering Analysis: 2 Conditions.
</caption>
<thead><tr>
<th style="text-align:left;">
Cond1
</th>
<th style="text-align:left;">
Cond2
</th>
<th style="text-align:left;">
RG1_All
</th>
<th style="text-align:left;">
RG2_Significant
</th>
<th style="text-align:left;">
RG3_SignificantChange
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Cond1 DOWN + Cond2 DOWN
</td>
<td style="text-align:left;">
Core_DOWN
</td>
<td style="text-align:left;">
Core_DOWN
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Cond1 DOWN + Cond2 Not Detected
</td>
<td style="text-align:left;">
Cond1_DOWN
</td>
<td style="text-align:left;">
Cond1_DOWN
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Cond1 DOWN + Cond2 Not Significant
</td>
<td style="text-align:left;">
Cond1_DOWN
</td>
<td style="text-align:left;">
Cond1_DOWN
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Cond1 DOWN + Cond2 Significant Negative
</td>
<td style="text-align:left;">
Core_DOWN
</td>
<td style="text-align:left;">
Cond1_DOWN
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Cond1 DOWN + Cond2 Significant Positive
</td>
<td style="text-align:left;">
Opposite
</td>
<td style="text-align:left;">
Cond1_DOWN
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Cond1 DOWN + Cond2 UP
</td>
<td style="text-align:left;">
Opposite
</td>
<td style="text-align:left;">
Opposite
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Cond1 UP + Cond2 DOWN
</td>
<td style="text-align:left;">
Opposite
</td>
<td style="text-align:left;">
Opposite
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Cond1 UP + Cond2 Not Detected
</td>
<td style="text-align:left;">
Cond1_UP
</td>
<td style="text-align:left;">
Cond1_UP
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Cond1 UP + Cond2 Not Significant
</td>
<td style="text-align:left;">
Cond1_UP
</td>
<td style="text-align:left;">
Cond1_UP
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Cond1 UP + Cond2 Significant Negative
</td>
<td style="text-align:left;">
Opposite
</td>
<td style="text-align:left;">
Cond1_UP
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Cond1 UP + Cond2 Significant Positive
</td>
<td style="text-align:left;">
Core_UP
</td>
<td style="text-align:left;">
Cond1_UP
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Cond1 UP + Cond2 UP
</td>
<td style="text-align:left;">
Core_UP
</td>
<td style="text-align:left;">
Core_UP
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Cond1 Not Detected + Cond2 DOWN
</td>
<td style="text-align:left;">
Cond2_DOWN
</td>
<td style="text-align:left;">
Cond2_DOWN
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Cond1 Not Detected + Cond2 Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Cond1 Not Detected + Cond2 Not Significant
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Cond1 Not Detected + Cond2 Significant Negative
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Cond1 Not Detected + Cond2 Significant Positive
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Cond1 Not Detected + Cond2 UP
</td>
<td style="text-align:left;">
Cond2_UP
</td>
<td style="text-align:left;">
Cond2_UP
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Cond1 Significant Negative + Cond2 DOWN
</td>
<td style="text-align:left;">
Core_DOWN
</td>
<td style="text-align:left;">
Cond2_DOWN
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Cond1 Significant Negative + Cond2 Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Cond1 Significant Negative + Cond2 Not Significant
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Cond1 Significant Negative + Cond2 Significant Negative
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Cond1 Significant Negative + Cond2 Significant Positive
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Cond1 Significant Negative + Cond2 UP
</td>
<td style="text-align:left;">
Opposite
</td>
<td style="text-align:left;">
Cond2_UP
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Cond1 Significant Positive + Cond2 DOWN
</td>
<td style="text-align:left;">
Opposite
</td>
<td style="text-align:left;">
Cond2_DOWN
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Cond1 Significant Positive + Cond2 Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Cond1 Significant Positive + Cond2 Not Significant
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Cond1 Significant Positive + Cond2 Significant Negative
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Cond1 Significant Positive + Cond2 Significant Positive
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Cond1 Significant Positive + Cond2 UP
</td>
<td style="text-align:left;">
Core_UP
</td>
<td style="text-align:left;">
Cond2_UP
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Cond1 Not Significant + Cond2 DOWN
</td>
<td style="text-align:left;">
Cond2_DOWN
</td>
<td style="text-align:left;">
Cond2_DOWN
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Cond1 Not Significant + Cond2 Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Cond1 Not Significant + Cond2 Not Significant
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Cond1 Not Significant + Cond2 Significant Negative
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Cond1 Not Significant + Cond2 Significant Positive
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Cond1 Not Significant + Cond2 UP
</td>
<td style="text-align:left;">
Cond1_UP
</td>
<td style="text-align:left;">
Cond1_UP
</td>
</tr>
</tbody>
</table>
<p><br>
Now let’s use the data and do clustering:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">MCAres</span> <span class="op">&lt;-</span> <span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/MCA_2Cond.html">MCA_2Cond</a></span><span class="op">(</span>Cond1_File<span class="op">=</span><span class="va">DMA_Annova</span><span class="op">[[</span><span class="st">"DF"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"DMA_result"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"786-O_vs_HK2"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                      Cond2_File<span class="op">=</span><span class="va">DMA_Annova</span><span class="op">[[</span><span class="st">"DF"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"DMA_result"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"786-M1A_vs_HK2"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                      MetaboliteID<span class="op">=</span> <span class="st">"Metabolite"</span>,</span>
<span>                      Cond1ValueCol<span class="op">=</span><span class="st">"Log2FC"</span>,</span>
<span>                      Cond1PadjCol<span class="op">=</span><span class="st">"p.adj"</span>,</span>
<span>                      Cond2ValueCol<span class="op">=</span><span class="st">"Log2FC"</span>,</span>
<span>                      Cond2PadjCol<span class="op">=</span><span class="st">"p.adj"</span>,</span>
<span>                      Cond1_padj_cutoff<span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                      Cond2_padj_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                      Cond1_FC_cutoff<span class="op">=</span> <span class="fl">1</span>,</span>
<span>                      Cond2_FC_cutoff <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                      backgroundMethod<span class="op">=</span><span class="st">"C1&amp;C2"</span>,</span>
<span>                      OutputFileName <span class="op">=</span> <span class="st">"MCA_2Cond"</span>,</span>
<span>                      Save_as_Results <span class="op">=</span> <span class="st">"xlsx"</span><span class="op">)</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Summary of MCA: 2 Conditions.
</caption>
<thead><tr>
<th style="text-align:left;">
Regulation Grouping
</th>
<th style="text-align:left;">
SiRCle cluster Name
</th>
<th style="text-align:right;">
Number of Genes
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
RG1_All
</td>
<td style="text-align:left;">
Cond1 DOWN + Cond2 DOWN
</td>
<td style="text-align:right;">
26
</td>
</tr>
<tr>
<td style="text-align:left;">
RG1_All
</td>
<td style="text-align:left;">
Cond1 DOWN + Cond2 Not Significant
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
RG1_All
</td>
<td style="text-align:left;">
Cond1 DOWN + Cond2 Significant Negative
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
RG1_All
</td>
<td style="text-align:left;">
Cond1 Not Significant + Cond2 Not Significant
</td>
<td style="text-align:right;">
74
</td>
</tr>
<tr>
<td style="text-align:left;">
RG1_All
</td>
<td style="text-align:left;">
Cond1 Not Significant + Cond2 Significant Negative
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
RG1_All
</td>
<td style="text-align:left;">
Cond1 Not Significant + Cond2 Significant Positive
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
RG1_All
</td>
<td style="text-align:left;">
Cond1 Not Significant + Cond2 UP
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
RG1_All
</td>
<td style="text-align:left;">
Cond1 Significant Negative + Cond2 DOWN
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
RG1_All
</td>
<td style="text-align:left;">
Cond1 Significant Negative + Cond2 Not Significant
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
RG1_All
</td>
<td style="text-align:left;">
Cond1 Significant Negative + Cond2 Significant Negative
</td>
<td style="text-align:right;">
25
</td>
</tr>
<tr>
<td style="text-align:left;">
RG1_All
</td>
<td style="text-align:left;">
Cond1 Significant Positive + Cond2 Not Significant
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
RG1_All
</td>
<td style="text-align:left;">
Cond1 Significant Positive + Cond2 Significant Positive
</td>
<td style="text-align:right;">
13
</td>
</tr>
<tr>
<td style="text-align:left;">
RG1_All
</td>
<td style="text-align:left;">
Cond1 Significant Positive + Cond2 UP
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
RG1_All
</td>
<td style="text-align:left;">
Cond1 UP + Cond2 Significant Positive
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
RG1_All
</td>
<td style="text-align:left;">
Cond1 UP + Cond2 UP
</td>
<td style="text-align:right;">
9
</td>
</tr>
<tr>
<td style="text-align:left;">
RG2_Significant
</td>
<td style="text-align:left;">
Cond1_DOWN
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
RG2_Significant
</td>
<td style="text-align:left;">
Cond2_UP
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
RG2_Significant
</td>
<td style="text-align:left;">
Core_DOWN
</td>
<td style="text-align:right;">
33
</td>
</tr>
<tr>
<td style="text-align:left;">
RG2_Significant
</td>
<td style="text-align:left;">
Core_UP
</td>
<td style="text-align:right;">
13
</td>
</tr>
<tr>
<td style="text-align:left;">
RG2_Significant
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:right;">
128
</td>
</tr>
<tr>
<td style="text-align:left;">
RG3_SignificantChange
</td>
<td style="text-align:left;">
Cond1_DOWN
</td>
<td style="text-align:right;">
7
</td>
</tr>
<tr>
<td style="text-align:left;">
RG3_SignificantChange
</td>
<td style="text-align:left;">
Cond1_UP
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
RG3_SignificantChange
</td>
<td style="text-align:left;">
Cond2_DOWN
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
RG3_SignificantChange
</td>
<td style="text-align:left;">
Cond2_UP
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
RG3_SignificantChange
</td>
<td style="text-align:left;">
Core_DOWN
</td>
<td style="text-align:right;">
26
</td>
</tr>
<tr>
<td style="text-align:left;">
RG3_SignificantChange
</td>
<td style="text-align:left;">
Core_UP
</td>
<td style="text-align:right;">
9
</td>
</tr>
<tr>
<td style="text-align:left;">
RG3_SignificantChange
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:right;">
128
</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="ora-on-each-metabolite-cluster">ORA on each metabolite cluster<a class="anchor" aria-label="anchor" href="#ora-on-each-metabolite-cluster"></a>
</h4>
<p>As explained in detail above, Over Representation Analysis (ORA) is a
pathway enrichment analysis (PEA) method. As ORA is based on the Fishers
exact test it is perfect to test if a set of features (=metabolic
pathways) are over-represented in the selection of features (= clusters
of metabolites) from the data in comparison to all measured features
(all metabolites). In detail, <code><a href="../reference/MC_ORA.html">MC_ORA()</a></code> will perform ORA on
each of the metabolite clusters using all metabolites as the
background.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#We need a pathway file with the columns Metabolite, term and description</span></span>
<span><span class="va">PathwayInput</span> <span class="op">&lt;-</span> <span class="va">MappingInfo</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">4</span><span class="op">)</span><span class="op">]</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span><span class="st">"term"</span><span class="op">=</span><span class="st">"Pathway"</span><span class="op">)</span></span>
<span><span class="va">PathwayInput</span><span class="op">$</span><span class="va">Description</span> <span class="op">&lt;-</span> <span class="va">PathwayInput</span><span class="op">$</span><span class="va">term</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Pathway Input for MetaProViz::MC_ORA.
</caption>
<thead><tr>
<th style="text-align:left;">
Metabolite
</th>
<th style="text-align:left;">
term
</th>
<th style="text-align:left;">
Description
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
N-acetylaspartate
</td>
<td style="text-align:left;">
Alanine, aspartate and glutamate metabolism
</td>
<td style="text-align:left;">
Alanine, aspartate and glutamate metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
argininosuccinate
</td>
<td style="text-align:left;">
Alanine, aspartate and glutamate metabolism
</td>
<td style="text-align:left;">
Alanine, aspartate and glutamate metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
N-acetylaspartylglutamate
</td>
<td style="text-align:left;">
Alanine, aspartate and glutamate metabolism
</td>
<td style="text-align:left;">
Alanine, aspartate and glutamate metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
creatinine
</td>
<td style="text-align:left;">
Arginine and proline metabolism
</td>
<td style="text-align:left;">
Arginine and proline metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
4-guanidinobutanoate
</td>
<td style="text-align:left;">
Arginine and proline metabolism
</td>
<td style="text-align:left;">
Arginine and proline metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
homoarginine
</td>
<td style="text-align:left;">
Arginine and proline metabolism
</td>
<td style="text-align:left;">
Arginine and proline metabolism
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="section level2">
<h2 id="run-metaproviz-visualisation">3. Run MetaProViz Visualisation<a class="anchor" aria-label="anchor" href="#run-metaproviz-visualisation"></a>
</h2>
<p>The big advantages of the <code>MetaProViz</code> visualization
module is its flexible and easy usage, which we will showcase below and
that the figures are saved in a publication ready style and format. For
instance, the x- and y-axis size will always be adjusted for the amount
of samples or features (=metabolites) plotted, or in the case of Volcano
plot and PCA plot the axis size is fixed and not affected by figure
legends or title. In this way, there is no need for many adjustments and
the figures can just be dropped into the presentation or paper and are
all in the same style.<br><br>
All the <code>VizPlotName()</code> functions are constructed in the same
way. Indeed, with the parameter <code>Plot_SettingsInfo</code> the user
can pass a named vector with information about the metadata column that
should be used to customize the plot by colour, shape or creating
individual plots, which will all be showcased for the different plot
types. Via the parameter <code>Plot_SettingsFile</code> the user can
pass the metadata DF, which can be dependent on the plot type for the
samples and/or the features (=metabolites). In case of both the
parameter is named <code>Plot_SettingsFile_Sample</code> and
<code>Plot_SettingsFile_Metab</code>.<br><br>
In each of those Plot_Settings, the user can label color and/or shape
based on additional information (e.g. Pathway information, Cluster
information or other other demographics like gender). Moreover, we also
enable to plot individual plots where applicable based on those MetaData
(e.g. one plot for each metabolic pathway).<br>
For this we need a metadata table including information about our
samples that could be relevant to e.g. color code:<br></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">MetaData_Sample</span> <span class="op">&lt;-</span> <span class="va">Intra_Preprocessed</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span><span class="op">]</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>   <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Celltype <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/case_when.html" class="external-link">case_when</a></span><span class="op">(</span><span class="va">Conditions</span><span class="op">==</span><span class="st">"HK2"</span> <span class="op">~</span> <span class="st">'Healthy'</span>,</span>
<span>                               <span class="va">Conditions</span><span class="op">==</span><span class="st">"786-O"</span> <span class="op">~</span> <span class="st">'Primary Tumour'</span>,</span>
<span>                               <span class="cn">TRUE</span> <span class="op">~</span> <span class="st">'Metastatic Tumour'</span><span class="op">)</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>   <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Status <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/case_when.html" class="external-link">case_when</a></span><span class="op">(</span><span class="va">Conditions</span><span class="op">==</span><span class="st">"HK2"</span> <span class="op">~</span> <span class="st">'Healthy'</span>,</span>
<span>                               <span class="cn">TRUE</span> <span class="op">~</span> <span class="st">'Cancer'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Metadata table including additional information about our Samples.
</caption>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Conditions
</th>
<th style="text-align:right;">
Biological_Replicates
</th>
<th style="text-align:left;">
Celltype
</th>
<th style="text-align:left;">
Status
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
786-M1A_1
</td>
<td style="text-align:left;">
786-M1A
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
Metastatic Tumour
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
786-M1A_2
</td>
<td style="text-align:left;">
786-M1A
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
Metastatic Tumour
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
786-M1A_3
</td>
<td style="text-align:left;">
786-M1A
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
Metastatic Tumour
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
786-M2A_1
</td>
<td style="text-align:left;">
786-M2A
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
Metastatic Tumour
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
786-M2A_2
</td>
<td style="text-align:left;">
786-M2A
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
Metastatic Tumour
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
786-M2A_3
</td>
<td style="text-align:left;">
786-M2A
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
Metastatic Tumour
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
786-O_1
</td>
<td style="text-align:left;">
786-O
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
Primary Tumour
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
786-O_2
</td>
<td style="text-align:left;">
786-O
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
Primary Tumour
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
786-O_3
</td>
<td style="text-align:left;">
786-O
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
Primary Tumour
</td>
<td style="text-align:left;">
Cancer
</td>
</tr>
<tr>
<td style="text-align:left;">
HK2_1
</td>
<td style="text-align:left;">
HK2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
Healthy
</td>
<td style="text-align:left;">
Healthy
</td>
</tr>
<tr>
<td style="text-align:left;">
HK2_2
</td>
<td style="text-align:left;">
HK2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
Healthy
</td>
<td style="text-align:left;">
Healthy
</td>
</tr>
<tr>
<td style="text-align:left;">
HK2_3
</td>
<td style="text-align:left;">
HK2
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
Healthy
</td>
<td style="text-align:left;">
Healthy
</td>
</tr>
</tbody>
</table>
<p><br>
Moreover, we can use MetaData for our features (=Metabolites), which we
loaded with the <code>MappingInfo</code> and we can also add the
information on which cluster a metabolite was assigned to in the
<code>MetaProViz::MCA()</code> analysis above:<br></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">MetaData_Metab</span> <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">MappingInfo</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">4</span><span class="op">)</span><span class="op">]</span>, <span class="va">MCAres</span><span class="op">[[</span><span class="st">"DF"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"MCA_result"</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">14</span><span class="op">)</span><span class="op">]</span>, by<span class="op">=</span><span class="st">"Metabolite"</span>, all.y<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Metadata table including additional information about the Metabolites.
</caption>
<thead><tr>
<th style="text-align:left;">
Metabolite
</th>
<th style="text-align:left;">
Pathway
</th>
<th style="text-align:left;">
RG2_Significant
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
2-ketoglutarate
</td>
<td style="text-align:left;">
Citrate cycle (TCA cycle)
</td>
<td style="text-align:left;">
Core_UP
</td>
</tr>
<tr>
<td style="text-align:left;">
2/3-phosphoglycerate
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
4-hydroxyphenyllactate
</td>
<td style="text-align:left;">
Not assigned
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
ATP
</td>
<td style="text-align:left;">
Purine metabolism
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
beta-alanine
</td>
<td style="text-align:left;">
Pyrimidine metabolism
</td>
<td style="text-align:left;">
Core_DOWN
</td>
</tr>
<tr>
<td style="text-align:left;">
beta-citrylglutamic acid
</td>
<td style="text-align:left;">
Not assigned
</td>
<td style="text-align:left;">
None
</td>
</tr>
</tbody>
</table>
<p><br>
Noteworthy, here we can also use the KEGG pathways we used for the
pathway analysis.</p>
<div class="section level4">
<h4 id="pca-plots">PCA plots<a class="anchor" aria-label="anchor" href="#pca-plots"></a>
</h4>
<p>Principal component analysis (PCA) is a dimensionality reduction
method that reduces all the measured features (=metabolites) of one
sample into a few features in the different principal components,
whereby each principal component can explain a certain percentage of the
variance between the different samples. Hence, this enables
interpretation of sample clustering based on the measured features
(=metabolites).<br>
As mentioned above, PCA plots can be quite useful for quality control,
but of course it offers us many more opportunities, which will be
showcased here.<br><br>
As input, we need a DF that contains the samples as rownames and the
features (=metabolites) as column names:<br></p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Input_PCA</span> <span class="op">&lt;-</span> <span class="va">Intra_Preprocessed</span><span class="op">[</span>,<span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">]</span><span class="co">#remove columns that include Metadata such as cell type,...</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Input_data for <code><a href="../reference/VizPCA.html">MetaProViz::VizPCA()</a></code>, with samples as
rownames and metabolites as column names.
</caption>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
hippuric acid-d5
</th>
<th style="text-align:right;">
2/3-phosphoglycerate
</th>
<th style="text-align:right;">
2-aminoadipic acid
</th>
<th style="text-align:right;">
2-hydroxyglutarate
</th>
<th style="text-align:right;">
2-ketoglutarate
</th>
<th style="text-align:right;">
4-guanidinobutanoate
</th>
<th style="text-align:right;">
4-hydroxyphenyllactate
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
786-M1A_1
</td>
<td style="text-align:right;">
4629332172
</td>
<td style="text-align:right;">
56926513
</td>
<td style="text-align:right;">
7523262
</td>
<td style="text-align:right;">
424773945
</td>
<td style="text-align:right;">
960112993
</td>
<td style="text-align:right;">
1921602
</td>
<td style="text-align:right;">
2202889
</td>
</tr>
<tr>
<td style="text-align:left;">
786-M1A_2
</td>
<td style="text-align:right;">
4344689204
</td>
<td style="text-align:right;">
48391908
</td>
<td style="text-align:right;">
7802414
</td>
<td style="text-align:right;">
433248034
</td>
<td style="text-align:right;">
980130169
</td>
<td style="text-align:right;">
1886961
</td>
<td style="text-align:right;">
2040746
</td>
</tr>
<tr>
<td style="text-align:left;">
786-M1A_3
</td>
<td style="text-align:right;">
4218419637
</td>
<td style="text-align:right;">
45848651
</td>
<td style="text-align:right;">
7249191
</td>
<td style="text-align:right;">
417901363
</td>
<td style="text-align:right;">
1004430291
</td>
<td style="text-align:right;">
2002094
</td>
<td style="text-align:right;">
2207484
</td>
</tr>
<tr>
<td style="text-align:left;">
786-M2A_1
</td>
<td style="text-align:right;">
4800921532
</td>
<td style="text-align:right;">
45829441
</td>
<td style="text-align:right;">
6142860
</td>
<td style="text-align:right;">
438809273
</td>
<td style="text-align:right;">
845124514
</td>
<td style="text-align:right;">
2625730
</td>
<td style="text-align:right;">
2255774
</td>
</tr>
<tr>
<td style="text-align:left;">
786-M2A_2
</td>
<td style="text-align:right;">
3850001995
</td>
<td style="text-align:right;">
44285426
</td>
<td style="text-align:right;">
6234439
</td>
<td style="text-align:right;">
432668677
</td>
<td style="text-align:right;">
886775267
</td>
<td style="text-align:right;">
1982761
</td>
<td style="text-align:right;">
2337265
</td>
</tr>
<tr>
<td style="text-align:left;">
786-M2A_3
</td>
<td style="text-align:right;">
4121917117
</td>
<td style="text-align:right;">
44464750
</td>
<td style="text-align:right;">
6352492
</td>
<td style="text-align:right;">
463539257
</td>
<td style="text-align:right;">
898862621
</td>
<td style="text-align:right;">
1638469
</td>
<td style="text-align:right;">
2247034
</td>
</tr>
<tr>
<td style="text-align:left;">
786-O_1
</td>
<td style="text-align:right;">
3900419287
</td>
<td style="text-align:right;">
36968914
</td>
<td style="text-align:right;">
8977305
</td>
<td style="text-align:right;">
390323670
</td>
<td style="text-align:right;">
888809328
</td>
<td style="text-align:right;">
2393129
</td>
<td style="text-align:right;">
2002372
</td>
</tr>
<tr>
<td style="text-align:left;">
786-O_2
</td>
<td style="text-align:right;">
4501256250
</td>
<td style="text-align:right;">
30523496
</td>
<td style="text-align:right;">
9099066
</td>
<td style="text-align:right;">
463781490
</td>
<td style="text-align:right;">
1058406623
</td>
<td style="text-align:right;">
2331351
</td>
<td style="text-align:right;">
2115981
</td>
</tr>
<tr>
<td style="text-align:left;">
786-O_3
</td>
<td style="text-align:right;">
4141232360
</td>
<td style="text-align:right;">
29334597
</td>
<td style="text-align:right;">
9034721
</td>
<td style="text-align:right;">
408324655
</td>
<td style="text-align:right;">
1013301175
</td>
<td style="text-align:right;">
1697183
</td>
<td style="text-align:right;">
2195370
</td>
</tr>
<tr>
<td style="text-align:left;">
HK2_1
</td>
<td style="text-align:right;">
3174566793
</td>
<td style="text-align:right;">
30962487
</td>
<td style="text-align:right;">
5807611
</td>
<td style="text-align:right;">
188606001
</td>
<td style="text-align:right;">
326693903
</td>
<td style="text-align:right;">
4423234
</td>
<td style="text-align:right;">
3967063
</td>
</tr>
<tr>
<td style="text-align:left;">
HK2_2
</td>
<td style="text-align:right;">
3183656156
</td>
<td style="text-align:right;">
32597393
</td>
<td style="text-align:right;">
7329704
</td>
<td style="text-align:right;">
229054002
</td>
<td style="text-align:right;">
367069890
</td>
<td style="text-align:right;">
4138012
</td>
<td style="text-align:right;">
3975926
</td>
</tr>
<tr>
<td style="text-align:left;">
HK2_3
</td>
<td style="text-align:right;">
3369292371
</td>
<td style="text-align:right;">
29536635
</td>
<td style="text-align:right;">
7166290
</td>
<td style="text-align:right;">
251312598
</td>
<td style="text-align:right;">
460404941
</td>
<td style="text-align:right;">
4370603
</td>
<td style="text-align:right;">
4230915
</td>
</tr>
</tbody>
</table>
<p><br>
Now lets check out the standard plot:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/VizPCA.html">VizPCA</a></span><span class="op">(</span>Input_data<span class="op">=</span><span class="va">Input_PCA</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-32-1.png" alt="Figure: Standard Settings." width="576"><p class="caption">
Figure: Standard Settings.
</p>
</div>
<p>Next, we can interactively choose shape and color using the
additional information of interest from our Metadata. Especially for
complex data, such as patient data, it can be valuable to use different
demographics (e.g. age, gender, medication,…) for this. First lets check
if we have any batch effect by colour coding for the biological
replicates, which would be the case if the replicates cluster
together.<br></p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/VizPCA.html">VizPCA</a></span><span class="op">(</span>Plot_SettingsInfo<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"Biological_Replicates"</span><span class="op">)</span>,</span>
<span>                   Plot_SettingsFile<span class="op">=</span> <span class="va">MetaData_Sample</span> ,</span>
<span>                   Input_data<span class="op">=</span><span class="va">Input_PCA</span>,</span>
<span>                   OutputPlotName <span class="op">=</span> <span class="st">"Batch Effect"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-33-1.png" alt="Figure: Do we have a batch effect?" width="576"><p class="caption">
Figure: Do we have a batch effect?
</p>
</div>
<p>Next, we can colour code for condition and use the biological
replicates in the shape parameter:<br></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/VizPCA.html">VizPCA</a></span><span class="op">(</span>Plot_SettingsInfo<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"Conditions"</span>, shape<span class="op">=</span><span class="st">"Biological_Replicates"</span><span class="op">)</span>,</span>
<span>                   Plot_SettingsFile<span class="op">=</span> <span class="va">MetaData_Sample</span>,</span>
<span>                   Input_data<span class="op">=</span><span class="va">Input_PCA</span>,</span>
<span>                   OutputPlotName <span class="op">=</span> <span class="st">"Sample Conditions"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-34-1.png" alt="Figure: Do the samples cluster for the conditions?" width="576"><p class="caption">
Figure: Do the samples cluster for the conditions?
</p>
</div>
<p>The different cell lines we have are either control or cancerous, so
we can display this too. Here is becomes apparent that the cell status
is responsible for 64% of the variance (x-axis).<br></p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/VizPCA.html">VizPCA</a></span><span class="op">(</span>Plot_SettingsInfo<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"Status"</span><span class="op">)</span>,</span>
<span>                   Plot_SettingsFile<span class="op">=</span> <span class="va">MetaData_Sample</span>,</span>
<span>                   Input_data<span class="op">=</span><span class="va">Input_PCA</span>,</span>
<span>                   OutputPlotName <span class="op">=</span> <span class="st">"Sample Status"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-35-1.png" alt="Figure: Do the samples cluster for the Cell status?" width="576"><p class="caption">
Figure: Do the samples cluster for the Cell status?
</p>
</div>
<p>We can separate the cancerous cell lines further into metastatic or
primary. This shows us that this is separated on the y-axis and accounts
for 30%of the variance.<br></p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/VizPCA.html">VizPCA</a></span><span class="op">(</span>Plot_SettingsInfo<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"Celltype"</span>, shape<span class="op">=</span><span class="st">"Status"</span><span class="op">)</span>,</span>
<span>                   Plot_SettingsFile<span class="op">=</span> <span class="va">MetaData_Sample</span>,</span>
<span>                   Input_data<span class="op">=</span><span class="va">Input_PCA</span>,</span>
<span>                   OutputPlotName <span class="op">=</span> <span class="st">"Cell type"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-36-1.png" alt="Figure: Do the samples cluster for the Cell type?" width="576"><p class="caption">
Figure: Do the samples cluster for the Cell type?
</p>
</div>
<p>Lastly, its worth mentioning that one can also change many style
parameters to customize the plot:<br></p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/VizPCA.html">VizPCA</a></span><span class="op">(</span>Plot_SettingsInfo<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"Celltype"</span>, shape<span class="op">=</span><span class="st">"Status"</span><span class="op">)</span>,</span>
<span>                   Plot_SettingsFile<span class="op">=</span> <span class="va">MetaData_Sample</span>,</span>
<span>                   Input_data<span class="op">=</span><span class="va">Input_PCA</span>,</span>
<span>                   color_palette<span class="op">=</span> <span class="cn">NULL</span>,<span class="co">#pass your own colour palette</span></span>
<span>                   shape_palette<span class="op">=</span><span class="cn">NULL</span>,<span class="co">#pass your own shape palette</span></span>
<span>                   Show_Loadings <span class="op">=</span> <span class="cn">FALSE</span>, <span class="co">#show loadings (default=FALSE)</span></span>
<span>                   Scaling <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co">#use scaling</span></span>
<span>                   Theme<span class="op">=</span><span class="cn">NULL</span>,<span class="co">#theme_classic() is default, use any: https://ggplot2.tidyverse.org/reference/ggtheme.html</span></span>
<span>                   OutputPlotName<span class="op">=</span> <span class="st">'Cell type'</span>,</span>
<span>                   Save_as_Plot <span class="op">=</span> <span class="st">"svg"</span><span class="op">)</span><span class="co">#file format</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="heatmaps">Heatmaps<a class="anchor" aria-label="anchor" href="#heatmaps"></a>
</h4>
<p>Clustered heatmaps can be useful to understand the patterns in the
data, which will be showcased on different examples.<br>
As input, we need a DF that contains the samples as rownames and the
features (=metabolites) as column names:<br></p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Input_Heatmap</span> <span class="op">&lt;-</span>  <span class="va">Intra_Preprocessed</span><span class="op">[</span>,<span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">]</span><span class="co">#remove columns that include Metadata such as cell type,...</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Input for <code><a href="../reference/VizHeatmap.html">MetaProViz::VizHeatmap()</a></code>, with samples as
rownames and metabolites as column names.
</caption>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
hippuric acid-d5
</th>
<th style="text-align:right;">
2/3-phosphoglycerate
</th>
<th style="text-align:right;">
2-aminoadipic acid
</th>
<th style="text-align:right;">
2-hydroxyglutarate
</th>
<th style="text-align:right;">
2-ketoglutarate
</th>
<th style="text-align:right;">
4-guanidinobutanoate
</th>
<th style="text-align:right;">
4-hydroxyphenyllactate
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
786-M1A_1
</td>
<td style="text-align:right;">
4629332172
</td>
<td style="text-align:right;">
56926513
</td>
<td style="text-align:right;">
7523262
</td>
<td style="text-align:right;">
424773945
</td>
<td style="text-align:right;">
960112993
</td>
<td style="text-align:right;">
1921602
</td>
<td style="text-align:right;">
2202889
</td>
</tr>
<tr>
<td style="text-align:left;">
786-M1A_2
</td>
<td style="text-align:right;">
4344689204
</td>
<td style="text-align:right;">
48391908
</td>
<td style="text-align:right;">
7802414
</td>
<td style="text-align:right;">
433248034
</td>
<td style="text-align:right;">
980130169
</td>
<td style="text-align:right;">
1886961
</td>
<td style="text-align:right;">
2040746
</td>
</tr>
<tr>
<td style="text-align:left;">
786-M1A_3
</td>
<td style="text-align:right;">
4218419637
</td>
<td style="text-align:right;">
45848651
</td>
<td style="text-align:right;">
7249191
</td>
<td style="text-align:right;">
417901363
</td>
<td style="text-align:right;">
1004430291
</td>
<td style="text-align:right;">
2002094
</td>
<td style="text-align:right;">
2207484
</td>
</tr>
<tr>
<td style="text-align:left;">
786-M2A_1
</td>
<td style="text-align:right;">
4800921532
</td>
<td style="text-align:right;">
45829441
</td>
<td style="text-align:right;">
6142860
</td>
<td style="text-align:right;">
438809273
</td>
<td style="text-align:right;">
845124514
</td>
<td style="text-align:right;">
2625730
</td>
<td style="text-align:right;">
2255774
</td>
</tr>
<tr>
<td style="text-align:left;">
786-M2A_2
</td>
<td style="text-align:right;">
3850001995
</td>
<td style="text-align:right;">
44285426
</td>
<td style="text-align:right;">
6234439
</td>
<td style="text-align:right;">
432668677
</td>
<td style="text-align:right;">
886775267
</td>
<td style="text-align:right;">
1982761
</td>
<td style="text-align:right;">
2337265
</td>
</tr>
<tr>
<td style="text-align:left;">
786-M2A_3
</td>
<td style="text-align:right;">
4121917117
</td>
<td style="text-align:right;">
44464750
</td>
<td style="text-align:right;">
6352492
</td>
<td style="text-align:right;">
463539257
</td>
<td style="text-align:right;">
898862621
</td>
<td style="text-align:right;">
1638469
</td>
<td style="text-align:right;">
2247034
</td>
</tr>
<tr>
<td style="text-align:left;">
786-O_1
</td>
<td style="text-align:right;">
3900419287
</td>
<td style="text-align:right;">
36968914
</td>
<td style="text-align:right;">
8977305
</td>
<td style="text-align:right;">
390323670
</td>
<td style="text-align:right;">
888809328
</td>
<td style="text-align:right;">
2393129
</td>
<td style="text-align:right;">
2002372
</td>
</tr>
<tr>
<td style="text-align:left;">
786-O_2
</td>
<td style="text-align:right;">
4501256250
</td>
<td style="text-align:right;">
30523496
</td>
<td style="text-align:right;">
9099066
</td>
<td style="text-align:right;">
463781490
</td>
<td style="text-align:right;">
1058406623
</td>
<td style="text-align:right;">
2331351
</td>
<td style="text-align:right;">
2115981
</td>
</tr>
<tr>
<td style="text-align:left;">
786-O_3
</td>
<td style="text-align:right;">
4141232360
</td>
<td style="text-align:right;">
29334597
</td>
<td style="text-align:right;">
9034721
</td>
<td style="text-align:right;">
408324655
</td>
<td style="text-align:right;">
1013301175
</td>
<td style="text-align:right;">
1697183
</td>
<td style="text-align:right;">
2195370
</td>
</tr>
<tr>
<td style="text-align:left;">
HK2_1
</td>
<td style="text-align:right;">
3174566793
</td>
<td style="text-align:right;">
30962487
</td>
<td style="text-align:right;">
5807611
</td>
<td style="text-align:right;">
188606001
</td>
<td style="text-align:right;">
326693903
</td>
<td style="text-align:right;">
4423234
</td>
<td style="text-align:right;">
3967063
</td>
</tr>
<tr>
<td style="text-align:left;">
HK2_2
</td>
<td style="text-align:right;">
3183656156
</td>
<td style="text-align:right;">
32597393
</td>
<td style="text-align:right;">
7329704
</td>
<td style="text-align:right;">
229054002
</td>
<td style="text-align:right;">
367069890
</td>
<td style="text-align:right;">
4138012
</td>
<td style="text-align:right;">
3975926
</td>
</tr>
<tr>
<td style="text-align:left;">
HK2_3
</td>
<td style="text-align:right;">
3369292371
</td>
<td style="text-align:right;">
29536635
</td>
<td style="text-align:right;">
7166290
</td>
<td style="text-align:right;">
251312598
</td>
<td style="text-align:right;">
460404941
</td>
<td style="text-align:right;">
4370603
</td>
<td style="text-align:right;">
4230915
</td>
</tr>
</tbody>
</table>
<p><br>
Now we can generate an overview heatmap. Since we plot all metabolites
the metabolite names are not plotted since this would get too crowded
(You can enforce this by changing the parameter
<code>enforce_FeatureNames = TRUE</code>).</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/VizHeatmap.html">VizHeatmap</a></span><span class="op">(</span>Input_data <span class="op">=</span> <span class="va">Input_Heatmap</span>, </span>
<span>                       OutputPlotName <span class="op">=</span> <span class="st">"Overview"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-40-1.png" alt="Overview heatmap." width="700"><p class="caption">
Overview heatmap.
</p>
</div>
<p><br>
Here we can add as many sample metadata information as needed at the
same time:<br></p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/VizHeatmap.html">VizHeatmap</a></span><span class="op">(</span>Input_data <span class="op">=</span> <span class="va">Input_Heatmap</span>,</span>
<span>                       Plot_SettingsFile_Sample <span class="op">=</span> <span class="va">MetaData_Sample</span>,</span>
<span>                       Plot_SettingsInfo <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>color_Sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"Conditions"</span>,<span class="st">"Biological_Replicates"</span>, <span class="st">"Celltype"</span>, <span class="st">"Status"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                       OutputPlotName <span class="op">=</span> <span class="st">"Colour Samples"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-41-1.png" alt="Colour for sample metadata." width="700"><p class="caption">
Colour for sample metadata.
</p>
</div>
<p><br>
Moreover, we can also add metabolite metadata information:<br></p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># row annotation: Color for Metabolites</span></span>
<span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/VizHeatmap.html">VizHeatmap</a></span><span class="op">(</span>Input_data <span class="op">=</span> <span class="va">Input_Heatmap</span>,</span>
<span>                       Plot_SettingsFile_Sample <span class="op">=</span> <span class="va">MetaData_Sample</span>,</span>
<span>                       Plot_SettingsInfo <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>color_Metab <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"Pathway"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                       Plot_SettingsFile_Metab <span class="op">=</span>  <span class="va">MappingInfo</span>, </span>
<span>                       OutputPlotName <span class="op">=</span> <span class="st">"Colour Metabolites"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in MetaProViz::VizHeatmap(Input_data = Input_Heatmap,</span></span>
<span><span class="co">#&gt; Plot_SettingsFile_Sample = MetaData_Sample, : The Input data contains</span></span>
<span><span class="co">#&gt; metabolites not found in Plot_SettingsFile_Metab.</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-42-1.png" alt="Colour for metabolite metadata." width="700"><p class="caption">
Colour for metabolite metadata.
</p>
</div>
<p><br>
Lastly, by generate individual plot for e.g. each pathway or the
metabolite clusters by adding individual (<code>individual_Sample</code>
or <code>individual_Metab</code>) to <code>Plot_SettingsInfo</code>. At
the same time we can still maintain the metadata information for both,
the samples and the metabolites. Together this can help us to draw
biological conclusions about the different pathways: Indeed, we can
observe for the <code>D-Amino acid metabolism</code> many metabolites
fall into the MCA-Cluster <code>Core_DOWN</code>, meaning in comparison
to HK2 cells we have a negative Log2FC for 786-O and 786-M1A.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># individual: One individual plot for each pathway, col annotation: Colour for samples</span></span>
<span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/VizHeatmap.html">VizHeatmap</a></span><span class="op">(</span>Input_data <span class="op">=</span> <span class="va">Input_Heatmap</span>, </span>
<span>                       Plot_SettingsFile_Sample <span class="op">=</span> <span class="va">MetaData_Sample</span>,</span>
<span>                       Plot_SettingsInfo <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>individual_Metab <span class="op">=</span> <span class="st">"Pathway"</span>,</span>
<span>                                             color_Sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"Conditions"</span>,<span class="st">"Biological_Replicates"</span><span class="op">)</span>,</span>
<span>                                             color_Metab <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"RG2_Significant"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                       Plot_SettingsFile_Metab <span class="op">=</span>  <span class="va">MetaData_Metab</span>,</span>
<span>                       OutputPlotName <span class="op">=</span> <span class="st">"Pathway"</span><span class="op">)</span></span></code></pre></div>
<div class="vscroll-plot">
<p><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-44-1.png" width="700" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-44-2.png" width="700" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-44-3.png" width="700" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-44-4.png" width="700" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-44-5.png" width="700" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-44-6.png" width="700" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-44-7.png" width="700" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-44-8.png" width="700" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-44-9.png" width="700" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-44-10.png" width="700" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-44-11.png" width="700" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-44-12.png" width="700" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-44-13.png" width="700" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-44-14.png" width="700" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-44-15.png" width="700" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-44-16.png" width="700" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-44-17.png" width="700" style="display: block; margin: auto auto auto 0;"></p>
</div>
<p><br><br><br>
You can also choose to make individual plots for any Sample Metadata
using <code>individual_Sample</code> (e.g. in patients you may want to
plot male and female separately). Moreover, you can also use both at the
same time.</p>
</div>
<div class="section level4">
<h4 id="volcano-plot">Volcano plot<a class="anchor" aria-label="anchor" href="#volcano-plot"></a>
</h4>
In general,we have three different <code>Plot_Settings</code>, which
will also be used for other plot types such as lollipop graphs.<br><code>1.</code> <code>"Standard"</code> is the standard version of the
plot, with one dataset being plotted.<br><code>2.</code> <code>"Conditions"</code> here two or more datasets will
be plotted together. How datasets can be plotted together depends on the
plot type.<br><code>3.</code> <code>"PEA"</code> stands for Pathway Enrichment
Analysis, and is used if the results of an GSE analysis should be
plotted as here the figure legends will be adapted.<br><br>
Here we will look at all the different options we have to display our
results from the different analysis (DMA, PEA), which will help us to
interpret our results as this can be sometimes difficult to do from the
many data tables.<br>
Just a quick reminder, how the input data look like:<br>
1. Results of Differential Metabolite Analysis (DMA): Log2FC and
stats:<br><table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Input_data for <code><a href="../reference/VizVolcano.html">MetaProViz::VizVolcano()</a></code> are for example
differential analysis results from <code><a href="../reference/DMA.html">MetaProViz::DMA()</a></code>.
</caption>
<thead><tr>
<th style="text-align:left;">
Metabolite
</th>
<th style="text-align:right;">
Log2FC
</th>
<th style="text-align:right;">
p.adj
</th>
<th style="text-align:right;">
t.val
</th>
<th style="text-align:right;">
786-M1A_1
</th>
<th style="text-align:right;">
786-M1A_2
</th>
<th style="text-align:right;">
786-M1A_3
</th>
<th style="text-align:right;">
HK2_1
</th>
<th style="text-align:right;">
HK2_2
</th>
<th style="text-align:right;">
HK2_3
</th>
<th style="text-align:left;">
KEGG.ID
</th>
<th style="text-align:left;">
KEGGCompound
</th>
<th style="text-align:left;">
Pathway
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
valine-d8
</td>
<td style="text-align:right;">
0.5165968
</td>
<td style="text-align:right;">
0.0027524
</td>
<td style="text-align:right;">
-674911156.2
</td>
<td style="text-align:right;">
2392542441.1
</td>
<td style="text-align:right;">
2229508295.5
</td>
<td style="text-align:right;">
2105059163.2
</td>
<td style="text-align:right;">
1484619272.3
</td>
<td style="text-align:right;">
1602336582.2
</td>
<td style="text-align:right;">
1615420576.6
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
5-aminovaleric acid*
</td>
<td style="text-align:right;">
0.6297507
</td>
<td style="text-align:right;">
0.0107010
</td>
<td style="text-align:right;">
-22646235.9
</td>
<td style="text-align:right;">
69317813.7
</td>
<td style="text-align:right;">
67050411.7
</td>
<td style="text-align:right;">
55705347.8
</td>
<td style="text-align:right;">
39938464.7
</td>
<td style="text-align:right;">
43488520.0
</td>
<td style="text-align:right;">
40707880.8
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Not assigned
</td>
</tr>
<tr>
<td style="text-align:left;">
indole
</td>
<td style="text-align:right;">
-0.5902244
</td>
<td style="text-align:right;">
0.0002600
</td>
<td style="text-align:right;">
312132.0
</td>
<td style="text-align:right;">
614014.8
</td>
<td style="text-align:right;">
620975.0
</td>
<td style="text-align:right;">
617495.7
</td>
<td style="text-align:right;">
992245.6
</td>
<td style="text-align:right;">
848342.7
</td>
<td style="text-align:right;">
948293.3
</td>
<td style="text-align:left;">
C00463
</td>
<td style="text-align:left;">
Indole
</td>
<td style="text-align:left;">
Not assigned
</td>
</tr>
<tr>
<td style="text-align:left;">
indole-3-lactate
</td>
<td style="text-align:right;">
-0.7901229
</td>
<td style="text-align:right;">
0.0054530
</td>
<td style="text-align:right;">
232172.0
</td>
<td style="text-align:right;">
332172.3
</td>
<td style="text-align:right;">
255106.9
</td>
<td style="text-align:right;">
367870.7
</td>
<td style="text-align:right;">
636497.7
</td>
<td style="text-align:right;">
501495.2
</td>
<td style="text-align:right;">
513673.2
</td>
<td style="text-align:left;">
C02043
</td>
<td style="text-align:left;">
Indolelactate
</td>
<td style="text-align:left;">
Not assigned
</td>
</tr>
<tr>
<td style="text-align:left;">
inosine
</td>
<td style="text-align:right;">
-0.1969354
</td>
<td style="text-align:right;">
0.8515773
</td>
<td style="text-align:right;">
100979.3
</td>
<td style="text-align:right;">
601970.7
</td>
<td style="text-align:right;">
782053.3
</td>
<td style="text-align:right;">
687193.2
</td>
<td style="text-align:right;">
720560.5
</td>
<td style="text-align:right;">
621020.3
</td>
<td style="text-align:right;">
1032574.3
</td>
<td style="text-align:left;">
C00294
</td>
<td style="text-align:left;">
Inosine
</td>
<td style="text-align:left;">
Purine metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
isoleucine
</td>
<td style="text-align:right;">
-0.2738529
</td>
<td style="text-align:right;">
0.0726755
</td>
<td style="text-align:right;">
85478665.6
</td>
<td style="text-align:right;">
420825940.6
</td>
<td style="text-align:right;">
402399470.5
</td>
<td style="text-align:right;">
403550788.8
</td>
<td style="text-align:right;">
471426303.6
</td>
<td style="text-align:right;">
462688021.0
</td>
<td style="text-align:right;">
549097872.0
</td>
<td style="text-align:left;">
C00407
</td>
<td style="text-align:left;">
L-Isoleucine
</td>
<td style="text-align:left;">
Valine, leucine and isoleucine biosynthesis
</td>
</tr>
<tr>
<td style="text-align:left;">
isovalerylcarnitine
</td>
<td style="text-align:right;">
0.9170474
</td>
<td style="text-align:right;">
0.0063473
</td>
<td style="text-align:right;">
-14524759.5
</td>
<td style="text-align:right;">
29686277.5
</td>
<td style="text-align:right;">
30650177.3
</td>
<td style="text-align:right;">
32294320.8
</td>
<td style="text-align:right;">
19772945.0
</td>
<td style="text-align:right;">
15552910.3
</td>
<td style="text-align:right;">
13730642.0
</td>
<td style="text-align:left;">
C20826
</td>
<td style="text-align:left;">
Isovalerylcarnitine
</td>
<td style="text-align:left;">
Fatty acyl carnitines
</td>
</tr>
</tbody>
</table>
<br>
2. Results from Pathway Enrichment Analysis (PEA): Score (e.g. NES
score) and stats<br><table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Input_data for <code><a href="../reference/VizVolcano.html">MetaProViz::VizVolcano()</a></code> are for example
pathway enrichment results from <code><a href="../reference/DM_ORA.html">MetaProViz::DM_ORA()</a></code>.
</caption>
<thead><tr>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:left;">
MetaboliteIDs_in_pathway
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
Metabolites_in_Pathway
</th>
<th style="text-align:right;">
Percentage_of_Pathway_detected
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
ABC transporters - Homo sapiens (human)
</td>
<td style="text-align:left;">
11/24
</td>
<td style="text-align:left;">
32/122
</td>
<td style="text-align:right;">
0.0172642
</td>
<td style="text-align:right;">
0.2956418
</td>
<td style="text-align:left;">
Betaine/Glutathione/L-Alanine/L-Aspartate/L-Glutamate/L-Histidine/L-Leucine/L-Proline/L-Threonine/myo-Inositol/Taurine
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
130
</td>
<td style="text-align:right;">
8.46
</td>
</tr>
<tr>
<td style="text-align:left;">
Alanine, aspartate and glutamate metabolism - Homo sapiens (human)
</td>
<td style="text-align:left;">
6/24
</td>
<td style="text-align:left;">
14/122
</td>
<td style="text-align:right;">
0.0313864
</td>
<td style="text-align:right;">
0.2956418
</td>
<td style="text-align:left;">
2-Oxoglutarate/Citrate/L-Alanine/L-Aspartate/L-Glutamate/N-Acetyl-L-aspartate
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
27
</td>
<td style="text-align:right;">
22.22
</td>
</tr>
<tr>
<td style="text-align:left;">
D-Amino acid metabolism - Homo sapiens (human)
</td>
<td style="text-align:left;">
7/24
</td>
<td style="text-align:left;">
18/122
</td>
<td style="text-align:right;">
0.0343769
</td>
<td style="text-align:right;">
0.2956418
</td>
<td style="text-align:left;">
2-Oxoglutarate/L-Alanine/L-Aspartate/L-Glutamate/L-Histidine/L-Proline/L-Threonine
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
69
</td>
<td style="text-align:right;">
10.14
</td>
</tr>
<tr>
<td style="text-align:left;">
Histidine metabolism - Homo sapiens (human)
</td>
<td style="text-align:left;">
4/24
</td>
<td style="text-align:left;">
6/122
</td>
<td style="text-align:right;">
0.0135586
</td>
<td style="text-align:right;">
0.2956418
</td>
<td style="text-align:left;">
2-Oxoglutarate/L-Aspartate/L-Glutamate/L-Histidine
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
47
</td>
<td style="text-align:right;">
8.51
</td>
</tr>
</tbody>
</table>
<div class="section level5">
<h5 id="standard">
<strong>Standard</strong><a class="anchor" aria-label="anchor" href="#standard"></a>
</h5>
<p>Here we will first look into the results from the differential
analysis (see section <code>DMA</code> above) for the comparison of
<code>786-M1A_vs_HK2</code>:</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run with default parameter --&gt; only need to provide Input_data and the title we like</span></span>
<span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/VizVolcano.html">VizVolcano</a></span><span class="op">(</span>Input_data<span class="op">=</span><span class="va">DMA_786M1A_vs_HK2</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-47-1.png" alt="Figure: Standard figure displaying DMA results." width="672"><p class="caption">
Figure: Standard figure displaying DMA results.
</p>
</div>
<p><br>
If you seek to plot the metabolite names you can change the paramter
<code>SelectLab</code> from its default (<code>SelectLab=""</code>) to
NULL and the metabolite names will be plotted randomly.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run with default parameter --&gt; only need to provide Input_data and the title we like</span></span>
<span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/VizVolcano.html">VizVolcano</a></span><span class="op">(</span>Input_data<span class="op">=</span><span class="va">DMA_786M1A_vs_HK2</span>,</span>
<span>                       SelectLab <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-48-1.png" alt="Figure: Standard figure displaying DMA results." width="672"><p class="caption">
Figure: Standard figure displaying DMA results.
</p>
</div>
<p><br>
With the parameter <code>SelectLab</code> you can also pass a vector
with Metabolite names that should be labeled:</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run with default parameter --&gt; only need to provide Input_data and the title we like</span></span>
<span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/VizVolcano.html">VizVolcano</a></span><span class="op">(</span>Input_data<span class="op">=</span><span class="va">DMA_786M1A_vs_HK2</span>,</span>
<span>                       SelectLab <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"N-acetylaspartylglutamate"</span>, <span class="st">"cystathionine"</span>, <span class="st">"orotidine"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-49-1.png" alt="Figure: Standard figure displaying DMA results." width="672"><p class="caption">
Figure: Standard figure displaying DMA results.
</p>
</div>
<p><br>
Next we may be interested to understand which metabolite clusters based
on our MCA the metabolites of the plot correspond to. In order to do
this we can provide a Plot_SettingsFile with this additional information
and use this information to color code and/or to shape the dots on the
volcano plot. In order to choose the right column we need to provide a
vector Plot_SettingsInfo with this information.<br></p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">#Now we need to add our Plot_SettingsFile and the Plot_SettingsInfo:</span></span>
<span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/VizVolcano.html">VizVolcano</a></span><span class="op">(</span>Plot_Settings<span class="op">=</span><span class="st">"Standard"</span>,</span>
<span>                       Plot_SettingsInfo<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"RG2_Significant"</span><span class="op">)</span>,</span>
<span>                       Plot_SettingsFile<span class="op">=</span> <span class="va">MetaData_Metab</span>,</span>
<span>                       Input_data<span class="op">=</span><span class="va">DMA_786M1A_vs_HK2</span>,</span>
<span>                       OutputPlotName<span class="op">=</span> <span class="st">"786M1A versus HK2"</span>,</span>
<span>                       Subtitle<span class="op">=</span> <span class="st">"Results of DMA. Colour coded for metabolic clusters"</span> <span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-50-1.png" alt="Figure: Standard figure displaying DMA results colour coded/shaped for metabolic clusters from MCA results." width="672"><p class="caption">
Figure: Standard figure displaying DMA results colour coded/shaped for
metabolic clusters from MCA results.
</p>
</div>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">#If we want to use the shape instead of the colour for the cluster info, we can just change our Plot_SettingsInfo</span></span>
<span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/VizVolcano.html">VizVolcano</a></span><span class="op">(</span>Plot_Settings<span class="op">=</span><span class="st">"Standard"</span>,</span>
<span>                       Plot_SettingsInfo<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>shape<span class="op">=</span><span class="st">"RG2_Significant"</span><span class="op">)</span>,</span>
<span>                       Plot_SettingsFile<span class="op">=</span> <span class="va">MetaData_Metab</span>,</span>
<span>                       Input_data<span class="op">=</span><span class="va">DMA_786M1A_vs_HK2</span>,</span>
<span>                       OutputPlotName<span class="op">=</span> <span class="st">"786M1A versus HK2"</span>,</span>
<span>                       Subtitle<span class="op">=</span> <span class="st">"Results of DMA. Shape for metabolic clusters, color for significance."</span> <span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-50-2.png" alt="Figure: Standard figure displaying DMA results colour coded/shaped for metabolic clusters from MCA results." width="672"><p class="caption">
Figure: Standard figure displaying DMA results colour coded/shaped for
metabolic clusters from MCA results.
</p>
</div>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">#Of course, we can also adapt both, color and shape for the same parameter:</span></span>
<span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/VizVolcano.html">VizVolcano</a></span><span class="op">(</span>Plot_Settings<span class="op">=</span><span class="st">"Standard"</span>,</span>
<span>                       Plot_SettingsInfo<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>shape<span class="op">=</span><span class="st">"RG2_Significant"</span>, color<span class="op">=</span><span class="st">"RG2_Significant"</span><span class="op">)</span>,</span>
<span>                       Plot_SettingsFile<span class="op">=</span> <span class="va">MetaData_Metab</span>,</span>
<span>                       Input_data<span class="op">=</span><span class="va">DMA_786M1A_vs_HK2</span>,</span>
<span>                       OutputPlotName<span class="op">=</span> <span class="st">"786M1A versus HK2"</span>,</span>
<span>                       Subtitle<span class="op">=</span> <span class="st">"Results of DMA. Shape and color for metabolic clusters."</span> <span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-50-3.png" alt="Figure: Standard figure displaying DMA results colour coded/shaped for metabolic clusters from MCA results." width="672"><p class="caption">
Figure: Standard figure displaying DMA results colour coded/shaped for
metabolic clusters from MCA results.
</p>
</div>
<p><br>
Given that we also know, which metabolic pathway the metabolites
correspond to, we can add this information into the plot. This is also a
good example to showcase the flexibility of the visualisation function:
Either you use the parameter
<code>Plot_SettingsFile= MetaData_Metab</code> as above, but as we have
the column “Pathway” also in our Input_data you can also pass
<code>Plot_SettingsFile= DMA_786-M1A_vs_HK2</code> or simply use the
default <code>Plot_SettingsFile=NULL</code>, in which case the
<code>Plot_SettingsInfo</code> information (here <code>color</code>)
will be used from Input_data.<br></p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Now we can use color for the pathways and shape for the metabolite clusters:</span></span>
<span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/VizVolcano.html">VizVolcano</a></span><span class="op">(</span>Plot_Settings<span class="op">=</span><span class="st">"Standard"</span>,</span>
<span>                       Plot_SettingsInfo<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"Pathway"</span><span class="op">)</span>,</span>
<span>                       Input_data<span class="op">=</span><span class="va">DMA_786M1A_vs_HK2</span>,</span>
<span>                       OutputPlotName<span class="op">=</span> <span class="st">"786M1A versus HK2 Results of DMA. Colour for metabolic pathways."</span>,</span>
<span>                       Subtitle<span class="op">=</span> <span class="st">"Results of DMA. Colour for metabolic pathways."</span> <span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-51-1.png" alt="Figure: Standard figure displaying DMA results colour coded for metabolic pathways and shaped for metabolic clusters from MCA results." width="672"><p class="caption">
Figure: Standard figure displaying DMA results colour coded for
metabolic pathways and shaped for metabolic clusters from MCA results.
</p>
</div>
<p><br>
We immediately see that there are many pathways displayed on the plot,
which can make it difficult to interpret. Hence, we will change our plot
settings in order to get individual plots for each of the pathways:</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Now we can generate a plot for each pathway and color for the metabolite clusters:</span></span>
<span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/VizVolcano.html">VizVolcano</a></span><span class="op">(</span>Plot_Settings<span class="op">=</span><span class="st">"Standard"</span>,</span>
<span>                       Plot_SettingsInfo<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"RG2_Significant"</span>, individual<span class="op">=</span><span class="st">"Pathway"</span><span class="op">)</span>,</span>
<span>                       Plot_SettingsFile<span class="op">=</span> <span class="va">MetaData_Metab</span>,</span>
<span>                       Input_data<span class="op">=</span><span class="va">DMA_786M1A_vs_HK2</span>,</span>
<span>                       OutputPlotName<span class="op">=</span> <span class="st">"786M1A versus HK2"</span>,</span>
<span>                       Subtitle<span class="op">=</span> <span class="st">"Results of DMA. Colour for metabolic pathways."</span> <span class="op">)</span></span></code></pre></div>
<div class="vscroll-plot">
<p><img src="Standard%20Metabolomics_files/figure-html/Run_not_Display-1.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/Run_not_Display-2.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/Run_not_Display-3.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/Run_not_Display-4.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/Run_not_Display-5.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/Run_not_Display-6.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/Run_not_Display-7.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/Run_not_Display-8.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/Run_not_Display-9.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/Run_not_Display-10.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/Run_not_Display-11.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/Run_not_Display-12.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/Run_not_Display-13.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/Run_not_Display-14.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/Run_not_Display-15.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/Run_not_Display-16.png" width="672" style="display: block; margin: auto auto auto 0;"></p>
</div>
<p><br><br><br></p>
</div>
<div class="section level5">
<h5 id="comparison">
<strong>Comparison</strong><a class="anchor" aria-label="anchor" href="#comparison"></a>
</h5>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/VizVolcano.html">VizVolcano</a></span><span class="op">(</span>Plot_Settings<span class="op">=</span><span class="st">"Compare"</span>,</span>
<span>                       Input_data<span class="op">=</span><span class="va">DMA_786M1A_vs_HK2</span>,</span>
<span>                       AdditionalInput_data<span class="op">=</span><span class="va">DMA_Annova</span><span class="op">[[</span><span class="st">"DF"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"DMA_result"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"786-O_vs_HK2"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                       OutputPlotName<span class="op">=</span> <span class="st">"786M1A vs HK2 compared to 7860 vs HK2"</span>,</span>
<span>                       Subtitle<span class="op">=</span> <span class="st">"Results of DMA"</span> <span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-52-1.png" alt="Figure: Comparison." width="672"><p class="caption">
Figure: Comparison.
</p>
</div>
<p>Now we do individual plots again:</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/VizVolcano.html">VizVolcano</a></span><span class="op">(</span>Plot_Settings<span class="op">=</span><span class="st">"Compare"</span>,</span>
<span>                       Plot_SettingsInfo<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>individual<span class="op">=</span><span class="st">"Pathway"</span><span class="op">)</span>,</span>
<span>                       Plot_SettingsFile<span class="op">=</span> <span class="va">MetaData_Metab</span>,</span>
<span>                       Input_data<span class="op">=</span><span class="va">DMA_786M1A_vs_HK2</span>,</span>
<span>                       AdditionalInput_data<span class="op">=</span><span class="va">DMA_Annova</span><span class="op">[[</span><span class="st">"DF"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"DMA_result"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"786-O_vs_HK2"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                       OutputPlotName<span class="op">=</span> <span class="st">"786M1A vs HK2 compared to 7860 vs HK2"</span>,</span>
<span>                       Subtitle<span class="op">=</span> <span class="st">"Results of DMA"</span> <span class="op">)</span></span></code></pre></div>
<div class="vscroll-plot">
<p><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-54-1.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-54-2.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-54-3.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-54-4.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-54-5.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-54-6.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-54-7.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-54-8.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-54-9.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-54-10.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-54-11.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-54-12.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-54-13.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-54-14.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-54-15.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-54-16.png" width="672" style="display: block; margin: auto auto auto 0;"></p>
</div>
<p><br><br></p>
</div>
<div class="section level5">
<h5 id="pathwayenrichmentanalysis">
<strong>PathwayEnrichmentAnalysis</strong><a class="anchor" aria-label="anchor" href="#pathwayenrichmentanalysis"></a>
</h5>
<p>If you have performed Pathway Enrichment Analysis (PEA) such as ORA
or GSEA, we can also plot the results or add the information into the
Figure legends.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/VizVolcano.html">VizVolcano</a></span><span class="op">(</span>Plot_Settings<span class="op">=</span><span class="st">"PEA"</span>,</span>
<span>                       Plot_SettingsInfo<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>individual<span class="op">=</span><span class="st">"term"</span>, PEA_Pathway<span class="op">=</span><span class="st">"ID"</span>, PEA_stat<span class="op">=</span><span class="st">"p.adjust"</span>, PEA_score<span class="op">=</span><span class="st">"GeneRatio"</span><span class="op">)</span>,</span>
<span>                       Plot_SettingsFile<span class="op">=</span> <span class="va">KEGG_Pathways</span><span class="op">[</span>,<span class="op">-</span><span class="fl">3</span><span class="op">]</span>,<span class="co">#Must be the pathways used for pathway analysis</span></span>
<span>                       Input_data<span class="op">=</span> <span class="va">DMA_786M1A_vs_HK2</span>,<span class="co"># Must be the data you have used as an input for the pathway analysis</span></span>
<span>                       AdditionalInput_data<span class="op">=</span><span class="va">DM_ORA_res</span><span class="op">[[</span><span class="st">"786-M1A_vs_HK2"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"ClusterGoSummary"</span><span class="op">]</span><span class="op">]</span>,<span class="co">#Must be the results of the pathway analysis</span></span>
<span>                       OutputPlotName<span class="op">=</span> <span class="st">"KEGG"</span>,</span>
<span>                       Subtitle<span class="op">=</span> <span class="st">"PEA"</span> ,</span>
<span>                       SelectLab <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<div class="vscroll-plot">
<p><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-1.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-2.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-3.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-4.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-5.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-6.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-7.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-8.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-9.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-10.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-11.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-12.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-13.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-14.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-15.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-16.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-17.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-18.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-19.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-20.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-21.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-22.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-23.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-24.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-25.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-26.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-27.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-28.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-29.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-30.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-31.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-32.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-33.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-34.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-35.png" width="672" style="display: block; margin: auto auto auto 0;"><img src="Standard%20Metabolomics_files/figure-html/unnamed-chunk-56-36.png" width="672" style="display: block; margin: auto auto auto 0;"></p>
</div>
<br><br><div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<pre><code><span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">#&gt;  version  R version 4.2.2 (2022-10-31 ucrt)</span></span>
<span><span class="co">#&gt;  os       Windows 10 x64 (build 19045)</span></span>
<span><span class="co">#&gt;  system   x86_64, mingw32</span></span>
<span><span class="co">#&gt;  ui       RTerm</span></span>
<span><span class="co">#&gt;  language en</span></span>
<span><span class="co">#&gt;  collate  English_United Kingdom.utf8</span></span>
<span><span class="co">#&gt;  ctype    English_United Kingdom.utf8</span></span>
<span><span class="co">#&gt;  tz       Europe/Berlin</span></span>
<span><span class="co">#&gt;  date     2023-11-07</span></span>
<span><span class="co">#&gt;  pandoc   2.19.2 @ C:/Program Files/RStudio/resources/app/bin/quarto/bin/tools/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">!</span> <span style="color: #555555; font-style: italic;">package         </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version   </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">#&gt;    abind              1.4-5      <span style="color: #555555;">2016-07-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;    AnnotationDbi      1.60.2     <span style="color: #555555;">2023-03-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    ape                5.7-1      <span style="color: #555555;">2023-03-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    aplot              0.1.10     <span style="color: #555555;">2023-03-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    backports          1.4.1      <span style="color: #555555;">2021-12-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;    Biobase            2.58.0     <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    BiocGenerics       0.44.0     <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    BiocParallel       1.32.5     <span style="color: #555555;">2022-12-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    Biostrings         2.66.0     <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    bit                4.0.5      <span style="color: #555555;">2022-11-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    bit64              4.0.5      <span style="color: #555555;">2020-08-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    bitops             1.0-7      <span style="color: #555555;">2021-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;    blob               1.2.4      <span style="color: #555555;">2023-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    broom              1.0.5      <span style="color: #555555;">2023-06-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    bslib              0.5.0      <span style="color: #555555;">2023-06-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    cachem             1.0.6      <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    callr              3.7.3      <span style="color: #555555;">2022-11-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    car                3.1-2      <span style="color: #555555;">2023-03-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    carData            3.0-5      <span style="color: #555555;">2022-01-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    cli                3.6.0      <span style="color: #555555;">2023-01-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    clusterProfiler    4.6.2      <span style="color: #555555;">2023-03-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    codetools          0.2-18     <span style="color: #555555;">2020-11-04</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    colorspace         2.1-0      <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    cowplot            1.1.1      <span style="color: #555555;">2020-12-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    crayon             1.5.2      <span style="color: #555555;">2022-09-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    data.table         1.14.8     <span style="color: #555555;">2023-02-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    DBI                1.1.3      <span style="color: #555555;">2022-06-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    desc               1.4.2      <span style="color: #555555;">2022-09-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    devtools           2.4.5      <span style="color: #555555;">2022-10-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    digest             0.6.31     <span style="color: #555555;">2022-12-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    DOSE               3.24.2     <span style="color: #555555;">2022-11-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    downloader         0.4        <span style="color: #555555;">2015-07-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    dplyr            * 1.1.2      <span style="color: #555555;">2023-04-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    ellipsis           0.3.2      <span style="color: #555555;">2021-04-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    EnhancedVolcano    1.16.0     <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    enrichplot         1.18.4     <span style="color: #555555;">2023-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    evaluate           0.21       <span style="color: #555555;">2023-05-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    factoextra         1.0.7      <span style="color: #555555;">2020-04-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    fansi              1.0.4      <span style="color: #555555;">2023-01-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    farver             2.1.1      <span style="color: #555555;">2022-07-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    fastmap            1.1.0      <span style="color: #555555;">2021-01-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    fastmatch          1.1-3      <span style="color: #555555;">2021-07-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;    fgsea              1.24.0     <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    forcats          * 1.0.0      <span style="color: #555555;">2023-01-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    fs                 1.6.0      <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    generics           0.1.3      <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    GenomeInfoDb       1.34.9     <span style="color: #555555;">2023-02-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    GenomeInfoDbData   1.2.9      <span style="color: #555555;">2023-02-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    ggforce            0.4.1      <span style="color: #555555;">2022-10-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    ggfortify        * 0.4.16     <span style="color: #555555;">2023-03-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    ggfun              0.1.1      <span style="color: #555555;">2023-06-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    ggplot2          * 3.4.2      <span style="color: #555555;">2023-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    ggplotify          0.1.1      <span style="color: #555555;">2023-06-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    ggpubr             0.6.0      <span style="color: #555555;">2023-02-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    ggraph             2.1.0      <span style="color: #555555;">2022-10-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    ggrepel            0.9.3      <span style="color: #555555;">2023-02-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    ggsignif           0.6.4      <span style="color: #555555;">2022-10-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    ggtree             3.6.2      <span style="color: #555555;">2022-11-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    ggupset            0.3.0      <span style="color: #555555;">2020-05-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    glue               1.6.2      <span style="color: #555555;">2022-02-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    GO.db              3.16.0     <span style="color: #555555;">2023-04-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    GOSemSim           2.24.0     <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    graphlayouts       0.8.4      <span style="color: #555555;">2022-11-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    gridExtra          2.3        <span style="color: #555555;">2017-09-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    gridGraphics       0.5-1      <span style="color: #555555;">2020-12-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    gson               0.1.0      <span style="color: #555555;">2023-03-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    gtable             0.3.3      <span style="color: #555555;">2023-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    gtools             3.9.4      <span style="color: #555555;">2022-11-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    hash               2.2.6.2    <span style="color: #555555;">2022-03-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    HDO.db             0.99.1     <span style="color: #555555;">2023-04-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    highr              0.10       <span style="color: #555555;">2022-12-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    hms                1.1.3      <span style="color: #555555;">2023-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    htmltools          0.5.4      <span style="color: #555555;">2022-12-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    htmlwidgets        1.6.2      <span style="color: #555555;">2023-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    httpuv             1.6.8      <span style="color: #555555;">2023-01-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    httr               1.4.6      <span style="color: #555555;">2023-05-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    igraph             1.3.5      <span style="color: #555555;">2022-09-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    inflection         1.3.6      <span style="color: #555555;">2022-06-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    IRanges            2.32.0     <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    jquerylib          0.1.4      <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    jsonlite           1.8.4      <span style="color: #555555;">2022-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    kableExtra         1.3.4      <span style="color: #555555;">2021-02-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    KEGGREST           1.41.0     <span style="color: #555555;">2023-07-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">Github (Bioconductor/KEGGREST@b810cd1)</span></span></span>
<span><span class="co">#&gt;    knitr              1.43       <span style="color: #555555;">2023-05-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    labeling           0.4.2      <span style="color: #555555;">2020-10-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;    later              1.3.0      <span style="color: #555555;">2021-08-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    lattice            0.20-45    <span style="color: #555555;">2021-09-22</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    lazyeval           0.2.2      <span style="color: #555555;">2019-03-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    lifecycle          1.0.3      <span style="color: #555555;">2022-10-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    lubridate        * 1.9.2      <span style="color: #555555;">2023-02-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    magrittr           2.0.3      <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    MASS               7.3-58.1   <span style="color: #555555;">2022-08-03</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    Matrix             1.6-0      <span style="color: #555555;">2023-07-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    memoise            2.0.1      <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> MetaProViz       * <span style="color: #BB00BB; font-weight: bold;">0.0.0.9000</span> <span style="color: #555555;">2023-02-16</span> <span style="color: #555555;">[?]</span> <span style="color: #BB00BB; font-weight: bold;">load_all()</span></span></span>
<span><span class="co">#&gt;    mime               0.12       <span style="color: #555555;">2021-09-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;    miniUI             0.1.1.1    <span style="color: #555555;">2018-05-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    munsell            0.5.0      <span style="color: #555555;">2018-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    nlme               3.1-160    <span style="color: #555555;">2022-10-10</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    patchwork          1.1.2      <span style="color: #555555;">2022-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    pheatmap           1.0.12     <span style="color: #555555;">2019-01-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    pillar             1.9.0      <span style="color: #555555;">2023-03-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    pkgbuild           1.4.2      <span style="color: #555555;">2023-06-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    pkgconfig          2.0.3      <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    pkgdown            2.0.7      <span style="color: #555555;">2022-12-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    pkgload            1.3.2.1    <span style="color: #555555;">2023-07-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    plyr               1.8.8      <span style="color: #555555;">2022-11-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    png                0.1-8      <span style="color: #555555;">2022-11-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    polyclip           1.10-4     <span style="color: #555555;">2022-10-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.1)</span></span></span>
<span><span class="co">#&gt;    prettyunits        1.1.1      <span style="color: #555555;">2020-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    processx           3.8.0      <span style="color: #555555;">2022-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    profvis            0.3.7      <span style="color: #555555;">2020-11-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    promises           1.2.0.1    <span style="color: #555555;">2021-02-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    ps                 1.7.2      <span style="color: #555555;">2022-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    purrr            * 1.0.1      <span style="color: #555555;">2023-01-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    qcc                2.7        <span style="color: #555555;">2017-07-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    qvalue             2.30.0     <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    R6                 2.5.1      <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    ragg               1.2.5      <span style="color: #555555;">2023-01-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    rappdirs           0.3.3      <span style="color: #555555;">2021-01-31</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    RColorBrewer       1.1-3      <span style="color: #555555;">2022-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;    Rcpp               1.0.10     <span style="color: #555555;">2023-01-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    RCurl              1.98-1.10  <span style="color: #555555;">2023-01-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    readr            * 2.1.4      <span style="color: #555555;">2023-02-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    remotes            2.4.2      <span style="color: #555555;">2021-11-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    reshape            0.8.9      <span style="color: #555555;">2022-04-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    reshape2           1.4.4      <span style="color: #555555;">2020-04-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    rlang              1.1.0      <span style="color: #555555;">2023-03-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    rmarkdown          2.23       <span style="color: #555555;">2023-07-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    rprojroot          2.0.3      <span style="color: #555555;">2022-04-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    RSQLite            2.2.20     <span style="color: #555555;">2022-12-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    rstatix            0.7.2      <span style="color: #555555;">2023-02-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    rstudioapi         0.15.0     <span style="color: #555555;">2023-07-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    rvest              1.0.3      <span style="color: #555555;">2022-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    S4Vectors          0.36.1     <span style="color: #555555;">2022-12-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    sass               0.4.6      <span style="color: #555555;">2023-05-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    scales             1.2.1      <span style="color: #555555;">2022-08-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    scatterpie         0.2.1      <span style="color: #555555;">2023-06-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    sessioninfo        1.2.2      <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    shadowtext         0.1.2      <span style="color: #555555;">2022-04-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    shiny              1.7.4.1    <span style="color: #555555;">2023-07-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    stringi            1.7.12     <span style="color: #555555;">2023-01-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    stringr          * 1.5.0      <span style="color: #555555;">2022-12-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    svglite            2.1.1      <span style="color: #555555;">2023-01-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    systemfonts        1.0.4      <span style="color: #555555;">2022-02-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    textshaping        0.3.6      <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    tibble           * 3.2.1      <span style="color: #555555;">2023-03-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    tidygraph          1.2.3      <span style="color: #555555;">2023-02-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    tidyr            * 1.3.0      <span style="color: #555555;">2023-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    tidyselect         1.2.0      <span style="color: #555555;">2022-10-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    tidytree           0.4.2      <span style="color: #555555;">2022-12-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    tidyverse        * 2.0.0      <span style="color: #555555;">2023-02-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    timechange         0.2.0      <span style="color: #555555;">2023-01-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    treeio             1.22.0     <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    tweenr             2.0.2      <span style="color: #555555;">2022-09-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    tzdb               0.4.0      <span style="color: #555555;">2023-05-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    urlchecker         1.0.1      <span style="color: #555555;">2021-11-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    usethis            2.2.2      <span style="color: #555555;">2023-07-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    utf8               1.2.2      <span style="color: #555555;">2021-07-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    vctrs              0.6.3      <span style="color: #555555;">2023-06-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    viridis            0.6.3      <span style="color: #555555;">2023-05-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    viridisLite        0.4.2      <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    vroom              1.6.3      <span style="color: #555555;">2023-04-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    webshot            0.5.5      <span style="color: #555555;">2023-06-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    withr              2.5.0      <span style="color: #555555;">2022-03-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    writexl            1.4.2      <span style="color: #555555;">2023-01-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    xfun               0.39       <span style="color: #555555;">2023-04-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    xml2               1.3.5      <span style="color: #555555;">2023-07-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.3)</span></span></span>
<span><span class="co">#&gt;    xtable             1.8-4      <span style="color: #555555;">2019-04-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    XVector            0.38.0     <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    yaml               2.3.7      <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    yulab.utils        0.0.6      <span style="color: #555555;">2022-12-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    zlibbioc           1.44.0     <span style="color: #555555;">2022-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [1] C:/Users/chris/AppData/Local/R/win-library/4.2</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [2] C:/Program Files/R/R-4.2.2/library</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> ── Loaded and on-disk path mismatch.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Badia-I-Mompel2022" class="csl-entry">
Badia-I-Mompel, Pau, Jesús Vélez Santiago, Jana Braunger, Celina Geiss,
Daniel Dimitrov, Sophia Müller-Dott, Petr Taus, et al. 2022.
<span>“decoupleR: Ensemble of Computational Methods to Infer Biological
Activities from Omics Data.”</span> <em>Bioinformatics Advances</em> 2
(1): vbac016. <a href="https://doi.org/10.1093/bioadv/vbac016" class="external-link">https://doi.org/10.1093/bioadv/vbac016</a>.
</div>
<div id="ref-Bijlsma2006" class="csl-entry">
Bijlsma, Sabina, Ivana Bobeldijk, Elwin R Verheij, Raymond Ramaker,
Sunil Kochhar, Ian A Macdonald, Ben van Ommen, and Age K Smilde. 2006.
<span>“Large-Scale Human Metabolomics Studies: A Strategy for Data
(Pre-) Processing and Validation.”</span> <em>Analytical Chemistry</em>
78 (2): 567–74. <a href="https://doi.org/10.1021/ac051495j" class="external-link">https://doi.org/10.1021/ac051495j</a>.
</div>
<div id="ref-Hotelling1931" class="csl-entry">
Hotelling, Harold. 1931. <span>“The Generalization of Student’s
Ratio.”</span> <em>The Annals of Mathematical Statistics</em> 2 (3):
360–78. <a href="https://doi.org/10.1214/aoms/1177732979" class="external-link">https://doi.org/10.1214/aoms/1177732979</a>.
</div>
<div id="ref-Kanehisa2000" class="csl-entry">
Kanehisa, M, and S Goto. 2000. <span>“KEGG: Kyoto Encyclopedia of Genes
and Genomes.”</span> <em>Nucleic Acids Research</em> 28 (1): 27–30. <a href="https://doi.org/10.1093/nar/28.1.27" class="external-link">https://doi.org/10.1093/nar/28.1.27</a>.
</div>
<div id="ref-Mora_Schmidt2022" class="csl-entry">
Mora, Ariane N, Christina Schmidt, Brad Balderson, Christian Frezza, and
Mikael Boden. 2022. <span>“SiRCle (Signature Regulatory Clustering)
Model Integration Reveals Mechanisms of Phenotype Regulation in Renal
Cancer.”</span> <em>BioRxiv</em>, July. <a href="https://doi.org/10.1101/2022.07.02.498058" class="external-link">https://doi.org/10.1101/2022.07.02.498058</a>.
</div>
<div id="ref-Sciacovelli_Dugourd2022" class="csl-entry">
Sciacovelli, Marco, Aurelien Dugourd, Lorea Valcarcel Jimenez, Ming
Yang, Efterpi Nikitopoulou, Ana S H Costa, Laura Tronci, et al. 2022.
<span>“Dynamic Partitioning of Branched-Chain Amino Acids-Derived
Nitrogen Supports Renal Cancer Progression.”</span> <em>Nature
Communications</em> 13 (1): 7830. <a href="https://doi.org/10.1038/s41467-022-35036-4" class="external-link">https://doi.org/10.1038/s41467-022-35036-4</a>.
</div>
<div id="ref-Wei2018" class="csl-entry">
Wei, Runmin, Jingye Wang, Mingming Su, Erik Jia, Shaoqiu Chen, Tianlu
Chen, and Yan Ni. 2018. <span>“Missing Value Imputation Approach for
Mass Spectrometry-Based Metabolomics Data.”</span> <em>Scientific
Reports</em> 8 (1): 663. <a href="https://doi.org/10.1038/s41598-017-19120-0" class="external-link">https://doi.org/10.1038/s41598-017-19120-0</a>.
</div>
<div id="ref-Wulff2018" class="csl-entry">
Wulff, Jacob E., and Matthew W. Mitchell. 2018. <span>“A Comparison of
Various Normalization Methods for LC/MS Metabolomics Data.”</span>
<em>Advances in Bioscience and Biotechnology</em> 09 (08): 339–51. <a href="https://doi.org/10.4236/abb.2018.98022" class="external-link">https://doi.org/10.4236/abb.2018.98022</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Christina Schmidt, Dimitrios Prymidis, Julio Saez-Rodriguez, Christian Frezza.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
