<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Enrichment Analysis • MetaProViz</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Enrichment Analysis">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MetaProViz</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.1.2</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-tutorials" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true"><span class="fa fa-book-open"></span> Tutorials</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-tutorials">
<li><a class="dropdown-item" href="../articles/Standard%20Metabolomics.html">Standard Metabolomics</a></li>
    <li><a class="dropdown-item" href="../articles/CoRe%20Metabolomics.html">CoRe Metabolomics</a></li>
    <li><a class="dropdown-item" href="../articles/EnrichmentAnalysis.html">Enrichment Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/Metadata%20Analysis.html">Metadata Analysis</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html"><span class="fa fa-solid fa-rocket"></span> Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/saezlab/MetaProViz"><span class="fa fa-brands fa-github"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://saezlab.org/"><span class="fa fas fa-university"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://frezza.cecad-labs.uni-koeln.de/home"><span class="fa fas fa-university"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">



<script src="EnrichmentAnalysis_files/kePrint-0.0.1/kePrint.js"></script><link href="EnrichmentAnalysis_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Enrichment Analysis</h1>
                        <h4 data-toc-skip class="author">Christina
Schmidt</h4>
            <address class="author_afil">
      Heidelberg
University<br><h4 data-toc-skip class="author">Macabe
Daley</h4>
            <address class="author_afil">
      Heidelberg University, DKFZ<br><small class="dont-index">Source: <a href="https://github.com/saezlab/MetaProViz/blob/main/vignettes/EnrichmentAnalysis.Rmd" class="external-link"><code>vignettes/EnrichmentAnalysis.Rmd</code></a></small>
      <div class="d-none name"><code>EnrichmentAnalysis.Rmd</code></div>
    </address>
</address>
</div>

    
    
<style>
.vscroll-plot {
    width: 850px;
    height: 500px;
    overflow-y: scroll;
    overflow-x: hidden;
}
</style>
<div class="section level2">
<h2 id="section">
<img src="Hexagon_MetaProViz.png" align="right" width="200"><a class="anchor" aria-label="anchor" href="#section"></a>
</h2>
<p><br><span style="text-decoration:underline">In this tutorial we showcase how
to use <strong>MetaProViz</strong></span>:<br>
- To access metabolite prior knowledge.<br>
- To deal with many-to-many mapping in your metabolite
identifiers.<br>
- To perform pathway enrichment analysis.<br></p>
<p><br>
First if you have not done yet, install the required dependencies and
load the libraries:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Install Rtools if you haven’t done this yet, using the appropriate version (e.g.windows or macOS). </span></span>
<span><span class="co"># 2. Install the latest development version from GitHub using devtools</span></span>
<span><span class="co">#devtools::install_github("https://github.com/saezlab/MetaProViz")</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://saezlab.github.io/MetaProViz" class="external-link">MetaProViz</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">#dependencies that need to be loaded:</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Please install the Biocmanager Dependencies:</span></span>
<span><span class="co">#BiocManager::install("clusterProfiler")</span></span>
<span><span class="co">#BiocManager::install("EnhancedVolcano")</span></span>
<span><span class="co">#BiocManager::install("cosmosR")</span></span></code></pre></div>
<p><br><br></p>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
</div>
<div class="section level2">
<h2 id="loading-the-example-data">1. Loading the example data<a class="anchor" aria-label="anchor" href="#loading-the-example-data"></a>
</h2>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
<p><br><span style="text-decoration:underline">As part of the
<strong>MetaProViz</strong> package you can load the example data into
your global environment using the function
<code>toy_data()</code></span>:<br><code>1.</code>Cell line experiment <strong>(CellLine)</strong><br>
Here we choose an example datasets, which is publicly available on <a href="https://www.metabolomicsworkbench.org/data/DRCCMetadata.php?Mode=Project&amp;ProjectID=PR001418" class="external-link">metabolomics
workbench project PR001418</a> including metabolic profiles of human
renal epithelial cells HK2 and cell renal cell carcinoma (ccRCC) cell
lines cultured in Plasmax cell culture media <span class="citation">(Sciacovelli et al. 2022)</span>. The raw data are
available via <a href="https://www.metabolomicsworkbench.org/data/DRCCMetadata.php?Mode=Study&amp;StudyID=ST002224&amp;StudyType=MS&amp;ResultType=1" class="external-link">metabolomics
workbench study ST002224</a> were intracellular metabolomics of HK2 and
ccRCC cell lines 786-O, 786-M1A and 786-M2A were performed.<br>
We have performed pre-processing and differential analysis (details can
be found in the vignette vignette <a href="https://saezlab.github.io/MetaProViz/articles/Standard%20Metabolomics.html" class="external-link">Standard
Metabolomics</a>) and and here we load the differential metabolite
analysis results for the comparison of 786M-1A versus HK2.<br></p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Load the Pre-processed intracellular data:</span></span>
<span><span class="va">Intra_DMA_786M1A_vs_HK2</span><span class="op">&lt;-</span> <span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/ToyData.html">ToyData</a></span><span class="op">(</span>Data<span class="op">=</span><span class="st">"IntraCells_DMA"</span><span class="op">)</span></span></code></pre></div>
<p><code>2.</code> Tissue experiment <strong>(Tissue)</strong><br>
Here we chose publicly available data from the <a href="https://www.cell.com/cancer-cell/comments/S1535-6108(15)00468-7#supplementaryMaterial" class="external-link">paper</a>
“An Integrated Metabolic Atlas of Clear Cell Renal Cell Carcinoma”,
which includes metabolomic profiling on 138 matched clear cell renal
cell carcinoma (ccRCC)/normal tissue pairs.<br>
We have performed differential analysis (details can be found in the
vignette <a href="https://saezlab.github.io/MetaProViz/articles/Metadata%20Analysis.html" class="external-link">Metadata
Analysis</a>) and and here we load the differential metabolite analysis
results for the comparison of Tumour versus Normal.<br></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the example data:</span></span>
<span><span class="va">Tissue_TvsN</span> <span class="op">&lt;-</span> <span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/ToyData.html">ToyData</a></span><span class="op">(</span>Data<span class="op">=</span><span class="st">"Tissue_DMA"</span><span class="op">)</span></span>
<span><span class="va">Tissue_TvsN_Old</span> <span class="op">&lt;-</span> <span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/ToyData.html">ToyData</a></span><span class="op">(</span>Data<span class="op">=</span><span class="st">"Tissue_DMA_Old"</span><span class="op">)</span></span>
<span><span class="va">Tissue_TvsN_Young</span> <span class="op">&lt;-</span> <span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/ToyData.html">ToyData</a></span><span class="op">(</span>Data<span class="op">=</span><span class="st">"Tissue_DMA_Young"</span><span class="op">)</span></span></code></pre></div>
<p><br><br></p>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
</div>
<div class="section level2">
<h2 id="accessing-prior-knowledge">2. Accessing Prior Knowledge<a class="anchor" aria-label="anchor" href="#accessing-prior-knowledge"></a>
</h2>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
<p><br>
Metabolite prior knowledge (PK) is essential for the interpretation of
metabolomics data. It can be used to perform pathway enrichment
analysis, compound class enrichment analysis, and by using specific PK
databases, it can be used to study the connection of metabolites and
receptors or transporters.Since the quality and content of the PK will
dictate the success of the downstream analysis and biological
interpretation, it is important to ensure the PK is used
correctly.<br>
Specifically in metabolite PK, the many different PK databases and
resources pose several issues. Indeed, the metabolite identifiers
(e.g. KEGG, HMDB, PubChem, etc.) are not standardized across databases,
and the same metabolite can have multiple identifiers in different
databases. This is known as the many-to-many mapping problem.<br>
Moreover, metabolic pathways that are the basis of the PK databases also
includes co-factors such as ions or other small molecules that are not
only part of most reactions, but can also not be detected in
experimentallly acquired data (e.g. H2O, CO2, etc).</p>
<div class="section level3">
<h3 id="kegg-pathway-metabolite-sets">KEGG pathway-metabolite sets<a class="anchor" aria-label="anchor" href="#kegg-pathway-metabolite-sets"></a>
</h3>
<p>KEGG pathways that are loaded via KEGG API using the package
<code>KEGGREST</code> and can be used to perform pathway analysis <span class="citation">(Kanehisa and Goto 2000)</span>.
<strong>(KEGG_Pathways)</strong><br></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#This will use KEGGREST to query the KEGG API to load the pathways:</span></span>
<span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/LoadKEGG.html">LoadKEGG</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Cached file loaded from: ~/.cache/KEGG_Metabolite.rds</span></span></code></pre></div>
<p><br></p>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>KEGG_Pathways</code>.
</caption>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
term
</th>
<th style="text-align:left;">
Metabolite
</th>
<th style="text-align:left;">
MetaboliteID
</th>
<th style="text-align:left;">
Description
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis - Homo sapiens (human)
</td>
<td style="text-align:left;">
Pyruvate
</td>
<td style="text-align:left;">
C00022
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis - Homo sapiens (human)
</td>
</tr>
<tr>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis - Homo sapiens (human)
</td>
<td style="text-align:left;">
Acetyl-CoA
</td>
<td style="text-align:left;">
C00024
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis - Homo sapiens (human)
</td>
</tr>
<tr>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis - Homo sapiens (human)
</td>
<td style="text-align:left;">
D-Glucose
</td>
<td style="text-align:left;">
C00031
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis - Homo sapiens (human)
</td>
</tr>
<tr>
<td style="text-align:left;">
52
</td>
<td style="text-align:left;">
Pentose phosphate pathway - Homo sapiens (human)
</td>
<td style="text-align:left;">
Pyruvate
</td>
<td style="text-align:left;">
C00022
</td>
<td style="text-align:left;">
Pentose phosphate pathway - Homo sapiens (human)
</td>
</tr>
<tr>
<td style="text-align:left;">
53
</td>
<td style="text-align:left;">
Pentose phosphate pathway - Homo sapiens (human)
</td>
<td style="text-align:left;">
D-Glucose
</td>
<td style="text-align:left;">
C00031
</td>
<td style="text-align:left;">
Pentose phosphate pathway - Homo sapiens (human)
</td>
</tr>
<tr>
<td style="text-align:left;">
54
</td>
<td style="text-align:left;">
Pentose phosphate pathway - Homo sapiens (human)
</td>
<td style="text-align:left;">
D-Fructose 6-phosphate
</td>
<td style="text-align:left;">
C00085
</td>
<td style="text-align:left;">
Pentose phosphate pathway - Homo sapiens (human)
</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="create-pathway-metabolite-sets">Create pathway-metabolite sets<a class="anchor" aria-label="anchor" href="#create-pathway-metabolite-sets"></a>
</h3>
<p>The function <code>Make_GeneMetabSet</code> can be used to translate
gene names to metabolite names by using a PK network of metabolic
reactions calls CosmosR <span class="citation">(Dugourd et al.
2021)</span>. This function is useful if you want to perform pathway
enrichment analysis on available gene-sets such as the Hallmarks
gene-sets from MSigDB <span class="citation">(Castanza et al.
2022)</span>. Moreover, it enables you to perform combined pathway
enrichment analysis on metabolite-gene sets, if you have other data
types such as proteomics measuring the enzymes expression.<br>
The Hallmarks gene-set is available in the package
<code>MetaProViz</code> and can be loaded using the function
<code><a href="../reference/LoadHallmarks.html">LoadHallmarks()</a></code>.<br></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Load the example data:</span></span>
<span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/LoadHallmarks.html">LoadHallmarks</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>Hallmarks_Pathways</code> including gene-sets
usable for pathway enrichment analysis.
</caption>
<thead><tr>
<th style="text-align:left;">
term
</th>
<th style="text-align:left;">
gene
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
HALLMARK_BILE_ACID_METABOLISM
</td>
<td style="text-align:left;">
GSTK1
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_BILE_ACID_METABOLISM
</td>
<td style="text-align:left;">
ABCG4
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
LDHC
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
ARPP19
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
LDHC
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
ARPP19
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
CENPA
</td>
</tr>
</tbody>
</table>
<p><br>
Now we can use the function <code>Make_GeneMetabSet</code> to translate
the gene names to metabolite names.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Translate gene names to metabolite names</span></span>
<span><span class="va">Hallmarks_GeneMetab</span> <span class="op">&lt;-</span> <span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/Make_GeneMetabSet.html">Make_GeneMetabSet</a></span><span class="op">(</span>Input_GeneSet<span class="op">=</span><span class="va">Hallmark_Pathways</span>,</span>
<span>                                                     SettingsInfo<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>Target<span class="op">=</span><span class="st">"gene"</span><span class="op">)</span>,</span>
<span>                                                     PKName<span class="op">=</span><span class="st">"Hallmarks"</span><span class="op">)</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>Hallmarks_GeneMetab</code> including gene-sets
usable for pathway enrichment analysis.
</caption>
<thead><tr>
<th style="text-align:left;">
term
</th>
<th style="text-align:left;">
feature
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
ME2
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
LDHC
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
FKBP4
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
HMDB0000241
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
HMDB0000570
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
HMDB0000122
</td>
</tr>
</tbody>
</table>
<p><br>
Given that we have the gene-metabolite-sets, we can now also run
enrichment analysis on combined data types, once including the
metabolite Log2FC and one including gene Log2FC from
e.g. transcriptomics or proteomics data. Yet, it is important to keep in
mind that generally we detect less metabolites than genes and hence this
may bias the results obtained from combined enrichment analysis.</p>
</div>
<div class="section level3">
<h3 id="metalinksdb-metabolite-receptor-sets">MetaLinksDB Metabolite-receptor sets<a class="anchor" aria-label="anchor" href="#metalinksdb-metabolite-receptor-sets"></a>
</h3>
<p>The MetaLinks database is a manually curated database of
metabolite-receptor and metabolite-transporter sets that can be used to
study the connection of metabolites and receptors or transporters <span class="citation">(Farr et al. 2024)</span>.<br></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">MetaLinksDB_Res</span> <span class="op">&lt;-</span> <span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/LoadMetalinks.html">LoadMetalinks</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Metalinks database downloaded and saved to: ~/.cache/metalinks.db</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>MetaLinksDB</code> including metabolite-receptor
sets.
</caption>
<thead><tr>
<th style="text-align:left;">
hmdb
</th>
<th style="text-align:left;">
metabolite
</th>
<th style="text-align:left;">
pubchem
</th>
<th style="text-align:left;">
metabolite_subclass
</th>
<th style="text-align:left;">
uniprot
</th>
<th style="text-align:left;">
gene_symbol
</th>
<th style="text-align:left;">
protein_type
</th>
<th style="text-align:left;">
source
</th>
<th style="text-align:right;">
db_score
</th>
<th style="text-align:right;">
experiment_score
</th>
<th style="text-align:right;">
combined_score
</th>
<th style="text-align:right;">
mor
</th>
<th style="text-align:left;">
type
</th>
<th style="text-align:left;">
transport_direction
</th>
<th style="text-align:left;">
mode_of_regulation
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
HMDB0002111
</td>
<td style="text-align:left;">
Water
</td>
<td style="text-align:left;">
962
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
A0A087X1C5
</td>
<td style="text-align:left;">
CYP2D7
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
rhea
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
Production-Degradation
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Activating
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0001377
</td>
<td style="text-align:left;">
Oxygen
</td>
<td style="text-align:left;">
977
</td>
<td style="text-align:left;">
Other non-metal oxides
</td>
<td style="text-align:left;">
A0A087X1C5
</td>
<td style="text-align:left;">
CYP2D7
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
rhea
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
-1
</td>
<td style="text-align:left;">
Production-Degradation
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Inhibiting
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0001377
</td>
<td style="text-align:left;">
Oxygen
</td>
<td style="text-align:left;">
977
</td>
<td style="text-align:left;">
Other non-metal oxides
</td>
<td style="text-align:left;">
A0A1W2PPD8
</td>
<td style="text-align:left;">
KDM4F
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
rhea
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
-1
</td>
<td style="text-align:left;">
Production-Degradation
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Inhibiting
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0001967
</td>
<td style="text-align:left;">
Carbon dioxide
</td>
<td style="text-align:left;">
280
</td>
<td style="text-align:left;">
Other non-metal oxides
</td>
<td style="text-align:left;">
A0A1W2PPD8
</td>
<td style="text-align:left;">
KDM4F
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
rhea
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
Production-Degradation
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Activating
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0002111
</td>
<td style="text-align:left;">
Water
</td>
<td style="text-align:left;">
962
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
A0A1W2PQ27
</td>
<td style="text-align:left;">
SSU72L1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
rhea
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
-1
</td>
<td style="text-align:left;">
Production-Degradation
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Inhibiting
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0002111
</td>
<td style="text-align:left;">
Water
</td>
<td style="text-align:left;">
962
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
A0A1W2PQ64
</td>
<td style="text-align:left;">
SSU72L5
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
rhea
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
-1
</td>
<td style="text-align:left;">
Production-Degradation
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Inhibiting
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0002111
</td>
<td style="text-align:left;">
Water
</td>
<td style="text-align:left;">
962
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
A0A1W2PQC6
</td>
<td style="text-align:left;">
SSU72L4
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
rhea
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
-1
</td>
<td style="text-align:left;">
Production-Degradation
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Inhibiting
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0002111
</td>
<td style="text-align:left;">
Water
</td>
<td style="text-align:left;">
962
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
A0A1W2PQD8
</td>
<td style="text-align:left;">
SSU72L2
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
rhea
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
-1
</td>
<td style="text-align:left;">
Production-Degradation
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Inhibiting
</td>
</tr>
</tbody>
</table>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>MetaLinksDB_Type</code> including
metabolite-receptor and metabolite-transporter sets.
</caption>
<thead><tr>
<th style="text-align:left;">
hmdb
</th>
<th style="text-align:left;">
metabolite
</th>
<th style="text-align:left;">
pubchem
</th>
<th style="text-align:left;">
term_specific
</th>
<th style="text-align:left;">
term
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
HMDB0000660
</td>
<td style="text-align:left;">
D-Fructose
</td>
<td style="text-align:left;">
439709
</td>
<td style="text-align:left;">
transporter_Production-Degradation
</td>
<td style="text-align:left;">
transporter
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0002212
</td>
<td style="text-align:left;">
Arachidic acid
</td>
<td style="text-align:left;">
10467
</td>
<td style="text-align:left;">
other_protein_Production-Degradation
</td>
<td style="text-align:left;">
other_protein
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0037790
</td>
<td style="text-align:left;">
Polyethylene glycol
</td>
<td style="text-align:left;">
174
</td>
<td style="text-align:left;">
catalytic_receptor_Ligand-Receptor
</td>
<td style="text-align:left;">
catalytic_receptor
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0034442
</td>
<td style="text-align:left;">
L-trans-alpha-Amino-2-carboxycyclopropaneacetic acid
</td>
<td style="text-align:left;">
1271
</td>
<td style="text-align:left;">
gpcr_Ligand-Receptor
</td>
<td style="text-align:left;">
gpcr
</td>
</tr>
</tbody>
</table>
<p><br><br></p>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="translate-ids">3. Translate IDs<a class="anchor" aria-label="anchor" href="#translate-ids"></a>
</h2>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
<p><br></p>
<div style="border: 1px solid red; padding: 10px; background-color: #ffe6e6;">
<p><strong>Warning:</strong> Translating IDs between formats e.g. KEGG
to HMDB is a non-trivial task, and it is possible for one original ID to
link to many translated IDs, and vice versa. We discuss this below and
in the InspectID subsection (which the TranslateID function will run by
default). We currently leave it to user discretion to select the most
appropriate ID based on their research question and data.</p>
</div>
<p><br></p>
<p>As visible from the prior knowledge tables above, it becomes clear
that we require a specific metabolite ID depending on the prior
knowledge resource. Indeed, the KEGG pathways include KEGG IDs, whilst
the Hallmarks metabolite sets include HMDB IDs.</p>
<p><br>
If we want to convert or ‘translate’ those IDs to another commonly used
form of ID, for instance because our measured data uses another type of
ID, we can make use of the <code><a href="../reference/TranslateID.html">MetaProViz::TranslateID()</a></code>
function. This is based on <a href="https://www.bioconductor.org/packages/release/bioc/html/OmnipathR.html" class="external-link">OmniPathR</a>
and RaMP DB <span class="citation">(Braisted et al. 2023)</span> on the
backend of our function and currently supports ID translation of
metabolites to and from the following formats:</p>
<ul>
<li><p>KEGG</p></li>
<li><p>HMDB</p></li>
<li><p>ChEBI</p></li>
<li><p>PubChem</p></li>
</ul>
<p>In the simplest case, we can use the function like so:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">translated_ids_example</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TranslateID.html">TranslateID</a></span><span class="op">(</span><span class="va">KEGG_Pathways</span><span class="op">[</span><span class="fl">39</span>,<span class="op">]</span><span class="op">)</span><span class="op">$</span><span class="va">Translated_DataFrame</span></span>
<span><span class="co">#&gt; [1] "Using method  GetAll"</span></span>
<span><span class="co">#&gt; [1] "Converting from  kegg  to  pubchem"</span></span>
<span><span class="co">#&gt; [1] "Converting from  kegg  to  chebi"</span></span>
<span><span class="co">#&gt; [1] "Converting from  kegg  to  hmdb"</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Example of 1-to-one translation of <code>KEGG_Pathways</code> (besides
HMDB)
</caption>
<thead><tr>
<th style="text-align:left;">
MetaboliteID
</th>
<th style="text-align:left;">
term
</th>
<th style="text-align:left;">
Metabolite
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
pubchem_collapsed
</th>
<th style="text-align:left;">
chebi_collapsed
</th>
<th style="text-align:left;">
hmdb_collapsed
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;">
C00091
</td>
<td style="text-align:left;">
Citrate cycle (TCA cycle) - Homo sapiens (human)
</td>
<td style="text-align:left;">
Succinyl-CoA
</td>
<td style="text-align:left;">
Citrate cycle (TCA cycle) - Homo sapiens (human)
</td>
<td style="text-align:left;">
92133
</td>
<td style="text-align:left;">
15380
</td>
<td style="text-align:left;">
HMDB0001022, HMDB01022
</td>
</tr></tbody>
</table>
<p><br>
Using the default translation settings for a single ID, we can see that
we receive back a single PubChem ID and ChEBI ID. However the HMDB
column has two HMDBs associated with it (HMDB0001022, HMDB01022). In
this case this difference is trivial and due to HMDB formatting
conventions: the longer version appears to be from the original HMDB ID
format versus the shorter more concise version. To avoid this we could
change the settings to remove HMDB completely, like so:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">translated_ids_example</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TranslateID.html">TranslateID</a></span><span class="op">(</span><span class="va">KEGG_Pathways</span><span class="op">[</span><span class="fl">39</span>,<span class="op">]</span>, </span>
<span>            SettingsInfo <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>IdColumn<span class="op">=</span><span class="st">"MetaboliteID"</span>, </span>
<span>                                FromFormat<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"kegg"</span><span class="op">)</span>, </span>
<span>                                ToFormat<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pubchem"</span>,<span class="st">"chebi"</span><span class="op">)</span>, </span>
<span>                                Method<span class="op">=</span><span class="st">"GetAll"</span>, </span>
<span>                                GroupingVariable<span class="op">=</span><span class="st">"term"</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">Translated_DataFrame</span></span>
<span><span class="co">#&gt; [1] "Using method  GetAll"</span></span>
<span><span class="co">#&gt; [1] "Converting from  kegg  to  pubchem"</span></span>
<span><span class="co">#&gt; [1] "Converting from  kegg  to  chebi"</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Example of 1-to-one translation of <code>KEGG_Pathways</code> (excluding
HMDB)
</caption>
<thead><tr>
<th style="text-align:left;">
MetaboliteID
</th>
<th style="text-align:left;">
term
</th>
<th style="text-align:left;">
Metabolite
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
pubchem_collapsed
</th>
<th style="text-align:left;">
chebi_collapsed
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;">
C00091
</td>
<td style="text-align:left;">
Citrate cycle (TCA cycle) - Homo sapiens (human)
</td>
<td style="text-align:left;">
Succinyl-CoA
</td>
<td style="text-align:left;">
Citrate cycle (TCA cycle) - Homo sapiens (human)
</td>
<td style="text-align:left;">
92133
</td>
<td style="text-align:left;">
15380
</td>
</tr></tbody>
</table>
<p><br></p>
<p>Or we could keep just the first item of every translation by changing
the `Method=‘GetFirst’` parameter like so:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">translated_ids_example</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TranslateID.html">TranslateID</a></span><span class="op">(</span><span class="va">KEGG_Pathways</span><span class="op">[</span><span class="fl">39</span>,<span class="op">]</span>, </span>
<span>            SettingsInfo <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>IdColumn<span class="op">=</span><span class="st">"MetaboliteID"</span>, </span>
<span>                                FromFormat<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"kegg"</span><span class="op">)</span>, </span>
<span>                                ToFormat<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pubchem"</span>,<span class="st">"chebi"</span>, <span class="st">"hmdb"</span><span class="op">)</span>, </span>
<span>                                Method<span class="op">=</span><span class="st">"GetFirst"</span>, </span>
<span>                                GroupingVariable<span class="op">=</span><span class="st">"term"</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">Translated_DataFrame</span></span>
<span><span class="co">#&gt; [1] "Using method  GetFirst"</span></span>
<span><span class="co">#&gt; [1] "Converting from  kegg  to  pubchem"</span></span>
<span><span class="co">#&gt; WARNING: Only the first translated ID from &lt;pubchem&gt; will be returned for each unique ID from &lt;kegg&gt;.</span></span>
<span><span class="co">#&gt; [1] "Converting from  kegg  to  chebi"</span></span>
<span><span class="co">#&gt; WARNING: Only the first translated ID from &lt;chebi&gt; will be returned for each unique ID from &lt;kegg&gt;.</span></span>
<span><span class="co">#&gt; [1] "Converting from  kegg  to  hmdb"</span></span>
<span><span class="co">#&gt; WARNING: Only the first translated ID from &lt;hmdb&gt; will be returned for each unique ID from &lt;kegg&gt;.</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Example of 1-to-one translation of <code>KEGG_Pathways</code> (using
GetFirst method)
</caption>
<thead><tr>
<th style="text-align:left;">
MetaboliteID
</th>
<th style="text-align:left;">
term
</th>
<th style="text-align:left;">
Metabolite
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
pubchem_first
</th>
<th style="text-align:left;">
chebi_first
</th>
<th style="text-align:left;">
hmdb_first
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;">
C00091
</td>
<td style="text-align:left;">
Citrate cycle (TCA cycle) - Homo sapiens (human)
</td>
<td style="text-align:left;">
Succinyl-CoA
</td>
<td style="text-align:left;">
Citrate cycle (TCA cycle) - Homo sapiens (human)
</td>
<td style="text-align:left;">
92133
</td>
<td style="text-align:left;">
15380
</td>
<td style="text-align:left;">
HMDB0001022
</td>
</tr></tbody>
</table>
<p><br></p>
<p>Ideally we would want a smarter and more consistent way to filter
these out. However, it is important to note that this translation
process is rarely as straightforward as the one-to-one translation above
(at least for PubChem and ChEBI IDs in the example).</p>
<p>Very commonly, an ID from one format will have a genuine one-to-many
relationship with the other format (e.g. one KEGG ID maps to multiple
HMDB IDs). Occasionally, it could also be a many-to-many relationship,
where some of the IDs from the new format link back to multiple IDs in
the original format (e.g. two different KEGG IDs map to multiple HMDS
IDs, some of which are shared between them). We can inspect instances of
where this is the case in our data:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">translated_ids_example</span> <span class="op">&lt;-</span> <span class="va">KEGG_Pathways</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">MetaboliteID</span> <span class="op">==</span> <span class="st">'C00009'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/nth.html" class="external-link">first</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/TranslateID.html">TranslateID</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Using method  GetAll"</span></span>
<span><span class="co">#&gt; [1] "Converting from  kegg  to  pubchem"</span></span>
<span><span class="co">#&gt; [1] "Converting from  kegg  to  chebi"</span></span>
<span><span class="co">#&gt; [1] "Converting from  kegg  to  hmdb"</span></span>
<span><span class="va">translated_ids_example_df</span> <span class="op">&lt;-</span> <span class="va">translated_ids_example</span><span class="op">$</span><span class="va">Translated_DataFrame</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Example of 1-to-many translation of <code>KEGG_Pathways</code> (besides
HMDB)
</caption>
<thead><tr>
<th style="text-align:left;">
MetaboliteID
</th>
<th style="text-align:left;">
term
</th>
<th style="text-align:left;">
Metabolite
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
pubchem_collapsed
</th>
<th style="text-align:left;">
chebi_collapsed
</th>
<th style="text-align:left;">
hmdb_collapsed
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;">
C00009
</td>
<td style="text-align:left;">
Oxidative phosphorylation - Homo sapiens (human)
</td>
<td style="text-align:left;">
Orthophosphate
</td>
<td style="text-align:left;">
Oxidative phosphorylation - Homo sapiens (human)
</td>
<td style="text-align:left;">
1004, 1061, 3681305, 1003
</td>
<td style="text-align:left;">
26078, 18367, 43474, 39745
</td>
<td style="text-align:left;">
HMDB0001429, HMDB0000973, HMDB0002105, HMDB0002142, HMDB0005947,
HMDB00973, HMDB01429, HMDB02105, HMDB02142, HMDB05947
</td>
</tr></tbody>
</table>
<p><br></p>
<p>The above example of the Orthophosphate metabolite
(<code>KEGG ID C00009</code>) shows the complexity of this one-to-many
multi-mapping issue: for the single KEGG ID, it returns:</p>
<ul>
<li><p>4 different PubChem IDs</p></li>
<li><p>4 different ChEBI IDs</p></li>
<li><p>10 different HMDB IDs</p></li>
</ul>
<p>For pathway analysis we would want to keep only one metabolite for
each pathway. But we would probably want to keep our metabolite choice
across pathways, avoid ambiguous mapping as much as possible
(many-to-many mapping), and have this metabolite ID selection guided by
any IDs we might have available to us in our input data.</p>
<p>This brings us to more formally discuss some of the important caveats
with the standard TranslateID function, which we will discuss now in the
InspectID subsection.</p>
<div class="section level3">
<h3 id="inspect-ids">Inspect IDs<a class="anchor" aria-label="anchor" href="#inspect-ids"></a>
</h3>
<p>Some of the complexities of Translating IDs which we have discussed
in the examples so far, can be demonstrated here (Fig.2). The InspectID
function (which is included by default when running TranslateID)
provides users with a summary of the relationships between the Original
and Translated IDs.</p>
<center>
<div class="float">
<img src="Fig.2.png" style="width:50.0%" alt="Fig. 2: Mapping problems in prior knowledge metabolite-sets when translating metabolite IDs."><div class="figcaption">Fig. 2: Mapping problems in prior knowledge
metabolite-sets when translating metabolite IDs.</div>
</div>
</center>
<p><br></p>
<p>As indicated by Figure 2, relationships between Original IDs
(e.g. KEGG) and Translated IDs (e.g. HMDB), can be quite complex,
resulting in One-to-None/One/Many mappings. Furthermore, the
relationships between the Translated IDs (e.g. HMDB) and Original IDs
(e.g. KEGG) must also be considered, because the Translated IDs could
also ambiguously map back to multiple different Original IDs.</p>
<p>Because of these complexities we leave it to the user to decide which
IDs exactly they wish to keep, which could also be influenced by which
identifiers they have in their experimental results. However we
automatically provide summary information to the user, with the
following two tables per Translated ID type:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Mapping_Orig2Trans_XXX_to_YYY</strong>
(e.g. Mapping_Orig2Trans_kegg_to_pubchem)</p></li>
<li><p><strong>Mapping_Trans2Orig_YYY_to_XXX</strong>
(e.g. Mapping_Trans2Orig_pubchem_to_kegg)</p></li>
</ol>
<p>Each table provides the summary of either the Original ID to
Translated IDs or vice versa, with counts of the number of Pathways the
ID was present in, and most importantly the number of
Original/Translated IDs that were mapped to the left-hand-side ID. This
is then summarised in the <code>Relationship</code> column, which is
specified as either:</p>
<ol style="list-style-type: decimal">
<li><p><code>One-to-None</code> : no matches were found for the
ID.</p></li>
<li><p><code>One-to-One</code> : a direct one-to-one relationship was
established between either Original ID to Translated ID, or Translated
ID to Original ID. Note however that this does not imply that the
reverse relationship is true. See Figure 2.</p></li>
<li><p><code>One-to-Many</code> : multiple matches were found for the
ID. i.e. it is ambiguously mapped.</p></li>
</ol>
<p><br></p>
<p>These tables are automatically returned as a DataFrame List when
running TranslateID, and can be accessed accordingly. Note that for the
full translation between all 4 ID types, it should take about 1 minute
to execute.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">translated_ids_example_full</span> <span class="op">&lt;-</span> <span class="va">KEGG_Pathways</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/TranslateID.html">TranslateID</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Using method  GetAll"</span></span>
<span><span class="co">#&gt; [1] "Converting from  kegg  to  pubchem"</span></span>
<span><span class="co">#&gt; [1] "Converting from  kegg  to  chebi"</span></span>
<span><span class="co">#&gt; [1] "Converting from  kegg  to  hmdb"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">translated_ids_example_full</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Translated_DataFrame"               "Mapping_Orig2Trans_kegg_to_pubchem"</span></span>
<span><span class="co">#&gt; [3] "Mapping_Trans2Orig_pubchem_to_kegg" "Mapping_Orig2Trans_kegg_to_chebi"  </span></span>
<span><span class="co">#&gt; [5] "Mapping_Trans2Orig_chebi_to_kegg"   "Mapping_Orig2Trans_kegg_to_hmdb"   </span></span>
<span><span class="co">#&gt; [7] "Mapping_Trans2Orig_hmdb_to_kegg"    "TranslationSummary"</span></span></code></pre></div>
<p><br></p>
<p>Finally, this information is summarised in the TranslationSummary
table.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">translated_ids_example_full</span><span class="op">$</span><span class="va">TranslationSummary</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 4</span></span></span>
<span><span class="co">#&gt;   Table                              `One-to-None` `One-to-One` `One-to-Many`</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Mapping_Orig2Trans_kegg_to_pubchem          <span style="text-decoration: underline;">1</span>681         <span style="text-decoration: underline;">1</span>300           556</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Mapping_Trans2Orig_pubchem_to_kegg             0         <span style="text-decoration: underline;">2</span>926           153</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Mapping_Orig2Trans_kegg_to_chebi            <span style="text-decoration: underline;">1</span>669          948           920</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> Mapping_Trans2Orig_chebi_to_kegg               0         <span style="text-decoration: underline;">3</span>043           112</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> Mapping_Orig2Trans_kegg_to_hmdb             <span style="text-decoration: underline;">1</span>688           97          <span style="text-decoration: underline;">1</span>752</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> Mapping_Trans2Orig_hmdb_to_kegg                0         <span style="text-decoration: underline;">5</span>685           308</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
TranslationSummary table from <code>KEGG_Pathways</code>
</caption>
<thead><tr>
<th style="text-align:left;">
Table
</th>
<th style="text-align:right;">
One-to-None
</th>
<th style="text-align:right;">
One-to-One
</th>
<th style="text-align:right;">
One-to-Many
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
Mapping_Orig2Trans_kegg_to_pubchem
</td>
<td style="text-align:right;">
1681
</td>
<td style="text-align:right;">
1300
</td>
<td style="text-align:right;">
556
</td>
</tr>
<tr>
<td style="text-align:left;">
Mapping_Trans2Orig_pubchem_to_kegg
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2926
</td>
<td style="text-align:right;">
153
</td>
</tr>
<tr>
<td style="text-align:left;">
Mapping_Orig2Trans_kegg_to_chebi
</td>
<td style="text-align:right;">
1669
</td>
<td style="text-align:right;">
948
</td>
<td style="text-align:right;">
920
</td>
</tr>
<tr>
<td style="text-align:left;">
Mapping_Trans2Orig_chebi_to_kegg
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3043
</td>
<td style="text-align:right;">
112
</td>
</tr>
<tr>
<td style="text-align:left;">
Mapping_Orig2Trans_kegg_to_hmdb
</td>
<td style="text-align:right;">
1688
</td>
<td style="text-align:right;">
97
</td>
<td style="text-align:right;">
1752
</td>
</tr>
<tr>
<td style="text-align:left;">
Mapping_Trans2Orig_hmdb_to_kegg
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
5685
</td>
<td style="text-align:right;">
308
</td>
</tr>
</tbody>
</table>
<p>Note that ordinarily, we should expect that the Trans2Orig tables do
not have any One-to-None relationships, because those IDs that have been
already translated naturally have an Original ID that was their
source.</p>
<p><br><br></p>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="run-enrichment-analysis">4. Run enrichment analysis<a class="anchor" aria-label="anchor" href="#run-enrichment-analysis"></a>
</h2>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
<p><br>
There are two options:<br>
1. Over Representation Analysis (ORA) that determines if a set of
features (=metabolic pathways) are over-represented in the selection of
features (=metabolites) from the data in comparison to all measured
features (metabolites) using the Fishers exact test
<code><a href="../reference/ClusterORA.html">MetaProViz::ClusterORA</a></code>. This can be applied to clusters of
metabolites as for example the results from
<code><a href="../reference/MCA_2Cond.html">MetaProViz::MCA_2Cond()</a></code> or <code>MetaProViz::CoRe()</code>
function. If you want to have more details on these clustering methods
please visit the vignette <a href="https://saezlab.github.io/MetaProViz/articles/Standard%20Metabolomics.html" class="external-link">Standard
Metabolomics</a> or <a href="https://saezlab.github.io/MetaProViz/articles/CoRe%20Metabolomics.html" class="external-link">CoRe
Metabolomics</a>.<br></p>
<p><br>
2. Enrichment analysis on standard differential analysis results. We
offer ORA <code><a href="../reference/StandardORA.html">MetaProViz::StandardORA</a></code>, but there are many other
statistical tests that can be used for enrichment analysis. The full
scope of different methods is beyond the scope of MetaProViz, but are
available in decoupleR <span class="citation">(Badia-I-Mompel et al.
2022)</span> packages from our group.</p>
<p><br><br></p>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<pre><code><span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">#&gt;  version  R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt;  os       Ubuntu 22.04.5 LTS</span></span>
<span><span class="co">#&gt;  system   x86_64, linux-gnu</span></span>
<span><span class="co">#&gt;  ui       X11</span></span>
<span><span class="co">#&gt;  language en</span></span>
<span><span class="co">#&gt;  collate  C.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    C.UTF-8</span></span>
<span><span class="co">#&gt;  tz       UTC</span></span>
<span><span class="co">#&gt;  date     2024-10-31</span></span>
<span><span class="co">#&gt;  pandoc   3.1.11 @ /opt/hostedtoolcache/pandoc/3.1.11/x64/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package    </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version  </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">#&gt;  abind         1.4-8     <span style="color: #555555;">2024-09-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  backports     1.5.0     <span style="color: #555555;">2024-05-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  beeswarm      0.4.0     <span style="color: #555555;">2021-06-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  bit           4.5.0     <span style="color: #555555;">2024-09-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  bit64         4.5.2     <span style="color: #555555;">2024-09-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  blob          1.2.4     <span style="color: #555555;">2023-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  broom         1.0.7     <span style="color: #555555;">2024-09-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  bslib         0.8.0     <span style="color: #555555;">2024-07-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  cachem        1.1.0     <span style="color: #555555;">2024-05-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  car           3.1-3     <span style="color: #555555;">2024-09-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  carData       3.0-5     <span style="color: #555555;">2022-01-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  cellranger    1.1.0     <span style="color: #555555;">2016-07-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  checkmate     2.3.2     <span style="color: #555555;">2024-07-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  cli           3.6.3     <span style="color: #555555;">2024-06-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  colorspace    2.1-1     <span style="color: #555555;">2024-07-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  crayon        1.5.3     <span style="color: #555555;">2024-06-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  curl          5.2.3     <span style="color: #555555;">2024-09-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  DBI           1.2.3     <span style="color: #555555;">2024-06-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  desc          1.4.3     <span style="color: #555555;">2023-12-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  digest        0.6.37    <span style="color: #555555;">2024-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  dplyr       * 1.1.4     <span style="color: #555555;">2023-11-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  evaluate      1.0.1     <span style="color: #555555;">2024-10-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  fansi         1.0.6     <span style="color: #555555;">2023-12-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  fastmap       1.2.0     <span style="color: #555555;">2024-05-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  forcats     * 1.0.0     <span style="color: #555555;">2023-01-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  Formula       1.2-5     <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  fs            1.6.5     <span style="color: #555555;">2024-10-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  generics      0.1.3     <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  ggbeeswarm    0.7.2     <span style="color: #555555;">2023-04-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  ggplot2     * 3.5.1     <span style="color: #555555;">2024-04-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  ggpubr        0.6.0     <span style="color: #555555;">2023-02-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  ggsignif      0.6.4     <span style="color: #555555;">2022-10-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  glue          1.8.0     <span style="color: #555555;">2024-09-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  gtable        0.3.6     <span style="color: #555555;">2024-10-25</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  highr         0.11      <span style="color: #555555;">2024-05-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  hms           1.1.3     <span style="color: #555555;">2023-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  htmltools     0.5.8.1   <span style="color: #555555;">2024-04-04</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  htmlwidgets   1.6.4     <span style="color: #555555;">2023-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  httr          1.4.7     <span style="color: #555555;">2023-08-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  igraph        2.1.1     <span style="color: #555555;">2024-10-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  jquerylib     0.1.4     <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  jsonlite      1.8.9     <span style="color: #555555;">2024-09-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  kableExtra    1.4.0     <span style="color: #555555;">2024-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  knitr         1.48      <span style="color: #555555;">2024-07-07</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  later         1.3.2     <span style="color: #555555;">2023-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  lifecycle     1.0.4     <span style="color: #555555;">2023-11-07</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  logger        0.4.0     <span style="color: #555555;">2024-10-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  lubridate   * 1.9.3     <span style="color: #555555;">2023-09-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  magrittr    * 2.0.3     <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  memoise       2.0.1     <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  MetaProViz  * 2.1.2     <span style="color: #555555;">2024-10-31</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">local</span></span></span>
<span><span class="co">#&gt;  munsell       0.5.1     <span style="color: #555555;">2024-04-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  OmnipathR     3.13.28   <span style="color: #555555;">2024-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">Github (saezlab/OmnipathR@4007299)</span></span></span>
<span><span class="co">#&gt;  pillar        1.9.0     <span style="color: #555555;">2023-03-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  pkgconfig     2.0.3     <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  pkgdown       2.1.1     <span style="color: #555555;">2024-09-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  prettyunits   1.2.0     <span style="color: #555555;">2023-09-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  progress      1.2.3     <span style="color: #555555;">2023-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  purrr       * 1.0.2     <span style="color: #555555;">2023-08-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  R.methodsS3   1.8.2     <span style="color: #555555;">2022-06-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  R.oo          1.26.0    <span style="color: #555555;">2024-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  R.utils       2.12.3    <span style="color: #555555;">2023-11-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  R6            2.5.1     <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  ragg          1.3.3     <span style="color: #555555;">2024-09-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  rappdirs      0.3.3     <span style="color: #555555;">2021-01-31</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  Rcpp          1.0.13    <span style="color: #555555;">2024-07-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  readr       * 2.1.5     <span style="color: #555555;">2024-01-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  readxl        1.4.3     <span style="color: #555555;">2023-07-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  rlang         1.1.4     <span style="color: #555555;">2024-06-04</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  rmarkdown     2.28      <span style="color: #555555;">2024-08-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  RSQLite       2.3.7     <span style="color: #555555;">2024-05-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  rstatix       0.7.2     <span style="color: #555555;">2023-02-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  rstudioapi    0.17.1    <span style="color: #555555;">2024-10-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  rvest         1.0.4     <span style="color: #555555;">2024-02-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  sass          0.4.9     <span style="color: #555555;">2024-03-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  scales        1.3.0     <span style="color: #555555;">2023-11-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  selectr       0.4-2     <span style="color: #555555;">2019-11-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  sessioninfo   1.2.2     <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">any (@1.2.2)</span></span></span>
<span><span class="co">#&gt;  stringi       1.8.4     <span style="color: #555555;">2024-05-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  stringr     * 1.5.1     <span style="color: #555555;">2023-11-14</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  svglite       2.1.3     <span style="color: #555555;">2023-12-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  systemfonts   1.1.0     <span style="color: #555555;">2024-05-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  textshaping   0.4.0     <span style="color: #555555;">2024-05-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  tibble      * 3.2.1     <span style="color: #555555;">2023-03-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  tidyr       * 1.3.1     <span style="color: #555555;">2024-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  tidyselect    1.2.1     <span style="color: #555555;">2024-03-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  tidyverse   * 2.0.0     <span style="color: #555555;">2023-02-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  timechange    0.3.0     <span style="color: #555555;">2024-01-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  tzdb          0.4.0     <span style="color: #555555;">2023-05-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  utf8          1.2.4     <span style="color: #555555;">2023-10-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  vctrs         0.6.5     <span style="color: #555555;">2023-12-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  vipor         0.4.7     <span style="color: #555555;">2023-12-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  viridisLite   0.4.2     <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  vroom         1.6.5     <span style="color: #555555;">2023-12-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  withr         3.0.2     <span style="color: #555555;">2024-10-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  xfun          0.48      <span style="color: #555555;">2024-10-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  XML           3.99-0.17 <span style="color: #555555;">2024-06-25</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  xml2          1.3.6     <span style="color: #555555;">2023-12-04</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  yaml          2.3.10    <span style="color: #555555;">2024-07-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt;  zip           2.3.1     <span style="color: #555555;">2024-01-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [1] /home/runner/work/_temp/Library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [2] /opt/R/4.4.1/lib/R/site-library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [3] /opt/R/4.4.1/lib/R/library</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-Badia-I-Mompel2022" class="csl-entry">
Badia-I-Mompel, Pau, Jesús Vélez Santiago, Jana Braunger, Celina Geiss,
Daniel Dimitrov, Sophia Müller-Dott, Petr Taus, et al. 2022.
<span>“decoupleR: Ensemble of Computational Methods to Infer Biological
Activities from Omics Data.”</span> <em>Bioinformatics Advances</em> 2
(1): vbac016. <a href="https://doi.org/10.1093/bioadv/vbac016" class="external-link">https://doi.org/10.1093/bioadv/vbac016</a>.
</div>
<div id="ref-Braisted2023" class="csl-entry">
Braisted, John, Andrew Patt, Cole Tindall, Timothy Sheils, Jorge Neyra,
Kyle Spencer, Tara Eicher, and Ewy A Mathé. 2023. <span>“RaMP-DB 2.0: A
Renovated Knowledgebase for Deriving Biological and Chemical Insight
from Metabolites, Proteins, and Genes.”</span> <em>Bioinformatics</em>,
no. 1 (January). <a href="https://doi.org/10.1093/bioinformatics/btac726" class="external-link">https://doi.org/10.1093/bioinformatics/btac726</a>.
</div>
<div id="ref-Castanza2022" class="csl-entry">
Castanza, Anthony Scott, Jill Marie Recla, David Eby, Helga
Thorvaldsdottir, Carol J. Bult, and Jill P Mesirov. 2022. <span>“The
Molecular Signatures Database Revisited: Extending Support for Mouse
Data.”</span> <em>BioRxiv</em>, October. <a href="https://doi.org/10.1101/2022.10.24.513539" class="external-link">https://doi.org/10.1101/2022.10.24.513539</a>.
</div>
<div id="ref-Dugourd2021" class="csl-entry">
Dugourd, Aurelien, Christoph Kuppe, Marco Sciacovelli, Enio Gjerga,
Attila Gabor, Kristina B Emdal, Vitor Vieira, et al. 2021. <span>“Causal
Integration of Multi-Omics Data with Prior Knowledge to Generate
Mechanistic Hypotheses.”</span> <em>Molecular Systems Biology</em>, no.
1 (January): e9730. <a href="https://doi.org/10.15252/msb.20209730" class="external-link">https://doi.org/10.15252/msb.20209730</a>.
</div>
<div id="ref-Farr_Dimitrov2024" class="csl-entry">
Farr, Elias, Daniel Dimitrov, Christina Schmidt, Denes Turei, Sebastian
Lobentanzer, Aurelien Dugourd, and Julio Saez-Rodriguez. 2024.
<span>“MetalinksDB: A Flexible and Contextualizable Resource of
Metabolite-Protein Interactions.”</span> <em>Briefings in
Bioinformatics</em>, no. 4 (May). <a href="https://doi.org/10.1093/bib/bbae347" class="external-link">https://doi.org/10.1093/bib/bbae347</a>.
</div>
<div id="ref-Kanehisa2000" class="csl-entry">
Kanehisa, M, and S Goto. 2000. <span>“KEGG: Kyoto Encyclopedia of Genes
and Genomes.”</span> <em>Nucleic Acids Research</em> 28 (1): 27–30. <a href="https://doi.org/10.1093/nar/28.1.27" class="external-link">https://doi.org/10.1093/nar/28.1.27</a>.
</div>
<div id="ref-Sciacovelli_Dugourd2022" class="csl-entry">
Sciacovelli, Marco, Aurelien Dugourd, Lorea Valcarcel Jimenez, Ming
Yang, Efterpi Nikitopoulou, Ana S H Costa, Laura Tronci, et al. 2022.
<span>“Dynamic Partitioning of Branched-Chain Amino Acids-Derived
Nitrogen Supports Renal Cancer Progression.”</span> <em>Nature
Communications</em> 13 (1): 7830. <a href="https://doi.org/10.1038/s41467-022-35036-4" class="external-link">https://doi.org/10.1038/s41467-022-35036-4</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Christina Schmidt, Dimitrios Prymidis, Denes Turei, Macabe Daley, Julio Saez-Rodriguez, Christian Frezza.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
