<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Enrichment Analysis • MetaProViz</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Enrichment Analysis">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MetaProViz</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.1.3</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-tutorials" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true"><span class="fa fa-book-open"></span> Tutorials</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-tutorials">
<li><a class="dropdown-item" href="../articles/Standard%20Metabolomics.html">Standard Metabolomics</a></li>
    <li><a class="dropdown-item" href="../articles/CoRe%20Metabolomics.html">CoRe Metabolomics</a></li>
    <li><a class="dropdown-item" href="../articles/EnrichmentAnalysis.html">Enrichment Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/Metadata%20Analysis.html">Metadata Analysis</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html"><span class="fa fa-solid fa-rocket"></span> Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/saezlab/MetaProViz"><span class="fa fa-brands fa-github"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://saezlab.org/"><span class="fa fas fa-university"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://frezza.cecad-labs.uni-koeln.de/home"><span class="fa fas fa-university"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">



<script src="EnrichmentAnalysis_files/kePrint-0.0.1/kePrint.js"></script><link href="EnrichmentAnalysis_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Enrichment Analysis</h1>
                        <h4 data-toc-skip class="author">Christina
Schmidt</h4>
            <address class="author_afil">
      Heidelberg
University<br><h4 data-toc-skip class="author">Macabe
Daley</h4>
            <address class="author_afil">
      Heidelberg University<br><small class="dont-index">Source: <a href="https://github.com/saezlab/MetaProViz/blob/main/vignettes/EnrichmentAnalysis.Rmd" class="external-link"><code>vignettes/EnrichmentAnalysis.Rmd</code></a></small>
      <div class="d-none name"><code>EnrichmentAnalysis.Rmd</code></div>
    </address>
</address>
</div>

    
    
<style>
.vscroll-plot {
    width: 850px;
    height: 500px;
    overflow-y: scroll;
    overflow-x: hidden;
}
</style>
<div class="section level2">
<h2 id="section">
<img src="Hexagon_MetaProViz.png" align="right" width="200"><a class="anchor" aria-label="anchor" href="#section"></a>
</h2>
<p><br><span style="text-decoration:underline">In this tutorial we showcase how
to use <strong>MetaProViz</strong></span>:<br>
- To access metabolite prior knowledge.<br>
- To deal with many-to-many mapping in your metabolite
identifiers.<br>
- To perform pathway enrichment analysis.<br></p>
<p><br>
First if you have not done yet, install the required dependencies and
load the libraries:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Install Rtools if you haven’t done this yet, using the appropriate version (e.g.windows or macOS).</span></span>
<span><span class="co"># 2. Install the latest development version from GitHub using devtools</span></span>
<span><span class="co">#devtools::install_github("https://github.com/saezlab/MetaProViz")</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://saezlab.github.io/MetaProViz" class="external-link">MetaProViz</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Error in get(paste0(generic, ".", class), envir = get_method_env()) : </span></span>
<span><span class="co">#&gt;   object 'type_sum.accel' not found</span></span>
<span></span>
<span><span class="co">#dependencies that need to be loaded:</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Please install the Biocmanager Dependencies:</span></span>
<span><span class="co">#BiocManager::install("clusterProfiler")</span></span>
<span><span class="co">#BiocManager::install("EnhancedVolcano")</span></span>
<span><span class="co">#BiocManager::install("cosmosR")</span></span></code></pre></div>
<p><br><br></p>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
</div>
<div class="section level2">
<h2 id="loading-the-example-data">1. Loading the example data<a class="anchor" aria-label="anchor" href="#loading-the-example-data"></a>
</h2>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
<p><br><span style="text-decoration:underline">As part of the
<strong>MetaProViz</strong> package you can load the example data into
your global environment using the function
<code>toy_data()</code></span>:<br><code>1.</code>Cell line experiment <strong>(CellLine)</strong><br>
Here we choose an example datasets, which is publicly available on <a href="https://www.metabolomicsworkbench.org/data/DRCCMetadata.php?Mode=Project&amp;ProjectID=PR001418" class="external-link">metabolomics
workbench project PR001418</a> including metabolic profiles of human
renal epithelial cells HK2 and cell renal cell carcinoma (ccRCC) cell
lines cultured in Plasmax cell culture media <span class="citation">(Sciacovelli et al. 2022)</span>. The raw data are
available via <a href="https://www.metabolomicsworkbench.org/data/DRCCMetadata.php?Mode=Study&amp;StudyID=ST002224&amp;StudyType=MS&amp;ResultType=1" class="external-link">metabolomics
workbench study ST002224</a> were intracellular metabolomics of HK2 and
ccRCC cell lines 786-O, 786-M1A and 786-M2A were performed.<br>
We have performed pre-processing and differential analysis (details can
be found in the vignette vignette <a href="https://saezlab.github.io/MetaProViz/articles/Standard%20Metabolomics.html" class="external-link">Standard
Metabolomics</a>) and and here we load the differential metabolite
analysis results for the comparison of 786M-1A versus HK2.<br></p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Load the Pre-processed intracellular data:</span></span>
<span><span class="va">Intra_DMA_786M1A_vs_HK2</span><span class="op">&lt;-</span> <span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/ToyData.html">ToyData</a></span><span class="op">(</span>Data<span class="op">=</span><span class="st">"IntraCells_DMA"</span><span class="op">)</span></span></code></pre></div>
<p><code>2.</code> Tissue experiment <strong>(Tissue)</strong><br>
Here we chose publicly available data from the <a href="https://www.cell.com/cancer-cell/comments/S1535-6108(15)00468-7#supplementaryMaterial" class="external-link">paper</a>
“An Integrated Metabolic Atlas of Clear Cell Renal Cell Carcinoma”,
which includes metabolomic profiling on 138 matched clear cell renal
cell carcinoma (ccRCC)/normal tissue pairs.<br>
We have performed differential analysis (details can be found in the
vignette <a href="https://saezlab.github.io/MetaProViz/articles/Metadata%20Analysis.html" class="external-link">Metadata
Analysis</a>) and and here we load the differential metabolite analysis
results for the comparison of Tumour versus Normal.<br></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the example data:</span></span>
<span><span class="va">Tissue_TvsN</span> <span class="op">&lt;-</span> <span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/ToyData.html">ToyData</a></span><span class="op">(</span>Data<span class="op">=</span><span class="st">"Tissue_DMA"</span><span class="op">)</span></span>
<span><span class="va">Tissue_TvsN_Old</span> <span class="op">&lt;-</span> <span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/ToyData.html">ToyData</a></span><span class="op">(</span>Data<span class="op">=</span><span class="st">"Tissue_DMA_Old"</span><span class="op">)</span></span>
<span><span class="va">Tissue_TvsN_Young</span> <span class="op">&lt;-</span> <span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/ToyData.html">ToyData</a></span><span class="op">(</span>Data<span class="op">=</span><span class="st">"Tissue_DMA_Young"</span><span class="op">)</span></span></code></pre></div>
<p><br><br></p>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
</div>
<div class="section level2">
<h2 id="accessing-prior-knowledge">2. Accessing Prior Knowledge<a class="anchor" aria-label="anchor" href="#accessing-prior-knowledge"></a>
</h2>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
<p><br>
Metabolite prior knowledge (PK) is essential for the interpretation of
metabolomics data. It can be used to perform pathway enrichment
analysis, compound class enrichment analysis, and by using specific PK
databases, it can be used to study the connection of metabolites and
receptors or transporters.Since the quality and content of the PK will
dictate the success of the downstream analysis and biological
interpretation, it is important to ensure the PK is used
correctly.<br>
Specifically in metabolite PK, the many different PK databases and
resources pose several issues. Indeed, the metabolite identifiers
(e.g. KEGG, HMDB, PubChem, etc.) are not standardized across databases,
and the same metabolite can have multiple identifiers in different
databases. This is known as the many-to-many mapping problem.<br>
Moreover, metabolic pathways that are the basis of the PK databases also
includes co-factors such as ions or other small molecules that are not
only part of most reactions, but can also not be detected in
experimentallly acquired data (e.g. H2O, CO2, etc).</p>
<div class="section level3">
<h3 id="kegg-pathway-metabolite-sets">KEGG pathway-metabolite sets<a class="anchor" aria-label="anchor" href="#kegg-pathway-metabolite-sets"></a>
</h3>
<p>KEGG pathways that are loaded via KEGG API using the package
<code>KEGGREST</code> and can be used to perform pathway analysis <span class="citation">(Kanehisa and Goto 2000)</span>.
<strong>(KEGG_Pathways)</strong><br></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#This will use KEGGREST to query the KEGG API to load the pathways:</span></span>
<span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/LoadKEGG.html">LoadKEGG</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Cached file loaded from: ~/.cache/KEGG_Metabolite.rds</span></span></code></pre></div>
<p><br></p>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>KEGG_Pathways</code>.
</caption>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
term
</th>
<th style="text-align:left;">
Metabolite
</th>
<th style="text-align:left;">
MetaboliteID
</th>
<th style="text-align:left;">
Description
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis - Homo sapiens (human)
</td>
<td style="text-align:left;">
Pyruvate
</td>
<td style="text-align:left;">
C00022
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis - Homo sapiens (human)
</td>
</tr>
<tr>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis - Homo sapiens (human)
</td>
<td style="text-align:left;">
Acetyl-CoA
</td>
<td style="text-align:left;">
C00024
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis - Homo sapiens (human)
</td>
</tr>
<tr>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis - Homo sapiens (human)
</td>
<td style="text-align:left;">
D-Glucose
</td>
<td style="text-align:left;">
C00031
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis - Homo sapiens (human)
</td>
</tr>
<tr>
<td style="text-align:left;">
52
</td>
<td style="text-align:left;">
Pentose phosphate pathway - Homo sapiens (human)
</td>
<td style="text-align:left;">
Pyruvate
</td>
<td style="text-align:left;">
C00022
</td>
<td style="text-align:left;">
Pentose phosphate pathway - Homo sapiens (human)
</td>
</tr>
<tr>
<td style="text-align:left;">
53
</td>
<td style="text-align:left;">
Pentose phosphate pathway - Homo sapiens (human)
</td>
<td style="text-align:left;">
D-Glucose
</td>
<td style="text-align:left;">
C00031
</td>
<td style="text-align:left;">
Pentose phosphate pathway - Homo sapiens (human)
</td>
</tr>
<tr>
<td style="text-align:left;">
54
</td>
<td style="text-align:left;">
Pentose phosphate pathway - Homo sapiens (human)
</td>
<td style="text-align:left;">
D-Fructose 6-phosphate
</td>
<td style="text-align:left;">
C00085
</td>
<td style="text-align:left;">
Pentose phosphate pathway - Homo sapiens (human)
</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="create-pathway-metabolite-sets">Create pathway-metabolite sets<a class="anchor" aria-label="anchor" href="#create-pathway-metabolite-sets"></a>
</h3>
<p>The function <code>Make_GeneMetabSet</code> can be used to translate
gene names to metabolite names by using a PK network of metabolic
reactions calls CosmosR <span class="citation">(Dugourd et al.
2021)</span>. This function is useful if you want to perform pathway
enrichment analysis on available gene-sets such as the Hallmarks
gene-sets from MSigDB <span class="citation">(Castanza et al.
2022)</span>. Moreover, it enables you to perform combined pathway
enrichment analysis on metabolite-gene sets, if you have other data
types such as proteomics measuring the enzymes expression.<br>
The Hallmarks gene-set is available in the package
<code>MetaProViz</code> and can be loaded using the function
<code><a href="../reference/LoadHallmarks.html">LoadHallmarks()</a></code>.<br></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Load the example data:</span></span>
<span><span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/LoadHallmarks.html">LoadHallmarks</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>Hallmarks_Pathways</code> including gene-sets
usable for pathway enrichment analysis.
</caption>
<thead><tr>
<th style="text-align:left;">
term
</th>
<th style="text-align:left;">
gene
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
HALLMARK_BILE_ACID_METABOLISM
</td>
<td style="text-align:left;">
GSTK1
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_BILE_ACID_METABOLISM
</td>
<td style="text-align:left;">
ABCG4
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
LDHC
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
ARPP19
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
LDHC
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
ARPP19
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
CENPA
</td>
</tr>
</tbody>
</table>
<p><br>
Now we can use the function <code>Make_GeneMetabSet</code> to translate
the gene names to metabolite names.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Translate gene names to metabolite names</span></span>
<span><span class="va">Hallmarks_GeneMetab</span> <span class="op">&lt;-</span> <span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/Make_GeneMetabSet.html">Make_GeneMetabSet</a></span><span class="op">(</span>Input_GeneSet<span class="op">=</span><span class="va">Hallmark_Pathways</span>,</span>
<span>                                                     SettingsInfo<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>Target<span class="op">=</span><span class="st">"gene"</span><span class="op">)</span>,</span>
<span>                                                     PKName<span class="op">=</span><span class="st">"Hallmarks"</span><span class="op">)</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>Hallmarks_GeneMetab</code> including gene-sets
usable for pathway enrichment analysis.
</caption>
<thead><tr>
<th style="text-align:left;">
term
</th>
<th style="text-align:left;">
feature
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
ME2
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
LDHC
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
FKBP4
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
HMDB0000241
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
HMDB0000570
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
HMDB0000122
</td>
</tr>
</tbody>
</table>
<p><br>
Given that we have the gene-metabolite-sets, we can now also run
enrichment analysis on combined data types, once including the
metabolite Log2FC and one including gene Log2FC from
e.g. transcriptomics or proteomics data. Yet, it is important to keep in
mind that generally we detect less metabolites than genes and hence this
may bias the results obtained from combined enrichment analysis.</p>
</div>
<div class="section level3">
<h3 id="metalinksdb-metabolite-receptor-sets">MetaLinksDB Metabolite-receptor sets<a class="anchor" aria-label="anchor" href="#metalinksdb-metabolite-receptor-sets"></a>
</h3>
<p>The MetaLinks database is a manually curated database of
metabolite-receptor and metabolite-transporter sets that can be used to
study the connection of metabolites and receptors or transporters <span class="citation">(Farr et al. 2024)</span>.<br></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">MetaLinksDB_Res</span> <span class="op">&lt;-</span> <span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/LoadMetalinks.html">LoadMetalinks</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Metalinks database downloaded and saved to: ~/.cache/metalinks.db</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>MetaLinksDB</code> including metabolite-receptor
sets.
</caption>
<thead><tr>
<th style="text-align:left;">
hmdb
</th>
<th style="text-align:left;">
metabolite
</th>
<th style="text-align:left;">
pubchem
</th>
<th style="text-align:left;">
metabolite_subclass
</th>
<th style="text-align:left;">
uniprot
</th>
<th style="text-align:left;">
gene_symbol
</th>
<th style="text-align:left;">
protein_type
</th>
<th style="text-align:left;">
source
</th>
<th style="text-align:right;">
db_score
</th>
<th style="text-align:right;">
experiment_score
</th>
<th style="text-align:right;">
combined_score
</th>
<th style="text-align:right;">
mor
</th>
<th style="text-align:left;">
type
</th>
<th style="text-align:left;">
transport_direction
</th>
<th style="text-align:left;">
mode_of_regulation
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
HMDB0002111
</td>
<td style="text-align:left;">
Water
</td>
<td style="text-align:left;">
962
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
A0A087X1C5
</td>
<td style="text-align:left;">
CYP2D7
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
rhea
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
Production-Degradation
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Activating
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0001377
</td>
<td style="text-align:left;">
Oxygen
</td>
<td style="text-align:left;">
977
</td>
<td style="text-align:left;">
Other non-metal oxides
</td>
<td style="text-align:left;">
A0A087X1C5
</td>
<td style="text-align:left;">
CYP2D7
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
rhea
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
-1
</td>
<td style="text-align:left;">
Production-Degradation
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Inhibiting
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0001377
</td>
<td style="text-align:left;">
Oxygen
</td>
<td style="text-align:left;">
977
</td>
<td style="text-align:left;">
Other non-metal oxides
</td>
<td style="text-align:left;">
A0A1W2PPD8
</td>
<td style="text-align:left;">
KDM4F
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
rhea
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
-1
</td>
<td style="text-align:left;">
Production-Degradation
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Inhibiting
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0001967
</td>
<td style="text-align:left;">
Carbon dioxide
</td>
<td style="text-align:left;">
280
</td>
<td style="text-align:left;">
Other non-metal oxides
</td>
<td style="text-align:left;">
A0A1W2PPD8
</td>
<td style="text-align:left;">
KDM4F
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
rhea
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
Production-Degradation
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Activating
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0002111
</td>
<td style="text-align:left;">
Water
</td>
<td style="text-align:left;">
962
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
A0A1W2PQ27
</td>
<td style="text-align:left;">
SSU72L1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
rhea
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
-1
</td>
<td style="text-align:left;">
Production-Degradation
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Inhibiting
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0002111
</td>
<td style="text-align:left;">
Water
</td>
<td style="text-align:left;">
962
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
A0A1W2PQ64
</td>
<td style="text-align:left;">
SSU72L5
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
rhea
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
-1
</td>
<td style="text-align:left;">
Production-Degradation
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Inhibiting
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0002111
</td>
<td style="text-align:left;">
Water
</td>
<td style="text-align:left;">
962
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
A0A1W2PQC6
</td>
<td style="text-align:left;">
SSU72L4
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
rhea
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
-1
</td>
<td style="text-align:left;">
Production-Degradation
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Inhibiting
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0002111
</td>
<td style="text-align:left;">
Water
</td>
<td style="text-align:left;">
962
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
A0A1W2PQD8
</td>
<td style="text-align:left;">
SSU72L2
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
rhea
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
-1
</td>
<td style="text-align:left;">
Production-Degradation
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Inhibiting
</td>
</tr>
</tbody>
</table>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>MetaLinksDB_Type</code> including
metabolite-receptor and metabolite-transporter sets.
</caption>
<thead><tr>
<th style="text-align:left;">
hmdb
</th>
<th style="text-align:left;">
metabolite
</th>
<th style="text-align:left;">
pubchem
</th>
<th style="text-align:left;">
term_specific
</th>
<th style="text-align:left;">
term
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
HMDB0000660
</td>
<td style="text-align:left;">
D-Fructose
</td>
<td style="text-align:left;">
439709
</td>
<td style="text-align:left;">
transporter_Production-Degradation
</td>
<td style="text-align:left;">
transporter
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0002212
</td>
<td style="text-align:left;">
Arachidic acid
</td>
<td style="text-align:left;">
10467
</td>
<td style="text-align:left;">
other_protein_Production-Degradation
</td>
<td style="text-align:left;">
other_protein
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0037790
</td>
<td style="text-align:left;">
Polyethylene glycol
</td>
<td style="text-align:left;">
174
</td>
<td style="text-align:left;">
catalytic_receptor_Ligand-Receptor
</td>
<td style="text-align:left;">
catalytic_receptor
</td>
</tr>
<tr>
<td style="text-align:left;">
HMDB0034442
</td>
<td style="text-align:left;">
L-trans-alpha-Amino-2-carboxycyclopropaneacetic acid
</td>
<td style="text-align:left;">
1271
</td>
<td style="text-align:left;">
gpcr_Ligand-Receptor
</td>
<td style="text-align:left;">
gpcr
</td>
</tr>
</tbody>
</table>
<p><br><br></p>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="translate-ids">3. Translate IDs<a class="anchor" aria-label="anchor" href="#translate-ids"></a>
</h2>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
<p><br></p>
<div style="border: 1px solid red; padding: 10px; background-color: #ffe6e6;">
<p><strong>Important Information:</strong> Translating IDs between
formats e.g. KEGG to HMDB is a non-trivial task, and it is expected for
one original ID to link to many translated IDs, and vice versa. We
discuss the implications throughout this vignette and leave it to user
discretion to select the most appropriate ID based on their research
question and data.</p>
</div>
<p><br></p>
<p>Across the different prior knowledge resources (see also tables
above) specific metabolite IDs are used and hence depending on the prior
knowledge resource a specific metabolite ID is required.<br>
If we want to convert or ‘translate’ those IDs to another commonly used
form of ID, for instance because our measured data uses another type of
ID, we can make use of the <code><a href="../reference/TranslateID.html">MetaProViz::TranslateID()</a></code>
function. This is based on <a href="https://www.bioconductor.org/packages/release/bioc/html/OmnipathR.html" class="external-link">OmniPathR</a>
and RaMP-DB <span class="citation">(Braisted et al. 2023)</span> on the
backend of our function and currently supports ID translation of
metabolites to and from the following formats:<br>
- KEGG<br>
- HMDB<br>
- ChEBI<br>
- PubChem<br><br>
As an example we are translating the KEGG pathways we loaded with
<code>metaproViz::LoadKEGG</code> into HMDB ids:<br></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">KEGG_Pathways_Translated</span> <span class="op">&lt;-</span> <span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/TranslateID.html">TranslateID</a></span><span class="op">(</span>InputData<span class="op">=</span> <span class="va">KEGG_Pathways</span>,</span>
<span>                                                     SettingsInfo <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>InputID<span class="op">=</span><span class="st">"MetaboliteID"</span>, GroupingVariable<span class="op">=</span><span class="st">"term"</span><span class="op">)</span>,</span>
<span>                                                     From <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"kegg"</span><span class="op">)</span>,</span>
<span>                                                     To <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hmdb"</span>, <span class="st">"pubchem"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Translation of KEGG IDs in KEGG pathways to HMDB &amp; pubchem IDs
</caption>
<thead><tr>
<th style="text-align:left;">
term
</th>
<th style="text-align:left;">
Metabolite
</th>
<th style="text-align:left;">
MetaboliteID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
hmdb
</th>
<th style="text-align:left;">
pubchem
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis - Homo sapiens (human)
</td>
<td style="text-align:left;">
Pyruvate
</td>
<td style="text-align:left;">
C00022
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis - Homo sapiens (human)
</td>
<td style="text-align:left;">
HMDB0000243, HMDB00243
</td>
<td style="text-align:left;">
1060, 107735
</td>
</tr>
<tr>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis - Homo sapiens (human)
</td>
<td style="text-align:left;">
Acetyl-CoA
</td>
<td style="text-align:left;">
C00024
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis - Homo sapiens (human)
</td>
<td style="text-align:left;">
HMDB0001206, HMDB01206, HMDB0247926
</td>
<td style="text-align:left;">
444493, 181
</td>
</tr>
<tr>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis - Homo sapiens (human)
</td>
<td style="text-align:left;">
D-Glucose
</td>
<td style="text-align:left;">
C00031
</td>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis - Homo sapiens (human)
</td>
<td style="text-align:left;">
HMDB0000122, HMDB0304632, HMDB0000516, HMDB0003340, HMDB0006564,
HMDB00122, HMDB00516, HMDB0062170, HMDB03340, HMDB06564, HMDB62170
</td>
<td style="text-align:left;">
64689, 5793, 107526
</td>
</tr>
<tr>
<td style="text-align:left;">
Ascorbate and aldarate metabolism - Homo sapiens (human)
</td>
<td style="text-align:left;">
beta-L-Galactose 1-phosphate
</td>
<td style="text-align:left;">
C15926
</td>
<td style="text-align:left;">
Ascorbate and aldarate metabolism - Homo sapiens (human)
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
Ascorbate and aldarate metabolism - Homo sapiens (human)
</td>
<td style="text-align:left;">
L-Galactonate
</td>
<td style="text-align:left;">
C15930
</td>
<td style="text-align:left;">
Ascorbate and aldarate metabolism - Homo sapiens (human)
</td>
<td style="text-align:left;">
HMDB0253886
</td>
<td style="text-align:left;">
604
</td>
</tr>
<tr>
<td style="text-align:left;">
Butanoate metabolism - Homo sapiens (human)
</td>
<td style="text-align:left;">
Succinate
</td>
<td style="text-align:left;">
C00042
</td>
<td style="text-align:left;">
Butanoate metabolism - Homo sapiens (human)
</td>
<td style="text-align:left;">
HMDB0000254, HMDB00254
</td>
<td style="text-align:left;">
1110, 160419
</td>
</tr>
<tr>
<td style="text-align:left;">
Butanoate metabolism - Homo sapiens (human)
</td>
<td style="text-align:left;">
Succinyl-CoA
</td>
<td style="text-align:left;">
C00091
</td>
<td style="text-align:left;">
Butanoate metabolism - Homo sapiens (human)
</td>
<td style="text-align:left;">
HMDB0001022, HMDB01022
</td>
<td style="text-align:left;">
92133
</td>
</tr>
<tr>
<td style="text-align:left;">
Butanoate metabolism - Homo sapiens (human)
</td>
<td style="text-align:left;">
Fumarate
</td>
<td style="text-align:left;">
C00122
</td>
<td style="text-align:left;">
Butanoate metabolism - Homo sapiens (human)
</td>
<td style="text-align:left;">
HMDB0000134, HMDB00134
</td>
<td style="text-align:left;">
444972, 5460307, 6183479
</td>
</tr>
</tbody>
</table>
<p><br>
Here it becomes apparent that the translation of IDs is not a one-to-one
mapping, but rather a one-to-many mapping. In fact it is very common
that an ID from one format will have a genuine one-to-many relationship
with the other format (e.g. one KEGG ID maps to multiple HMDB IDs) or
even a many-to-many relationship, where some of the IDs from the new
format link back to multiple IDs in the original format (e.g. two
different KEGG IDs map to multiple HMDS IDs, some of which are shared
between them).<br>
This comes with many implications for the analysis that will be
discussed in the next section.</p>
<div class="section level3">
<h3 id="mapping-problems">3.1 Mapping problems<a class="anchor" aria-label="anchor" href="#mapping-problems"></a>
</h3>
<p>The complexities of translating metabolite IDs are demonstrated here
(Fig.2). The relationships between Original IDs (e.g. KEGG) and
Translated IDs (e.g. HMDB), can be quite complex and in fact we
encounter <code>one-to-one</code> (no matches were found for the ID),
<code>one-to-none</code> ( direct #relationship was established),
<code>one-to-many</code> (multiple matches were found for the ID.
i.e. it is ambiguously mapped) and <code>many-to-many</code> ( considers
the relationships between the Translated IDs to Original IDs , where a
translated ID ambiguously maps back to multiple different Original IDs)
mappings.<br>
For enrichment analysis the translation from KEGG IDs to HMDB IDs
increases the pathways size, i.e. how many metabolites are in the
pathway “Glycolysis / Gluconeogenesis - Homo sapiens (human)”, which
would in turn inflate/deflate the enrichment results and hence it would
be desired to keep the number of metabolites in a pathway
consistent.<br></p>
<center>
<div class="float">
<img src="Fig.2.png" style="width:50.0%" alt="Fig. 2: Mapping problems in prior knowledge metabolite-sets when translating metabolite IDs."><div class="figcaption">Fig. 2: Mapping problems in prior knowledge
metabolite-sets when translating metabolite IDs.</div>
</div>
</center>
<p><br></p>
<p>Because of this complexity the output of
<code><a href="../reference/TranslateID.html">MetaProViz::TranslateID()</a></code> includes not only the Translation
table showcased above, but additionally information about the mapping
ambiguity as well as a summary of the relationships between the Original
and Translated IDs.<br>
Indeed, the translation of e.g. KEGG to hmdb and pubchem includes
multiple data frames including a summary of the mapping
occurrences:<br></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">KEGG_Pathways_Translated</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "TranslatedDF"             "TranslatedDF_MappingInfo"</span></span>
<span><span class="co">#&gt; [3] "MappingSummary_hmdb"      "MappingSummary_pubchem"</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
MappingSummary_hmdb
</caption>
<thead><tr>
<th style="text-align:left;">
term
</th>
<th style="text-align:right;">
many_to_many
</th>
<th style="text-align:right;">
one_to_many
</th>
<th style="text-align:right;">
one_to_none
</th>
<th style="text-align:right;">
total
</th>
<th style="text-align:right;">
to_many
</th>
<th style="text-align:right;">
to_none
</th>
<th style="text-align:right;">
to_total
</th>
<th style="text-align:right;">
from_many
</th>
<th style="text-align:right;">
from_one
</th>
<th style="text-align:right;">
from_total
</th>
<th style="text-align:right;">
one_to_one
</th>
<th style="text-align:right;">
to_one
</th>
<th style="text-align:left;">
scope
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
Glycolysis / Gluconeogenesis - Homo sapiens (human)
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
68
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
96
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
31
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
69
</td>
<td style="text-align:right;">
80
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
group
</td>
</tr>
<tr>
<td style="text-align:left;">
Citrate cycle (TCA cycle) - Homo sapiens (human)
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
48
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
51
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
49
</td>
<td style="text-align:right;">
49
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
group
</td>
</tr>
<tr>
<td style="text-align:left;">
Pentose phosphate pathway - Homo sapiens (human)
</td>
<td style="text-align:right;">
38
</td>
<td style="text-align:right;">
60
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
109
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
37
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
62
</td>
<td style="text-align:right;">
81
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
group
</td>
</tr>
<tr>
<td style="text-align:left;">
Ascorbate and aldarate metabolism - Homo sapiens (human)
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
96
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
143
</td>
<td style="text-align:right;">
28
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
57
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
100
</td>
<td style="text-align:right;">
109
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
group
</td>
</tr>
<tr>
<td style="text-align:left;">
Starch and sucrose metabolism - Homo sapiens (human)
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
78
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
98
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
37
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
80
</td>
<td style="text-align:right;">
80
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
group
</td>
</tr>
<tr>
<td style="text-align:left;">
Amino sugar and nucleotide sugar metabolism - Homo sapiens (human)
</td>
<td style="text-align:right;">
98
</td>
<td style="text-align:right;">
134
</td>
<td style="text-align:right;">
72
</td>
<td style="text-align:right;">
304
</td>
<td style="text-align:right;">
47
</td>
<td style="text-align:right;">
72
</td>
<td style="text-align:right;">
119
</td>
<td style="text-align:right;">
49
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
184
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
group
</td>
</tr>
</tbody>
</table>
<p><br><br>
We also have the ability to extract a long version of the DF that
includes a row for each mapping occurrence, which can be useful for
downstream analysis. Yet, this can become very large dependent on the
amount of many-to-many mappings, hence by default we do not generate
this summary. Within the <code><a href="../reference/TranslateID.html">MetaProViz::TranslateID()</a></code> you can
set the parameter <code>Summary =TRUE</code> or in case you have a
dataframe that includes both, original and translated ID, you can use
the function <code><a href="../reference/MappingAmbiguity.html">MetaProViz::MappingAmbiguity()</a></code> to generate
this long summary as well as the mapping summary in general.<br></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Option 1:</span></span>
<span><span class="va">KEGG_Pathways_TranslatedSum</span> <span class="op">&lt;-</span> <span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/TranslateID.html">TranslateID</a></span><span class="op">(</span>InputData<span class="op">=</span> <span class="va">KEGG_Pathways</span>,</span>
<span>                                                    SettingsInfo <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>InputID<span class="op">=</span><span class="st">"MetaboliteID"</span>, GroupingVariable<span class="op">=</span><span class="st">"term"</span><span class="op">)</span>,</span>
<span>                                                    From <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"kegg"</span><span class="op">)</span>,</span>
<span>                                                    To <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hmdb"</span>, <span class="st">"pubchem"</span><span class="op">)</span>,</span>
<span>                                                    Summary <span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Option 2:</span></span>
<span><span class="va">MappingProblems</span> <span class="op">&lt;-</span> <span class="fu">MetaProViz</span><span class="fu">::</span><span class="fu"><a href="../reference/MappingAmbiguity.html">MappingAmbiguity</a></span><span class="op">(</span>InputData<span class="op">=</span> <span class="va">KEGG_Pathways_Translated</span><span class="op">[[</span><span class="st">"TranslatedDF"</span><span class="op">]</span><span class="op">]</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span><span class="st">"KEGG"</span><span class="op">=</span><span class="st">"MetaboliteID"</span><span class="op">)</span>,</span>
<span>                                                From <span class="op">=</span> <span class="st">"KEGG"</span>,</span>
<span>                                                To <span class="op">=</span> <span class="st">"hmdb"</span>,</span>
<span>                                                GroupingVariable <span class="op">=</span> <span class="st">"term"</span>,</span>
<span>                                                Summary<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Long summary of mapping problems taking into account both directions,
From-to-To and To-to-From.
</caption>
<thead><tr>
<th style="text-align:left;">
KEGG
</th>
<th style="text-align:left;">
hmdb
</th>
<th style="text-align:left;">
term
</th>
<th style="text-align:left;">
KEGG_to_hmdb
</th>
<th style="text-align:right;">
Count(KEGG_to_hmdb)
</th>
<th style="text-align:left;">
hmdb_to_KEGG
</th>
<th style="text-align:right;">
Count(hmdb_to_KEGG)
</th>
<th style="text-align:left;">
Mapping
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
C00002
</td>
<td style="text-align:left;">
HMDB0000538
</td>
<td style="text-align:left;">
Calcium signaling pathway - Homo sapiens (human)
</td>
<td style="text-align:left;">
C00002 –&gt; HMDB0000538, HMDB00538, HMDB0257997
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
HMDB0000538 –&gt; C00002
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
one-to-many
</td>
</tr>
<tr>
<td style="text-align:left;">
C00002
</td>
<td style="text-align:left;">
HMDB0000538
</td>
<td style="text-align:left;">
Diabetic cardiomyopathy - Homo sapiens (human)
</td>
<td style="text-align:left;">
C00002 –&gt; HMDB0000538, HMDB00538, HMDB0257997
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
HMDB0000538 –&gt; C00002
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
one-to-many
</td>
</tr>
<tr>
<td style="text-align:left;">
C00024
</td>
<td style="text-align:left;">
HMDB01206
</td>
<td style="text-align:left;">
Insulin resistance - Homo sapiens (human)
</td>
<td style="text-align:left;">
C00024 –&gt; HMDB0001206, HMDB01206, HMDB0247926
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
HMDB01206 –&gt; C00024
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
one-to-many
</td>
</tr>
<tr>
<td style="text-align:left;">
C00024
</td>
<td style="text-align:left;">
HMDB01206
</td>
<td style="text-align:left;">
Lysine degradation - Homo sapiens (human)
</td>
<td style="text-align:left;">
C00024 –&gt; HMDB0001206, HMDB01206, HMDB0247926
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
HMDB01206 –&gt; C00024
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
one-to-many
</td>
</tr>
</tbody>
</table>
<p><br>
in the table it is shown that the KEGG ID C00002 maps to 3 different
HMDB IDs, and it is shown that one of those HMDB IDs HMDB0000538 maps to
one KEGG ID, hence this Mapping is one-to-many. The other two HMDB Ids
are also in the table and it is checked to how many KEGG IDs they map.
Additionally, we have passed <code>GroupingVariable = "term"</code> as
we have pathways, which also means each of those mappings is checked
within a pathway and across pathways (e.g. C00002 is shown twice, for
two different terms).<br>
In fact, to perform enrichment analysis we need a column source
(=e.g. term) and we would want to keep the metabolite IDs across
pathways consistent, avoid ambiguous mapping as much as possible
(many-to-many mapping), and have this metabolite ID selection guided by
any IDs we might have available to us in our measured input data (Fig.
3). This is crucial to circumvent inflation or deflation of
metabolite-sets, which in turn will affect the enrichment analysis
results. In the next section 4., we will elaborate on this.<br></p>
<center>
<div class="float">
<img src="Fig.3.png" style="width:50.0%" alt="Fig. 3: Mapping problems in prior knowledge metabolite-sets when translating metabolite IDs and the connection to detected (measured input metabolites)."><div class="figcaption">Fig. 3: Mapping problems in prior knowledge
metabolite-sets when translating metabolite IDs and the connection to
detected (measured input metabolites).</div>
</div>
</center>
<p><br></p>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="metabolite-ids-in-measured-data">4. Metabolite IDs in measured data<a class="anchor" aria-label="anchor" href="#metabolite-ids-in-measured-data"></a>
</h2>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
<div class="section level3">
<h3 id="use-measured-metabolite-ids-to-guide-the-selection-of-pk-ids">4.1 Use measured Metabolite IDs to guide the selection of PK
IDs<a class="anchor" aria-label="anchor" href="#use-measured-metabolite-ids-to-guide-the-selection-of-pk-ids"></a>
</h3>
<p>The measured metabolite IDs can be used to guide the selection of PK
IDs. This is crucial to circumvent inflation or deflation of
metabolite-sets, which in turn will affect the enrichment analysis
results.<br>
This is something we are currently working on and hope to provide within
the next release, so stay tuned.</p>
</div>
<div class="section level3">
<h3 id="assigning-metabolite-ids-to-measured-data">4.2. Assigning Metabolite IDs to measured data<a class="anchor" aria-label="anchor" href="#assigning-metabolite-ids-to-measured-data"></a>
</h3>
The difficulty with assigning metabolite IDs to measured data is the
uncertainty in the detection of metabolites. Indeed, differentiation of
structural isomers (both constitutional isomers and stereoisomers) as
for example the distinction between enantiomers. This leads to loss of
information and hence uncertainty is assigning metabolite IDs.<br>
One example is the metabolite Alanine, which can occur in its L- or D-
form. If in an experiment those enantiomers have not been distinguished,
the correct way would be to either assign two metabolite IDs (L- and
D-Alanine) or a more general Alanine ID without chiral information. Yet,
in reality this is not as trivial:<br><table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Available Alanine IDs in HMDB and ChEBI.
</caption>
<thead><tr>
<th style="text-align:left;">
TrivialName
</th>
<th style="text-align:left;">
HMDB
</th>
<th style="text-align:left;">
ChEBI
</th>
<th style="text-align:left;">
KEGG
</th>
<th style="text-align:left;">
PubChem
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
D-Alanine
</td>
<td style="text-align:left;">
HMDB0001310
</td>
<td style="text-align:left;">
15570
</td>
<td style="text-align:left;">
C00133
</td>
<td style="text-align:left;">
71080
</td>
</tr>
<tr>
<td style="text-align:left;">
L-Alanine
</td>
<td style="text-align:left;">
HMDB0000161
</td>
<td style="text-align:left;">
16977
</td>
<td style="text-align:left;">
C00041
</td>
<td style="text-align:left;">
5950
</td>
</tr>
<tr>
<td style="text-align:left;">
Alanine
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
16449
</td>
<td style="text-align:left;">
C01401
</td>
<td style="text-align:left;">
602
</td>
</tr>
<tr>
<td style="text-align:left;">
Alanine zwitterion
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
66916
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
57383916
</td>
</tr>
</tbody>
</table>
<br>
Indeed, dependent on the database, the Alanine metabolite can have
different IDs available:<br><img src="EnrichmentAnalysis_files/figure-html/unnamed-chunk-21-1.png" width="700"><br>
For instance, if we want to assign a HMDB ID, we have to assign both
“HMDB0001310”, “HMDB0000161” to the metabolite Alanine, for ChEBI we
could assign only one, “16449”, but this may lead to other problems as
the ChEBI ID is not specific and may not be part of certain metabolic
pathways. The reason for this is that substrate chirality is critical to
enzymatic processes and stereo selectivity of enzymes to be homochiral
with predominance of one particular enantiomer (e.g. D-sugars, L-amino
acids, etc.).<br>
To showcase the severity of this problem, we can look at the occurrence
of those metabolites in metabolic pathways across different databases.
To do so we searched for those metabolite IDs in the RaMP database <span class="citation">(Braisted et al. 2023)</span> and extracted the
pathways they are part of:<br><table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Alanine IDs in HMDB and ChEBI mapped to pathways from wiki, KEGG and
Reactome using RamP.
</caption>
<thead><tr>
<th style="text-align:left;">
TrivialName
</th>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Database
</th>
<th style="text-align:right;">
PathwayCount
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
L-Alanine
</td>
<td style="text-align:left;">
16977
</td>
<td style="text-align:left;">
ChEBI
</td>
<td style="text-align:right;">
44
</td>
</tr>
<tr>
<td style="text-align:left;">
L-Alanine
</td>
<td style="text-align:left;">
5950
</td>
<td style="text-align:left;">
PubChem
</td>
<td style="text-align:right;">
44
</td>
</tr>
<tr>
<td style="text-align:left;">
L-Alanine
</td>
<td style="text-align:left;">
C00041
</td>
<td style="text-align:left;">
KEGG
</td>
<td style="text-align:right;">
44
</td>
</tr>
<tr>
<td style="text-align:left;">
L-Alanine
</td>
<td style="text-align:left;">
HMDB0000161
</td>
<td style="text-align:left;">
HMDB
</td>
<td style="text-align:right;">
44
</td>
</tr>
<tr>
<td style="text-align:left;">
D-Alanine
</td>
<td style="text-align:left;">
15570
</td>
<td style="text-align:left;">
ChEBI
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
D-Alanine
</td>
<td style="text-align:left;">
71080
</td>
<td style="text-align:left;">
PubChem
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
D-Alanine
</td>
<td style="text-align:left;">
C00133
</td>
<td style="text-align:left;">
KEGG
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
D-Alanine
</td>
<td style="text-align:left;">
HMDB0001310
</td>
<td style="text-align:left;">
HMDB
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
Alanine zwitterion
</td>
<td style="text-align:left;">
57383916
</td>
<td style="text-align:left;">
PubChem
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
Alanine zwitterion
</td>
<td style="text-align:left;">
66916
</td>
<td style="text-align:left;">
ChEBI
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
Alanine
</td>
<td style="text-align:left;">
16449
</td>
<td style="text-align:left;">
ChEBI
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
Alanine
</td>
<td style="text-align:left;">
602
</td>
<td style="text-align:left;">
PubChem
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
Alanine
</td>
<td style="text-align:left;">
C01401
</td>
<td style="text-align:left;">
KEGG
</td>
<td style="text-align:right;">
4
</td>
</tr>
</tbody>
</table>
<p><br><img src="EnrichmentAnalysis_files/figure-html/unnamed-chunk-23-1.png" width="700"></p>
<p><br>
This showcases if we choose the ChEBI ID for Alanine (ChEBI ID 16449),
if experimentally the distinction was not possible, we will not map to
any pathway even though the metabolite is part of many pathways. Hence,
we recommend to assign multiple IDs to a measured ID, where specificity
in detection is not given.<br></p>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="run-enrichment-analysis">5. Run enrichment analysis<a class="anchor" aria-label="anchor" href="#run-enrichment-analysis"></a>
</h2>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
<p><br>
There are two options:<br>
1. Over Representation Analysis (ORA) that determines if a set of
features (=metabolic pathways) are over-represented in the selection of
features (=metabolites) from the data in comparison to all measured
features (metabolites) using the Fishers exact test
<code><a href="../reference/ClusterORA.html">MetaProViz::ClusterORA</a></code>. This can be applied to clusters of
metabolites as for example the results from
<code><a href="../reference/MCA_2Cond.html">MetaProViz::MCA_2Cond()</a></code> or <code>MetaProViz::CoRe()</code>
function. If you want to have more details on these clustering methods
please visit the vignette <a href="https://saezlab.github.io/MetaProViz/articles/Standard%20Metabolomics.html" class="external-link">Standard
Metabolomics</a> or <a href="https://saezlab.github.io/MetaProViz/articles/CoRe%20Metabolomics.html" class="external-link">CoRe
Metabolomics</a>.<br></p>
<p><br>
2. Enrichment analysis on standard differential analysis results. We
offer ORA <code><a href="../reference/StandardORA.html">MetaProViz::StandardORA</a></code>, but there are many other
statistical tests that can be used for enrichment analysis. The full
scope of different methods is beyond the scope of MetaProViz, but are
available in decoupleR <span class="citation">(Badia-I-Mompel et al.
2022)</span> packages from our group.</p>
<p><br><br></p>
<div class="progress progress-striped active">
<div class="progress-bar progress-bar-success" style="width: 100%">

</div>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<pre><code><span><span class="co">#&gt; R version 4.4.2 (2024-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.1 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8        LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [6] LC_MESSAGES=C.UTF-8    LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C           LC_TELEPHONE=C        </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] lubridate_1.9.4  forcats_1.0.0    stringr_1.5.1    purrr_1.0.2      readr_2.1.5      tibble_3.2.1    </span></span>
<span><span class="co">#&gt;  [7] tidyverse_2.0.0  tidyr_1.3.1      dplyr_1.1.4      magrittr_2.0.3   MetaProViz_2.1.3 ggplot2_3.5.1   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] DBI_1.2.3         gridExtra_2.3     logger_0.4.0      readxl_1.4.3      rlang_1.1.4       compiler_4.4.2   </span></span>
<span><span class="co">#&gt;   [7] RSQLite_2.3.9     systemfonts_1.1.0 vctrs_0.6.5       reshape2_1.4.4    rvest_1.0.4       pkgconfig_2.0.3  </span></span>
<span><span class="co">#&gt;  [13] crayon_1.5.3      fastmap_1.2.0     backports_1.5.0   labeling_0.4.3    rmarkdown_2.29    tzdb_0.4.0       </span></span>
<span><span class="co">#&gt;  [19] ggbeeswarm_0.7.2  ragg_1.3.3        bit_4.5.0.1       xfun_0.49         cachem_1.1.0      jsonlite_1.8.9   </span></span>
<span><span class="co">#&gt;  [25] progress_1.2.3    blob_1.2.4        later_1.4.1       broom_1.0.7       parallel_4.4.2    prettyunits_1.2.0</span></span>
<span><span class="co">#&gt;  [31] R6_2.5.1          bslib_0.8.0       stringi_1.8.4     limma_3.58.1      car_3.1-3         jquerylib_0.1.4  </span></span>
<span><span class="co">#&gt;  [37] cellranger_1.1.0  Rcpp_1.0.13-1     knitr_1.49        R.utils_2.12.3    igraph_2.1.2      timechange_0.3.0 </span></span>
<span><span class="co">#&gt;  [43] tidyselect_1.2.1  rstudioapi_0.17.1 abind_1.4-8       yaml_2.3.10       curl_6.0.1        plyr_1.8.9       </span></span>
<span><span class="co">#&gt;  [49] withr_3.0.2       inflection_1.3.6  evaluate_1.0.1    desc_1.4.3        zip_2.3.1         xml2_1.3.6       </span></span>
<span><span class="co">#&gt;  [55] pillar_1.10.0     ggpubr_0.6.0      carData_3.0-5     checkmate_2.3.2   generics_0.1.3    vroom_1.6.5      </span></span>
<span><span class="co">#&gt;  [61] hms_1.1.3         munsell_0.5.1     scales_1.3.0      gtools_3.9.5      OmnipathR_3.15.2  glue_1.8.0       </span></span>
<span><span class="co">#&gt;  [67] tools_4.4.2       ggsignif_0.6.4    fs_1.6.5          XML_3.99-0.17     grid_4.4.2        qcc_2.7          </span></span>
<span><span class="co">#&gt;  [73] colorspace_2.1-1  patchwork_1.3.0   beeswarm_0.4.0    vipor_0.4.7       Formula_1.2-5     cli_3.6.3        </span></span>
<span><span class="co">#&gt;  [79] rappdirs_0.3.3    kableExtra_1.4.0  textshaping_0.4.1 viridisLite_0.4.2 svglite_2.1.3     gtable_0.3.6     </span></span>
<span><span class="co">#&gt;  [85] R.methodsS3_1.8.2 rstatix_0.7.2     hash_2.2.6.3      selectr_0.4-2     sass_0.4.9        digest_0.6.37    </span></span>
<span><span class="co">#&gt;  [91] ggrepel_0.9.6     htmlwidgets_1.6.4 farver_2.1.2      memoise_2.0.1     htmltools_0.5.8.1 pkgdown_2.1.1    </span></span>
<span><span class="co">#&gt;  [97] R.oo_1.27.0       factoextra_1.0.7  lifecycle_1.0.4   httr_1.4.7        statmod_1.5.0     bit64_4.5.2      </span></span>
<span><span class="co">#&gt; [103] MASS_7.3-61</span></span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-Badia-I-Mompel2022" class="csl-entry">
Badia-I-Mompel, Pau, Jesús Vélez Santiago, Jana Braunger, Celina Geiss,
Daniel Dimitrov, Sophia Müller-Dott, Petr Taus, et al. 2022.
<span>“decoupleR: Ensemble of Computational Methods to Infer Biological
Activities from Omics Data.”</span> <em>Bioinformatics Advances</em> 2
(1): vbac016. <a href="https://doi.org/10.1093/bioadv/vbac016" class="external-link">https://doi.org/10.1093/bioadv/vbac016</a>.
</div>
<div id="ref-Braisted2023" class="csl-entry">
Braisted, John, Andrew Patt, Cole Tindall, Timothy Sheils, Jorge Neyra,
Kyle Spencer, Tara Eicher, and Ewy A Mathé. 2023. <span>“RaMP-DB 2.0: A
Renovated Knowledgebase for Deriving Biological and Chemical Insight
from Metabolites, Proteins, and Genes.”</span> <em>Bioinformatics</em>,
no. 1 (January). <a href="https://doi.org/10.1093/bioinformatics/btac726" class="external-link">https://doi.org/10.1093/bioinformatics/btac726</a>.
</div>
<div id="ref-Castanza2022" class="csl-entry">
Castanza, Anthony Scott, Jill Marie Recla, David Eby, Helga
Thorvaldsdottir, Carol J. Bult, and Jill P Mesirov. 2022. <span>“The
Molecular Signatures Database Revisited: Extending Support for Mouse
Data.”</span> <em>BioRxiv</em>, October. <a href="https://doi.org/10.1101/2022.10.24.513539" class="external-link">https://doi.org/10.1101/2022.10.24.513539</a>.
</div>
<div id="ref-Dugourd2021" class="csl-entry">
Dugourd, Aurelien, Christoph Kuppe, Marco Sciacovelli, Enio Gjerga,
Attila Gabor, Kristina B Emdal, Vitor Vieira, et al. 2021. <span>“Causal
Integration of Multi-Omics Data with Prior Knowledge to Generate
Mechanistic Hypotheses.”</span> <em>Molecular Systems Biology</em>, no.
1 (January): e9730. <a href="https://doi.org/10.15252/msb.20209730" class="external-link">https://doi.org/10.15252/msb.20209730</a>.
</div>
<div id="ref-Farr_Dimitrov2024" class="csl-entry">
Farr, Elias, Daniel Dimitrov, Christina Schmidt, Denes Turei, Sebastian
Lobentanzer, Aurelien Dugourd, and Julio Saez-Rodriguez. 2024.
<span>“MetalinksDB: A Flexible and Contextualizable Resource of
Metabolite-Protein Interactions.”</span> <em>Briefings in
Bioinformatics</em>, no. 4 (May). <a href="https://doi.org/10.1093/bib/bbae347" class="external-link">https://doi.org/10.1093/bib/bbae347</a>.
</div>
<div id="ref-Kanehisa2000" class="csl-entry">
Kanehisa, M, and S Goto. 2000. <span>“KEGG: Kyoto Encyclopedia of Genes
and Genomes.”</span> <em>Nucleic Acids Research</em> 28 (1): 27–30. <a href="https://doi.org/10.1093/nar/28.1.27" class="external-link">https://doi.org/10.1093/nar/28.1.27</a>.
</div>
<div id="ref-Sciacovelli_Dugourd2022" class="csl-entry">
Sciacovelli, Marco, Aurelien Dugourd, Lorea Valcarcel Jimenez, Ming
Yang, Efterpi Nikitopoulou, Ana S H Costa, Laura Tronci, et al. 2022.
<span>“Dynamic Partitioning of Branched-Chain Amino Acids-Derived
Nitrogen Supports Renal Cancer Progression.”</span> <em>Nature
Communications</em> 13 (1): 7830. <a href="https://doi.org/10.1038/s41467-022-35036-4" class="external-link">https://doi.org/10.1038/s41467-022-35036-4</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Christina Schmidt, Dimitrios Prymidis, Denes Turei, Macabe Daley, Julio Saez-Rodriguez, Christian Frezza.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
