<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Getting started with MetaProViz • MetaProViz</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Getting started with MetaProViz">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MetaProViz</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">3.99.25</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true"><span class="fa fa-book-open"></span></button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-">
<li><a class="dropdown-item" href="../articles/quick-start.html">Getting started</a></li>
    <li><a class="dropdown-item" href="../articles/vignettes/pkgdown/standard-metabolomics.html">Standard Metabolomics</a></li>
    <li><a class="dropdown-item" href="../articles/vignettes/pkgdown/core-metabolomics.html">CoRe Metabolomics</a></li>
    <li><a class="dropdown-item" href="../articles/vignettes/pkgdown/prior-knowledge.html">Prior Knowledge</a></li>
    <li><a class="dropdown-item" href="../articles/vignettes/pkgdown/sample-metadata.html">Sample Metadata Analysis</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html"><span class="fa fa-solid fa-rocket"></span> Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/saezlab/MetaProViz"><span class="fa fa-brands fa-github"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://saezlab.org/"><span class="fa fas fa-university"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://frezza.cecad-labs.uni-koeln.de/home"><span class="fa fas fa-university"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">



<script src="quick-start_files/kePrint-0.0.1/kePrint.js"></script><link href="quick-start_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Getting started with MetaProViz</h1>
                        <h4 data-toc-skip class="author">Christina
Schmidt</h4>
            <address class="author_afil">
      Heidelberg University<br><small class="dont-index">Source: <a href="https://github.com/saezlab/MetaProViz/blob/main/vignettes/quick-start.Rmd" class="external-link"><code>vignettes/quick-start.Rmd</code></a></small>
      <div class="d-none name"><code>quick-start.Rmd</code></div>
    </address>
</div>

    
    
<style>
.vscroll-plot {
    width: 850px;
    height: 500px;
    overflow-y: scroll;
    overflow-x: hidden;
}
</style>
<div class="section level2">
<h2 id="section">
<img src="Hexagon_MetaProViz.png" align="right" width="200"><a class="anchor" aria-label="anchor" href="#section"></a>
</h2>
<p>First if you have not done yet, install the required dependencies and
load the libraries:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Install Rtools if you haven’t done this yet, using the appropriate version (e.g.windows or macOS).</span></span>
<span><span class="co"># 2. Install the latest development version from GitHub using devtools</span></span>
<span><span class="co"># devtools::install_github("https://github.com/saezlab/MetaProViz")</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://saezlab.github.io/MetaProViz" class="external-link">MetaProViz</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># dependencies that need to be loaded:</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rlang.r-lib.org" class="external-link">rlang</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/sinhrks/ggfortify" class="external-link">ggfortify</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span></code></pre></div>
<br><br><div class="progress progress-striped active">
<pre><code>&lt;div class="progress-bar progress-bar-success" style="width: 100%"&gt;&lt;/div&gt;</code></pre>
</div>
</div>
<div class="section level2">
<h2 id="loading-the-example-data">1. Loading the example data<a class="anchor" aria-label="anchor" href="#loading-the-example-data"></a>
</h2>
<div class="progress progress-striped active">
<pre><code>&lt;div class="progress-bar progress-bar-success" style="width: 100%"&gt;&lt;/div&gt;</code></pre>
</div>
<p><br>
Here we choose an example datasets, which is publicly available on <a href="https://www.metabolomicsworkbench.org/data/DRCCMetadata.php?Mode=Project&amp;ProjectID=PR001418" class="external-link">metabolomics
workbench project PR001418</a> including metabolic profiles of human
renal epithelial cells HK2 and cell renal cell carcinoma (ccRCC) cell
lines cultured in Plasmax cell culture media <span class="citation">(Sciacovelli et al. 2022)</span>. Here we use the
integrated raw peak data as example data using the trivial metabolite
name in combination with the KEGG ID as the metabolite
identifiers.<br><br><span style="text-decoration:underline">As part of the
<strong>MetaProViz</strong> package you can load the example data into
your global environment using the function
<code>toy_data()</code></span>:<br>
Intracellular experiment <strong>(Intra)</strong><br>
The raw data are available via <a href="https://www.metabolomicsworkbench.org/data/DRCCMetadata.php?Mode=Study&amp;StudyID=ST002224&amp;StudyType=MS&amp;ResultType=1" class="external-link">metabolomics
workbench study ST002224</a> were intracellular metabolomics of HK2 and
ccRCC cell lines 786-O, 786-M1A and 786-M2A were performed.<br>
We can access the built-in dataset <code>intracell_raw</code>, which
includes columns with Sample information and columns with the measured
metabolite integrated peaks.<br></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">intracell_raw</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Intra</span> <span class="op">&lt;-</span> <span class="va">intracell_raw</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Code"</span><span class="op">)</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the DF <code>Intra</code> including columns with sample
information and metabolite ids with their measured values.
</caption>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Conditions
</th>
<th style="text-align:right;">
Analytical_Replicates
</th>
<th style="text-align:right;">
Biological_Replicates
</th>
<th style="text-align:right;">
valine-d8
</th>
<th style="text-align:right;">
ADP-ribose
</th>
<th style="text-align:right;">
citrulline
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
MS55_01
</td>
<td style="text-align:left;">
HK2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1910140239
</td>
<td style="text-align:right;">
2417484
</td>
<td style="text-align:right;">
514024322
</td>
</tr>
<tr>
<td style="text-align:left;">
MS55_02
</td>
<td style="text-align:left;">
HK2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
2030901280
</td>
<td style="text-align:right;">
2159520
</td>
<td style="text-align:right;">
507001076
</td>
</tr>
<tr>
<td style="text-align:left;">
MS55_03
</td>
<td style="text-align:left;">
HK2
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
2001950756
</td>
<td style="text-align:right;">
2427805
</td>
<td style="text-align:right;">
551503662
</td>
</tr>
<tr>
<td style="text-align:left;">
MS55_04
</td>
<td style="text-align:left;">
HK2
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1971520079
</td>
<td style="text-align:right;">
1988317
</td>
<td style="text-align:right;">
483751307
</td>
</tr>
<tr>
<td style="text-align:left;">
MS55_05
</td>
<td style="text-align:left;">
786-O
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
2150817213
</td>
<td style="text-align:right;">
1732016
</td>
<td style="text-align:right;">
272896668
</td>
</tr>
</tbody>
</table>
<br><br><div class="progress progress-striped active">
<pre><code>&lt;div class="progress-bar progress-bar-success" style="width: 550%"&gt;&lt;/div&gt;</code></pre>
</div>
</div>
<div class="section level2">
<h2 id="pre-processing">2. Pre-processing<a class="anchor" aria-label="anchor" href="#pre-processing"></a>
</h2>
<div class="progress progress-striped active">
<pre><code>&lt;div class="progress-bar progress-bar-success" style="width: 550%"&gt;&lt;/div&gt;</code></pre>
</div>
<p><strong>MetaProViz</strong> includes a pre-processing module with the
function <code><a href="../reference/processing.html">processing()</a></code> that has multiple parameters to
perform customize data processing.<br><code>Feature_Filtering</code> applies the 80%-filtering rule on the
metabolite features either on the whole dataset (=“Standard”) <span class="citation">(Bijlsma et al. 2006)</span> or per condition
(=“Modified”) <span class="citation">(Wei et al. 2018)</span>. This
means that metabolites are removed were more than 20% of the samples
(all or per condition) have no detection. With the parameter
<code>cutoff_featurefilt</code> we enable the adaptation of the
stringency of the filtering based on the experimental context. For
instance, patient tumour samples can contain many unknown subgroups due
to gender, age, stage etc., which leads to a metabolite being detected
in only 50% (or even less) of the tumour samples, hence in this context
it could be considered to change the <code>cutoff_featurefilt</code>
from the default (=0.8). If <code>featurefilt = "None"</code>, no
feature filtering is performed. In the context of
<code>featurefilt</code> it is also noteworthy that the function
<code><a href="../reference/pool_estimation.html">pool_estimation()</a></code> can be used to estimate the quality of
the metabolite detection and will return a list of metabolites that are
variable across the different pool measurements (pool = mixture of all
experimental samples measured several times during the LC-MS run) .
Variable metabolite in the pool sample should be removed from the
data.<br>
The parameter <code>tic</code> refers to total Ion Count (tic)
normalisation, which is often used with LC-MS derived metabolomics data.
If <code>tic = TRUE</code>, each feature (=metabolite) in a sample is
divided by the sum of all intensity value (= total number of ions) for
the sample and finally multiplied by a constant ( = the mean of all
samples total number of ions). Noteworthy, tic normalisation should not
be used with small number of features (= metabolites), since tic assumes
that on “average” the ion count of each sample is equal if there were no
instrument batch effects <span class="citation">(Wulff and Mitchell
2018)</span>.<br>
The parameter <code>mvi</code> refers to Missing Value Imputation (mvi)
and if <code>mvi = TRUE</code> half minimum (HM) missing value
imputation is performed per feature (= per metabolite). Here it is
important to mention that HM has been shown to perform well for missing
vales that are missing not at random (MNAR) <span class="citation">(Wei
et al. 2018)</span>.<br>
Lastly, the function <code><a href="../reference/processing.html">processing()</a></code> performs outlier
detection and adds a column “Outliers” into the DF, which can be used to
remove outliers. The parameter <code>hotellins_confidence</code> can be
used to choose the confidence interval that should be used for the
Hotellins T2 outlier test <span class="citation">(Hotelling
1931)</span>.<br><br>
If your data contain pool samples, you can do
<code><a href="../reference/pool_estimation.html">pool_estimation()</a></code> before applying the
<code><a href="../reference/processing.html">processing()</a></code> function. This is important, since one should
remove the features (=metabolites) that are too variable prior to
performing any data transformations such as tic as part of the
<code><a href="../reference/processing.html">processing()</a></code> function. If there is a high variability (high
CVs), you should consider to remove those features from the data. If you
have used internal standard in your experiment you should specifically
check their CV as this would indicate technical issues.You can find
details on this in the extended vignettes:<br>
- <a href="https://saezlab.github.io/MetaProViz/articles/standard-metabolomics.html" class="external-link">Standard
metabolomics data</a><br>
- <a href="https://saezlab.github.io/MetaProViz/articles/core-metabolomics.html" class="external-link">Consumption-Release
(CoRe) metabolomics data from cell culture media</a><br><br>
Now we will apply the <code><a href="../reference/processing.html">processing()</a></code> function to example data
and have a look at the output produced. You will notice that all the
chosen parameters and results are documented in messages. All the
results data tables, the Quality Control (QC) plots and outlier
detection plots are returned and can be easily viewed.<br></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">PreprocessingResults</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/processing.html">processing</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">Intra</span><span class="op">[</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">49</span><span class="op">:</span><span class="fl">58</span><span class="op">)</span> ,<span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span><span class="op">]</span>, <span class="co">#remove pool samples and columns with sample information</span></span>
<span>metadata_sample<span class="op">=</span><span class="va">Intra</span><span class="op">[</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">49</span><span class="op">:</span><span class="fl">58</span><span class="op">)</span> , <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span><span class="op">]</span>, <span class="co">#remove pool samples and columns with metabolite measurements</span></span>
<span>metadata_info <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>Conditions <span class="op">=</span> <span class="st">"Conditions"</span>,</span>
<span>Biological_Replicates <span class="op">=</span> <span class="st">"Biological_Replicates"</span><span class="op">)</span>,</span>
<span>featurefilt <span class="op">=</span> <span class="st">"Modified"</span>,</span>
<span>cutoff_featurefilt <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>tic <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>mvi <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>hotellins_confidence <span class="op">=</span> <span class="fl">0.99</span>,<span class="co"># We perform outlier testing using 0.99 confidence intervall</span></span>
<span>core <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>save_plot <span class="op">=</span> <span class="st">"svg"</span>,</span>
<span>save_table<span class="op">=</span> <span class="st">"csv"</span>,</span>
<span>print_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>path <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; feature_filtering: Here we apply the modified 80%-filtering rule that takes the class information (Column `Conditions`) into account, which additionally reduces the effect of missing values (REF: Yang et. al., (2015), doi: 10.3389/fmolb.2015.00004). Filtering value selected: 0.8</span></span>
<span><span class="co">#&gt; 3 metabolites where removed: AICAR, FAICAR, SAICAR</span></span>
<span><span class="co">#&gt; Missing Value Imputation: Missing value imputation is performed, as a complementary approach to address the missing value problem, where the missing values are imputing using the `half minimum value`. REF: Wei et. al., (2018), Reports, 8, 663, doi:https://doi.org/10.1038/s41598-017-19120-0</span></span>
<span><span class="co">#&gt; Warning: `aes_string()` was deprecated in ggplot2 3.0.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use tidy evaluation idioms with `aes()`.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> See also `vignette("ggplot2-in-packages")` for more information.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">MetaProViz</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue at <span style="color: #0000BB; font-style: italic;">&lt;https://github.com/saezlab/MetaProViz/issues&gt;</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; total Ion Count (tic) normalization: total Ion Count (tic) normalization is used to reduce the variation from non-biological sources, while maintaining the biological variation. REF: Wulff et. al., (2018), Advances in Bioscience and Biotechnology, 9, 339-351, doi:https://doi.org/10.4236/abb.2018.98022</span></span>
<span><span class="co">#&gt; Outlier detection: Identification of outlier samples is performed using Hotellin's T2 test to define sample outliers in a mathematical way (Confidence = 0.99 ~ p.val &lt; 0.01) (REF: Hotelling, H. (1931), Annals of Mathematical Statistics. 2 (3), 360-378, doi:https://doi.org/10.1214/aoms/1177732979). hotellins_confidence value selected: 0.99</span></span>
<span><span class="co">#&gt; There are possible outlier samples in the data</span></span>
<span><span class="co">#&gt; Filtering round  1  Outlier Samples:  MS55_29</span></span></code></pre></div>
<p><img src="quick-start_files/figure-html/processing-1.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"><img src="quick-start_files/figure-html/processing-2.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Warning: ggrepel: 26 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre>
<p><img src="quick-start_files/figure-html/processing-3.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Warning: ggrepel: 25 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span>
<span><span class="co">#&gt; Warning in geom_bar(stat = "identity", fill = barfill, color = barcolor, :</span></span>
<span><span class="co">#&gt; Ignoring empty aesthetic: `width`.</span></span></code></pre>
<p><img src="quick-start_files/figure-html/processing-4.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Warning in geom_bar(stat = "identity", fill = barfill, color = barcolor, :</span></span>
<span><span class="co">#&gt; Ignoring empty aesthetic: `width`.</span></span></code></pre>
<p><img src="quick-start_files/figure-html/processing-5.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Warning: ggrepel: 16 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre>
<p><img src="quick-start_files/figure-html/processing-6.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Warning: ggrepel: 12 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></span>
<span><span class="co">#&gt; Ignoring empty aesthetic: `width`.</span></span></code></pre>
<p><img src="quick-start_files/figure-html/processing-7.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Warning in geom_bar(stat = "identity", fill = barfill, color = barcolor, :</span></span>
<span><span class="co">#&gt; Ignoring empty aesthetic: `width`.</span></span></code></pre>
<p><img src="quick-start_files/figure-html/processing-8.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Warning: ggrepel: 26 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre>
<p><img src="quick-start_files/figure-html/processing-9.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Warning: ggrepel: 24 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 26 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre>
<p><img src="quick-start_files/figure-html/processing-10.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Warning: ggrepel: 24 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre>
<p><img src="quick-start_files/figure-html/processing-11.png" class="r-plt" width="576" style="display: block; margin: auto auto auto 0;"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span></span>
<span><span class="co"># This is the results table:</span></span>
<span><span class="va">Intra_Preprocessed</span> <span class="op">&lt;-</span> <span class="va">PreprocessingResults</span><span class="op">[[</span><span class="st">"DF"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"Preprocessing_output"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p><br><br><br></p>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the pre-processing results, which has an additional column
<code>Outlier</code> including the results of Hotellins T2.
</caption>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Conditions
</th>
<th style="text-align:right;">
Analytical_Replicates
</th>
<th style="text-align:right;">
Biological_Replicates
</th>
<th style="text-align:left;">
Outliers
</th>
<th style="text-align:right;">
valine-d8
</th>
<th style="text-align:right;">
hippuric acid-d5
</th>
<th style="text-align:right;">
2/3-phosphoglycerate
</th>
<th style="text-align:right;">
2-aminoadipic acid
</th>
<th style="text-align:right;">
2-hydroxyglutarate
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
MS55_29
</td>
<td style="text-align:left;">
786-M2A
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
Outlier_filtering_round_1
</td>
<td style="text-align:right;">
2387588900
</td>
<td style="text-align:right;">
4569088590
</td>
<td style="text-align:right;">
40184147
</td>
<td style="text-align:right;">
6064712
</td>
<td style="text-align:right;">
447702444
</td>
</tr>
<tr>
<td style="text-align:left;">
MS55_30
</td>
<td style="text-align:left;">
786-M2A
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
2129509827
</td>
<td style="text-align:right;">
3909434732
</td>
<td style="text-align:right;">
40901362
</td>
<td style="text-align:right;">
5928248
</td>
<td style="text-align:right;">
438592007
</td>
</tr>
<tr>
<td style="text-align:left;">
MS55_31
</td>
<td style="text-align:left;">
786-M2A
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
2008257641
</td>
<td style="text-align:right;">
3820133317
</td>
<td style="text-align:right;">
45656317
</td>
<td style="text-align:right;">
6122422
</td>
<td style="text-align:right;">
423960574
</td>
</tr>
<tr>
<td style="text-align:left;">
MS55_32
</td>
<td style="text-align:left;">
786-M2A
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
2023353119
</td>
<td style="text-align:right;">
3808913048
</td>
<td style="text-align:right;">
46166031
</td>
<td style="text-align:right;">
6633984
</td>
<td style="text-align:right;">
434158266
</td>
</tr>
</tbody>
</table>
<p><br>
In the output table you can now see the column “Outliers” and for the
Condition 786-M2A, we can see that based on Hotellin’s T2 test, one
sample was detected as an outlier in the first round of filtering.<br>
As part of the <code><a href="../reference/processing.html">processing()</a></code> function several plots are
generated and saved. Additionally, the ggplots are returned into the
list to enable further modifiaction using the ggplot syntax. These plots
include plots showing the outliers for each filtering round and other QC
plots.<br><br>
Before we proceed, we will remove the outlier:<br></p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Intra_Preprocessed</span> <span class="op">&lt;-</span> <span class="va">Intra_Preprocessed</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Outliers</span><span class="op">==</span><span class="st">"no"</span><span class="op">)</span><span class="co">#remove MS55_29</span></span></code></pre></div>
<p><br>
As you may have noticed, in this example dataset we have several
biological replicates that were injected (=measured) several times,
which can be termed as analytical replicates. The
<strong>MetaProViz</strong> pre-processing module includes the function
<code><a href="../reference/replicate_sum.html">replicate_sum()</a></code>, which will do this task and save the
results:<br></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Intra_Preprocessed</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/replicate_sum.html">replicate_sum</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">Intra_Preprocessed</span><span class="op">[</span>,<span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">]</span>,</span>
<span>metadata_sample<span class="op">=</span><span class="va">Intra_Preprocessed</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">]</span>,</span>
<span>metadata_info <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>Conditions<span class="op">=</span><span class="st">"Conditions"</span>, Biological_Replicates<span class="op">=</span><span class="st">"Biological_Replicates"</span>, Analytical_Replicates<span class="op">=</span><span class="st">"Analytical_Replicates"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><br><br>
In case you have performed a Consumption-Release (core) metabolomics
experiment, which usually refers to a cell culture experiment where
metabolomics is performed on the cell culture media, you will also need
to set the parameter <code>core=TRUE</code> in the
<code><a href="../reference/processing.html">processing()</a></code> function. Now additional data processing steps
are applied:<br>
1. Blank sample: This refers to media samples where no cells have been
cultured in, which will be used as blank. In detail, the mean of the
blank sample of a feature (= metabolite) will be substracted from the
values measured in each sample for the same feature. In the column
“Condition” of the Experimental_design DF, you will need to label your
blank samples with “blank”.<br>
2. Growth factor or growth rate: This refers to the different conditions
and is either based on cell count or protein quantification at the start
of the experiment (t0) and at the end of the experiment (t1) resulting
in the growth factor (t1/t0). Otherwise, one can experimentally estimate
the growth rate of each condition. Ultimately, this measure is used to
normalize the data, since the amount of growth will impact the
consumption and release of metabolites from the media and hence we need
to account for this. If you do not have this information, this will be
set to 1, yet be aware that this may affect the results.<br>
For details see extensive vignette <a href="https://saezlab.github.io/MetaProViz/articles/core-metabolomics.html" class="external-link">Consumption-Release
(CoRe) metabolomics data from cell culture media</a>.</p>
<div class="section level3">
<h3 id="pca-plot-and-heatmap">PCA plot and Heatmap<a class="anchor" aria-label="anchor" href="#pca-plot-and-heatmap"></a>
</h3>
<p>Using the processed data, we can now use the
<strong>MetaProViz</strong> visualization module and generate some
overview Heatmaps <code><a href="../reference/viz_heatmap.html">viz_heatmap()</a></code> or PCA plots
<code><a href="../reference/viz_pca.html">viz_pca()</a></code>. 1. PCA<br>
Principal component analysis (PCA) is a dimensionality reduction method
that reduces all the measured features (=metabolites) of one sample into
a few features in the different principal components, whereby each
principal component can explain a certain percentage of the variance
between the different samples. Hence, this enables interpretation of
sample clustering based on the measured features (=metabolites).<br>
We can interactively choose shape and color using the additional
information of interest from our Metadata. Especially for complex data,
such as patient data, it can be valuable to use different demographics
(e.g. age, gender, medication,…) for this.<br>
Here the different cell lines are either control or cancerous, so we can
display this as colour. The cancerous cell lines can further be divided
into metastatic or primary, whcih we display by shape. This shows us
that this is separated on the y-axis and accounts for 30%of the
variance.Here is becomes apparent that the cell status is responsible
for 64% of the variance (x-axis).<br></p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Create the metadata file:</span></span>
<span><span class="va">MetaData_Sample</span> <span class="op">&lt;-</span> <span class="va">Intra_Preprocessed</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span><span class="op">]</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Celltype <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/case_when.html" class="external-link">case_when</a></span><span class="op">(</span><span class="va">Conditions</span><span class="op">==</span><span class="st">"HK2"</span> <span class="op">~</span> <span class="st">'Healthy'</span>,</span>
<span><span class="va">Conditions</span><span class="op">==</span><span class="st">"786-O"</span> <span class="op">~</span> <span class="st">'Primary Tumour'</span>,</span>
<span><span class="cn">TRUE</span> <span class="op">~</span> <span class="st">'Metastatic Tumour'</span><span class="op">)</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Status <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/case_when.html" class="external-link">case_when</a></span><span class="op">(</span><span class="va">Conditions</span><span class="op">==</span><span class="st">"HK2"</span> <span class="op">~</span> <span class="st">'Healthy'</span>,</span>
<span><span class="cn">TRUE</span> <span class="op">~</span> <span class="st">'Cancer'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#Make PCA plot</span></span>
<span><span class="fu"><a href="../reference/viz_pca.html">viz_pca</a></span><span class="op">(</span>metadata_info<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"Celltype"</span>, shape<span class="op">=</span><span class="st">"Status"</span><span class="op">)</span>,</span>
<span>                    metadata_sample<span class="op">=</span> <span class="va">MetaData_Sample</span>,</span>
<span>                    data<span class="op">=</span> <span class="va">Intra_Preprocessed</span><span class="op">[</span>,<span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span><span class="op">]</span>,</span>
<span>                    plot_name <span class="op">=</span> <span class="st">"Cell type"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="quick-start_files/figure-html/pca_plot-1.png" class="r-plt" alt="Figure: Do the samples cluster for the Cell type?" width="576"><p class="caption">
Figure: Do the samples cluster for the Cell type?
</p>
</div>
<p><br>
Similarly, we can use the data and sample information to make a
heatmap:<br></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/viz_heatmap.html">viz_heatmap</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">Intra_Preprocessed</span><span class="op">[</span>,<span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">]</span>,</span>
<span>                        metadata_sample <span class="op">=</span> <span class="va">MetaData_Sample</span>,</span>
<span>                        metadata_info <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>color_Sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"Conditions"</span>,<span class="st">"Biological_Replicates"</span>, <span class="st">"Celltype"</span>, <span class="st">"Status"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="quick-start_files/figure-html/heatmap_plot-1.png" class="r-plt" alt="Colour for sample metadata." width="700"><p class="caption">
Colour for sample metadata.
</p>
</div>
<div class="progress progress-striped active">
<pre><code>&lt;div class="progress-bar progress-bar-success" style="width: 550%"&gt;&lt;/div&gt;</code></pre>
</div>
</div>
</div>
<div class="section level2">
<h2 id="differential-metabolite-analysis">3. Differential Metabolite Analysis<a class="anchor" aria-label="anchor" href="#differential-metabolite-analysis"></a>
</h2>
<div class="progress progress-striped active">
<pre><code>&lt;div class="progress-bar progress-bar-success" style="width: 550%"&gt;&lt;/div&gt;</code></pre>
</div>
<p>Differential Metabolite Analysis (<code>dma</code>) is used to
compare two conditions (e.g. Tumour versus Healthy) by calculating the
Log2FC, p-value, adjusted p-value and t-value. With the different
parameters <code>pval</code> and <code>padj</code> one can choose the
statistical tests such as t.test, wilcoxon test, limma, annova, kruskal
walles, etc. (see function reference for more information).<br>
As input one can use the processed data we have generated using the
<code>processing</code> module, but here one can of course use any DF
including metabolite values, even though we recommend to normalize the
data and remove outliers prior to dma. Moreover, we require information
which condition a sample corresponds to.<br><br>
By defining the numerator and denominator as part of the
<code>metadata_info</code> parameter, it is defined which comparisons
are performed:<br>
1. <strong>one_vs_one</strong> (single comparison):
numerator=“Condition1”, denominator =“Condition2”<br>
2. <strong>all_vs_one</strong> (multiple comparison): numerator=NULL,
denominator =“Condition”<br>
3. <strong>all_vs_all</strong> (multiple comparison): numerator=NULL,
denominator =NULL (=default)<br><br>
Noteworthy, if you have not performed missing value imputation and hence
your data includes NAs or 0 values for some features, this is how we
deal with this in the <code><a href="../reference/dma.html">dma()</a></code> function:<br>
1. If you use the parameter <code>pval="lmFit"</code>, limma is
performed. Limma does a baesian fit of the data and substracts
Mean(Condition1 fit) - Mean(Condition2 fit). As such, unless all values
of a feature are NA, Limma can deal with NAs. 2. Standard Log2FC:
log2(Mean(Condition1)) - log2(Mean(Condition2)) a. If all values of the
replicates of one condition are NA/0 for a feature (=metabolite):
Log2FC= Inf/-Inf and the statistics will be NA<br>
b. If some values of the replicates of one condition are NA/0 for a
feature (=metabolite): Log2FC= positive or negative value, but the
statistics will be NA<br><br>
It is important to mention that in case of <code>pval="lmFit"</code>, we
perform log2 transformation of the data as prior to running limma to
enable the calculation of the log2FC, hence do not provide log2
transformed data.<br><br>
Here, the example data we have four different cell lines, healthy (HK2)
and cancer (ccRCC: 786-M1A, 786-M2A and 786-O), hence we can perform
multiple different comparisons. For simplicity, we will compare 786-M1A
versus HK2. The results can be automatically saved and all the results
are returned in a list with the different data frames. If parameter
plot=TRUE, an overview Volcano plot is generated and saved.<br></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Perform multiple comparison All_vs_One using annova:</span></span>
<span><span class="va">DMA_Res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/dma.html">dma</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">Intra_Preprocessed</span><span class="op">[</span>,<span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span><span class="op">]</span>, <span class="co">#we need to remove columns that do not include metabolite measurements</span></span>
<span>metadata_sample<span class="op">=</span><span class="va">Intra_Preprocessed</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span><span class="op">]</span>,<span class="co">#only maintain the information about condition and replicates</span></span>
<span>metadata_info <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>Conditions<span class="op">=</span><span class="st">"Conditions"</span>, Numerator<span class="op">=</span><span class="st">"786-M1A"</span> , Denominator <span class="op">=</span> <span class="st">"HK2"</span><span class="op">)</span>,<span class="co"># we compare 786-M1A_vs_HK2</span></span>
<span>pval <span class="op">=</span><span class="st">"t.test"</span>,</span>
<span>padj<span class="op">=</span><span class="st">"fdr"</span><span class="op">)</span></span>
<span><span class="co">#&gt; There are no NA/0 values</span></span>
<span><span class="co">#&gt; Warning: Computation failed in `stat_bin()`.</span></span>
<span><span class="co">#&gt; Caused by error in `bin_breaks_width()`:</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> The number of histogram bins must be less than 1,000,000.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Did you make `binwidth` too small?</span></span>
<span><span class="co">#&gt; Warning: Removed 142 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_bin()`).</span></span>
<span><span class="co">#&gt; Warning: Computation failed in `stat_bin()`.</span></span>
<span><span class="co">#&gt; Caused by error in `bin_breaks_width()`:</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> The number of histogram bins must be less than 1,000,000.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Did you make `binwidth` too small?</span></span>
<span><span class="co">#&gt; Warning: Removed 142 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_density()`).</span></span>
<span><span class="co">#&gt; Warning: Computation failed in `stat_bin()`.</span></span>
<span><span class="co">#&gt; Caused by error in `bin_breaks_width()`:</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> The number of histogram bins must be less than 1,000,000.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Did you make `binwidth` too small?</span></span>
<span><span class="co">#&gt; Warning: Removed 137 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_bin()`).</span></span>
<span><span class="co">#&gt; Warning: Computation failed in `stat_bin()`.</span></span>
<span><span class="co">#&gt; Caused by error in `bin_breaks_width()`:</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> The number of histogram bins must be less than 1,000,000.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Did you make `binwidth` too small?</span></span>
<span><span class="co">#&gt; Warning: Removed 137 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_density()`).</span></span>
<span><span class="co">#&gt; Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use `linewidth` instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">EnhancedVolcano</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue to the authors.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; Warning: The `size` argument of `element_line()` is deprecated as of ggplot2 3.4.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use the `linewidth` argument instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">EnhancedVolcano</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue to the authors.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; Warning: Removed 142 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_bin()`).</span></span>
<span><span class="co">#&gt; Warning: Computation failed in `stat_bin()`.</span></span>
<span><span class="co">#&gt; Caused by error in `bin_breaks_width()`:</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> The number of histogram bins must be less than 1,000,000.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Did you make `binwidth` too small?</span></span>
<span><span class="co">#&gt; Warning: Removed 142 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_density()`).</span></span>
<span><span class="co">#&gt; Warning: Removed 142 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_bin()`).</span></span>
<span><span class="co">#&gt; Warning: Computation failed in `stat_bin()`.</span></span>
<span><span class="co">#&gt; Caused by error in `bin_breaks_width()`:</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> The number of histogram bins must be less than 1,000,000.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Did you make `binwidth` too small?</span></span>
<span><span class="co">#&gt; Warning: Removed 142 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_density()`).</span></span>
<span><span class="co">#&gt; Warning: Removed 137 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_bin()`).</span></span>
<span><span class="co">#&gt; Warning: Computation failed in `stat_bin()`.</span></span>
<span><span class="co">#&gt; Caused by error in `bin_breaks_width()`:</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> The number of histogram bins must be less than 1,000,000.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Did you make `binwidth` too small?</span></span>
<span><span class="co">#&gt; Warning: Removed 137 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_density()`).</span></span></code></pre></div>
<p><img src="quick-start_files/figure-html/dma-1.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"></p>
<pre><code><span><span class="co">#&gt; Warning: Removed 137 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_bin()`).</span></span>
<span><span class="co">#&gt; Warning: Computation failed in `stat_bin()`.</span></span>
<span><span class="co">#&gt; Caused by error in `bin_breaks_width()`:</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> The number of histogram bins must be less than 1,000,000.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Did you make `binwidth` too small?</span></span>
<span><span class="co">#&gt; Warning: Removed 137 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_density()`).</span></span></code></pre>
<p><img src="quick-start_files/figure-html/dma-2.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"><img src="quick-start_files/figure-html/dma-3.png" class="r-plt" width="672" style="display: block; margin: auto auto auto 0;"></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Inspect the dma results tables:</span></span>
<span><span class="va">DMA_786M1A_vs_HK2</span> <span class="op">&lt;-</span> <span class="va">DMA_Res</span><span class="op">[[</span><span class="st">"dma"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"786-M1A _vs_ HK2"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p><br><br><br></p>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<ol start="2" style="list-style-type: decimal"><li>Preview of the dma results for the comparison of 786-M1A versus HK2
cells.
</li></ol>
</caption>
<thead><tr>
<th style="text-align:left;">
Metabolite
</th>
<th style="text-align:right;">
Log2FC
</th>
<th style="text-align:right;">
786-M1A_1
</th>
<th style="text-align:right;">
786-M1A_2
</th>
<th style="text-align:right;">
786-M1A_3
</th>
<th style="text-align:right;">
HK2_1
</th>
<th style="text-align:right;">
HK2_2
</th>
<th style="text-align:right;">
HK2_3
</th>
<th style="text-align:right;">
p.val
</th>
<th style="text-align:right;">
p.adj
</th>
<th style="text-align:left;">
t.val
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
isovalerylcarnitine
</td>
<td style="text-align:right;">
0.9170474
</td>
<td style="text-align:right;">
29656656
</td>
<td style="text-align:right;">
30619594
</td>
<td style="text-align:right;">
32262097
</td>
<td style="text-align:right;">
19753215
</td>
<td style="text-align:right;">
15537391
</td>
<td style="text-align:right;">
13716941
</td>
<td style="text-align:right;">
0.0070624
</td>
<td style="text-align:right;">
0.0234206
</td>
<td style="text-align:left;">
-7.4689200974348
</td>
</tr>
<tr>
<td style="text-align:left;">
lactate
</td>
<td style="text-align:right;">
0.3015843
</td>
<td style="text-align:right;">
721790170
</td>
<td style="text-align:right;">
654376727
</td>
<td style="text-align:right;">
761914317
</td>
<td style="text-align:right;">
565048879
</td>
<td style="text-align:right;">
499136450
</td>
<td style="text-align:right;">
670570164
</td>
<td style="text-align:right;">
0.0973061
</td>
<td style="text-align:right;">
0.1662329
</td>
<td style="text-align:left;">
-2.27995828643078
</td>
</tr>
<tr>
<td style="text-align:left;">
lysine
</td>
<td style="text-align:right;">
0.4006104
</td>
<td style="text-align:right;">
39846551
</td>
<td style="text-align:right;">
37827659
</td>
<td style="text-align:right;">
40500396
</td>
<td style="text-align:right;">
27919226
</td>
<td style="text-align:right;">
25870671
</td>
<td style="text-align:right;">
35731826
</td>
<td style="text-align:right;">
0.0775608
</td>
<td style="text-align:right;">
0.1388337
</td>
<td style="text-align:left;">
-3.07082976561226
</td>
</tr>
<tr>
<td style="text-align:left;">
malate
</td>
<td style="text-align:right;">
0.0981352
</td>
<td style="text-align:right;">
2244208988
</td>
<td style="text-align:right;">
2276823115
</td>
<td style="text-align:right;">
2058245048
</td>
<td style="text-align:right;">
1940397414
</td>
<td style="text-align:right;">
2064435997
</td>
<td style="text-align:right;">
2141789027
</td>
<td style="text-align:right;">
0.1853322
</td>
<td style="text-align:right;">
0.2764539
</td>
<td style="text-align:left;">
-1.6049019168411
</td>
</tr>
<tr>
<td style="text-align:left;">
malonate
</td>
<td style="text-align:right;">
0.1701143
</td>
<td style="text-align:right;">
7457170
</td>
<td style="text-align:right;">
8327513
</td>
<td style="text-align:right;">
6230908
</td>
<td style="text-align:right;">
5918003
</td>
<td style="text-align:right;">
6198577
</td>
<td style="text-align:right;">
7450266
</td>
<td style="text-align:right;">
0.3518985
</td>
<td style="text-align:right;">
0.4700734
</td>
<td style="text-align:left;">
-1.06115449290904
</td>
</tr>
</tbody>
</table>
<p><br>
In case you have performed a Consumption-Release (core) metabolomics
experiment, which usually refers to a cell culture experiment where
metabolomics is performed on the cell culture media, you will also need
to set the parameter <code>core=TRUE</code> in the <code><a href="../reference/dma.html">dma()</a></code>
function. In a core experiment the normalized metabolite values can be
either a negative value, if the metabolite has been consumed from the
media, or a positive value, if the metabolite has been released from the
cell into the culture media. Since we can not calculate a Log2FC using
negative values, we calculate the absolute difference between the mean
of Condition 1 versus the mean of Condition 2. The absolute difference
is log2 transformed in order to make the values comparable between the
different metabolites, resulting in the Log2Dist. The result doesn’t
consider whether one product is larger than the other; it only looks at
the magnitude of their difference. to reflect the direction of change
between the two conditions we multiply with -1 if C1 &lt; C2. By setting
the paramteter core = TRUE, instead of calclulating the Log2FC, the Log2
Distance is calculated. For details see extensive vignette <a href="https://saezlab.github.io/MetaProViz/articles/core-metabolomics.html#dma" class="external-link">Consumption-Release
(CoRe) metabolomics data from cell culture media</a>.</p>
<div class="section level3">
<h3 id="volcano-plots">Volcano Plots<a class="anchor" aria-label="anchor" href="#volcano-plots"></a>
</h3>
<p>In general,we have three different <code>Plot_Settings</code>, which
will also be used for other plot types such as lollipop graphs.<br><code>1.</code> <code>"Standard"</code> is the standard version of the
plot, with one dataset being plotted.<br><code>2.</code> <code>"Conditions"</code> here two or more datasets will
be plotted together.<br><code>3.</code> <code>"PEA"</code> stands for Pathway Enrichment
Analysis, and is used if the results of an GSE analysis should be
plotted as here the figure legends will be adapted.<br><br>
Using the dma results, we can now use the <strong>MetaProViz</strong>
visualization module and generate further customized Volcano plots
<code><a href="../reference/viz_volcano.html">viz_volcano()</a></code>:<br>
- To plot the metabolite names you can change the paramter
<code>select_label</code> from its default
(<code>select_label=""</code>) to NULL and the metabolite names will be
plotted randomly or you can also pass a vector with Metabolite names
that should be labeled.<br>
- By providing additional feature or sample information, you can color
code and/or shape the dots on the volcano plot.<br>
- Based on feature information (i.e. pathways), you can also create
individual plots, one for each pathway.<br>
For detailed exaples check out
<code>3. Run MetaProViz Visualisation</code> in the vignettes <a href="https://saezlab.github.io/MetaProViz/articles/standard-metabolomics.html#run-metaproviz-visualisation" class="external-link">Standard
metabolomics data</a> or <a href="https://saezlab.github.io/MetaProViz/articles/core-metabolomics.html#run-metaproviz-visualisation" class="external-link">Consumption-Release
(CoRe) metabolomics data from cell culture media</a><br></p>
<div class="progress progress-striped active">
<pre><code>&lt;div class="progress-bar progress-bar-success" style="width: 550%"&gt;&lt;/div&gt;</code></pre>
</div>
</div>
</div>
<div class="section level2">
<h2 id="enrichment-analysis-and-prior-knowledge">3. Enrichment Analysis and Prior knowledge<a class="anchor" aria-label="anchor" href="#enrichment-analysis-and-prior-knowledge"></a>
</h2>
<div class="progress progress-striped active">
<pre><code>&lt;div class="progress-bar progress-bar-success" style="width: 550%"&gt;&lt;/div&gt;</code></pre>
</div>
<p>Over Representation Analysis (ORA) is a enrichment method that
determines if a set of features (i.e. metabolic pathways) are
over-represented in the selection of features (=metabolites) from the
data in comparison to all measured features (metabolites) using the
Fishers exact test. The selection of metabolites are usually the most
altered metabolites in the data, which can be selected by the top and
bottom t-values. Before we can perform ORA on the dma results, we have
to ensure that the metabolite names match with the metabolite IDs of the
prior knowledge (PK).<br></p>
<div class="section level3">
<h3 id="match-ids-with-pk">Match IDs with PK<a class="anchor" aria-label="anchor" href="#match-ids-with-pk"></a>
</h3>
<p>As part of the <strong>MetaProViz</strong> package you can access
metabolite prior knowledge with the collection of metabolite sets
MetSigDB (Metabolite signature database) for pathway enrichment
analysis, compound class enrichment analysis, and by using specific PK
databases, it can be used to study the connection of metabolites and
receptors or transporters. In metabolite PK, the many different PK
databases and resources pose issues like metabolite identifiers
(e.g. KEGG, HMDB, PubChem, etc.) are not standardized across databases,
and the same metabolite can have multiple identifiers in different
databases. This is known as the many-to-many mapping problem. If you
want to know more on how to translate ids, quantify the mapping of your
data to the prior knwoeldge resource, increase the mapping, etc. have a
look at our dedicated vignette, <a href="https://saezlab.github.io/MetaProViz/articles/prior-knowledge.html" class="external-link">Prior
Knowledge Access &amp; Integration</a>.<br><br>
Here we will use the KEGG pathways <span class="citation">(Kanehisa and
Goto 2000)</span>, hence we have to ensure that the metabolite names
match with the KEGG IDs or KEGG trivial names.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#--------Add metabolite IDs to our example data:</span></span>
<span><span class="co"># 1. Load Feature metainformation of our example data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">cellular_meta</span><span class="op">)</span></span>
<span></span>
<span><span class="va">MappingInfo</span> <span class="op">&lt;-</span> <span class="va">cellular_meta</span></span>
<span></span>
<span><span class="co"># 2. Merge with our differential results (FYI: you can also do this automatically as part of the dma function using the parameter metadata_feature)</span></span>
<span><span class="va">ORA_Input</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">DMA_786M1A_vs_HK2</span>,</span>
<span><span class="va">MappingInfo</span>,</span>
<span>by<span class="op">=</span> <span class="st">"Metabolite"</span>,</span>
<span>all.x<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">KEGGCompound</span><span class="op">)</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="co">#remove features without KEGG ID</span></span>
<span><span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"KEGGCompound"</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">Metabolite</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#--------Load KEGG pathways:</span></span>
<span><span class="va">KEGG_Pathways</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/metsigdb_kegg.html">metsigdb_kegg</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="run-ora">Run ORA<a class="anchor" aria-label="anchor" href="#run-ora"></a>
</h3>
<p>In general, the <code>input_pathway</code> requirements are column
“term”, “Metabolite” and “Description”, and the <code>data</code>
requirements are column “t.val” and column “Metabolite”.<br></p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Perform ORA</span></span>
<span><span class="va">DM_ORA_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/standard_ora.html">standard_ora</a></span><span class="op">(</span>data<span class="op">=</span> <span class="va">ORA_Input</span> , <span class="co">#Input data requirements: column `t.val` and column `Metabolite`</span></span>
<span>metadata_info<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>pvalColumn<span class="op">=</span><span class="st">"p.adj"</span>, percentageColumn<span class="op">=</span><span class="st">"t.val"</span>, PathwayTerm<span class="op">=</span> <span class="st">"term"</span>, PathwayFeature<span class="op">=</span> <span class="st">"Metabolite"</span><span class="op">)</span>,</span>
<span>input_pathway<span class="op">=</span><span class="va">KEGG_Pathways</span>,<span class="co">#Pathway file requirements: column `term`, `Metabolite` and `Description`. Above we loaded the Kegg_Pathways using Load_KEGG()</span></span>
<span>pathway_name<span class="op">=</span><span class="st">"KEGG"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Lets check how the results look like:</span></span>
<span><span class="va">DM_ORA_786M1A_vs_HK2</span> <span class="op">&lt;-</span> <span class="va">DM_ORA_res</span><span class="op">[[</span><span class="st">"ClusterGosummary"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Preview of the ORA results for the comparison of 786-M1A versus HK2
cells.
</caption>
<thead><tr>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
RichFactor
</th>
<th style="text-align:right;">
FoldEnrichment
</th>
<th style="text-align:right;">
zScore
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
Metabolites_in_pathway
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
Metabolites_in_Pathway
</th>
<th style="text-align:right;">
percentage_of_Pathway_detected
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
2/14
</td>
<td style="text-align:left;">
18/130
</td>
<td style="text-align:right;">
0.1111111
</td>
<td style="text-align:right;">
1.0317460
</td>
<td style="text-align:right;">
0.0502165
</td>
<td style="text-align:right;">
0.6103571
</td>
<td style="text-align:right;">
0.9360814
</td>
<td style="text-align:right;">
0.9360814
</td>
<td style="text-align:left;">
L-Alanine/L-Tryptophan
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
144
</td>
<td style="text-align:right;">
1.39
</td>
</tr>
<tr>
<td style="text-align:left;">
6/14
</td>
<td style="text-align:left;">
33/130
</td>
<td style="text-align:right;">
0.1818182
</td>
<td style="text-align:right;">
1.6883117
</td>
<td style="text-align:right;">
1.5841129
</td>
<td style="text-align:right;">
0.1058932
</td>
<td style="text-align:right;">
0.9360814
</td>
<td style="text-align:right;">
0.9360814
</td>
<td style="text-align:left;">
Glutathione/Hydroxyproline/L-Alanine/L-Threonine/sn-Glycerol
3-phosphate/Thiamine
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
129
</td>
<td style="text-align:right;">
4.65
</td>
</tr>
<tr>
<td style="text-align:left;">
1/14
</td>
<td style="text-align:left;">
14/130
</td>
<td style="text-align:right;">
0.0714286
</td>
<td style="text-align:right;">
0.6632653
</td>
<td style="text-align:right;">
-0.4615861
</td>
<td style="text-align:right;">
0.8148272
</td>
<td style="text-align:right;">
0.9360814
</td>
<td style="text-align:right;">
0.9360814
</td>
<td style="text-align:left;">
L-Alanine
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
27
</td>
<td style="text-align:right;">
3.70
</td>
</tr>
<tr>
<td style="text-align:left;">
3/14
</td>
<td style="text-align:left;">
19/130
</td>
<td style="text-align:right;">
0.1578947
</td>
<td style="text-align:right;">
1.4661654
</td>
<td style="text-align:right;">
0.7610010
</td>
<td style="text-align:right;">
0.3343601
</td>
<td style="text-align:right;">
0.9360814
</td>
<td style="text-align:right;">
0.9360814
</td>
<td style="text-align:left;">
L-Alanine/L-Threonine/L-Tryptophan
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
52
</td>
<td style="text-align:right;">
5.77
</td>
</tr>
<tr>
<td style="text-align:left;">
1/14
</td>
<td style="text-align:left;">
11/130
</td>
<td style="text-align:right;">
0.0909091
</td>
<td style="text-align:right;">
0.8441558
</td>
<td style="text-align:right;">
-0.1869577
</td>
<td style="text-align:right;">
0.7295219
</td>
<td style="text-align:right;">
0.9360814
</td>
<td style="text-align:right;">
0.9360814
</td>
<td style="text-align:left;">
Hydroxyproline
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
70
</td>
<td style="text-align:right;">
1.43
</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="volcano-plot">Volcano plot<a class="anchor" aria-label="anchor" href="#volcano-plot"></a>
</h3>
<p>If you have performed Pathway Enrichment Analysis (PEA) such as ORA
or GSEA, we can also plot the results and add the information into the
Figure legends.<br>
For this we need to prepare the correct input data including the
pathways used to run the pathway analysis, the differential metabolite
data used as input for the pathway analysis and the results of the
pathway analysis:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Here we select only a few pathways to make only the most important plots:</span></span>
<span><span class="va">InputPEA2</span> <span class="op">&lt;-</span> <span class="va">DM_ORA_786M1A_vs_HK2</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">GeneRatio</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">pvalue</span> <span class="op">&lt;=</span> <span class="fl">0.1</span><span class="op">)</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span><span class="st">"term"</span><span class="op">=</span><span class="st">"ID"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/viz_volcano.html">viz_volcano</a></span><span class="op">(</span>plot_types<span class="op">=</span><span class="st">"PEA"</span>,</span>
<span>                        metadata_info<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>PEA_Pathway<span class="op">=</span><span class="st">"term"</span>,<span class="co"># Needs to be the same in both, metadata_feature and data2.</span></span>
<span>                        PEA_stat<span class="op">=</span><span class="st">"pvalue"</span>,<span class="co">#Column data2</span></span>
<span>                        PEA_score<span class="op">=</span><span class="st">"GeneRatio"</span>,<span class="co">#Column data2</span></span>
<span>                        PEA_Feature<span class="op">=</span><span class="st">"Metabolite"</span><span class="op">)</span>,<span class="co"># Column metadata_feature (needs to be the same as row names in data)</span></span>
<span>                        metadata_feature<span class="op">=</span> <span class="va">KEGG_Pathways</span>,<span class="co">#Must be the pathways used for pathway analysis</span></span>
<span>                        data<span class="op">=</span> <span class="va">ORA_Input</span>, <span class="co">#Must be the data you have used as an input for the pathway analysis</span></span>
<span>                        data2<span class="op">=</span> <span class="va">InputPEA2</span>, <span class="co">#Must be the results of the pathway analysis</span></span>
<span>                        plot_name<span class="op">=</span> <span class="st">"KEGG"</span>,</span>
<span>                        select_label <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<p><img src="quick-start_files/figure-html/volcano_pea-1.png" class="r-plt" width="700"><img src="quick-start_files/figure-html/volcano_pea-2.png" class="r-plt" width="700"></p>
<div class="progress progress-striped active">
<pre><code>&lt;div class="progress-bar progress-bar-success" style="width: 550%"&gt;&lt;/div&gt;</code></pre>
</div>
</div>
</div>
<div class="section level2">
<h2 id="metabolite-clustering-analysis">4. Metabolite Clustering Analysis<a class="anchor" aria-label="anchor" href="#metabolite-clustering-analysis"></a>
</h2>
<div class="progress progress-striped active">
<pre><code>&lt;div class="progress-bar progress-bar-success" style="width: 550%"&gt;&lt;/div&gt;</code></pre>
</div>
<p>Metabolite Clustering Analysis (<code>MCA</code>) is a module, which
includes different functions to enable clustering of metabolites into
groups either based on logical regulatory rules. This can be
particularly useful if one has multiple conditions and aims to find
patterns in the data.</p>
<div class="section level3">
<h3 id="mca-2cond">MCA-2Cond<a class="anchor" aria-label="anchor" href="#mca-2cond"></a>
</h3>
<p>This metabolite clustering method is based on the Regulatory
Clustering method (RCM) that was developed as part of the Signature
Regulatory Clustering (SiRCle) model (<span class="citation">Mora et al.
(2024)</span>). As part of the <a href="https://github.com/ArianeMora/SiRCleR/tree/main" class="external-link">SiRCleR
package</a>, also variation of the initial RCM method are proposed as
clustering based on two comparisons (e.g. KO versus WT in hypoxia and in
normoxia).<br>
Here we set two different thresholds, one for the differential
metabolite abundance (<code>Log2FC</code>) and one for the
<code>significance</code> (e.g. p.adj). This will define if a feature (=
metabolite) is assigned into:<br>
1. <strong><em>“UP”</em></strong>, which means a metabolite is
significantly up-regulated in the underlying comparison.<br>
2. <strong><em>“DOWN”</em></strong>, which means a metabolite is
significantly down-regulated in the underlying comparison.<br>
3. <strong><em>“No Change”</em></strong>, which means a metabolite does
not change significantly in the underlying comparison and/or is not
defined as up-regulated/down-regulated based on the Log2FC threshold
chosen.<br><br>
Therebye “No Change” is further subdivided into four states:<br>
1. <strong><em>“Not Detected”</em></strong>, which means a metabolite is
not detected in the underlying comparison.<br>
2. <strong><em>“Not Significant”</em></strong>, which means a metabolite
is not significant in the underlying comparison.<br>
3. <strong><em>“Significant positive”</em></strong>, which means a
metabolite is significant in the underlying comparison and the
differential metabolite abundance is positive, yet does not meet the
threshold set for “UP” (e.g. Log2FC &gt;1 = “UP” and we have a
significant Log2FC=0.8).<br>
4. <strong><em>“Significant negative”</em></strong>, which means a
metabolite is significant in the underlying comparison and the
differential metabolite abundance is negative, yet does not meet the
threshold set for “DOWN”.<br><br>
This definition is done individually for each comparison and will impact
in which metabolite cluster a metabolite is sorted into.<br>
Since we have two comparisons, we can choose between different
Background settings, which defines which features will be considered for
the clusters (e.g. you could include only features (= metabolites) that
are detected in both comparisons, removing the rest of the features).The
background methods <code>method_background</code> are the following from
<strong><em>1.1. - 1.4.</em></strong> from most restrictive to least
restrictive:<br><strong><em>1.1. C1&amp;C2</em></strong>: Most stringend background
setting and will lead to a small number of metabolites.<br><strong><em>1.2. C1</em></strong>: Focus is on the metabolite abundance
of Condition 1 (C1).<br><strong><em>1.3. C2</em></strong>: Focus is on the metabolite abundance
of Condition 2 (C2).<br><strong><em>1.4. C1|C2</em></strong>: Least stringent background method,
since a metabolite will be included in the input if it has been detected
on one of the two conditions.<br><br>
Lastly, we will get clusters of metabolites that are defined by the
metabolite change in the two conditions. For example, if Alanine is “UP”
based on the thresholds in both comparisons it will be sorted into the
cluster “core_UP”. As there are two 6-state6 transitions between the
comparisons, the flows are summarised into smaller amount of metabolite
clusters using different Regulation Groupings (RG): 1. RG1_All<br>
2. RG2_Significant taking into account genes that are significant (UP,
DOWN, significant positive, significant negative)<br>
3. RG3_SignificantChange only takes into account genes that have
significant changes (UP, DOWN).<br><br></p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example of all possible flows:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">mca_twocond_rules</span><span class="op">)</span></span>
<span></span>
<span><span class="va">MCA2Cond_Rules</span> <span class="op">&lt;-</span> <span class="va">mca_twocond_rules</span></span></code></pre></div>
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Metabolite Clustering Analysis: 2 Conditions.
</caption>
<thead><tr>
<th style="text-align:left;">
Cond1
</th>
<th style="text-align:left;">
Cond2
</th>
<th style="text-align:left;">
RG1_All
</th>
<th style="text-align:left;">
RG2_Significant
</th>
<th style="text-align:left;">
RG3_SignificantChange
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Cond1 DOWN + Cond2 DOWN
</td>
<td style="text-align:left;">
Core_DOWN
</td>
<td style="text-align:left;">
Core_DOWN
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Cond1 DOWN + Cond2 Not Detected
</td>
<td style="text-align:left;">
Cond1_DOWN
</td>
<td style="text-align:left;">
Cond1_DOWN
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Cond1 DOWN + Cond2 Not Significant
</td>
<td style="text-align:left;">
Cond1_DOWN
</td>
<td style="text-align:left;">
Cond1_DOWN
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Cond1 DOWN + Cond2 Significant Negative
</td>
<td style="text-align:left;">
Core_DOWN
</td>
<td style="text-align:left;">
Cond1_DOWN
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Cond1 DOWN + Cond2 Significant Positive
</td>
<td style="text-align:left;">
Opposite
</td>
<td style="text-align:left;">
Cond1_DOWN
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Cond1 DOWN + Cond2 UP
</td>
<td style="text-align:left;">
Opposite
</td>
<td style="text-align:left;">
Opposite
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Cond1 UP + Cond2 DOWN
</td>
<td style="text-align:left;">
Opposite
</td>
<td style="text-align:left;">
Opposite
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Cond1 UP + Cond2 Not Detected
</td>
<td style="text-align:left;">
Cond1_UP
</td>
<td style="text-align:left;">
Cond1_UP
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Cond1 UP + Cond2 Not Significant
</td>
<td style="text-align:left;">
Cond1_UP
</td>
<td style="text-align:left;">
Cond1_UP
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Cond1 UP + Cond2 Significant Negative
</td>
<td style="text-align:left;">
Opposite
</td>
<td style="text-align:left;">
Cond1_UP
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Cond1 UP + Cond2 Significant Positive
</td>
<td style="text-align:left;">
Core_UP
</td>
<td style="text-align:left;">
Cond1_UP
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Cond1 UP + Cond2 UP
</td>
<td style="text-align:left;">
Core_UP
</td>
<td style="text-align:left;">
Core_UP
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Cond1 Not Detected + Cond2 DOWN
</td>
<td style="text-align:left;">
Cond2_DOWN
</td>
<td style="text-align:left;">
Cond2_DOWN
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Cond1 Not Detected + Cond2 Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Cond1 Not Detected + Cond2 Not Significant
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Cond1 Not Detected + Cond2 Significant Negative
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Cond1 Not Detected + Cond2 Significant Positive
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Cond1 Not Detected + Cond2 UP
</td>
<td style="text-align:left;">
Cond2_UP
</td>
<td style="text-align:left;">
Cond2_UP
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Cond1 Significant Negative + Cond2 DOWN
</td>
<td style="text-align:left;">
Core_DOWN
</td>
<td style="text-align:left;">
Cond2_DOWN
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Cond1 Significant Negative + Cond2 Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Cond1 Significant Negative + Cond2 Not Significant
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Cond1 Significant Negative + Cond2 Significant Negative
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Cond1 Significant Negative + Cond2 Significant Positive
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Cond1 Significant Negative + Cond2 UP
</td>
<td style="text-align:left;">
Opposite
</td>
<td style="text-align:left;">
Cond2_UP
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Cond1 Significant Positive + Cond2 DOWN
</td>
<td style="text-align:left;">
Opposite
</td>
<td style="text-align:left;">
Cond2_DOWN
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Cond1 Significant Positive + Cond2 Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Cond1 Significant Positive + Cond2 Not Significant
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Cond1 Significant Positive + Cond2 Significant Negative
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Cond1 Significant Positive + Cond2 Significant Positive
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Cond1 Significant Positive + Cond2 UP
</td>
<td style="text-align:left;">
Core_UP
</td>
<td style="text-align:left;">
Cond2_UP
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Cond1 Not Significant + Cond2 DOWN
</td>
<td style="text-align:left;">
Cond2_DOWN
</td>
<td style="text-align:left;">
Cond2_DOWN
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Cond1 Not Significant + Cond2 Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Cond1 Not Significant + Cond2 Not Significant
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Cond1 Not Significant + Cond2 Significant Negative
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Cond1 Not Significant + Cond2 Significant Positive
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Cond1 Not Significant + Cond2 UP
</td>
<td style="text-align:left;">
Cond1_UP
</td>
<td style="text-align:left;">
Cond1_UP
</td>
</tr>
</tbody>
</table>
<p><br>
For a detailed example of the <code><a href="../reference/mca_2cond.html">mca_2cond()</a></code> function visit
the extended vignette <a href="https://saezlab.github.io/MetaProViz/articles/standard-metabolomics.html#mca" class="external-link">Standard
metabolomics data</a>.</p>
</div>
<div class="section level3">
<h3 id="mca-core">MCA-CoRe<a class="anchor" aria-label="anchor" href="#mca-core"></a>
</h3>
<p>This metabolite clustering method is based on logical regulatory
rules to sort metabolites into metabolite clusters. Here you need
intracellular metabolomics and corresponding consumption-release
metabolomics.<br>
Here we will define if a feature (= metabolite) is assigned into:<br>
1. <strong><em>“UP”</em></strong>, which means a metabolite is
significantly up-regulated in the underlying comparison.<br>
2. <strong><em>“DOWN”</em></strong>, which means a metabolite is
significantly down-regulated in the underlying comparison.<br>
3. <strong><em>“No Change”</em></strong>, which means a metabolite does
not change significantly in the underlying comparison and/or is not
defined as up-regulated/down-regulated based on the Log2FC threshold
chosen.<br><br>
Therebye “No Change” is further subdivided into four states:<br>
1. <strong><em>“Not Detected”</em></strong>, which means a metabolite is
not detected in the underlying comparison.<br>
2. <strong><em>“Not Significant”</em></strong>, which means a metabolite
is not significant in the underlying comparison.<br>
3. <strong><em>“Significant positive”</em></strong>, which means a
metabolite is significant in the underlying comparison and the
differential metabolite abundance is positive, yet does not meet the
threshold set for “UP” (e.g. Log2FC &gt;1 = “UP” and we have a
significant Log2FC=0.8).<br>
4. <strong><em>“Significant negative”</em></strong>, which means a
metabolite is significant in the underlying comparison and the
differential metabolite abundance is negative, yet does not meet the
threshold set for “DOWN”.<br><br>
Lastly, we also take into account the core direction, meaning if a
metabolite was:<br>
1. <strong><em>“Released”</em></strong>, which means is released into
the media in both conditions of the underlying comparison.<br>
2. <strong><em>“Consumed”</em></strong>, which means is consumed from
the media in both conditions of the underlying comparison.<br>
3. <strong><em>“Released/Consumed”</em></strong>, which means is
consumed/released in one condition, whilst the opposite occurs in the
second condition of the underlying comparison.<br>
4. <strong><em>“Not Detected”</em></strong>, which means a metabolite is
not detected in the underlying comparison.<br>
This definition is done individually for each comparison and will impact
in which metabolite cluster a metabolite is sorted into.<br>
Since we have two comparisons (Intracellular and core), we can choose
between different Background settings, which defines which features will
be considered for the clusters (e.g. you could include only features (=
metabolites) that are detected in both comparisons, removing the rest of
the features).The background methods <code>method_background</code> are
the following from <strong><em>1.1. - 1.4.</em></strong> from most
restrictive to least restrictive:<br><strong><em>1.1. Intra&amp;core</em></strong>: Most stringend background
setting and will lead to a small number of metabolites.<br><strong><em>1.2. core</em></strong>: Focus is on the metabolite
abundance of the core.<br><strong><em>1.3. Intra</em></strong>: Focus is on the metabolite
abundance of intracellular.<br><strong><em>1.4. Intra|core</em></strong>: Least stringent background
method, since a metabolite will be included in the input if it has been
detected on one of the two conditions.<br><br>
Lastly, we will get clusters of metabolites that are defined by the
metabolite change in the two conditions. For example, if Alanine is “UP”
based on the thresholds in both comparisons it will be sorted into the
cluster “core_UP”. As there are three 6-state6-state4 transitions
between the comparisons, the flows are summarised into smaller amount of
metabolite clusters using different Regulation Groupings (RG): 1.
RG1_All<br>
2. RG2_Significant taking into account genes that are significant (UP,
DOWN, significant positive, significant negative)<br>
3. RG3_SignificantChange only takes into account genes that have
significant changes (UP, DOWN).<br><br>
In order to define which group a metabolite is assigned to, we set two
different thresholds. For intracellular those are based on the
differential metabolite abundance (<code>Log2FC</code>) and the
<code>significance</code> (e.g. p.adj). For the core data this is based
on the <code>Log2 Distance</code> and the <code>significance</code>
(e.g. p.adj). For <code>Log2FC</code> we recommend a threshold of 0.5 or
1, whilst for the <code>Log2 Distance</code> one should check the
distance ranges and base the threshold on this.<br><br>
Regulatory rules:<br><br></p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example of all possible flows:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">mca_core_rules</span><span class="op">)</span></span>
<span></span>
<span><span class="va">MCA_CoRe_Rule</span> <span class="op">&lt;-</span> <span class="va">mca_core_rules</span></span></code></pre></div>
<div class="vscroll-plot">
<table class="table  lightable-classic" style="font-size: 12px; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Metabolite Clustering Analysis: core.
</caption>
<thead><tr>
<th style="text-align:left;">
Intra
</th>
<th style="text-align:left;">
CoRe
</th>
<th style="text-align:left;">
Core_Direction
</th>
<th style="text-align:left;">
RG1_All
</th>
<th style="text-align:left;">
R2_Significant
</th>
<th style="text-align:left;">
RG3_Change
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra DOWN+ CoRe DOWN_Released
</td>
<td style="text-align:left;">
Both_DOWN (Released)
</td>
<td style="text-align:left;">
Both_DOWN (Released)
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra DOWN+ CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra DOWN+ CoRe Not Significant_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra DOWN+ CoRe Significant Negative_Released
</td>
<td style="text-align:left;">
Both_DOWN (Released)
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra DOWN+ CoRe Significant Positive_Released
</td>
<td style="text-align:left;">
Opposite (Released UP)
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra DOWN+ CoRe UP_Released
</td>
<td style="text-align:left;">
Opposite (Released UP)
</td>
<td style="text-align:left;">
Opposite (Released UP)
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra UP+ CoRe DOWN_Released
</td>
<td style="text-align:left;">
Opposite (Released DOWN)
</td>
<td style="text-align:left;">
Opposite (Released DOWN)
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra UP+ CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra UP+ CoRe Not Significant_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra UP+ CoRe Significant Negative_Released
</td>
<td style="text-align:left;">
Opposite (Released UP)
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra UP+ CoRe Significant Positive_Released
</td>
<td style="text-align:left;">
Both_UP (Released)
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra UP+ CoRe UP_Released
</td>
<td style="text-align:left;">
Both_UP (Released)
</td>
<td style="text-align:left;">
Both_UP (Released)
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Not Detected+ CoRe DOWN_Released
</td>
<td style="text-align:left;">
CoRe_DOWN (Released)
</td>
<td style="text-align:left;">
CoRe_DOWN (Released)
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra Not Detected+ CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Not Detected+ CoRe Not Significant_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Not Detected+ CoRe Significant Negative_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Not Detected+ CoRe Significant Positive_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Not Detected+ CoRe UP_Released
</td>
<td style="text-align:left;">
CoRe_UP (Released)
</td>
<td style="text-align:left;">
CoRe_UP (Released)
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Significant Negative+ CoRe DOWN_Released
</td>
<td style="text-align:left;">
Both_DOWN (Released)
</td>
<td style="text-align:left;">
CoRe_DOWN (Released)
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra Significant Negative+ CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Significant Negative+ CoRe Not Significant_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Significant Negative+ CoRe Significant Negative_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Significant Negative+ CoRe Significant Positive_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Significant Negative+ CoRe UP_Released
</td>
<td style="text-align:left;">
Opposite (Released UP)
</td>
<td style="text-align:left;">
CoRe_UP (Released)
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Significant Positive+ CoRe DOWN_Released
</td>
<td style="text-align:left;">
Opposite (Released DOWN)
</td>
<td style="text-align:left;">
CoRe_DOWN (Released)
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra Significant Positive+ CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Significant Positive+ CoRe Not Significant_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Significant Positive+ CoRe Significant Negative_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Significant Positive+ CoRe Significant Positive_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Significant Positive+ CoRe UP_Released
</td>
<td style="text-align:left;">
Both_UP (Released)
</td>
<td style="text-align:left;">
CoRe_UP (Released)
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Not Significant+ CoRe DOWN_Released
</td>
<td style="text-align:left;">
CoRe_DOWN (Released)
</td>
<td style="text-align:left;">
CoRe_DOWN (Released)
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra Not Significant+ CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Not Significant+ CoRe Not Significant_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Not Significant+ CoRe Significant Negative_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Not Significant+ CoRe Significant Positive_Released
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Released
</td>
<td style="text-align:left;">
Intra Not Significant+ CoRe UP_Released
</td>
<td style="text-align:left;">
CoRe_UP (Released)
</td>
<td style="text-align:left;">
CoRe_UP (Released)
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra DOWN+ CoRe DOWN_Consumed
</td>
<td style="text-align:left;">
Both_DOWN (Consumed)
</td>
<td style="text-align:left;">
Both_DOWN (Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra DOWN+ CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra DOWN+ CoRe Not Significant_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra DOWN+ CoRe Significant Negative_Consumed
</td>
<td style="text-align:left;">
Both_DOWN (Consumed)
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra DOWN+ CoRe Significant Positive_Consumed
</td>
<td style="text-align:left;">
Opposite (Consumed UP)
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra DOWN+ CoRe UP_Consumed
</td>
<td style="text-align:left;">
Opposite (Consumed UP)
</td>
<td style="text-align:left;">
Opposite (Consumed UP)
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra UP+ CoRe DOWN_Consumed
</td>
<td style="text-align:left;">
Opposite (Consumed DOWN)
</td>
<td style="text-align:left;">
Opposite (Consumed DOWN)
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra UP+ CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra UP+ CoRe Not Significant_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra UP+ CoRe Significant Negative_Consumed
</td>
<td style="text-align:left;">
Opposite (Consumed UP)
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra UP+ CoRe Significant Positive_Consumed
</td>
<td style="text-align:left;">
Both_UP (Consumed)
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra UP+ CoRe UP_Consumed
</td>
<td style="text-align:left;">
Both_UP (Consumed)
</td>
<td style="text-align:left;">
Both_UP (Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Not Detected+ CoRe DOWN_Consumed
</td>
<td style="text-align:left;">
CoRe_DOWN (Consumed)
</td>
<td style="text-align:left;">
CoRe_DOWN (Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra Not Detected+ CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Not Detected+ CoRe Not Significant_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Not Detected+ CoRe Significant Negative_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Not Detected+ CoRe Significant Positive_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Not Detected+ CoRe UP_Consumed
</td>
<td style="text-align:left;">
CoRe_UP (Consumed)
</td>
<td style="text-align:left;">
CoRe_UP (Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Significant Negative+ CoRe DOWN_Consumed
</td>
<td style="text-align:left;">
Both_DOWN (Consumed)
</td>
<td style="text-align:left;">
CoRe_DOWN (Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra Significant Negative+ CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Significant Negative+ CoRe Not Significant_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Significant Negative+ CoRe Significant Negative_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Significant Negative+ CoRe Significant Positive_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Significant Negative+ CoRe UP_Consumed
</td>
<td style="text-align:left;">
Opposite (Consumed UP)
</td>
<td style="text-align:left;">
CoRe_UP (Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Significant Positive+ CoRe DOWN_Consumed
</td>
<td style="text-align:left;">
Opposite (Consumed DOWN)
</td>
<td style="text-align:left;">
CoRe_DOWN (Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra Significant Positive+ CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Significant Positive+ CoRe Not Significant_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Significant Positive+ CoRe Significant Negative_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Significant Positive+ CoRe Significant Positive_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Significant Positive+ CoRe UP_Consumed
</td>
<td style="text-align:left;">
Both_UP (Consumed)
</td>
<td style="text-align:left;">
CoRe_UP (Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Not Significant+ CoRe DOWN_Consumed
</td>
<td style="text-align:left;">
CoRe_DOWN (Consumed)
</td>
<td style="text-align:left;">
CoRe_DOWN (Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra Not Significant+ CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Not Significant+ CoRe Not Significant_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Not Significant+ CoRe Significant Negative_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Not Significant+ CoRe Significant Positive_Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Consumed
</td>
<td style="text-align:left;">
Intra Not Significant+ CoRe UP_Consumed
</td>
<td style="text-align:left;">
CoRe_UP (Consumed)
</td>
<td style="text-align:left;">
CoRe_UP (Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra DOWN + CoRe DOWN_Released/Consumed
</td>
<td style="text-align:left;">
Both_DOWN (Released/Consumed)
</td>
<td style="text-align:left;">
Both_DOWN (Released/Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra DOWN + CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra DOWN + CoRe Not Significant_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra DOWN + CoRe Significant Negative_Released/Consumed
</td>
<td style="text-align:left;">
Both_DOWN (Released/Consumed)
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra DOWN + CoRe Significant Positive_Released/Consumed
</td>
<td style="text-align:left;">
Opposite (Released/Consumed UP)
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra DOWN + CoRe UP_Released/Consumed
</td>
<td style="text-align:left;">
Opposite (Released/Consumed UP)
</td>
<td style="text-align:left;">
Opposite (Released/Consumed UP)
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra UP + CoRe DOWN_Released/Consumed
</td>
<td style="text-align:left;">
Opposite (Released/Consumed DOWN)
</td>
<td style="text-align:left;">
Opposite (Released/Consumed DOWN)
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra UP + CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra UP + CoRe Not Significant_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra UP + CoRe Significant Negative_Released/Consumed
</td>
<td style="text-align:left;">
Opposite (Released/Consumed UP)
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra UP + CoRe Significant Positive_Released/Consumed
</td>
<td style="text-align:left;">
Both_UP (Released/Consumed)
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra UP + CoRe UP_Released/Consumed
</td>
<td style="text-align:left;">
Both_UP (Released/Consumed)
</td>
<td style="text-align:left;">
Both_UP (Released/Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Not Detected + CoRe DOWN_Released/Consumed
</td>
<td style="text-align:left;">
CoRe_DOWN (Released/Consumed)
</td>
<td style="text-align:left;">
CoRe_DOWN (Released/Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra Not Detected + CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Not Detected + CoRe Not Significant_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Not Detected + CoRe Significant Negative_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Not Detected + CoRe Significant Positive_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Not Detected + CoRe UP_Released/Consumed
</td>
<td style="text-align:left;">
CoRe_UP (Released/Consumed)
</td>
<td style="text-align:left;">
CoRe_UP (Released/Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Significant Negative + CoRe DOWN_Released/Consumed
</td>
<td style="text-align:left;">
Both_DOWN (Released/Consumed)
</td>
<td style="text-align:left;">
CoRe_DOWN (Released/Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra Significant Negative + CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Significant Negative + CoRe Not Significant_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Significant Negative + CoRe Significant Negative_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Significant Negative + CoRe Significant Positive_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Significant Negative + CoRe UP_Released/Consumed
</td>
<td style="text-align:left;">
Opposite (Released/Consumed UP)
</td>
<td style="text-align:left;">
CoRe_UP (Released/Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Significant Positive + CoRe DOWN_Released/Consumed
</td>
<td style="text-align:left;">
Opposite (Released/Consumed DOWN)
</td>
<td style="text-align:left;">
CoRe_DOWN (Released/Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra Significant Positive + CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Significant Positive + CoRe Not Significant_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Significant Positive + CoRe Significant Negative_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Significant Positive + CoRe Significant Positive_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Significant Positive + CoRe UP_Released/Consumed
</td>
<td style="text-align:left;">
Both_UP (Released/Consumed)
</td>
<td style="text-align:left;">
CoRe_UP (Released/Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
DOWN
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Not Significant + CoRe DOWN_Released/Consumed
</td>
<td style="text-align:left;">
CoRe_DOWN (Released/Consumed)
</td>
<td style="text-align:left;">
CoRe_DOWN (Released/Consumed)
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Not Detected
</td>
<td style="text-align:left;">
Intra Not Significant + CoRe Not Detected
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Not Significant + CoRe Not Significant_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Significant Negative
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Not Significant + CoRe Significant Negative_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
Significant Positive
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Not Significant + CoRe Significant Positive_Released/Consumed
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
</tr>
<tr>
<td style="text-align:left;">
Not Significant
</td>
<td style="text-align:left;">
UP
</td>
<td style="text-align:left;">
Released/Consumed
</td>
<td style="text-align:left;">
Intra Not Significant + CoRe UP_Released/Consumed
</td>
<td style="text-align:left;">
CoRe_UP (Released/Consumed)
</td>
<td style="text-align:left;">
CoRe_UP (Released/Consumed)
</td>
</tr>
</tbody>
</table>
</div>
<p><br><br><br></p>
<p>For a detailed example of the <code><a href="../reference/mca_core.html">mca_core()</a></code> function visit
the extended vignette <a href="https://saezlab.github.io/MetaProViz/articles/core-metabolomics.html#mca" class="external-link">Consumption-Release
(CoRe) metabolomics data from cell culture media</a>.</p>
</div>
<div class="section level3">
<h3 id="ora-on-each-metabolite-cluster">ORA on each metabolite cluster<a class="anchor" aria-label="anchor" href="#ora-on-each-metabolite-cluster"></a>
</h3>
<p>As explained in detail above, Over Representation Analysis (ORA) is a
pathway enrichment analysis (PEA) method. As ORA is based on the Fishers
exact test it is perfect to test if a set of features (=metabolic
pathways) are over-represented in the selection of features (= clusters
of metabolites) from the data in comparison to all measured features
(all metabolites). In detail, <code><a href="../reference/cluster_ora.html">cluster_ora()</a></code> will perform
ORA on each of the metabolite clusters we got as a result of performing
<code>mca_2cond</code> or <code>mca_core</code> using all metabolites as
the background. For a detailed example of the <code><a href="../reference/cluster_ora.html">cluster_ora()</a></code>
function visit the extended vignette <a href="https://saezlab.github.io/MetaProViz/articles/core-metabolomics.html#ora-on-each-metabolite-cluster" class="external-link">Consumption-Release
(CoRe) metabolomics data from cell culture media</a> or <a href="https://saezlab.github.io/MetaProViz/articles/standard-metabolomics.html#ora-on-each-metabolite-cluster" class="external-link">Standard
metabolomics data</a>.</p>
</div>
</div>
<div class="section level2">
<h2 id="viz">Viz<a class="anchor" aria-label="anchor" href="#viz"></a>
</h2>
<p>The big advantages of the <code>MetaProViz</code> visualization
module is its flexible and easy usage, which we will showcase below and
that the figures are saved in a publication ready style and format. For
instance, the x- and y-axis size will always be adjusted for the amount
of samples or features (=metabolites) plotted, or in the case of Volcano
plot and PCA plot the axis size is fixed and not affected by figure
legends or title. In this way, there is no need for many adjustments and
the figures can just be dropped into the presentation or paper and are
all in the same style.<br><br>
All the <code>VizPlotName()</code> functions are constructed in the same
way. Indeed, with the parameter <code>Plot_metadata_info</code> the user
can pass a named vector with information about the metadata column that
should be used to customize the plot by colour, shape or creating
individual plots, which will all be showcased for the different plot
types. Via the parameter <code>Plot_SettingsFile</code> the user can
pass the metadata DF, which can be dependent on the plot type for the
samples and/or the features (=metabolites). In case of both the
parameter is named <code>Plot_metadata_sample</code> and
<code>Plot_metadata_feature</code>.<br><br>
In each of those Plot_Settings, the user can label color and/or shape
based on additional information (e.g. Pathway information, Cluster
information or other other demographics like gender). Moreover, we also
enable to plot individual plots where applicable based on those MetaData
(e.g. one plot for each metabolic pathway).<br><br>
We support the plot types:<br>
- PCA plot - Superplots (Bar, Box and Violin plots) - Heatmaps - Volcano
Plots</p>
<p>For a detailed example of the visualisation functions visit the
extended vignette <a href="https://saezlab.github.io/MetaProViz/articles/core-metabolomics.html#run-metaproviz-visualisation" class="external-link">Consumption-Release
(CoRe) metabolomics data from cell culture media</a> or <a href="https://saezlab.github.io/MetaProViz/articles/standard-metabolomics.html#run-metaproviz-visualisation" class="external-link">Standard
metabolomics data</a>.</p>
<div class="progress progress-striped active">
<pre><code>&lt;div class="progress-bar progress-bar-success" style="width: 100%"&gt;&lt;/div&gt;</code></pre>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<pre><code><span><span class="co">#&gt; R version 4.5.2 (2025-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.3 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Etc/UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] tibble_3.3.0       ggfortify_0.4.19   ggplot2_4.0.1      rlang_1.1.6        dplyr_1.1.4        magrittr_2.0.4    </span></span>
<span><span class="co">#&gt; [7] MetaProViz_3.99.25 BiocStyle_2.38.0  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] RColorBrewer_1.1-3          rstudioapi_0.17.1           jsonlite_2.0.0              ggbeeswarm_0.7.2           </span></span>
<span><span class="co">#&gt;   [5] farver_2.1.2                rmarkdown_2.30              fs_1.6.6                    ragg_1.5.0                 </span></span>
<span><span class="co">#&gt;   [9] vctrs_0.6.5                 memoise_2.0.1               rstatix_0.7.3               htmltools_0.5.8.1          </span></span>
<span><span class="co">#&gt;  [13] S4Arrays_1.10.0             progress_1.2.3              curl_7.0.0                  ComplexUpset_1.3.3         </span></span>
<span><span class="co">#&gt;  [17] decoupleR_2.16.0            broom_1.0.10                cellranger_1.1.0            SparseArray_1.10.1         </span></span>
<span><span class="co">#&gt;  [21] Formula_1.2-5               sass_0.4.10                 parallelly_1.45.1           bslib_0.9.0                </span></span>
<span><span class="co">#&gt;  [25] htmlwidgets_1.6.4           desc_1.4.3                  plyr_1.8.9                  httr2_1.2.1                </span></span>
<span><span class="co">#&gt;  [29] lubridate_1.9.4             cachem_1.1.0                igraph_2.2.1                lifecycle_1.0.4            </span></span>
<span><span class="co">#&gt;  [33] pkgconfig_2.0.3             Matrix_1.7-4                R6_2.6.1                    fastmap_1.2.0              </span></span>
<span><span class="co">#&gt;  [37] MatrixGenerics_1.22.0       digest_0.6.38               colorspace_2.1-2            patchwork_1.3.2            </span></span>
<span><span class="co">#&gt;  [41] S4Vectors_0.48.0            textshaping_1.0.4           GenomicRanges_1.62.0        RSQLite_2.4.4              </span></span>
<span><span class="co">#&gt;  [45] ggpubr_0.6.2                labeling_0.4.3              timechange_0.3.0            httr_1.4.7                 </span></span>
<span><span class="co">#&gt;  [49] abind_1.4-8                 compiler_4.5.2              bit64_4.6.0-1               withr_3.0.2                </span></span>
<span><span class="co">#&gt;  [53] S7_0.2.1                    backports_1.5.0             BiocParallel_1.44.0         carData_3.0-5              </span></span>
<span><span class="co">#&gt;  [57] DBI_1.2.3                   logger_0.4.1                OmnipathR_3.19.2            R.utils_2.13.0             </span></span>
<span><span class="co">#&gt;  [61] ggsignif_0.6.4              cosmosR_1.18.0              MASS_7.3-65                 rappdirs_0.3.3             </span></span>
<span><span class="co">#&gt;  [65] DelayedArray_0.36.0         sessioninfo_1.2.3           scatterplot3d_0.3-44        gtools_3.9.5               </span></span>
<span><span class="co">#&gt;  [69] tools_4.5.2                 vipor_0.4.7                 beeswarm_0.4.0              qcc_2.7                    </span></span>
<span><span class="co">#&gt;  [73] zip_2.3.3                   R.oo_1.27.1                 glue_1.8.0                  grid_4.5.2                 </span></span>
<span><span class="co">#&gt;  [77] checkmate_2.3.3             reshape2_1.4.5              generics_0.1.4              gtable_0.3.6               </span></span>
<span><span class="co">#&gt;  [81] tzdb_0.5.0                  R.methodsS3_1.8.2           tidyr_1.3.1                 hms_1.1.4                  </span></span>
<span><span class="co">#&gt;  [85] xml2_1.5.0                  car_3.1-3                   XVector_0.50.0              BiocGenerics_0.56.0        </span></span>
<span><span class="co">#&gt;  [89] ggrepel_0.9.6               pillar_1.11.1               stringr_1.6.0               vroom_1.6.6                </span></span>
<span><span class="co">#&gt;  [93] limma_3.66.0                later_1.4.4                 splines_4.5.2               lattice_0.22-7             </span></span>
<span><span class="co">#&gt;  [97] bit_4.6.0                   tidyselect_1.2.1            knitr_1.50                  gridExtra_2.3              </span></span>
<span><span class="co">#&gt; [101] bookdown_0.45               IRanges_2.44.0              Seqinfo_1.0.0               SummarizedExperiment_1.40.0</span></span>
<span><span class="co">#&gt; [105] svglite_2.2.2               stats4_4.5.2                xfun_0.54                   Biobase_2.70.0             </span></span>
<span><span class="co">#&gt; [109] statmod_1.5.1               factoextra_1.0.7            matrixStats_1.5.0           pheatmap_1.0.13            </span></span>
<span><span class="co">#&gt; [113] stringi_1.8.7               yaml_2.3.10                 kableExtra_1.4.0            evaluate_1.0.5             </span></span>
<span><span class="co">#&gt; [117] codetools_0.2-20            tcltk_4.5.2                 qvalue_2.42.0               hash_2.2.6.3               </span></span>
<span><span class="co">#&gt; [121] BiocManager_1.30.27         Polychrome_1.5.4            cli_3.6.5                   systemfonts_1.3.1          </span></span>
<span><span class="co">#&gt; [125] jquerylib_0.1.4             EnhancedVolcano_1.13.2      Rcpp_1.1.0                  readxl_1.4.5               </span></span>
<span><span class="co">#&gt; [129] XML_3.99-0.20               parallel_4.5.2              pkgdown_2.2.0               readr_2.1.6                </span></span>
<span><span class="co">#&gt; [133] blob_1.2.4                  prettyunits_1.2.0           viridisLite_0.4.2           scales_1.4.0               </span></span>
<span><span class="co">#&gt; [137] writexl_1.5.4               inflection_1.3.7            purrr_1.2.0                 crayon_1.5.3               </span></span>
<span><span class="co">#&gt; [141] rvest_1.0.5</span></span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-Bijlsma2006" class="csl-entry">
Bijlsma, Sabina, Ivana Bobeldijk, Elwin R Verheij, Raymond Ramaker,
Sunil Kochhar, Ian A Macdonald, Ben van Ommen, and Age K Smilde. 2006.
<span>“Large-Scale Human Metabolomics Studies: A Strategy for Data
(Pre-) Processing and Validation.”</span> <em>Analytical Chemistry</em>
78 (2): 567–74. <a href="https://doi.org/10.1021/ac051495j" class="external-link">https://doi.org/10.1021/ac051495j</a>.
</div>
<div id="ref-Hotelling1931" class="csl-entry">
Hotelling, Harold. 1931. <span>“The Generalization of Student’s
Ratio.”</span> <em>The Annals of Mathematical Statistics</em> 2 (3):
360–78. <a href="https://doi.org/10.1214/aoms/1177732979" class="external-link">https://doi.org/10.1214/aoms/1177732979</a>.
</div>
<div id="ref-Kanehisa2000" class="csl-entry">
Kanehisa, M, and S Goto. 2000. <span>“KEGG: Kyoto Encyclopedia of Genes
and Genomes.”</span> <em>Nucleic Acids Research</em> 28 (1): 27–30. <a href="https://doi.org/10.1093/nar/28.1.27" class="external-link">https://doi.org/10.1093/nar/28.1.27</a>.
</div>
<div id="ref-Mora_Schmidt2024" class="csl-entry">
Mora, Ariane, Christina Schmidt, Brad Balderson, Christian Frezza, and
Mikael Bodén. 2024. <span>“SiRCle (Signature Regulatory Clustering)
Model Integration Reveals Mechanisms of Phenotype Regulation in Renal
Cancer.”</span> <em>Genome Medicine</em> 16 (1): 144. <a href="https://doi.org/10.1186/s13073-024-01415-3" class="external-link">https://doi.org/10.1186/s13073-024-01415-3</a>.
</div>
<div id="ref-Sciacovelli_Dugourd2022" class="csl-entry">
Sciacovelli, Marco, Aurelien Dugourd, Lorea Valcarcel Jimenez, Ming
Yang, Efterpi Nikitopoulou, Ana S H Costa, Laura Tronci, et al. 2022.
<span>“Dynamic Partitioning of Branched-Chain Amino Acids-Derived
Nitrogen Supports Renal Cancer Progression.”</span> <em>Nature
Communications</em> 13 (1): 7830. <a href="https://doi.org/10.1038/s41467-022-35036-4" class="external-link">https://doi.org/10.1038/s41467-022-35036-4</a>.
</div>
<div id="ref-Wei2018" class="csl-entry">
Wei, Runmin, Jingye Wang, Mingming Su, Erik Jia, Shaoqiu Chen, Tianlu
Chen, and Yan Ni. 2018. <span>“Missing Value Imputation Approach for
Mass Spectrometry-Based Metabolomics Data.”</span> <em>Scientific
Reports</em> 8 (1): 663. <a href="https://doi.org/10.1038/s41598-017-19120-0" class="external-link">https://doi.org/10.1038/s41598-017-19120-0</a>.
</div>
<div id="ref-Wulff2018" class="csl-entry">
Wulff, Jacob E., and Matthew W. Mitchell. 2018. <span>“A Comparison of
Various Normalization Methods for LC/MS Metabolomics Data.”</span>
<em>Advances in Bioscience and Biotechnology</em> 09 (08): 339–51. <a href="https://doi.org/10.4236/abb.2018.98022" class="external-link">https://doi.org/10.4236/abb.2018.98022</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Christina Schmidt, Denes Turei, Dimitrios Prymidis, Macabe Daley, Jannik Franken, Julio Saez-Rodriguez, Christian Frezza.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
